// @opentiny/vue-country/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/country/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Country = defineComponent({
  name: $prefix + "Country",
  props: _extends({}, $props, {
    disabled: {
      type: Boolean,
      default: function _default() {
        return false;
      }
    },
    clearable: {
      type: Boolean,
      default: function _default2() {
        return true;
      }
    },
    fetchCountry: Function,
    fields: Object,
    placeholder: {
      type: String,
      default: function _default3() {
        return "";
      }
    },
    modelValue: {},
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: function _default4() {
        return true;
      }
    },
    size: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Country.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Country.install = function(Vue) {
  Vue.component(Country.name, Country);
};
Country.version = version;
export {
  Country as default
};
