// @opentiny/vue-fall-menu/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/fall-menu/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Row from "@opentiny/vue-row";
import Col from "@opentiny/vue-col";
import { iconChevronLeft, iconChevronRight } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, createElementVNode, withDirectives, renderSlot, createVNode, vShow, normalizeStyle, Fragment, renderList, normalizeClass, toDisplayString, withCtx, createBlock } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["hook-updated"],
  props: [].concat(props, ["data"]),
  components: {
    TinyRow: Row,
    TinyCol: Col,
    IconChevronLeft: iconChevronLeft(),
    IconChevronRight: iconChevronRight()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-fall-menu"
};
var _hoisted_2 = {
  class: "tiny-fall-menu__wrap"
};
var _hoisted_3 = {
  class: "tiny-fall-menu__nav"
};
var _hoisted_4 = {
  class: "tiny-fall-menu__subnav"
};
var _hoisted_5 = ["onMouseover"];
var _hoisted_6 = ["href"];
var _hoisted_7 = {
  class: "contbox"
};
var _hoisted_8 = {
  class: "sublist clearfix"
};
var _hoisted_9 = {
  class: "mcate-item-hd"
};
var _hoisted_10 = {
  class: "mcate-item-bd"
};
var _hoisted_11 = ["href"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_chevron_left = resolveComponent("icon-chevron-left");
  var _component_icon_chevron_right = resolveComponent("icon-chevron-right");
  var _component_tiny_col = resolveComponent("tiny-col");
  var _component_tiny_row = resolveComponent("tiny-row");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [createElementVNode("div", _hoisted_4, [withDirectives(createElementVNode(
    "div",
    {
      class: "icon-slot-left",
      onClick: _cache[0] || (_cache[0] = function($event) {
        return _ctx.arrowClick(-1);
      })
    },
    [renderSlot(_ctx.$slots, "left", {}, function() {
      return [createVNode(_component_icon_chevron_left, {
        class: "tiny-svg-size"
      })];
    })],
    512
    /* NEED_PATCH */
  ), [[vShow, _ctx.state.pager != 1]]), createElementVNode(
    "div",
    {
      ref: "list",
      style: normalizeStyle({
        left: _ctx.state.left
      }),
      class: "tiny-fall-menu__list"
    },
    [createElementVNode("ul", null, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.data, function(level1, index) {
        return openBlock(), createElementBlock("li", {
          key: index,
          onMouseover: function onMouseover($event) {
            return _ctx.mouseover(index);
          },
          onMouseout: _cache[1] || (_cache[1] = function() {
            return _ctx.mouseout && _ctx.mouseout.apply(_ctx, arguments);
          }),
          class: normalizeClass({
            "fall-hide": index >= _ctx.state.pagerData.index[_ctx.state.pager - 1] && _ctx.state.pager !== _ctx.state.pagerData.size
          })
        }, [renderSlot(_ctx.$slots, "level1", {
          slotScope: level1
        }, function() {
          return [createElementVNode("a", {
            href: level1.url,
            class: normalizeClass({
              now: index === _ctx.state.activeNode
            })
          }, toDisplayString(level1.title), 11, _hoisted_6)];
        })], 42, _hoisted_5);
      }),
      128
      /* KEYED_FRAGMENT */
    ))])],
    4
    /* STYLE */
  ), withDirectives(createElementVNode(
    "div",
    {
      class: "icon-slot-right",
      onClick: _cache[2] || (_cache[2] = function($event) {
        return _ctx.arrowClick(1);
      })
    },
    [renderSlot(_ctx.$slots, "right", {}, function() {
      return [createVNode(_component_icon_chevron_right, {
        class: "tiny-svg-size"
      })];
    })],
    512
    /* NEED_PATCH */
  ), [[vShow, _ctx.state.pager !== _ctx.state.pagerData.size]])]), createVNode(_component_tiny_row, null, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          ref: "fall",
          class: normalizeClass([{
            active: _ctx.state.isActive
          }, "tiny-fall-menu__box"]),
          onMouseover: _cache[3] || (_cache[3] = function($event) {
            return _ctx.overContent(true);
          }),
          onMouseout: _cache[4] || (_cache[4] = function($event) {
            return _ctx.overContent(false);
          })
        },
        [createElementVNode("div", _hoisted_7, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.level2data, function(level2, index) {
            return openBlock(), createBlock(
              _component_tiny_col,
              {
                key: index,
                xs: 6,
                sm: 3,
                md: 3,
                lg: 3,
                xl: 2,
                class: "cont"
              },
              {
                default: withCtx(function() {
                  return [createElementVNode("ul", _hoisted_8, [createElementVNode("li", null, [createElementVNode("h3", _hoisted_9, [renderSlot(_ctx.$slots, "level2", {
                    slotScope: level2
                  }, function() {
                    return [createElementVNode(
                      "span",
                      null,
                      toDisplayString(level2.title),
                      1
                      /* TEXT */
                    )];
                  })]), createElementVNode("p", _hoisted_10, [renderSlot(_ctx.$slots, "level3", {
                    slotScope: level2.children
                  }, function() {
                    return [(openBlock(true), createElementBlock(
                      Fragment,
                      null,
                      renderList(level2.children, function(level3, index2) {
                        return openBlock(), createElementBlock("a", {
                          href: level3.url,
                          key: index2
                        }, toDisplayString(level3.title), 9, _hoisted_11);
                      }),
                      128
                      /* KEYED_FRAGMENT */
                    ))];
                  })])])])];
                }),
                _: 2
                /* DYNAMIC */
              },
              1024
              /* DYNAMIC_SLOTS */
            );
          }),
          128
          /* KEYED_FRAGMENT */
        ))])],
        34
        /* CLASS, NEED_HYDRATION */
      ), [[vShow, _ctx.state.isActive]])];
    }),
    _: 3
    /* FORWARDED */
  })])])]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
