// @opentiny/vue-radio-button/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/radio-button/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconHelpCircle } from "@opentiny/vue-icon";
import Tooltip from "@opentiny/vue-tooltip";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, withKeys, withModifiers, withDirectives, createElementVNode, vModelRadio, renderSlot, Fragment, createTextVNode, toDisplayString, createBlock, withCtx, createVNode, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change"],
  props: [].concat(props, ["label", "events", "text", "disabled", "name", "tipContent"]),
  components: {
    TinyTooltip: Tooltip,
    IconHelpCircle: iconHelpCircle()
  },
  inheritAttrs: false,
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-checked", "aria-disabled", "tabindex"];
var _hoisted_2 = ["value", "name", "disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconHelpCircle = resolveComponent("IconHelpCircle");
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  return openBlock(), createElementBlock("label", {
    "data-tag": "tiny-radio-button",
    class: normalizeClass(["relative outline-0 inline-flex rounded-none cursor-pointer items-center [&:first-of-type_span]:border-l-0.5 sm:[&:first-of-type_span]:border-l [&:first-of-type_span]:rounded-l-sm [&:last-of-type_span]:rounded-r-sm", _ctx.state.size === "small" ? "sm:h-6 h-10 text-xs" : "sm:h-7 sm:leading-7 h-11 text-sm sm:text-xs", !_ctx.disabled && !_ctx.state.radioGroup.disabled && _ctx.state.value === _ctx.label && "sm:[&:not(:first-of-type)_span]:shadow-[calc(-1*theme(borderWidth.DEFAULT))_0_0_0_theme(colors.color.border.focus)] [&:not(:first-of-type)_span]:shadow-[calc(-1*theme(borderWidth[0.5]))_0_0_0_theme(colors.color.border.focus)]"]),
    role: "radio",
    "aria-checked": _ctx.state.value === _ctx.label,
    "aria-disabled": _ctx.state.isDisabled,
    tabindex: _ctx.state.tabIndex,
    onKeydown: _cache[5] || (_cache[5] = withKeys(withModifiers(function() {
      return _ctx.keydownHandle && _ctx.keydownHandle.apply(_ctx, arguments);
    }, ["stop", "prevent"]), ["space"]))
  }, [withDirectives(createElementVNode("input", {
    ref: "radio",
    class: "opacity-0 outline-0 absolute -z-10",
    value: _ctx.label,
    type: "radio",
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.value = $event;
    }),
    name: _ctx.name,
    onChange: _cache[1] || (_cache[1] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    disabled: _ctx.state.isDisabled,
    tabindex: "-1",
    onFocus: _cache[2] || (_cache[2] = function() {
      return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
    }),
    onBlur: _cache[3] || (_cache[3] = function() {
      return _ctx.handleBlur && _ctx.handleBlur.apply(_ctx, arguments);
    })
  }, null, 40, _hoisted_2), [[vModelRadio, _ctx.state.value]]), createElementVNode(
    "span",
    {
      "data-tag": "tiny-radio-button-content",
      class: normalizeClass(_ctx.m("relative sm:border-y border-y-0.5 sm:border-r border-r-0.5 border-solid inline-block leading-7 items-center min-w-[theme(spacing.10)] max-w-[theme(spacing.20)] sm:min-w-[theme(spacing.12)] sm:max-w-[theme(spacing.36)] text-center overflow-hidden text-ellipsis whitespace-nowrap px-3 cursor-pointer border-color-border box-border", _ctx.state.radioGroup.disabled && _ctx.state.value === _ctx.label && "text-color-bg-1 bg-color-brand-disabled border-y-0 cursor-not-allowed border-transparent shadow-none", _ctx.state.radioGroup.disabled && _ctx.state.value !== _ctx.label && "bg-color-bg-4 cursor-not-allowed border-color-border-disabled text-color-text-disabled", _ctx.disabled && !_ctx.state.radioGroup.disabled && "text-color-text-disabled bg-none cursor-not-allowed", _ctx.state.value === _ctx.label && !_ctx.disabled && !_ctx.state.radioGroup.disabled && "border-color-border-focus text-color-brand-focus", _ctx.state.value !== _ctx.label && !_ctx.disabled && !_ctx.state.radioGroup.disabled && "text-color-text-secondary hover:text-color-brand-hover active:text-color-brand", _ctx.state.size === "small" ? "sm:h-6 leading-6 px-2 h-6" : "sm:h-7 h-7", _ctx.state.showTips && "sm:pr-7")),
      onKeydown: _cache[4] || (_cache[4] = withModifiers(function() {
      }, ["stop"]))
    },
    [renderSlot(_ctx.$slots, "default"), !_ctx.slots.default ? (openBlock(), createElementBlock(
      Fragment,
      {
        key: 0
      },
      [createTextVNode(
        toDisplayString(_ctx.text || _ctx.label) + " ",
        1
        /* TEXT */
      ), _ctx.state.showTips ? (openBlock(), createBlock(_component_tiny_tooltip, {
        key: 0,
        effect: "light",
        content: _ctx.tipContent,
        placement: "top",
        class: "sm:block hidden"
      }, {
        default: withCtx(function() {
          return [createVNode(_component_IconHelpCircle, {
            class: "absolute right-3 top-1.5 bottom-1.5 text-sm cursor-pointer w-3.5 h-3.5 fill-color-border hover:fill-color-brand-focus"
          })];
        }),
        _: 1
        /* STABLE */
      }, 8, ["content"])) : createCommentVNode("v-if", true)],
      64
      /* STABLE_FRAGMENT */
    )) : createCommentVNode("v-if", true)],
    34
    /* CLASS, NEED_HYDRATION */
  )], 42, _hoisted_1);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
