// @opentiny/vue-radio-group/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/radio-group/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var radioGroupProps = _extends({}, $props, {
  modelValue: {},
  size: {
    type: String,
    default: ""
  },
  fill: String,
  textColor: String,
  disabled: Boolean,
  vertical: Boolean,
  options: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  type: {
    type: String,
    default: "radio"
  },
  showTips: {
    type: Boolean,
    default: false
  },
  displayOnly: {
    type: Boolean,
    default: false
  }
});
var RadioGroup = defineComponent({
  name: $prefix + "RadioGroup",
  componentName: "RadioGroup",
  props: radioGroupProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
RadioGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
RadioGroup.install = function(Vue) {
  Vue.component(RadioGroup.name, RadioGroup);
};
RadioGroup.version = version;
export {
  RadioGroup as default
};
