// @opentiny/vue-radio-group/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/radio-group/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Radio from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { resolveComponent, openBlock, createBlock, resolveDynamicComponent, normalizeClass, withCtx, renderSlot, createElementBlock, Fragment, renderList, mergeProps, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    Radio,
    RadioButton
  },
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["options", "modelValue", "type", "size", "fill", "textColor", "disabled", "vertical"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_radio = resolveComponent("radio");
  var _component_radio_button = resolveComponent("radio-button");
  return openBlock(), createBlock(resolveDynamicComponent(_ctx.state.tag), {
    class: normalizeClass(["tiny-radio-group", [_ctx.vertical ? "list-inline" : ""]]),
    role: "radiogroup",
    onKeydown: _ctx.handleKeydown,
    ref: "group"
  }, {
    default: withCtx(function() {
      return [renderSlot(_ctx.$slots, "default", {}, function() {
        return [_ctx.type === "radio" ? (openBlock(true), createElementBlock(
          Fragment,
          {
            key: 0
          },
          renderList(_ctx.options, function(item, index) {
            return openBlock(), createBlock(
              _component_radio,
              mergeProps({
                key: index
              }, item),
              null,
              16
              /* FULL_PROPS */
            );
          }),
          128
          /* KEYED_FRAGMENT */
        )) : createCommentVNode("v-if", true), _ctx.type === "button" ? (openBlock(true), createElementBlock(
          Fragment,
          {
            key: 1
          },
          renderList(_ctx.options, function(item, index) {
            return openBlock(), createBlock(
              _component_radio_button,
              mergeProps({
                key: index
              }, item),
              null,
              16
              /* FULL_PROPS */
            );
          }),
          128
          /* KEYED_FRAGMENT */
        )) : createCommentVNode("v-if", true)];
      })];
    }),
    _: 3
    /* FORWARDED */
  }, 40, ["class", "onKeydown"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
