// @opentiny/vue-dropdown-item/lib/mobile.js
import { renderless, api } from "@opentiny/vue-renderless/dropdown-item/vue";
import { defineComponent, directive, props, setup as _setup } from "@opentiny/vue-common";
import { iconYes } from "@opentiny/vue-icon";
import Popup from "@opentiny/vue-popup";
import Button from "@opentiny/vue-button";
import Clickoutside from "@opentiny/vue-renderless/common/deps/clickoutside";
import "@opentiny/vue-theme-mobile/dropdown-item/index.css";
import { resolveComponent, resolveDirective, withDirectives, openBlock, createElementBlock, normalizeClass, normalizeStyle, createVNode, withCtx, renderSlot, Fragment, renderList, createElementVNode, createBlock, resolveDynamicComponent, toDisplayString, createCommentVNode, createTextVNode, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  directives: directive({
    Clickoutside
  }),
  props: [].concat(props, ["modelValue", "title", "disabled", "titleClass", "options", "icon", "type"]),
  components: {
    IconYes: iconYes(),
    TinyPopup: Popup,
    TinyButton: Button
  },
  emits: ["update:modelValue", "open", "opened", "click", "change", "closed", "close", "reset", "confirm", "item-click"],
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  class: "tiny-mobile-dropdown-item__options"
};
var _hoisted_2 = ["onClick"];
var _hoisted_3 = {
  class: "tiny-mobile-dropdown-item__option-icon"
};
var _hoisted_4 = {
  class: "tiny-mobile-dropdown-item__option-title"
};
var _hoisted_5 = {
  class: "tiny-mobile-dropdown-item__option-value"
};
var _hoisted_6 = {
  key: 1,
  class: "tiny-mobile-dropdown-item__filter"
};
var _hoisted_7 = {
  class: "tiny-mobile-dropdown-item__filter-wrap"
};
var _hoisted_8 = {
  class: "tiny-mobile-dropdown-item__filter-title"
};
var _hoisted_9 = ["onClick"];
var _hoisted_10 = {
  class: "tiny-mobile-dropdown-item__filter-operate"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_yes = resolveComponent("icon-yes");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_popup = resolveComponent("tiny-popup");
  var _directive_clickoutside = resolveDirective("clickoutside");
  return withDirectives((openBlock(), createElementBlock(
    "div",
    {
      ref: "wrapper",
      class: normalizeClass(["tiny-mobile-dropdown-item--" + _ctx.state.direction, "tiny-mobile-dropdown-item"]),
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.clickWrapper && _ctx.clickWrapper.apply(_ctx, arguments);
      }),
      style: normalizeStyle(_ctx.state.itemStyle)
    },
    [createVNode(_component_tiny_popup, {
      modelValue: _ctx.state.showPopup,
      "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.showPopup = $event;
      }),
      "popup-class": "tiny-mobile-dropdown-item__content",
      overlay: _ctx.state.overlay,
      "close-on-click-overlay": _ctx.state.closeOnClickOverlay,
      closeable: false,
      position: _ctx.state.direction === "down" ? "top" : "bottom",
      duration: _ctx.state.transition ? _ctx.state.duration : 0,
      onOpen: _ctx.open,
      onClose: _ctx.close,
      onOpened: _ctx.opened,
      onClosed: _ctx.closed
    }, {
      default: withCtx(function() {
        return [renderSlot(_ctx.$slots, "default", {}, function() {
          return [_ctx.type === "selection" ? (openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.options, function(item, index) {
              return openBlock(), createElementBlock("div", {
                class: normalizeClass(["tiny-mobile-dropdown-item__cell tiny-mobile-dropdown-item__option", item.value === _ctx.modelValue ? "is-active" : ""]),
                key: index,
                tabindex: "0",
                onClick: function onClick($event) {
                  return _ctx.clickItem(item.value);
                }
              }, [createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "icon", {}, function() {
                return [(openBlock(), createBlock(resolveDynamicComponent(_ctx.icon)))];
              })]), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "title", {
                titleData: item
              }, function() {
                return [createElementVNode(
                  "span",
                  {
                    style: normalizeStyle({
                      color: item.value === _ctx.modelValue && _ctx.state.activeColor ? _ctx.state.activeColor : ""
                    })
                  },
                  toDisplayString(item.text),
                  5
                  /* TEXT, STYLE */
                )];
              })]), createElementVNode("div", _hoisted_5, [item.value === _ctx.modelValue ? (openBlock(), createBlock(_component_icon_yes, {
                key: 0,
                fill: _ctx.state.activeColor ? _ctx.state.activeColor : "#f36f64"
              }, null, 8, ["fill"])) : createCommentVNode("v-if", true)])], 10, _hoisted_2);
            }),
            128
            /* KEYED_FRAGMENT */
          ))])) : createCommentVNode("v-if", true), _ctx.type === "filter" ? withDirectives((openBlock(), createElementBlock("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.options, function(item, key) {
              return openBlock(), createElementBlock("div", {
                key,
                class: "tiny-mobile-dropdown-item__cell tiny-mobile-dropdown-item__filter-item"
              }, [createElementVNode(
                "span",
                _hoisted_8,
                toDisplayString(item.title),
                1
                /* TEXT */
              ), createElementVNode("ul", null, [(openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(_ctx.options.length === 0 ? item.data = [] : item.data, function(tag, tagkey) {
                  return openBlock(), createElementBlock("li", {
                    class: normalizeClass(["tiny-mobile-dropdown-item__filter-li", [_ctx.modelValue[key].indexOf(tag.value) > -1 ? "checked" : ""]]),
                    style: normalizeStyle(_ctx.getOptionStyle(tag, _ctx.modelValue[key])),
                    onClick: function onClick($event) {
                      return _ctx.tagClick(key, tag, $event);
                    },
                    key: tagkey
                  }, toDisplayString(tag.text), 15, _hoisted_9);
                }),
                128
                /* KEYED_FRAGMENT */
              ))])]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))]), createElementVNode("div", _hoisted_10, [createVNode(_component_tiny_button, {
            onClick: _ctx.reset
          }, {
            default: withCtx(function() {
              return [createTextVNode(" Reset ")];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"]), createVNode(_component_tiny_button, {
            type: "primary",
            onClick: _ctx.confirm
          }, {
            default: withCtx(function() {
              return [createTextVNode(" OK ")];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"])])])), [[_directive_clickoutside, _ctx.clickOutside]]) : createCommentVNode("v-if", true)];
        })];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["modelValue", "overlay", "close-on-click-overlay", "position", "duration", "onOpen", "onClose", "onOpened", "onClosed"])],
    6
    /* CLASS, STYLE */
  )), [[vShow, _ctx.state.showWrapper]]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
