// @opentiny/vue-slide-bar/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/slide-bar/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var SlideBar = defineComponent({
  name: $prefix + "SlideBar",
  props: _extends({}, $props, {
    initBlocks: Number,
    wheelBlocks: {
      type: Number,
      default: 1
    },
    modelValue: Array,
    tag: {
      type: String,
      default: "ul"
    },
    subTag: {
      type: String,
      default: "li"
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
SlideBar.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
SlideBar.install = function(Vue) {
  Vue.component(SlideBar.name, SlideBar);
};
SlideBar.version = version;
export {
  SlideBar as default
};
