// @opentiny/vue-badge/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/badge/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { openBlock, createElementBlock, normalizeClass, toDisplayString, renderSlot, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "primary-status": "inline-flex items-center justify-center leading-4 relative",
  "dot-showleft": "ml-2.5",
  "dot-showleft-child": "w-1.5 h-1.5 absolute -left-2.5 ml-0",
  "primary-status-child": "rounded-xl bg-color-error",
  "type-danger": "bg-color-error",
  "type-primary": "bg-color-brand",
  "type-success": "bg-color-success",
  "type-warning": "bg-color-alert",
  "type-default": "bg-color-error",
  "type-info": "bg-color-border",
  "type-icon": "h-1.5 self-start -ml-1.5 z-10",
  "circle": "min-w-[theme(spacing.4)] h-4 px-1 py-0 ml-1 text-color-bg-1 text-xs text-center box-border",
  "circle-icon-label": "box-border -ml-2 -mt-1.5",
  "dot-default-icon-label": "w-1.5 h-1.5 self-start z-10",
  "dot-default": "w-1.5 h-1.5 ml-1",
  "state-href": "!text-color-bg-1 !no-underline",
  "state-not-href": "inline-block align-top"
};
var _sfc_main = defineComponent({
  props: [].concat(props, ["showLeft", "isDot", "hidden", "type", "max", "value", "modelValue", "href", "target", "data"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = {
  key: 0
};
var _hoisted_2 = ["href", "target"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-badge",
      class: normalizeClass(_ctx.m(_ctx.gcls("primary-status"), _ctx.gcls({
        "dot-showleft": _ctx.isDot && _ctx.showLeft
      })))
    },
    [_ctx.data ? (openBlock(), createElementBlock(
      "span",
      _hoisted_1,
      toDisplayString(_ctx.data),
      1
      /* TEXT */
    )) : renderSlot(_ctx.$slots, "default", {
      key: 1
    }), !_ctx.hidden && (_ctx.state.content || _ctx.state.content === 0 || _ctx.isDot || _ctx.showLeft) ? (openBlock(), createElementBlock(
      "div",
      {
        key: 2,
        "data-tag": "tiny-badge-content",
        class: normalizeClass(_ctx.m(_ctx.gcls("primary-status-child"), _ctx.gcls({
          "dot-showleft-child": _ctx.showLeft && _ctx.isDot
        }), _ctx.gcls("type-" + (_ctx.type || "default")), _ctx.gcls({
          "circle": !_ctx.isDot
        }), _ctx.gcls({
          "dot-default-icon-label": !_ctx.showLeft && _ctx.isDot && ["icon", "label"].includes(_ctx.type)
        }), _ctx.gcls({
          "dot-default": !_ctx.showLeft && _ctx.isDot && !["icon", "label"].includes(_ctx.type)
        }), _ctx.gcls({
          "circle-icon-label": !_ctx.isDot && ["icon", "label"].includes(_ctx.type)
        })))
      },
      [renderSlot(_ctx.$slots, "content", {}, function() {
        return [_ctx.state.href ? (openBlock(), createElementBlock("a", {
          key: 0,
          class: normalizeClass(_ctx.m(_ctx.gcls("state-href"))),
          href: _ctx.state.href,
          target: _ctx.target,
          rel: "noopener noreferrer"
        }, toDisplayString(_ctx.state.content), 11, _hoisted_2)) : (openBlock(), createElementBlock(
          "span",
          {
            key: 1,
            class: normalizeClass(_ctx.m(_ctx.gcls("state-not-href")))
          },
          toDisplayString(_ctx.state.content),
          3
          /* TEXT, CLASS */
        ))];
      })],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
