// @opentiny/vue-menu/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless as renderless$1, api as api$1 } from "@opentiny/vue-renderless/menu/vue";
import { defineComponent, $props, setup as _setup, $prefix } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/menu-node/vue";
import { IconRightward, IconSearch } from "@opentiny/vue-icon";
import { addClass, removeClass } from "@opentiny/vue-renderless/common/deps/dom";
import { openBlock, createBlock, Transition, toHandlers, withCtx, renderSlot, resolveComponent, createElementBlock, createElementVNode, withModifiers, normalizeClass, normalizeStyle, resolveDynamicComponent, createCommentVNode, toDisplayString, createVNode, withDirectives, Fragment, renderList, vShow, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import Input from "@opentiny/vue-input";
import Exception from "@opentiny/vue-exception";
var _sfc_main$2 = {
  name: "menuTransition",
  setup: function setup() {
    return {
      on: {
        beforeEnter: function beforeEnter(elem) {
          addClass(elem, "transition-[height] duration-300 ease-in-out");
          if (!elem.dataset)
            elem.dataset = {};
          elem.style.height = "0";
        },
        enter: function enter(elem) {
          elem.dataset.oldOverflow = elem.style.overflow;
          if (elem.scrollHeight === 0) {
            elem.style.height = "";
          } else {
            elem.style.height = elem.scrollHeight + "px";
          }
          elem.style.overflow = "hidden";
        },
        afterEnter: function afterEnter(elem) {
          removeClass(elem, "transition-[height] duration-300 ease-in-out");
          elem.style.height = "";
          elem.style.overflow = elem.dataset.oldOverflow;
        },
        beforeLeave: function beforeLeave(elem) {
          if (!elem.dataset)
            elem.dataset = {};
          elem.dataset.oldOverflow = elem.style.overflow;
          elem.style.height = elem.scrollHeight + "px";
          elem.style.overflow = "hidden";
        },
        leave: function leave(elem) {
          if (elem.scrollHeight === 0)
            return;
          addClass(elem, "transition-[height] duration-300 ease-in-out");
          elem.style.transitionProperty = "height";
          elem.style.height = 0;
        },
        afterLeave: function afterLeave(elem) {
          removeClass(elem, "transition-[height] duration-300 ease-in-out");
          elem.style.height = "";
          elem.style.overflow = elem.dataset.oldOverflow;
        }
      }
    };
  }
};
function _sfc_render$2(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createBlock(
    Transition,
    toHandlers($setup.on),
    {
      default: withCtx(function() {
        return [renderSlot(_ctx.$slots, "default")];
      }),
      _: 3
      /* FORWARDED */
    },
    16
    /* FULL_PROPS */
  );
}
var MenuTransition = /* @__PURE__ */ _export_sfc(_sfc_main$2, [["render", _sfc_render$2]]);
var _sfc_main$1 = defineComponent({
  name: "MenuNode",
  emits: [],
  components: {
    MenuTransition,
    IconRightward: IconRightward()
  },
  props: _extends({}, $props, {
    activedKeys: [Number, String],
    nodes: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    currentPaths: {
      type: Array,
      default: function _default2() {
        return [];
      }
    },
    depth: {
      type: Number,
      default: 0
    },
    activedKeysPath: {
      type: Array,
      default: function _default3() {
        return [];
      }
    },
    onlyCheckChildren: {
      type: Boolean,
      default: false
    },
    expandedKeysPath: {
      type: Array,
      default: function _default4() {
        return [];
      }
    }
  }),
  setup: function setup2(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1$1 = {
  class: "w-full"
};
var _hoisted_2$1 = {
  class: "w-full hidden sm:block"
};
var _hoisted_3$1 = ["title", "href"];
var _hoisted_4$1 = {
  class: "line-clamp-1"
};
var _hoisted_5$1 = {
  class: "w-full block sm:hidden"
};
var _hoisted_6 = ["href"];
var _hoisted_7 = {
  class: "relative px-4 line-clamp-2"
};
var _hoisted_8 = {
  key: 0,
  class: "absolute left-0 top-0.5 w-0.5 bg-color-brand",
  style: {
    "height": "calc(100% - 6px)"
  }
};
var _hoisted_9 = {
  key: 0
};
var _hoisted_10 = ["href", "onClick"];
var _hoisted_11 = {
  class: "w-full relative pl-6 pr-4 line-clamp-2"
};
var _hoisted_12 = {
  key: 0,
  class: "absolute left-0 top-0.5 w-0.5 bg-color-brand",
  style: {
    "height": "calc(100% - 6px)"
  }
};
function _sfc_render$1(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_icon_rightward = resolveComponent("icon-rightward");
  var _component_menu_node = resolveComponent("menu-node", true);
  var _component_menu_transition = resolveComponent("menu-transition");
  return openBlock(), createElementBlock("div", _hoisted_1$1, [createElementVNode("div", _hoisted_2$1, [createElementVNode("div", {
    class: "w-full px-2 mb-1",
    onClick: _cache[0] || (_cache[0] = withModifiers(function($event) {
      return _ctx.handelNodeClick(_ctx.nodes);
    }, ["stop"]))
  }, [createElementVNode(
    "div",
    {
      class: normalizeClass(["relative w-full h-12 flex text-sm overflow-hidden cursor-pointer hover:bg-color-bg-2 rounded", [_ctx.nodes.isCurrent ? "text-color-brand text-sm bg-color-fill-6 rounded hover:bg-color-brand-disabled" : "", _ctx.currentPaths.includes(_ctx.nodes.id) ? "text-color-brand" : "", _ctx.nodes.disabled ? "text-color-text-disabled cursor-not-allowed" : ""]]),
      style: normalizeStyle({
        "padding-left": _ctx.depth * 24 + "px"
      })
    },
    [createElementVNode("a", {
      class: normalizeClass(["flex items-center whitespace-normal", [_ctx.nodes.disabled ? "text-color-text-disabled cursor-not-allowed" : ""]]),
      title: _ctx.nodes.label,
      href: _ctx.nodes.url || void 0,
      style: {
        "text-decoration": "none"
      }
    }, [_ctx.nodes.customIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.nodes.customIcon), {
      key: 0,
      "custom-class": "mr-3 w-4.5 h-4.5 flex-shrink-0",
      class: normalizeClass([_ctx.nodes.isCurrent ? "fill-color-brand" : _ctx.nodes.disabled ? "fill-color-icon-disabled" : "fill-color-icon-primary"])
    }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
      "span",
      _hoisted_4$1,
      toDisplayString(_ctx.nodes.label),
      1
      /* TEXT */
    )], 10, _hoisted_3$1), _ctx.nodes.children && _ctx.nodes.children.length ? (openBlock(), createBlock(_component_icon_rightward, {
      key: 0,
      class: normalizeClass(["absolute right-3 top-4 transition-transform duration-300 ease-in-out", [_ctx.nodes.isCurrent ? "fill-color-brand" : _ctx.nodes.disabled ? "fill-color-icon-disabled" : "fill-color-icon-secondary", _ctx.state.showNodes || _ctx.nodes.visible ? "rotate-90" : ""]])
    }, null, 8, ["class"])) : createCommentVNode("v-if", true)],
    6
    /* CLASS, STYLE */
  )]), createVNode(_component_menu_transition, null, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        null,
        [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.nodes.children, function(node, index) {
            return openBlock(), createBlock(_component_menu_node, {
              key: index,
              nodes: node,
              label: node.label,
              depth: _ctx.depth + 1,
              "current-paths": _ctx.currentPaths,
              "expanded-keys-path": _ctx.expandedKeysPath,
              "actived-keys-path": _ctx.activedKeysPath,
              "actived-keys": _ctx.activedKeys
            }, null, 8, ["nodes", "label", "depth", "current-paths", "expanded-keys-path", "actived-keys-path", "actived-keys"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.showNodes || _ctx.nodes.visible]])];
    }),
    _: 1
    /* STABLE */
  })]), createElementVNode("div", _hoisted_5$1, [createElementVNode("a", {
    class: normalizeClass(["w-full whitespace-normal text-sm leading-5.5 py-3 px-0 overflow-hidden line-clamp-2", [_ctx.nodes.disabled || _ctx.onlyCheckChildren && _ctx.nodes.children ? "text-color-text-disabled cursor-not-allowed" : "", _ctx.nodes.isCurrent ? "text-color-brand bg-color-bg-1" : ""]]),
    href: _ctx.nodes.url || void 0,
    style: {
      "text-decoration": "none"
    },
    onClick: _cache[1] || (_cache[1] = withModifiers(function($event) {
      return _ctx.handelNodeClick(_ctx.nodes);
    }, ["stop"]))
  }, [createElementVNode("span", _hoisted_7, [createTextVNode(
    toDisplayString(_ctx.nodes.label) + " ",
    1
    /* TEXT */
  ), _ctx.nodes.isCurrent ? (openBlock(), createElementBlock("i", _hoisted_8)) : createCommentVNode("v-if", true)])], 10, _hoisted_6), _ctx.nodes.children ? (openBlock(), createElementBlock("ul", _hoisted_9, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.nodes.children, function(node, index) {
      return openBlock(), createElementBlock("li", {
        key: index
      }, [createElementVNode("a", {
        class: normalizeClass(["block w-full whitespace-normal text-xs leading-5.5 py-3 px-0 overflow-hidden", [node.disabled ? "text-color-text-disabled cursor-not-allowed" : "", node.isCurrent ? "text-color-brand bg-color-bg-1" : ""]]),
        href: node.url || void 0,
        style: {
          "text-decoration": "none"
        },
        onClick: withModifiers(function($event) {
          return _ctx.handelNodeClick(node);
        }, ["stop"])
      }, [createElementVNode("span", _hoisted_11, [createTextVNode(
        toDisplayString(node.label) + " ",
        1
        /* TEXT */
      ), node.isCurrent ? (openBlock(), createElementBlock("i", _hoisted_12)) : createCommentVNode("v-if", true)])], 10, _hoisted_10)]);
    }),
    128
    /* KEYED_FRAGMENT */
  ))])) : createCommentVNode("v-if", true)])]);
}
var MenuNode = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var _sfc_main = defineComponent({
  name: $prefix + "Menu",
  componentName: "Menu",
  emits: ["current-change", "node-click"],
  components: {
    MenuNode,
    TinyInput: Input,
    TinyException: Exception
  },
  props: _extends({}, $props, {
    showFilter: {
      type: Boolean,
      default: false
    },
    searchIcon: {
      type: Object,
      default: function _default5() {
        return IconSearch();
      }
    },
    emptyText: {
      type: String,
      default: ""
    },
    expandedKeys: {
      type: Array,
      default: function _default6() {
        return [];
      }
    },
    activedKeys: {
      type: [Number, String],
      default: ""
    },
    onlyCheckChildren: {
      type: Boolean,
      default: false
    },
    expandAll: {
      type: Boolean,
      default: false
    },
    data: {
      type: Array,
      default: function _default7() {
        return [];
      }
    }
  }),
  setup: function setup3(props, context) {
    return _setup({
      props,
      context,
      renderless: renderless$1,
      api: api$1,
      mono: true
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-menu",
  class: "w-28 sm:w-56 h-full pt-0 sm:pt-4 bg-color-bg-2 sm:bg-color-bg-1 overflow-y-auto"
};
var _hoisted_2 = {
  key: 0,
  "data-tag": "tiny-menu-filter",
  class: "px-6 pb-3 pt-0 hidden sm:block [&_svg]:fill-color-text-placeholder"
};
var _hoisted_3 = {
  key: 1,
  "data-tag": "tiny-menu-empty",
  class: "relative w-full min-h-[theme(spacing.16)]"
};
var _hoisted_4 = {
  key: 2,
  "data-tag": "tiny-slot-empty"
};
var _hoisted_5 = {
  class: "relative w-full min-h-[theme(spacing.16)] flex justify-center items-center"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_input = resolveComponent("tiny-input");
  var _component_tiny_exception = resolveComponent("tiny-exception");
  var _component_menu_node = resolveComponent("menu-node");
  return openBlock(), createElementBlock("div", _hoisted_1, [_ctx.showFilter ? (openBlock(), createElementBlock("div", _hoisted_2, [createVNode(_component_tiny_input, {
    modelValue: _ctx.state.filterText,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.filterText = $event;
    }),
    onInput: _cache[1] || (_cache[1] = function($event) {
      return _ctx.filterInput($event);
    }),
    "suffix-icon": _ctx.searchIcon,
    placeholder: _ctx.t("ui.treeMenu.placeholder")
  }, null, 8, ["modelValue", "suffix-icon", "placeholder"])])) : createCommentVNode("v-if", true), _ctx.state.isEmpty ? (openBlock(), createElementBlock("div", _hoisted_3, [createVNode(_component_tiny_exception, {
    "component-page": "",
    type: "noresult"
  })])) : createCommentVNode("v-if", true), !_ctx.data.length ? (openBlock(), createElementBlock("div", _hoisted_4, [renderSlot(_ctx.$slots, "empty", {}, function() {
    return [createElementVNode("div", _hoisted_5, [createElementVNode(
      "span",
      null,
      toDisplayString(_ctx.emptyText || _ctx.t("ui.tree.emptyText")),
      1
      /* TEXT */
    )])];
  })])) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.filterValue ? _ctx.state.filterData : _ctx.state.data, function(item, index) {
      return withDirectives((openBlock(), createBlock(_component_menu_node, {
        key: item.id || index,
        nodes: item,
        depth: 1,
        "current-paths": _ctx.state.currentPaths,
        "expanded-keys-path": _ctx.state.expandedKeysPath,
        "actived-keys-path": _ctx.state.activedKeysPath,
        "actived-keys": _ctx.state.activedKeys,
        "only-check-children": _ctx.onlyCheckChildren
      }, null, 8, ["nodes", "current-paths", "expanded-keys-path", "actived-keys-path", "actived-keys", "only-check-children"])), [[vShow, !item.hidden]]);
    }),
    128
    /* KEYED_FRAGMENT */
  ))]);
}
var Menu = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
Menu.install = function(Vue) {
  Vue.component(Menu.name, Menu);
};
Menu.version = version;
export {
  Menu as default
};
