// @opentiny/vue-color-select-panel/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {};
var ColorSelectPanel = defineComponent({
  name: $prefix + "ColorSelectPanel",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: String,
    visible: Boolean,
    alpha: Boolean,
    history: Array,
    predefine: Array
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
ColorSelectPanel.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ColorSelectPanel.install = function(Vue) {
  Vue.component(ColorSelectPanel.name, ColorSelectPanel);
};
ColorSelectPanel.version = version;
export {
  ColorSelectPanel as default
};
