// @opentiny/vue-renderless/tree/vue.js
import "../chunk-G2ADBYYC.js";
import debounce from "../common/deps/debounce";
import {
  computedShowEmptyText,
  closeMenu,
  setChildren,
  getChildren,
  computedTreeItemArray,
  computedIsEmpty,
  watchDefaultCheckedKeys,
  watchDefaultExpandedKeys,
  watchData,
  watchCheckboxItems,
  watchCheckStrictly,
  created,
  mounted,
  updated,
  filter,
  getNode,
  remove,
  append,
  getNodeKey,
  getNodePath,
  getHalfCheckedNodes,
  getHalfCheckedKeys,
  getCheckedNodes,
  getCheckedKeys,
  getCurrentNode,
  getCurrentKey,
  setCheckedNodes,
  setCheckedKeys,
  setChecked,
  setCurrentNode,
  setCurrentKey,
  insertBefore,
  insertAfter,
  handleNodeExpand,
  updateKeyChildren,
  initTabIndex,
  handleKeydown,
  setCurrentRadio,
  expandAllNodes,
  dragStart,
  dragOver,
  dragEnd,
  clearCurrentStore,
  initIsCurrent,
  beforeUnmount,
  wrapMounted,
  switchToggle,
  initTreeStore,
  getSameDataIndex,
  loopGetTreeData,
  cancelDelete,
  deleteAction,
  deleteConfirm,
  openEdit,
  saveNode,
  addNode,
  editNode,
  closeEdit,
  saveEdit,
  setAddDisabledKeys,
  setEditDisabledKeys,
  setDeleteDisabledKeys,
  initPlainNodeStore,
  handleCheckPlainNode,
  handleClickPlainNode,
  setCheckedByNodeKey
} from "./index";
const api = [
  "state",
  "closeMenu",
  "filter",
  "getNodeKey",
  "getNodePath",
  "getCheckedNodes",
  "getCheckedKeys",
  "getCurrentNode",
  "getCurrentKey",
  "setCheckedNodes",
  "setCheckedKeys",
  "setChecked",
  "getHalfCheckedNodes",
  "getHalfCheckedKeys",
  "setCurrentNode",
  "setCurrentKey",
  "getNode",
  "remove",
  "append",
  "insertBefore",
  "insertAfter",
  "handleNodeExpand",
  "updateKeyChildren",
  "initTabIndex",
  "handleKeydown",
  "setCurrentRadio",
  "expandAllNodes",
  "clearCurrentStore",
  "switchToggle",
  "initTreeStore",
  "getSameDataIndex",
  "loopGetTreeData",
  "cancelDelete",
  "deleteAction",
  "deleteConfirm",
  "openEdit",
  "saveNode",
  "addNode",
  "editNode",
  "closeEdit",
  "saveEdit",
  "setAddDisabledKeys",
  "setEditDisabledKeys",
  "setDeleteDisabledKeys",
  "handleCheckPlainNode",
  "handleClickPlainNode",
  "setCheckedByNodeKey"
];
const initState = ({ reactive, emitter, props, computed, api: api2 }) => {
  const state = reactive({
    loaded: false,
    checkEasily: false,
    root: null,
    store: null,
    treeItems: null,
    currentNode: null,
    checkboxItems: [],
    isEmpty: computed(() => api2.computedIsEmpty(props, state)),
    emitter: emitter(),
    expandIcon: props.expandIcon,
    shrinkIcon: props.shrinkIcon,
    expandIconColor: props.expandIconColor,
    shrinkIconColor: props.shrinkIconColor,
    dragState: { showDropIndicator: false, draggingNode: null, dropNode: null, allowDrop: true },
    currentRadio: { value: null },
    children: computed({
      get: () => api2.getChildren(props, state),
      set: (data) => api2.setChildren(data)
    }),
    treeItemArray: computed(() => api2.computedTreeItemArray(props, state)),
    showEmptyText: computed(() => api2.computedShowEmptyText(props, state)),
    currentStore: { flag: false, node: null },
    action: {
      deleteDisabled: [],
      editDisabled: [],
      addDisabled: [],
      show: false,
      node: null,
      data: [],
      label: "",
      type: "",
      isLeaf: false,
      isSaveChildNode: false,
      referenceElm: null,
      popoverVisible: false
    },
    treeEdit: {
      addData: [],
      deleteData: [],
      editData: []
    },
    plainNodeStore: {}
  });
  return state;
};
const initApi = ({ state, dispatch, broadcast, props, vm, constants, t, emit, refs }) => ({
  state,
  dispatch,
  broadcast,
  setChildren: setChildren(props),
  getChildren: getChildren(),
  computedTreeItemArray: computedTreeItemArray(),
  computedIsEmpty: computedIsEmpty(),
  watchDefaultCheckedKeys: watchDefaultCheckedKeys(state),
  watchData: watchData({ state }),
  watchCheckboxItems: watchCheckboxItems(),
  watchCheckStrictly: watchCheckStrictly(state),
  updated: updated({ vm, state }),
  filter: filter({ props, state }),
  getNodeKey: getNodeKey(props),
  getNodePath: getNodePath({ props, state }),
  getCheckedNodes: getCheckedNodes(state),
  getCheckedKeys: getCheckedKeys(state),
  getCurrentNode: getCurrentNode(state),
  setCheckedNodes: setCheckedNodes({ props, state }),
  setCheckedKeys: setCheckedKeys({ props, state }),
  setChecked: setChecked(state),
  getHalfCheckedNodes: getHalfCheckedNodes(state),
  getHalfCheckedKeys: getHalfCheckedKeys(state),
  setCurrentNode: setCurrentNode({ props, state }),
  setCurrentKey: setCurrentKey({ props, state }),
  getNode: getNode(state),
  remove: remove(state),
  append: append(state),
  insertBefore: insertBefore(state),
  insertAfter: insertAfter(state),
  updateKeyChildren: updateKeyChildren({ props, state }),
  initTabIndex: initTabIndex({ vm, state }),
  handleKeydown: handleKeydown({ vm, state }),
  computedShowEmptyText: computedShowEmptyText({ constants, t }),
  setCurrentRadio: setCurrentRadio({ props, state }),
  expandAllNodes: expandAllNodes({ state }),
  dragStart: dragStart({ props, state, emit }),
  dragOver: dragOver({ props, state, emit, vm }),
  dragEnd: dragEnd({ state, emit }),
  clearCurrentStore: clearCurrentStore(state),
  initIsCurrent: debounce(20, initIsCurrent({ props, state })),
  setCheckedByNodeKey: setCheckedByNodeKey({ props, state })
});
const initWatcher = ({ watch, props, api: api2, state }) => {
  watch(() => props.defaultCheckedKeys, api2.watchDefaultCheckedKeys);
  watch(() => props.defaultExpandedKeys, api2.watchDefaultExpandedKeys);
  watch(() => props.defaultExpandedKeysHighlight, api2.initIsCurrent);
  watch(() => props.data, api2.watchData, { deep: true });
  watch(() => props.checkboxItems, api2.watchCheckboxItems);
  watch(() => props.checkStrictly, api2.watchCheckStrictly);
  watch(() => props.defaultCheckedKeys, api2.setCurrentRadio, { immediate: true });
  watch(
    () => props.deleteDisabledKeys,
    (value) => state.action.deleteDisabled = value || [],
    { immediate: true }
  );
  watch(
    () => props.editDisabledKeys,
    (value) => state.action.editDisabled = value || [],
    { immediate: true }
  );
  watch(
    () => props.addDisabledKeys,
    (value) => state.action.addDisabled = value || [],
    { immediate: true }
  );
  watch(() => state.root, api2.initPlainNodeStore, { deep: true });
};
const renderless = (props, { computed, onMounted, onUpdated, reactive, watch, provide, onBeforeUnmount }, { vm, t, emit, constants, broadcast, dispatch, service, emitter, nextTick }) => {
  const api2 = {};
  const state = initState({ reactive, emitter, props, computed, api: api2 });
  provide("parentEmitter", state.emitter);
  Object.assign(api2, initApi({ state, dispatch, broadcast, props, vm, constants, t, emit }), {
    closeMenu: closeMenu(state),
    mounted: mounted({ api: api2, vm }),
    created: created({ api: api2, props, state }),
    getCurrentKey: getCurrentKey({ api: api2, props }),
    handleNodeExpand: handleNodeExpand(emit),
    beforeUnmount: beforeUnmount({ api: api2, vm, state }),
    wrapMounted: wrapMounted({ api: api2, props, service }),
    initTreeStore: initTreeStore({ api: api2, props, state, emit }),
    deleteAction: deleteAction({ state, api: api2, emit }),
    deleteConfirm: deleteConfirm({ state }),
    getSameDataIndex,
    loopGetTreeData,
    cancelDelete: cancelDelete({ state }),
    openEdit: openEdit({ props, state, api: api2, emit }),
    saveNode: saveNode({ state, emit, api: api2 }),
    addNode: addNode({ api: api2 }),
    editNode: editNode({ state }),
    closeEdit: closeEdit({ props, state, api: api2, emit }),
    saveEdit: saveEdit({ props, state, emit }),
    setAddDisabledKeys: setAddDisabledKeys({ state }),
    setEditDisabledKeys: setEditDisabledKeys({ state }),
    setDeleteDisabledKeys: setDeleteDisabledKeys({ state }),
    watchDefaultExpandedKeys: watchDefaultExpandedKeys({ state, api: api2, nextTick }),
    switchToggle: switchToggle({ state }),
    initPlainNodeStore: initPlainNodeStore({ props, state }),
    handleCheckPlainNode: handleCheckPlainNode({ props, emit }),
    handleClickPlainNode: handleClickPlainNode(emit)
  });
  api2.created();
  initWatcher({ watch, props, api: api2, state });
  onMounted(api2.wrapMounted);
  onUpdated(api2.updated);
  onBeforeUnmount(api2.beforeUnmount);
  return api2;
};
export {
  api,
  renderless
};
