// @opentiny/vue-drop-times/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/drop-times/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var DropTimes = defineComponent({
  name: $prefix + "DropTimes",
  props: _extends({}, $props, {
    modelValue: {},
    placeholder: {
      type: String,
      default: ""
    },
    tabindex: {
      type: String,
      default: "1"
    },
    size: String,
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    start: {
      type: Number,
      default: 0
    },
    end: {
      type: Number,
      default: 24 * 60
    },
    step: {
      type: Number,
      default: 15
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
DropTimes.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
DropTimes.install = function(Vue) {
  Vue.component(DropTimes.name, DropTimes);
};
DropTimes.version = version;
export {
  DropTimes as default
};
