// @opentiny/vue-breadcrumb/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/breadcrumb/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { resolveComponent, openBlock, createElementBlock, renderSlot, Fragment, renderList, createBlock } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    TinyBreadcrumbItem: BreadcrumbItem
  },
  props: [].concat(props, ["separator", "separatorIcon", "textField", "options"]),
  provide: function provide() {
    return {
      breadcrumb: this
    };
  },
  emits: ["select"],
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-breadcrumb",
  "aria-label": "Breadcrumb",
  role: "navigation"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_breadcrumb_item = resolveComponent("tiny-breadcrumb-item");
  return openBlock(), createElementBlock("div", _hoisted_1, [renderSlot(_ctx.$slots, "default", {}, function() {
    return [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.options, function(item, index) {
        return openBlock(), createBlock(_component_tiny_breadcrumb_item, {
          option: item,
          label: item[_ctx.textField],
          key: index,
          to: item.to,
          replace: item.replace
        }, null, 8, ["option", "label", "to", "replace"]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))];
  })]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
