// @opentiny/vue-credit-card-form/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/credit-card-form/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var CreditCardForm = defineComponent({
  name: $prefix + "CreditCardForm",
  props: _extends({}, $props, {
    backgroundImage: [String, Object],
    formData: {
      type: Object,
      default: function _default() {
        return {
          cardYear: "",
          cardNumber: "",
          cardName: "",
          cardMonth: "",
          cardCvv: ""
        };
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
CreditCardForm.install = function(Vue) {
  Vue.component(CreditCardForm.name, CreditCardForm);
};
CreditCardForm.version = version;
export {
  CreditCardForm as default
};
