// @opentiny/vue-cascader/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/cascader/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  placeholder: "ui.cascader.placeholder",
  COMPONENT_NAME: {
    FormItem: "FormItem"
  },
  EVENT_NAME: {
    FormBlur: "form.blur"
  }
};
var Cascader = defineComponent({
  name: $prefix + "Cascader",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    autoSize: Boolean,
    beforeFilter: {
      type: Function,
      default: function _default2() {
        return function() {
        };
      }
    },
    clearable: Boolean,
    collapseTags: Boolean,
    debounce: {
      type: Number,
      default: 300
    },
    disabled: Boolean,
    filterMethod: Function,
    filterable: Boolean,
    modelValue: {},
    options: Array,
    placeholder: {
      type: String,
      default: ""
    },
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    popperClass: String,
    props: Object,
    separator: {
      type: String,
      default: "/"
    },
    showAllLevels: {
      type: Boolean,
      default: true
    },
    size: String,
    shape: String,
    label: String,
    tip: String,
    displayOnly: {
      type: Boolean,
      default: false
    },
    hoverExpand: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.1";
Cascader.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Cascader.install = function(Vue) {
  Vue.component(Cascader.name, Cascader);
};
Cascader.version = version;
export {
  Cascader as default
};
