// @opentiny/vue-cascader-mobile/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/cascader-mobile/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { IconCheck, IconYes, IconLoading, IconCheckedSur, IconHalfselect, IconSearch, IconClose } from "@opentiny/vue-icon";
import Button from "@opentiny/vue-button";
import ActionSheet from "@opentiny/vue-action-sheet";
import Exception from "@opentiny/vue-exception";
import { resolveComponent, openBlock, createElementBlock, createBlock, resolveDynamicComponent, normalizeClass, createCommentVNode, createElementVNode, renderSlot, toDisplayString, createVNode, withDirectives, withCtx, Fragment, renderList, vShow, createTextVNode, vModelText } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import { t } from "@opentiny/vue-locale";
var _sfc_main$1 = {
  emits: ["click"],
  components: {
    IconCheck: IconCheck(),
    IconYes: IconYes(),
    IconLoading: IconLoading(),
    IconCheckedSur: IconCheckedSur(),
    IconHalfselect: IconHalfselect()
  },
  props: ["selected", "ellipsis", "multiple", "option", "textField", "textField2", "textField3", "loading"],
  computed: {
    data: function data() {
      return this.option.data || this.option;
    }
  }
};
var _hoisted_1$1 = {
  class: "flex"
};
var _hoisted_2$1 = {
  key: 0,
  class: "flex flex-none w-5 items-center justify-center ml-2"
};
var _hoisted_3$1 = {
  key: 1,
  class: "flex flex-none w-5 items-center justify-center ml-2"
};
var _hoisted_4$1 = /* @__PURE__ */ createElementVNode(
  "svg",
  {
    class: "animate-[spin_1.5s_linear_infinite] fill-current",
    viewBox: "25 25 50 50"
  },
  [/* @__PURE__ */ createElementVNode("circle", {
    class: "stroke-color-brand tiny-loading-path",
    cx: "50",
    cy: "50",
    r: "24",
    fill: "none"
  })],
  -1
  /* HOISTED */
);
var _hoisted_5$1 = [_hoisted_4$1];
function _sfc_render$1(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconYes = resolveComponent("IconYes");
  return openBlock(), createElementBlock("div", {
    class: "flex items-start leading-6 py-3 cursor-pointer select-none",
    onClick: _cache[0] || (_cache[0] = function($event) {
      return _ctx.$emit("click", $event);
    })
  }, [$props2.multiple ? (openBlock(), createBlock(resolveDynamicComponent($props2.selected ? "icon-checked-sur" : "icon-check"), {
    key: 0,
    class: normalizeClass(["flex-none mt-1 mr-2", $props2.selected ? "fill-color-brand" : "fill-color-icon-disabled"]),
    "custom-class": "w-4.5 h-4.5"
  }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      class: normalizeClass(["flex-auto flex w-full text-sm", {
        "truncate": $props2.ellipsis
      }, {
        "text-color-brand": !$props2.multiple && $props2.selected
      }])
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createElementVNode(
        "div",
        {
          class: normalizeClass(["flex-auto", {
            "truncate": $props2.ellipsis
          }])
        },
        [createElementVNode("div", _hoisted_1$1, [createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "min-w-[theme(spacing.14)]": !$props2.textField3
            }, {
              "truncate": $props2.ellipsis
            }])
          },
          toDisplayString($options.data[$props2.textField]),
          3
          /* TEXT, CLASS */
        ), $props2.textField2 ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass(["ml-5 text-color-text-placeholder", {
              "truncate": $props2.ellipsis
            }])
          },
          toDisplayString($options.data[$props2.textField2]),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true)]), $props2.textField3 ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass(["text-color-text-placeholder", {
              "truncate": $props2.ellipsis
            }])
          },
          toDisplayString($options.data[$props2.textField3]),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      ), !$props2.loading && !$props2.multiple && $props2.selected ? (openBlock(), createElementBlock("div", _hoisted_2$1, [createVNode(_component_IconYes, {
        "custom-class": "w-5 h-5",
        class: "fill-color-brand"
      })])) : createCommentVNode("v-if", true), $props2.option.loading ? (openBlock(), createElementBlock("div", _hoisted_3$1, _hoisted_5$1)) : createCommentVNode("v-if", true)];
    })],
    2
    /* CLASS */
  )]);
}
var Option = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var $constants = {
  defaultNodeConfig: {
    lazy: false,
    load: null,
    isLeaf: "leaf",
    afterLoad: null,
    currentNodeKey: null,
    checkStrictly: false,
    checkDescendants: null,
    defaultCheckedKeys: null,
    defaultExpandedKeys: null,
    autoExpandParent: null,
    defaultExpandAll: null,
    filterNodeMethod: null
  }
};
var _sfc_main = defineComponent({
  name: $prefix + "CascaderMobile",
  emits: ["click", "search-click", "update:visible", "update:text"],
  components: {
    TinyActionSheet: ActionSheet,
    TinyOption: Option,
    TinyButton: Button,
    TinyException: Exception,
    IconCheck: IconCheck(),
    IconCheckedSur: IconCheckedSur(),
    IconHalfselect: IconHalfselect(),
    IconSearch: IconSearch(),
    IconClose: IconClose()
  },
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    data: {
      type: Array,
      default: function _default2() {
        return [];
      }
    },
    modelValue: [Number, String, Array],
    visible: {
      type: Boolean,
      default: false
    },
    ellipsis: {
      type: Boolean,
      default: false
    },
    valueField: {
      type: String,
      default: "id"
    },
    placeholder: {
      type: [String, Array],
      default: t("ui.select.placeholder")
    },
    childrenField: {
      type: String,
      default: "children"
    },
    textField: {
      type: String,
      default: "label"
    },
    textField2: {
      type: String,
      default: ""
    },
    textField3: {
      type: String,
      default: ""
    },
    title: String,
    textSplit: {
      type: String,
      default: " / "
    },
    showHeader: {
      type: Boolean,
      default: true
    },
    showFooter: {
      type: Boolean,
      default: false
    },
    multiple: Boolean,
    searchConfig: {
      type: Object,
      default: function _default3() {
        return {
          options: [],
          searchMethod: null,
          openSearchSlot: false
        };
      }
    },
    nodeConfig: {
      type: Object,
      default: function _default4() {
        return $constants.defaultNodeConfig;
      }
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-cascader-mobile"
};
var _hoisted_2 = {
  class: "h-11 flex-none flex items-center border-b-0.5 border-color-border-separator overflow-auto whitespace-nowrap scrollbar-size-0"
};
var _hoisted_3 = {
  class: "px-4 flex"
};
var _hoisted_4 = {
  key: 0,
  class: "px-2 text-color-text-placeholder"
};
var _hoisted_5 = ["onClick"];
var _hoisted_6 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "flex-auto"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_7 = {
  "data-tag": "tabs",
  class: "flex-auto"
};
var _hoisted_8 = {
  "data-tag": "search-header",
  class: "flex leading-6 pt-4 pb-4 px-4 text-base items-center"
};
var _hoisted_9 = {
  class: "flex-auto flex items-center h-8 py-1 px-3 bg-color-bg-4 rounded"
};
var _hoisted_10 = ["placeholder"];
var _hoisted_11 = {
  class: "flex items-center pl-3 cursor-pointer"
};
var _hoisted_12 = {
  "data-tag": "search-body",
  class: "flex-auto overflow-auto flex flex-col"
};
var _hoisted_13 = {
  class: /* @__PURE__ */ normalizeClass(["flex flex-col px-4"])
};
var _hoisted_14 = {
  key: 0,
  class: "text-color-brand"
};
var _hoisted_15 = {
  class: "w-full flex justify-center items-center text-center flex-auto"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_option = resolveComponent("tiny-option");
  var _component_IconSearch = resolveComponent("IconSearch");
  var _component_tiny_exception = resolveComponent("tiny-exception");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  return withDirectives((openBlock(), createElementBlock(
    "div",
    _hoisted_1,
    [createVNode(_component_tiny_action_sheet, {
      tiny_mode: "mobile-first",
      title: _ctx.title,
      "show-footer": "",
      visible: _ctx.state.toggle,
      "onUpdate:visible": _cache[4] || (_cache[4] = function($event) {
        return _ctx.$emit("update:visible", $event);
      }),
      onClose: _ctx.close,
      "show-header": !_ctx.state.search.show,
      "custom-class": ["min-h-[95%] pb-4"]
    }, {
      "body-top": withCtx(function() {
        return [withDirectives(createElementVNode(
          "div",
          _hoisted_2,
          [createElementVNode("div", _hoisted_3, [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.computedNavList, function(option, index) {
              return openBlock(), createElementBlock("div", {
                key: index,
                class: "flex-none flex"
              }, [index !== 0 ? (openBlock(), createElementBlock("span", _hoisted_4, "/")) : createCommentVNode("v-if", true), createElementVNode("span", {
                class: normalizeClass(["flex-none max-w-[theme(spacing.24)] truncate", {
                  "text-color-brand": index === _ctx.state.level
                }, _ctx.disabledNavOption(index, option) ? "text-color-text-disabled" : "cursor-pointer"]),
                onClick: function onClick($event) {
                  return _ctx.setLevel(index, option);
                }
              }, toDisplayString(option.data[_ctx.textField]), 11, _hoisted_5)]);
            }),
            128
            /* KEYED_FRAGMENT */
          )), _hoisted_6])],
          512
          /* NEED_PATCH */
        ), [[vShow, !_ctx.state.search.show]])];
      }),
      "header-left": withCtx(function() {
        return [createVNode(_component_IconSearch, {
          "custom-class": "h-5 w-5",
          class: "cursor-pointer",
          onClick: _cache[3] || (_cache[3] = function($event) {
            return _ctx.searchBoxToggle(true);
          })
        })];
      }),
      footer: withCtx(function() {
        return [withDirectives(createVNode(_component_tiny_button, {
          tiny_mode: "mobile-first",
          class: "flex-1",
          type: "primary",
          "reset-time": 0,
          onClick: _ctx.confirm
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.button.confirm")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onClick"]), [[vShow, _ctx.state.showButton]])];
      }),
      default: withCtx(function() {
        return [withDirectives(createElementVNode(
          "div",
          _hoisted_7,
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.options, function(currentList, index) {
              return withDirectives((openBlock(), createElementBlock("div", {
                "data-tag": "tab-item",
                key: index,
                class: "flex flex-col px-4 overflow-auto pt-2"
              }, [(openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(currentList, function(item, index2) {
                  return openBlock(), createBlock(_component_tiny_option, {
                    key: item.data[_ctx.valueField],
                    multiple: _ctx.multiple,
                    ellipsis: _ctx.ellipsis,
                    "text-field": _ctx.textField,
                    "text-field2": _ctx.textField2,
                    "text-field3": _ctx.textField3,
                    loading: _ctx.state.loading,
                    option: item,
                    selected: _ctx.isSelected(item),
                    onClick: function onClick($event) {
                      return _ctx.selectOption(item, $event);
                    }
                  }, {
                    default: withCtx(function() {
                      return [renderSlot(_ctx.$slots, "default", {
                        item,
                        index: index2
                      })];
                    }),
                    _: 2
                    /* DYNAMIC */
                  }, 1032, ["multiple", "ellipsis", "text-field", "text-field2", "text-field3", "loading", "option", "selected", "onClick"]);
                }),
                128
                /* KEYED_FRAGMENT */
              ))])), [[vShow, index === _ctx.state.level]]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          512
          /* NEED_PATCH */
        ), [[vShow, !_ctx.state.search.show]]), createElementVNode(
          "div",
          {
            "data-tag": "search-box",
            class: normalizeClass([_ctx.state.search.show ? "flex flex-col flex-auto" : "hidden"])
          },
          [createElementVNode("div", _hoisted_8, [createElementVNode("div", _hoisted_9, [createVNode(_component_IconSearch, {
            "custom-class": "h-4 w-4",
            class: "mr-1 fill-color-icon-disabled"
          }), withDirectives(createElementVNode("input", {
            "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
              return _ctx.state.search.input = $event;
            }),
            class: "h-5 flex-auto text-xs bg-transparent outline-0",
            placeholder: _ctx.t("ui.select.pleaseSearch"),
            onInput: _cache[1] || (_cache[1] = function() {
              return _ctx.searchMethod && _ctx.searchMethod.apply(_ctx, arguments);
            })
          }, null, 40, _hoisted_10), [[vModelText, _ctx.state.search.input]])]), createElementVNode("div", _hoisted_11, [createElementVNode(
            "div",
            {
              onClick: _cache[2] || (_cache[2] = function($event) {
                return _ctx.searchBoxToggle(false);
              })
            },
            toDisplayString(_ctx.t("ui.base.cancel")),
            1
            /* TEXT */
          )])]), createElementVNode("div", _hoisted_12, [withDirectives(createElementVNode(
            "div",
            _hoisted_13,
            [(openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(_ctx.state.search.filterOptions, function(item, index) {
                return openBlock(), createBlock(_component_tiny_option, {
                  key: index,
                  multiple: _ctx.multiple,
                  ellipsis: _ctx.ellipsis,
                  "text-field": _ctx.textField,
                  "text-field2": _ctx.textField2,
                  "text-field3": _ctx.textField3,
                  option: item,
                  selected: _ctx.isSelected(item),
                  onClick: function onClick($event) {
                    return _ctx.searchSelectHandler(item, index);
                  }
                }, {
                  default: withCtx(function() {
                    return [_ctx.searchConfig.openSearchSlot ? renderSlot(_ctx.$slots, "search-item", {
                      key: 0,
                      item,
                      index
                    }) : renderSlot(_ctx.$slots, "default", {
                      key: 1,
                      item,
                      index
                    }, function() {
                      return [createElementVNode(
                        "div",
                        {
                          class: normalizeClass(["flex-auto", {
                            "truncate": _ctx.ellipsis
                          }])
                        },
                        [(openBlock(true), createElementBlock(
                          Fragment,
                          null,
                          renderList(_ctx.renderSearchOption(item[_ctx.textField]), function(text, index2) {
                            return openBlock(), createElementBlock("span", {
                              key: index2
                            }, [index2 ? (openBlock(), createElementBlock(
                              "span",
                              _hoisted_14,
                              toDisplayString(_ctx.state.search.input),
                              1
                              /* TEXT */
                            )) : createCommentVNode("v-if", true), createTextVNode(
                              toDisplayString(text),
                              1
                              /* TEXT */
                            )]);
                          }),
                          128
                          /* KEYED_FRAGMENT */
                        ))],
                        2
                        /* CLASS */
                      )];
                    })];
                  }),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["multiple", "ellipsis", "text-field", "text-field2", "text-field3", "option", "selected", "onClick"]);
              }),
              128
              /* KEYED_FRAGMENT */
            ))],
            512
            /* NEED_PATCH */
          ), [[vShow, _ctx.state.search.filterOptions.length]]), withDirectives(createElementVNode(
            "div",
            _hoisted_15,
            [createVNode(_component_tiny_exception, {
              "component-page": "",
              type: "noresult"
            })],
            512
            /* NEED_PATCH */
          ), [[vShow, _ctx.state.search.input && !_ctx.state.search.filterOptions.length]])])],
          2
          /* CLASS */
        )];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["title", "visible", "onClose", "show-header"])],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.visible]]);
}
var CascaderMobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
CascaderMobile.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
CascaderMobile.install = function(Vue) {
  Vue.component(CascaderMobile.name, CascaderMobile);
};
CascaderMobile.version = version;
export {
  CascaderMobile as default
};
