// @opentiny/vue-panel/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Panel = defineComponent({
  name: $prefix + "Panel",
  props: _extends({}, $props, {
    expand: {
      type: Boolean,
      default: true
    },
    foot: {
      type: String
    },
    head: {
      type: String
    },
    iconOpen: {
      type: String,
      default: "icon-chevron-down"
    },
    iconClose: {
      type: String,
      default: "icon-chevron-right"
    },
    isToggle: {
      type: Boolean,
      default: false
    },
    transition: {
      type: String,
      default: "panel"
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Panel.install = function(Vue) {
  Vue.component(Panel.name, Panel);
};
Panel.version = version;
export {
  Panel as default
};
