// @opentiny/vue-renderless/autocomplete/vue.js
import "../chunk-G2ADBYYC.js";
import debounce from "../common/deps/debounce";
import userPopper from "../common/deps/vue-popper";
import { guid } from "../common/string";
import {
  computedVisible,
  watchVisible,
  mounted,
  getData,
  handleChange,
  handleFocus,
  handleBlur,
  handleClear,
  close,
  handleKeyEnter,
  select,
  highlight
} from "./index";
const api = [
  "state",
  "select",
  "suggestionState",
  "getInput",
  "handleChange",
  "close",
  "handleFocus",
  "handleBlur",
  "handleClear",
  "handleKeyEnter",
  "highlight",
  "select",
  "doDestroy"
];
const initState = ({ reactive, $prefix, computed }) => {
  const state = reactive({
    activated: false,
    suggestions: [],
    loading: false,
    highlightedIndex: -1,
    suggestionDisabled: false,
    id: $prefix + "-" + guid(),
    suggestionVisible: computed(() => computedVisible(state))
  });
  return state;
};
const initSuggestionState = ({ reactive, parent, showPopper, popperElm, referenceElm }) => reactive({
  parent,
  dropdownWidth: "",
  showPopper,
  popperElm,
  referenceElm
});
const initApi = ({ api: api2, state, doDestroy, suggestionState, emit, refs, props, updatePopper, nextTick, constants }) => {
  api2.getData = getData({ props, state, updatePopper, nextTick });
  Object.assign(api2, {
    state,
    doDestroy,
    suggestionState,
    close: close(state),
    handleBlur: handleBlur({ emit, state }),
    mounted: mounted({ refs, state, suggestionState }),
    highlight: highlight({ constants, refs, state }),
    handleClear: handleClear({ emit, state }),
    select: select({ emit, nextTick, props, state }),
    watchVisible: watchVisible({ suggestionState, refs }),
    handleChange: handleChange({ api: api2, emit, state, props }),
    handleFocus: handleFocus({ api: api2, emit, props, state }),
    handleKeyEnter: handleKeyEnter({ api: api2, emit, nextTick, props, state }),
    debouncedGetData: debounce(props.debounce, api2.getData)
  });
};
const renderless = (props, { computed, onBeforeUnmount, onMounted, reactive, watch, toRefs, onDeactivated }, { $prefix, refs, parent, emit, constants, nextTick, slots }) => {
  const api2 = {};
  const state = initState({ reactive, $prefix, computed });
  const { showPopper, popperElm, referenceElm, doDestroy, updatePopper } = userPopper({
    reactive,
    watch,
    refs,
    emit,
    slots,
    nextTick,
    props,
    onBeforeUnmount,
    toRefs,
    onDeactivated
  });
  const suggestionState = initSuggestionState({ reactive, parent, showPopper, popperElm, referenceElm });
  initApi({ api: api2, state, doDestroy, suggestionState, emit, refs, props, updatePopper, nextTick, constants });
  watch(() => state.suggestionVisible, api2.watchVisible);
  onMounted(api2.mounted);
  return api2;
};
export {
  api,
  renderless
};
