// @opentiny/vue-label/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
import "@opentiny/vue-theme-mobile/label/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileTemplate;
};
var labelProps = _extends({}, $props, {
  label: {
    type: String,
    default: ""
  },
  color: {
    type: String,
    default: "primary"
  },
  size: {
    type: String,
    default: "normal"
  },
  type: {
    type: String,
    default: "base"
  },
  wholeline: {
    type: Boolean,
    default: false
  },
  position: {
    type: String,
    default: "left"
  },
  ellipsis: {
    type: Number,
    default: 0
  },
  decimal: {
    type: Number,
    default: 2
  },
  limit: {
    type: Number,
    default: 0
  },
  isRequired: {
    type: Boolean,
    default: false
  },
  bold: {
    type: Boolean,
    default: false
  }
});
var Label = defineComponent({
  name: $prefix + "Label",
  props: labelProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Label.install = function(Vue) {
  Vue.component(Label.name, Label);
};
Label.version = version;
export {
  Label as default
};
