// @opentiny/vue-hrapprover/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/hrapprover/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Hrapprover = defineComponent({
  name: $prefix + "Hrapprover",
  props: _extends({}, $props, {
    modelValue: String,
    approvalPerson: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: false
    },
    category: {
      type: String,
      default: "0303"
    },
    title: String,
    fetchHrapprover: Function,
    fetchDeptList: Function,
    fetchDeptByValue: Function,
    fetchDept: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Hrapprover.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Hrapprover.install = function(Vue) {
  Vue.component(Hrapprover.name, Hrapprover);
};
Hrapprover.version = version;
export {
  Hrapprover as default
};
