// @opentiny/vue-exception/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/exception/vue";
import { defineComponent, $props, setup as _setup } from "@opentiny/vue-common";
import Button from "@opentiny/vue-button";
import noperm from "@opentiny/vue-theme/images/empty-noaccess.png";
import nodata from "@opentiny/vue-theme/images/empty-nodata.png";
import weaknet from "@opentiny/vue-theme/images/empty-nonetwork.png";
import nonews from "@opentiny/vue-theme/images/empty-nonotice.png";
import noresult from "@opentiny/vue-theme/images/empty-noresult.png";
import pagenoperm from "@opentiny/vue-theme/images/empty-page-noaccess.png";
import pageweaknet from "@opentiny/vue-theme/images/empty-page-nonetwork.png";
import pagenothing from "@opentiny/vue-theme/images/empty-page-nothing.png";
import pageservererror from "@opentiny/vue-theme/images/empty-page-servererror.png";
import { openBlock, createElementBlock, normalizeClass, createElementVNode, createCommentVNode, renderSlot, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var $constants = {
  nodata: "ui.exception.nodatamf",
  noperm: "ui.exception.nopermmf",
  weaknet: "ui.exception.weaknetmf",
  noresult: "ui.exception.noresult",
  nonews: "ui.exception.nonews",
  pagenoperm: "ui.exception.pagenoperm",
  pageweaknet: "ui.exception.pageweaknet",
  pagenothing: "ui.exception.pagenothing",
  pageservererror: "ui.exception.pageservererror"
};
var _sfc_main = defineComponent({
  components: {
    TinyButton: Button
  },
  emits: ["click"],
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    type: {
      type: String,
      default: "nodata"
    },
    subMessage: String,
    exceptionClass: String,
    pageEmpty: {
      type: Boolean,
      default: false
    },
    componentPage: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true,
      extendOptions: {
        images: {
          noperm,
          nodata,
          weaknet,
          noresult,
          nonews,
          pagenoperm,
          pageweaknet,
          pagenothing,
          pageservererror
        }
      }
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-exception-body"
};
var _hoisted_2 = {
  class: "bg-cover",
  "data-tag": "tiny-exception-image"
};
var _hoisted_3 = ["src"];
var _hoisted_4 = {
  "data-tag": "tiny-exception-description"
};
var _hoisted_5 = {
  key: 0,
  "data-tag": "tiny-exception-submessage",
  class: "leading-5.5 sm:leading-3 sm:text-xs mt-1 sm:mt-2 text-color-icon-placeholder"
};
var _hoisted_6 = {
  class: "sm:mt-4 mt-6 inline-block"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-exception",
      class: normalizeClass(_ctx.m("flex t-0 justify-center items-center w-full h-full bg-color-bg-1 text-center", _ctx.exceptionClass))
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [_ctx.state.urlType ? (openBlock(), createElementBlock("img", {
      key: 0,
      class: normalizeClass(["inline-block", _ctx.componentPage ? "w-24" : "w-52"]),
      src: _ctx.state.images[_ctx.state.urlType],
      alt: ""
    }, null, 10, _hoisted_3)) : createCommentVNode("v-if", true)]), renderSlot(_ctx.$slots, "content", {}, function() {
      return [createElementVNode("div", _hoisted_4, [createElementVNode(
        "div",
        {
          "data-tag": "tiny-exception-componentpage",
          class: normalizeClass(["text-color-text-primary mt-1 text-center text-sm", _ctx.componentPage ? "sm:text-xs" : "sm:text-sm"])
        },
        toDisplayString(_ctx.t(_ctx._constants[_ctx.type])),
        3
        /* TEXT, CLASS */
      ), _ctx.subMessage ? (openBlock(), createElementBlock(
        "div",
        _hoisted_5,
        toDisplayString(_ctx.subMessage),
        1
        /* TEXT */
      )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_6, [renderSlot(_ctx.$slots, "default")])])];
    })])],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
