// @opentiny/vue-renderless/time-line/vue.js
import "../chunk-G2ADBYYC.js";
import {
  handleClick,
  getStatusCls,
  getStatus,
  computedData,
  getDate,
  computedCurrent,
  computedIsReverse,
  changeStatus,
  computedStackNodes,
  computedSpace,
  computedWrapperClass,
  toggleFold
} from "./index";
const api = ["state", "handleClick", "getStatusCls", "getStatus", "getDate", "changeStatus", "toggleFold"];
const renderless = (props, { computed, reactive, provide, watch }, { t, emit, constants }) => {
  const api2 = {};
  const state = reactive({
    nodes: computed(() => api2.computedData()),
    timelineItems: [],
    current: computed(() => api2.computedCurrent()),
    isReverse: computed(() => api2.computedIsReverse()),
    stackNodes: computed(() => state.showAll ? state.nodes : api2.computedStackNodes()),
    computedSpace: computed(() => api2.computedSpace()),
    showData: false,
    showAll: false,
    computedWrapperClass: computed(() => api2.computedWrapperClass())
  });
  Object.assign(api2, {
    state,
    getDate,
    computedData: computedData({ props, state }),
    computedCurrent: computedCurrent({ props, state }),
    computedIsReverse: computedIsReverse(props),
    computedSpace: computedSpace({ props }),
    getStatus: getStatus({ state, t }),
    handleClick: handleClick({ emit, state }),
    getStatusCls: getStatusCls({ constants, state }),
    computedStackNodes: computedStackNodes({ state, constants }),
    changeStatus: changeStatus({ state }),
    computedWrapperClass: computedWrapperClass({ props }),
    toggleFold: toggleFold({ props })
  });
  provide("nodesInject", { timelineItems: state.timelineItems, nodes: state.nodes, props });
  watch(
    () => state.timelineItems,
    (newVal) => {
      newVal.forEach((item, i) => item.index = i);
    },
    {
      immediate: true,
      deep: true
    }
  );
  return api2;
};
export {
  api,
  renderless
};
