// @opentiny/vue-renderless/common/deps/vue-popper.js
import {
  __spreadValues
} from "../../chunk-G2ADBYYC.js";
import PopupManager from "./popup-manager";
import PopperJS from "./popper";
import { on } from "./dom";
const stop = (e) => e.stopPropagation();
const isServer = typeof window === "undefined";
const getReference = ({
  state,
  props,
  refs,
  slots
}) => {
  let reference = state.referenceElm || props.reference || refs.reference && refs.reference.$el || refs.reference;
  if (!reference && slots.reference && slots.reference()[0]) {
    state.referenceElm = slots.reference()[0].elm || slots.reference()[0].el;
    reference = state.referenceElm;
  }
  return reference;
};
var vue_popper_default = (options) => {
  const { parent, emit, nextTick, onBeforeUnmount, onDeactivated, props, watch, reactive, refs, slots, toRefs } = options;
  const state = reactive({
    popperJS: null,
    appended: false,
    // arrow 是否添加
    popperElm: null,
    showPopper: props.manual ? Boolean(props.modelValue) : false,
    referenceElm: null,
    currentPlacement: ""
  });
  const appendArrow = (el) => {
    if (state.appended) {
      return;
    }
    state.appended = true;
    const div = document.createElement("div");
    div.setAttribute("x-arrow", "");
    div.className = "popper__arrow";
    el.appendChild(div);
  };
  const followHide = (popperInstance) => {
    const { followReferenceHide = true } = (props == null ? void 0 : props.popperOptions) || {};
    const { _popper: popper, _reference: reference } = popperInstance;
    if (followReferenceHide && getComputedStyle(reference).position !== "fixed" && reference.offsetParent === null) {
      popper.style.display = "none";
    }
  };
  const createPopper = () => {
    if (isServer) {
      return;
    }
    state.currentPlacement = state.currentPlacement || props.placement;
    if (!/^(top|bottom|left|right)(-start|-end)?$/g.test(state.currentPlacement)) {
      return;
    }
    const options2 = props.popperOptions || {};
    state.popperElm = state.popperElm || props.popper || refs.popper;
    const popper = state.popperElm;
    let reference = getReference({ state, props, refs, slots });
    if (!popper || !reference || reference.nodeType !== Node.ELEMENT_NODE) {
      return;
    }
    if (props.visibleArrow) {
      appendArrow(popper);
    }
    if (props.appendToBody || props.popperAppendToBody) {
      document.body.appendChild(state.popperElm);
    } else {
      parent && parent.$el && parent.$el.appendChild(state.popperElm);
      options2.forceAbsolute = true;
    }
    options2.placement = state.currentPlacement;
    options2.offset = props.offset || 0;
    options2.arrowOffset = props.arrowOffset || 0;
    options2.adjustArrow = props.adjustArrow || false;
    state.popperJS = new PopperJS(reference, popper, options2);
    emit("created", state);
    if (typeof options2.onUpdate === "function") {
      state.popperJS.onUpdate(options2.onUpdate);
    }
    state.popperJS._popper.style.zIndex = PopupManager.nextZIndex().toString();
    followHide(state.popperJS);
    on(state.popperElm, "click", stop);
  };
  const updatePopper = (popperElm) => {
    if (popperElm) {
      state.popperElm = popperElm;
    }
    const popperJS = state.popperJS;
    if (popperJS) {
      popperJS.update();
    } else {
      createPopper();
    }
  };
  const doDestroy = (forceDestroy) => {
    if (!state.popperJS || state.showPopper && !forceDestroy) {
      return;
    }
    state.popperJS.destroy();
    state.popperJS = null;
  };
  const destroyPopper = (remove) => {
    if (remove) {
      if (state.popperElm) {
        state.popperElm.remove();
      }
    }
  };
  watch(
    () => state.showPopper,
    (val) => {
      if (props.disabled) {
        return;
      }
      if (val) {
        nextTick(updatePopper);
      }
      props.trigger === "manual" && emit("update:modelValue", val);
    }
  );
  onBeforeUnmount(() => {
    nextTick(() => {
      doDestroy(true);
      destroyPopper("remove");
    });
  });
  onDeactivated(() => {
    doDestroy(true);
    destroyPopper("remove");
  });
  return __spreadValues({ updatePopper, destroyPopper, doDestroy }, toRefs(state));
};
export {
  vue_popper_default as default
};
