// @opentiny/vue-renderless/floating-button/vue.js
import "../chunk-G2ADBYYC.js";
import { getClientWidth, handleClick, clearTimer, useTouchEvent, onScroll, computedStyle } from "./index";
const api = ["state", "handleClick"];
const renderless = (props, { computed, reactive, onMounted, onBeforeUnmount }, { vm, emit }) => {
  const state = reactive({
    centerSpace: 0,
    disabled: false,
    initTimer: null,
    scrolling: false,
    commHiddenSpace: "",
    specialHiddenSpace: "",
    stayTime: null,
    lastScrollTop: 0,
    style: computed(() => api2.computedStyle())
  });
  const api2 = {};
  Object.assign(api2, {
    state,
    getClientWidth: getClientWidth({ state, vm }),
    handleClick: handleClick({ props, state, emit }),
    clearTimer: clearTimer(state),
    useTouchEvent: useTouchEvent({ state, vm, props }),
    onScroll: onScroll({ state, api: api2 }),
    computedStyle: computedStyle({ props, state })
  });
  onMounted(() => {
    api2.getClientWidth(), api2.onScroll();
  });
  onBeforeUnmount(api2.clearTimer);
  return api2;
};
export {
  api,
  renderless
};
