// @opentiny/vue-upload/lib/mobile-first.js
import { createVNode, isVNode } from "vue";
import { defineComponent, $prefix, props, setup as _setup, h } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/upload/vue";
import UploadDragger from "@opentiny/vue-upload-dragger";
import Modal from "@opentiny/vue-modal";
function _isSlot(s) {
  return typeof s === "function" || Object.prototype.toString.call(s) === "[object Object]" && !isVNode(s);
}
var _sfc_main = /* @__PURE__ */ defineComponent({
  inheritAttrs: false,
  name: $prefix + "Upload",
  props: [].concat(props, ["accept", "action", "autoUpload", "beforeUpload", "data", "disabled", "drag", "edmToken", "fileList", "headers", "httpRequest", "isFolder", "limit", "listType", "multiple", "name", "onError", "onExceed", "onPreview", "onProgress", "onRemove", "onStart", "onSuccess", "type", "withCredentials", "scale", "sourceType", "displayOnly", "customClass", "handleTriggerClick", "mode"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      h,
      extendOptions: {
        Modal
      }
    });
  },
  render: function render() {
    var accept = this.accept, disabled = this.disabled, handleChange = this.handleChange, handleClick = this.handleClick, handleKeydown = this.handleKeydown, isFolder = this.isFolder, multiple = this.multiple, name = this.name, uploadFiles = this.uploadFiles, listType = this.listType, displayOnly = this.displayOnly, customClass = this.customClass, sourceType = this.sourceType, mode = this.mode;
    var defaultSlot = this.slots.default && this.slots.default();
    var tipSlot = this.slots.tip && this.slots.tip();
    var operateSlot = this.slots.operate && this.slots.operate();
    var isBubbleMode = mode === "bubble";
    return createVNode("div", {
      "class": !displayOnly && listType === "text" ? "flex justify-between mt-4 mb-2 " + (isBubbleMode ? "sm:my-0" : "sm:my-3") : "h-full"
    }, [tipSlot && createVNode("div", {
      "class": "flex items-center sm:hidden inline-block text-sm"
    }, [tipSlot]), createVNode("div", {
      "class": "h-full",
      "onClick": function onClick($event) {
        return handleClick($event, sourceType);
      },
      "onKeydown": handleKeydown,
      "tabindex": "0"
    }, [listType === "drag-single" ? createVNode(UploadDragger, {
      "customClass": customClass,
      "disabled": disabled,
      "onFile": uploadFiles
    }, _isSlot(defaultSlot) ? defaultSlot : {
      default: function _default() {
        return [defaultSlot];
      }
    }) : defaultSlot]), operateSlot, tipSlot && createVNode("div", {
      "class": "hidden sm:inline-block flex-1 w-0 self-center"
    }, [tipSlot]), createVNode("input", {
      "class": "hidden",
      "type": "file",
      "webkitdirectory": isFolder,
      "ref": "input",
      "name": name,
      "onChange": handleChange,
      "multiple": isFolder ? true : multiple,
      "accept": accept
    }, null)]);
  }
});
export {
  _sfc_main as default
};
