// @opentiny/vue-user-account/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/user-account/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var UserAccount = defineComponent({
  name: $prefix + "UserAccount",
  props: _extends({}, $props, {
    getLangData: Function,
    getUserInfo: Function,
    getUserImageUrl: Function,
    getLogoutUrl: Function,
    redirectUrl: String,
    isLocal: Boolean,
    network: Function,
    isMock: Boolean,
    isGuestUser: Function,
    showLogin: Function,
    showArrow: {
      type: Boolean,
      default: false
    },
    popperClass: String,
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    beforeLogout: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
UserAccount.install = function(Vue) {
  Vue.component(UserAccount.name, UserAccount);
};
UserAccount.version = version;
export {
  UserAccount as default
};
