// @opentiny/vue-action-menu/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/action-menu/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var actionMenuProps = _extends({}, $props, {
  options: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  maxShowNum: {
    type: Number,
    default: 2
  },
  moreText: String,
  spacing: {
    type: [String, Number],
    default: "5px"
  },
  textField: {
    type: String,
    default: "label"
  },
  popperClass: {
    type: String,
    default: ""
  },
  popperAppendToBody: {
    type: Boolean,
    default: false
  },
  trigger: {
    type: String,
    default: "hover"
  },
  suffixIcon: Object,
  showIcon: {
    type: Boolean,
    default: true
  }
});
var ActionMenu = defineComponent({
  name: $prefix + "ActionMenu",
  props: actionMenuProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
ActionMenu.install = function(Vue) {
  Vue.component(ActionMenu.name, ActionMenu);
};
ActionMenu.version = version;
export {
  ActionMenu as default
};
