// @opentiny/vue-timeline-item/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/timeline-item/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconYes, iconClose } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/steps/index.css";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, Fragment, createElementVNode, createBlock, toDisplayString, renderSlot, createTextVNode, withDirectives, vShow, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["click"],
  props: [].concat(props, ["node", "space", "lineWidth", "shape"]),
  components: {
    IconYes: iconYes(),
    IconClose: iconClose()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["custom-title"];
var _hoisted_2 = {
  key: 1
};
var _hoisted_3 = {
  class: "node-description"
};
var _hoisted_4 = ["textContent"];
var _hoisted_5 = {
  class: "tiny-timeline-item__description",
  ref: "description"
};
var _hoisted_6 = {
  class: "date-time"
};
var _hoisted_7 = {
  class: "time"
};
var _hoisted_8 = {
  class: "tiny-steps__node"
};
var _hoisted_9 = ["custom-title"];
var _hoisted_10 = {
  key: 1
};
var _hoisted_11 = {
  class: "node-description"
};
var _hoisted_12 = ["textContent"];
var _hoisted_13 = {
  key: 0,
  class: "status"
};
var _hoisted_14 = {
  key: 0,
  class: "date-time"
};
var _hoisted_15 = {
  class: "date"
};
var _hoisted_16 = {
  class: "time"
};
var _hoisted_17 = {
  class: "tiny-timeline-item__pillar"
};
var _hoisted_18 = {
  key: 1
};
var _hoisted_19 = {
  key: 1,
  class: "dot-container"
};
var _hoisted_20 = /* @__PURE__ */ createElementVNode(
  "span",
  {
    class: "dot"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_21 = [_hoisted_20];
var _hoisted_22 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "line"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_23 = {
  class: "tiny-timeline-item__content"
};
var _hoisted_24 = {
  class: "name"
};
var _hoisted_25 = {
  key: 0,
  class: "time"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  var _component_icon_yes = resolveComponent("icon-yes");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-timeline-item", _ctx.state.computedItemCls]),
      style: normalizeStyle(_ctx.state.computedItemStyle)
    },
    [!_ctx.rootProps.vertical ? (openBlock(), createElementBlock(
      Fragment,
      {
        key: 0
      },
      [_ctx.rootProps.textPosition === "right" ? (openBlock(), createElementBlock(
        Fragment,
        {
          key: 0
        },
        [createElementVNode("div", {
          class: "icon step-icon",
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.handleClick(_ctx.node);
          })
        }, [(_ctx.node.index < _ctx.state.current || _ctx.node.error) && !_ctx.rootProps.onlyNumber ? (openBlock(), createElementBlock("span", {
          key: 0,
          "custom-title": _ctx.node.index + _ctx.rootProps.start,
          class: "icon-wrap"
        }, [_ctx.node.error ? (openBlock(), createBlock(_component_icon_close, {
          key: 0,
          class: "tiny-svg-size fixicon"
        })) : (openBlock(), createBlock(_component_icon_yes, {
          key: 1,
          class: "tiny-svg-size fixicon"
        }))], 8, _hoisted_1)) : (openBlock(), createElementBlock(
          "span",
          _hoisted_2,
          toDisplayString(_ctx.rootProps.showNumber ? _ctx.node.index + _ctx.rootProps.start : ""),
          1
          /* TEXT */
        ))]), createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "text", {
          slotScope: _ctx.node
        }, function() {
          return [createElementVNode("div", {
            class: "name",
            textContent: toDisplayString(_ctx.node[_ctx.rootProps.nameField]),
            onClick: _cache[1] || (_cache[1] = function($event) {
              return _ctx.handleClick(_ctx.node);
            })
          }, null, 8, _hoisted_4)];
        }), createElementVNode(
          "div",
          {
            class: normalizeClass(["line", {
              "line-done": _ctx.node.index < _ctx.state.current,
              "line-end": _ctx.node.index === _ctx.state.nodesLength - 1
            }]),
            style: normalizeStyle({
              width: _ctx.state.computedLineWidth
            })
          },
          null,
          6
          /* CLASS, STYLE */
        ), createElementVNode(
          "div",
          _hoisted_5,
          [renderSlot(_ctx.$slots, "description", {
            slotScope: _ctx.node
          }, function() {
            return [createTextVNode(
              toDisplayString(_ctx.node.description),
              1
              /* TEXT */
            )];
          }), withDirectives(createElementVNode(
            "div",
            null,
            [renderSlot(_ctx.$slots, "active-node-desc", {
              slotScope: _ctx.node
            })],
            512
            /* NEED_PATCH */
          ), [[vShow, _ctx.node.index === _ctx.rootProps.active]])],
          512
          /* NEED_PATCH */
        )])],
        64
        /* STABLE_FRAGMENT */
      )) : (openBlock(), createElementBlock(
        Fragment,
        {
          key: 1
        },
        [renderSlot(_ctx.$slots, "top", {
          slotScope: _ctx.node
        }, function() {
          return [createElementVNode("div", _hoisted_6, [createElementVNode(
            "span",
            _hoisted_7,
            toDisplayString(_ctx.getDate(_ctx.node[_ctx.rootProps.timeField]).date) + " " + toDisplayString(_ctx.getDate(_ctx.node[_ctx.rootProps.timeField]).time),
            1
            /* TEXT */
          )])];
        }), createElementVNode("div", _hoisted_8, [createElementVNode("div", {
          class: "icon",
          onClick: _cache[2] || (_cache[2] = function($event) {
            return _ctx.handleClick(_ctx.node);
          })
        }, [(_ctx.node.index < _ctx.state.current || _ctx.node.error) && !_ctx.rootProps.onlyNumber ? (openBlock(), createElementBlock("span", {
          key: 0,
          "custom-title": _ctx.node.index + _ctx.rootProps.start,
          class: "icon-wrap"
        }, [_ctx.node.error ? (openBlock(), createBlock(_component_icon_close, {
          key: 0,
          class: "tiny-svg-size fixicon"
        })) : (openBlock(), createBlock(_component_icon_yes, {
          key: 1,
          class: "tiny-svg-size fixicon"
        }))], 8, _hoisted_9)) : (openBlock(), createElementBlock(
          "span",
          _hoisted_10,
          toDisplayString(_ctx.rootProps.showNumber ? _ctx.node.index + _ctx.rootProps.start : ""),
          1
          /* TEXT */
        ))]), createElementVNode(
          "div",
          {
            class: normalizeClass(["line", {
              "line-done": _ctx.node.index < _ctx.state.current,
              "line-end": _ctx.node.index === _ctx.state.nodesLength - 1
            }])
          },
          null,
          2
          /* CLASS */
        )]), createElementVNode("div", _hoisted_11, [renderSlot(_ctx.$slots, "bottom", {
          slotScope: _ctx.node
        }, function() {
          return [createElementVNode("div", {
            class: "name",
            textContent: toDisplayString(_ctx.node[_ctx.rootProps.nameField]),
            onClick: _cache[3] || (_cache[3] = function($event) {
              return _ctx.handleClick(_ctx.node);
            })
          }, null, 8, _hoisted_12), _ctx.rootProps.showStatus ? (openBlock(), createElementBlock(
            "div",
            _hoisted_13,
            toDisplayString(_ctx.getStatus(_ctx.node.index)),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)];
        })])],
        64
        /* STABLE_FRAGMENT */
      ))],
      64
      /* STABLE_FRAGMENT */
    )) : (openBlock(), createElementBlock(
      Fragment,
      {
        key: 1
      },
      [renderSlot(_ctx.$slots, "left", {
        slotScope: _ctx.node
      }, function() {
        return [_ctx.shape === "circle" ? (openBlock(), createElementBlock("div", _hoisted_14, [createElementVNode(
          "span",
          _hoisted_15,
          toDisplayString(_ctx.getDate(_ctx.node[_ctx.rootProps.timeField]).date),
          1
          /* TEXT */
        ), createElementVNode(
          "span",
          _hoisted_16,
          toDisplayString(_ctx.getDate(_ctx.node[_ctx.rootProps.timeField]).time),
          1
          /* TEXT */
        )])) : createCommentVNode("v-if", true)];
      }), createElementVNode("div", _hoisted_17, [_ctx.shape === "circle" ? (openBlock(), createElementBlock("div", {
        key: 0,
        class: "icon",
        onClick: _cache[4] || (_cache[4] = function($event) {
          return _ctx.handleClick(_ctx.node);
        })
      }, [(_ctx.state.isReverse ? _ctx.node.index > _ctx.state.current : _ctx.node.index < _ctx.state.current) ? (openBlock(), createBlock(_component_icon_yes, {
        key: 0,
        class: "tiny-svg-size"
      })) : (openBlock(), createElementBlock(
        "span",
        _hoisted_18,
        toDisplayString(_ctx.rootProps.showNumber ? _ctx.state.isReverse ? _ctx.state.nodesLength - 1 - _ctx.node.index + _ctx.rootProps.start : _ctx.node.index + _ctx.rootProps.start : ""),
        1
        /* TEXT */
      ))])) : (openBlock(), createElementBlock("div", _hoisted_19, _hoisted_21)), _hoisted_22]), renderSlot(_ctx.$slots, "right", {
        slotScope: _ctx.node
      }, function() {
        return [createElementVNode("div", _hoisted_23, [createElementVNode(
          "div",
          _hoisted_24,
          toDisplayString(_ctx.node[_ctx.rootProps.nameField]),
          1
          /* TEXT */
        ), _ctx.shape === "dot" ? (openBlock(), createElementBlock(
          "div",
          _hoisted_25,
          toDisplayString(_ctx.node[_ctx.rootProps.timeField]),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)])];
      })],
      64
      /* STABLE_FRAGMENT */
    ))],
    6
    /* CLASS, STYLE */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
