// @opentiny/vue-scroll-text/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/scroll-text/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  ANIMATION_DURATION: "animation-duration"
};
var ScrollText = defineComponent({
  name: $prefix + "ScrollText",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    text: {
      type: String,
      default: "ScrollText"
    },
    time: {
      type: Number,
      default: 5
    },
    styles: {
      type: String,
      default: ""
    },
    direction: {
      type: String,
      default: "left",
      validator: function validator(value) {
        return Boolean(~["up", "left", "down", "right"].indexOf(value));
      }
    },
    hoverStop: {
      type: Boolean,
      default: true
    },
    widthAdapt: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
ScrollText.install = function(Vue) {
  Vue.component(ScrollText.name, ScrollText);
};
ScrollText.version = version;
export {
  ScrollText as default
};
