// @opentiny/vue-cascader-menu/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { createVNode, resolveComponent, mergeProps } from "vue";
import { renderless, api } from "@opentiny/vue-renderless/cascader-menu/vue";
import { defineComponent, $prefix, setup as _setup, h } from "@opentiny/vue-common";
import { t } from "@opentiny/vue-locale";
import Scrollbar from "@opentiny/vue-scrollbar";
import CascaderNode from "@opentiny/vue-cascader-node";
import "@opentiny/vue-theme/cascader-menu/index.css";
var renderNodeListFunc = function renderNodeListFunc2(_vm) {
  return function() {
    var events = {
      on: {}
    };
    var menuId = _vm.state.menuId;
    var isHoverMenu = _vm.panel.state.isHoverMenu;
    if (isHoverMenu) {
      events.on.expand = _vm.handleExpand;
    }
    var mapHandler = function mapHandler2(node, index) {
      var uid = node.uid;
      var hasChildren = node.hasChildren;
      return createVNode(resolveComponent("cascader-node"), mergeProps({
        "key": uid,
        "node": node,
        "node-id": menuId + "-" + index,
        "data-haspopup": hasChildren,
        "data-owns": hasChildren ? menuId : null
      }, events), null);
    };
    var nodes = _vm.nodes.map(mapHandler);
    return [].concat(nodes, [isHoverMenu ? createVNode("svg", {
      "ref": "hoverZone",
      "class": "tiny-cascader-menu__hover-zone"
    }, null) : null]);
  };
};
var CascaderMenu = /* @__PURE__ */ defineComponent({
  name: $prefix + "CascaderMenu",
  components: {
    TinyScrollbar: Scrollbar,
    CascaderNode
  },
  props: {
    nodes: {
      type: Array,
      required: true
    },
    index: Number
  },
  inject: ["panel"],
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  },
  render: function render() {
    var renderEmptyText = function renderEmptyText2() {
      return createVNode("div", {
        "class": "tiny-cascader-menu__empty-text"
      }, [t("ui.cascader.noData")]);
    };
    var renderNodeList = renderNodeListFunc(this);
    var state = this.state;
    var events = {
      nativeOn: {}
    };
    if (this.panel.state.isHoverMenu) {
      events.nativeOn.mousemove = this.handleMouseMove;
    }
    return h("tiny-scrollbar", _extends({
      attrs: {
        role: "menu",
        id: state.menuId
      },
      ref: "cascaderMenu",
      class: "tiny-cascader-menu",
      props: {
        tag: "ul",
        "wrap-class": "tiny-cascader-menu__wrap",
        "view-class": {
          "tiny-cascader-menu__list": true,
          "is-empty": state.isEmpty
        }
      }
    }, events), [state.isEmpty ? renderEmptyText() : renderNodeList()]);
  }
});
var CascaderMenu$1 = CascaderMenu;
var version = "3.13.0";
CascaderMenu$1.install = function(Vue) {
  Vue.component(CascaderMenu$1.name, CascaderMenu$1);
};
CascaderMenu$1.version = version;
export {
  CascaderMenu$1 as default
};
