// @opentiny/vue-steps/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, props, setup as _setup, $props, $prefix, $setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/steps/vue";
import { IconFinish, IconWarn, IconChevronLeft, IconChevronRight } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, withDirectives, createElementVNode, normalizeClass, toDisplayString, Fragment, renderList, normalizeStyle, createCommentVNode, vShow, createBlock, createTextVNode, renderSlot, createVNode, withCtx } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import { renderless as renderless$1, api as api$1 } from "@opentiny/vue-renderless/steps/slide-bar";
var classes$1 = {
  "steps-icon": "flex-none flex items-center justify-center rounded-full border-0.5 border-current text-gray-300 bg-gray-50",
  "steps-icon-left": "w-full h-full flex items-center justify-center rounded-full border-0.5 border-color-brand text-color-brand bg-color-bg-1 relative z-10",
  "steps-icon-right": "w-full h-full flex items-center justify-center rounded-full border-0.5 border-color-brand text-color-brand bg-color-bg-1 relative z-[9999]",
  "icon-mini": "w-2 h-2 text-xs",
  "icon-small": "w-4 h-4 text-xs",
  "icon-medium": "w-6 h-6 text-sm",
  "icon-large": "w-8 h-8 text-base",
  "icon-error": "text-color-error bg-color-bg-1",
  "icon-doing": "text-color-brand bg-blue-50",
  "icon-done": "text-color-brand bg-blue-50",
  "icon-disabled": "bg-gray-50 text-gray-200",
  "icon-error-active": "bg-red-50 border-red-500",
  "icon-doing-active": "bg-color-brand border-color-brand text-color-text-inverse",
  "icon-done-active": "bg-color-brand border-color-brand text-color-text-inverse",
  "icon-disabled-active": "bg-gray-50 text-gray-200",
  "steps-line": "border-t-0.5 sm:border-t border-color-text flex-auto",
  "left-line-mini": "mr-1",
  "left-line-small": "mr-2",
  "left-line-medium": "mr-2",
  "left-line-large": "mr-3",
  "right-line-mini": "ml-1",
  "right-line-small": "ml-2",
  "right-line-medium": "ml-2",
  "right-line-large": "ml-3",
  "steps-title": "hidden sm:block text-sm ml-2 font-medium truncate",
  "title-font-mini": "text-xs",
  "title-font-small": "text-xs",
  "title-font-medium": "text-sm",
  "title-font-large": "text-base",
  "circle-wrap": "flex-none flex relative z-10",
  "icon-circle": "flex-none rounded-full border-0.5 border-color-brand bg-color-bg-1 font-medium absolute"
};
var _sfc_main$3 = defineComponent({
  emits: ["click"],
  components: {
    IconFinish: IconFinish(),
    IconWarn: IconWarn()
  },
  props: [].concat(props, ["vertical", "nameField", "statusField", "data", "active", "visibleNum", "descriptionField", "size"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      classes: classes$1
    });
  }
});
var _hoisted_1$2 = {
  "data-tag": "tiny-steps",
  class: "flex text-color-text-primary"
};
var _hoisted_2$2 = {
  "data-tag": "tiny-steps-main",
  class: "flex items-center relative"
};
var _hoisted_3$2 = ["onClick"];
var _hoisted_4$2 = {
  "data-tag": "tiny-steps-main",
  class: "flex items-center"
};
var _hoisted_5$1 = ["title"];
var _hoisted_6 = ["title"];
var _hoisted_7 = {
  "data-tag": "tiny-steps-main",
  class: "flex items-center relative"
};
function _sfc_render$2(_ctx, _cache, $props2, $setup2, $data, $options) {
  var _component_icon_finish = resolveComponent("icon-finish");
  var _component_icon_warn = resolveComponent("icon-warn");
  return openBlock(), createElementBlock("div", _hoisted_1$2, [withDirectives(createElementVNode(
    "div",
    {
      "data-tag": "tiny-steps-left",
      class: normalizeClass(["flex-1 flex flex-col cursor-pointer", _ctx.vertical ? "text-center" : "sm:max-w-[theme(spacing.80)] sm:mr-4"]),
      onClick: _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.startIndex--;
      })
    },
    [createElementVNode("div", _hoisted_2$2, [createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-line",
        class: normalizeClass(["border-t-0.5 sm:border-t  flex-auto opacity-0", {
          "sm:min-w-[theme(spacing.6)] sm:ml-4 sm:hidden": !_ctx.vertical
        }])
      },
      null,
      2
      /* CLASS */
    ), createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-icon",
        class: normalizeClass([_ctx.gcls("circle-wrap"), _ctx.gcls("icon-" + _ctx.size)])
      },
      [createElementVNode(
        "div",
        {
          class: normalizeClass(_ctx.gcls("steps-icon-left"))
        },
        toDisplayString(_ctx.size === "mini" ? "" : _ctx.state.startIndex),
        3
        /* TEXT, CLASS */
      ), _ctx.state.startIndex - 1 > 0 ? (openBlock(true), createElementBlock(
        Fragment,
        {
          key: 0
        },
        renderList(_ctx.state.startIndex - 1, function(item, index) {
          return openBlock(), createElementBlock(
            "div",
            {
              class: normalizeClass([_ctx.gcls("icon-circle"), _ctx.gcls("icon-" + _ctx.size)]),
              style: normalizeStyle({
                right: (_ctx.state.startIndex - 1 - index) * 4 + "px"
              }),
              key: index
            },
            null,
            6
            /* CLASS, STYLE */
          );
        }),
        128
        /* KEYED_FRAGMENT */
      )) : createCommentVNode("v-if", true)],
      2
      /* CLASS */
    ), createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-line",
        class: normalizeClass(["border-t-0.5 sm:border-t flex-auto", {
          "sm:min-w-[theme(spacing.6)] sm:ml-4": !_ctx.vertical
        }, {
          "border-color-brand": _ctx.data[_ctx.state.startIndex - 1] && _ctx.data[_ctx.state.startIndex - 1].status === "done"
        }])
      },
      null,
      2
      /* CLASS */
    )])],
    2
    /* CLASS */
  ), [[vShow, _ctx.state.startIndex !== 0]]), (openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.data, function(node, index) {
      return withDirectives((openBlock(), createElementBlock("div", {
        "data-tag": "tiny-steps-block",
        key: index,
        class: normalizeClass(_ctx.m("group flex-1 flex flex-col cursor-pointer text-center", {
          "text-color-brand": index === _ctx.active
        }, _ctx.vertical ? "" : "sm:min-w-[theme(spacing.32)] sm:max-w-[theme(spacing.80)] sm:mr-4 sm:text-left", {
          "flex-none": index === _ctx.data.length - 1 && !_ctx.vertical
        }, {
          " hover:text-color-brand-hover active:text-color-brand-active": node[_ctx.statusField] !== "disabled" && node[_ctx.statusField] !== "error"
        }, {
          "text-color-none-hover": node[_ctx.statusField] === "disabled"
        }, {
          "text-color-error": node[_ctx.statusField] === "error"
        })),
        onClick: function onClick($event) {
          return _ctx.$emit("click", index, node, $event);
        }
      }, [createElementVNode("div", _hoisted_4$2, [createElementVNode(
        "div",
        {
          "data-tag": "tiny-steps-line",
          class: normalizeClass(_ctx.m(_ctx.gcls("steps-line"), _ctx.gcls("left-line-" + _ctx.size), {
            "opacity-0": index === 0 && _ctx.state.startIndex === 0
          }, {
            "sm:hidden sm:min-w-[theme(spacing.6)] sm:ml-4": !_ctx.vertical
          }, {
            "border-color-brand": ["done", "doing"].includes(node[_ctx.statusField])
          }))
        },
        null,
        2
        /* CLASS */
      ), createElementVNode(
        "div",
        {
          "data-tag": "tiny-steps-icon",
          class: normalizeClass(_ctx.m(_ctx.gcls("steps-icon"), _ctx.gcls("icon-" + _ctx.size), {
            "bg-gray-300 border-gray-300 text-color-text-inverse": index === _ctx.active
          }, _ctx.gcls("icon-" + node[_ctx.statusField]), index === _ctx.active && _ctx.gcls("icon-" + node[_ctx.statusField] + "-active"), {
            "bg-red-500 border-red-500": index === _ctx.active && _ctx.size === "mini" && node[_ctx.statusField] === "error"
          }))
        },
        [_ctx.size !== "mini" ? (openBlock(), createElementBlock(
          Fragment,
          {
            key: 0
          },
          [node[_ctx.statusField] === "done" ? (openBlock(), createBlock(_component_icon_finish, {
            key: 0,
            "custom-class": _ctx.m("w-3.5 h-3.5 fill-color-brand", {
              "fill-color-icon-inverse": index === _ctx.active
            }, {
              "w-2.5 h-2.5": _ctx.size === "small"
            }, {
              "w-4.5 h-4.5": _ctx.size === "large"
            })
          }, null, 8, ["custom-class"])) : node[_ctx.statusField] === "error" ? (openBlock(), createBlock(_component_icon_warn, {
            key: 1,
            "custom-class": _ctx.m("w-3.5 h-3.5 fill-color-error", {
              "w-2.5 h-2.5": _ctx.size === "small"
            }, {
              "w-4.5 h-4.5": _ctx.size === "large"
            })
          }, null, 8, ["custom-class"])) : (openBlock(), createElementBlock(
            Fragment,
            {
              key: 2
            },
            [createTextVNode(
              toDisplayString(index + 1),
              1
              /* TEXT */
            )],
            64
            /* STABLE_FRAGMENT */
          ))],
          64
          /* STABLE_FRAGMENT */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      ), !_ctx.vertical ? (openBlock(), createElementBlock("div", {
        key: 0,
        "data-tag": "tiny-steps-title",
        title: node[_ctx.nameField],
        class: normalizeClass([_ctx.gcls("steps-title"), _ctx.gcls("title-font-" + _ctx.size)])
      }, toDisplayString(node[_ctx.nameField]), 11, _hoisted_5$1)) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        {
          "data-tag": "tiny-steps-line",
          class: normalizeClass(_ctx.m(_ctx.gcls("steps-line"), _ctx.gcls("right-line-" + _ctx.size), {
            "sm:hidden": !_ctx.vertical && index === _ctx.data.length - 1
          }, {
            "opacity-0": index === _ctx.data.length - 1
          }, {
            "sm:min-w-[theme(spacing.6)] sm:ml-4": !_ctx.vertical
          }, {
            "border-color-brand": node[_ctx.statusField] === "done"
          }))
        },
        null,
        2
        /* CLASS */
      )]), renderSlot(_ctx.$slots, "item", {
        slotScope: node,
        index
      }, function() {
        return [createElementVNode("div", {
          "data-tag": "tiny-steps-title",
          title: node[_ctx.nameField],
          class: normalizeClass(["mt-3 font-medium truncate mx-5 ", _ctx.gcls("title-font-" + _ctx.size), {
            "sm:hidden sm:mx-0": !_ctx.vertical
          }])
        }, toDisplayString(node[_ctx.nameField]), 11, _hoisted_6), node[_ctx.descriptionField] ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            "data-tag": "tiny-steps-description",
            class: normalizeClass(_ctx.m("text-xs text-color-text-secondary mt-2 line-clamp-2 mx-5 break-all", {
              "text-color-brand": index === _ctx.active
            }, {
              "group-hover:text-color-brand-hover group-active:text-color-brand-active": node[_ctx.statusField] !== "disabled" && node[_ctx.statusField] !== "error"
            }, {
              "text-color-none-hover": node[_ctx.statusField] === "disabled"
            }, {
              "text-color-brand": index === _ctx.active
            }, {
              "text-color-error": node[_ctx.statusField] === "error"
            }, !_ctx.vertical ? "sm:ml-8" : ""))
          },
          toDisplayString(node[_ctx.descriptionField]),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true)];
      })], 10, _hoisted_3$2)), [[vShow, _ctx.isVisibleHandler(index) === "visible"]]);
    }),
    128
    /* KEYED_FRAGMENT */
  )), withDirectives(createElementVNode(
    "div",
    {
      "data-tag": "tiny-steps-right",
      class: normalizeClass(["flex-1 flex flex-col cursor-pointer text-center", {
        "sm:min-w-[theme(spacing.32)] sm:max-w-[theme(spacing.80)] sm:text-left": !_ctx.vertical
      }]),
      onClick: _cache[1] || (_cache[1] = function($event) {
        return _ctx.state.startIndex++;
      })
    },
    [createElementVNode("div", _hoisted_7, [createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-line",
        class: normalizeClass(["border-t-0.5 sm:border-t  flex-auto", {
          "sm:hidden sm:min-w-[theme(spacing.6)] sm:ml-4": !_ctx.vertical
        }, {
          "border-color-brand": _ctx.data[_ctx.state.endIndex] && ["done", "doing"].includes(_ctx.data[_ctx.state.endIndex].status)
        }])
      },
      null,
      2
      /* CLASS */
    ), createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-icon",
        class: normalizeClass([_ctx.gcls("circle-wrap"), _ctx.gcls("icon-" + _ctx.size)])
      },
      [createElementVNode(
        "div",
        {
          class: normalizeClass(_ctx.gcls("steps-icon-right"))
        },
        toDisplayString(_ctx.size === "mini" ? "" : _ctx.state.endIndex + 1),
        3
        /* TEXT, CLASS */
      ), _ctx.state.rightNodePositions.length > 0 ? (openBlock(true), createElementBlock(
        Fragment,
        {
          key: 0
        },
        renderList(_ctx.state.rightNodePositions, function(item, index) {
          return openBlock(), createElementBlock(
            "div",
            {
              class: normalizeClass([_ctx.gcls("icon-circle"), _ctx.gcls("icon-" + _ctx.size)]),
              style: normalizeStyle(item),
              key: index
            },
            null,
            6
            /* CLASS, STYLE */
          );
        }),
        128
        /* KEYED_FRAGMENT */
      )) : createCommentVNode("v-if", true)],
      2
      /* CLASS */
    ), createElementVNode(
      "div",
      {
        "data-tag": "tiny-steps-line",
        class: normalizeClass(["border-t-0.5 sm:border-t  flex-auto opacity-0", {
          "sm:hidden sm:min-w-[theme(spacing.6)] sm:ml-4": !_ctx.vertical
        }])
      },
      null,
      2
      /* CLASS */
    )])],
    2
    /* CLASS */
  ), [[vShow, _ctx.state.startIndex + _ctx.visibleNum < _ctx.data.length]])]);
}
var MobileFirstTemplate = /* @__PURE__ */ _export_sfc(_sfc_main$3, [["render", _sfc_render$2]]);
var mode = {
  tiny_mode: "mobile-first"
};
var _sfc_main$2 = defineComponent({
  emits: ["click"],
  components: {
    IconChevronLeft: IconChevronLeft(mode),
    IconChevronRight: IconChevronRight(mode)
  },
  props: _extends({}, $props, {
    data: Array,
    duration: {
      type: Number,
      default: 300
    },
    delay: {
      type: Number,
      default: 5
    },
    noArrow: Boolean,
    flex: Boolean
  }),
  setup: function setup2(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless: renderless$1,
      api: api$1
    });
  }
});
var _hoisted_1$1 = {
  "data-tag": "tiny-steps-slide-bar",
  class: "w-full flex"
};
var _hoisted_2$1 = {
  class: "w-4 h-full flex items-center justify-center mr-1.5 bg-color-bg-2"
};
var _hoisted_3$1 = {
  ref: "slideMain",
  "data-tag": "slide-main",
  class: "flex-auto overflow-auto scrollbar-size-0"
};
var _hoisted_4$1 = {
  class: "flex"
};
var _hoisted_5 = {
  class: "w-4 h-full flex items-center justify-center ml-1.5 bg-color-bg-2"
};
function _sfc_render$1(_ctx, _cache, $props2, $setup2, $data, $options) {
  var _component_IconChevronLeft = resolveComponent("IconChevronLeft");
  var _component_IconChevronRight = resolveComponent("IconChevronRight");
  return openBlock(), createElementBlock("div", _hoisted_1$1, [createElementVNode(
    "div",
    {
      ref: "leftButton",
      "data-tag": "left-button",
      class: normalizeClass(["hidden flex-none h-8 cursor-pointer items-center select-none", {
        "sm:flex": _ctx.slideBarState.canLeftScroll || _ctx.slideBarState.canRightScroll
      }]),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.leftSlideHandler && _ctx.leftSlideHandler.apply(_ctx, arguments);
      })
    },
    [renderSlot(_ctx.$slots, "left-button", {
      show: _ctx.slideBarState.canLeftScroll,
      index: _ctx.slideBarState.boundingIndex.left
    }, function() {
      return [createElementVNode("div", _hoisted_2$1, [createVNode(_component_IconChevronLeft, {
        class: "h-3 w-3"
      })])];
    })],
    2
    /* CLASS */
  ), createElementVNode(
    "div",
    _hoisted_3$1,
    [createElementVNode("div", _hoisted_4$1, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.data, function(node, index) {
        return openBlock(), createElementBlock(
          "div",
          {
            ref_for: true,
            ref: "block" + index,
            key: index,
            style: normalizeStyle({
              width: _ctx.flex ? 100 / _ctx.data.length + "%" : null
            })
          },
          [renderSlot(_ctx.$slots, "block", {
            slotScoped: {
              node,
              index,
              show: index > _ctx.slideBarState.boundingIndex.left && (_ctx.slideBarState.boundingIndex.right === -1 || index < _ctx.slideBarState.boundingIndex.right)
            }
          }, function() {
            return [createTextVNode(
              toDisplayString(index),
              1
              /* TEXT */
            )];
          }), renderSlot(_ctx.$slots, "block-bottom", {
            node,
            index
          })],
          4
          /* STYLE */
        );
      }),
      128
      /* KEYED_FRAGMENT */
    ))])],
    512
    /* NEED_PATCH */
  ), createElementVNode(
    "div",
    {
      ref: "rightButton",
      "data-tag": "right-button",
      class: normalizeClass(["hidden flex-none h-8 cursor-pointer items-center select-none", {
        "sm:flex": _ctx.slideBarState.canLeftScroll || _ctx.slideBarState.canRightScroll
      }]),
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.rightSlideHandler && _ctx.rightSlideHandler.apply(_ctx, arguments);
      })
    },
    [renderSlot(_ctx.$slots, "right-button", {
      show: _ctx.slideBarState.canRightScroll,
      index: _ctx.slideBarState.boundingIndex.right
    }, function() {
      return [createElementVNode("div", _hoisted_5, [createVNode(_component_IconChevronRight, {
        class: "h-3 w-3"
      })])];
    })],
    2
    /* CLASS */
  )]);
}
var SlideBar = /* @__PURE__ */ _export_sfc(_sfc_main$2, [["render", _sfc_render$1]]);
var classes = {
  "steps-advanced": "w-full flex text-color-text text-xs leading-4",
  "steps-block": "h-8 flex rounded-sm relative cursor-pointer group text-color-text-primary bg-color-bg-2 hover:bg-gray-100",
  "steps-block-error": "text-color-error bg-color-error-subtler hover:bg-red-100",
  "steps-block-disabled": "text-color-text-disabled hover:bg-color-bg-2",
  "steps-block-done": "bg-blue-50 hover:bg-blue-100",
  "steps-block-doing": "bg-blue-50 hover:bg-brand-400",
  "steps-block-error-active": "bg-red-500 hover:bg-red-500",
  "steps-block-disabled-active": "bg-color-bg-2 hover:bg-color-bg-2",
  "steps-block-done-active": "bg-color-brand hover:bg-color-brand",
  "steps-block-doing-active": "bg-color-brand hover:bg-color-brand",
  "steps-text": "flex-auto max-w-full truncate text-gray-500",
  "steps-text-error": "text-gray-900",
  "steps-text-disabled": "text-gray-200",
  "steps-text-done": "text-gray-900",
  "steps-text-doing": "group-hover:text-color-text-inverse text-gray-900",
  "steps-content": "flex-auto flex max-w-full items-center justify-center px-4 overflow-hidden h-full"
};
var _sfc_main$1 = defineComponent({
  emits: ["click"],
  components: {
    SlideBar,
    IconFinish: IconFinish(),
    IconWarn: IconWarn(),
    IconChevronLeft: IconChevronLeft(),
    IconChevronRight: IconChevronRight()
  },
  props: [].concat(props, ["vertical", "nameField", "statusField", "data", "active", "visibleNum", "duration", "noArrow", "flex"]),
  setup: function setup3(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["onClick"];
var _hoisted_2 = {
  key: 2
};
var _hoisted_3 = {
  "data-tag": "tiny-steps-prefixslot",
  class: "flex-none"
};
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "-mr-2"
  },
  null,
  -1
  /* HOISTED */
);
function _sfc_render(_ctx, _cache, $props2, $setup2, $data, $options) {
  var _component_IconChevronLeft = resolveComponent("IconChevronLeft");
  var _component_IconChevronRight = resolveComponent("IconChevronRight");
  var _component_icon_finish = resolveComponent("icon-finish");
  var _component_icon_warn = resolveComponent("icon-warn");
  var _component_SlideBar = resolveComponent("SlideBar");
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-steps-advanced",
      class: normalizeClass(_ctx.gcls("steps-advanced"))
    },
    [createVNode(_component_SlideBar, {
      ref: "slideBar",
      data: _ctx.data,
      duration: _ctx.duration,
      "no-arrow": _ctx.noArrow,
      flex: _ctx.flex
    }, {
      "left-button": withCtx(function(slotScoped) {
        return [createElementVNode(
          "div",
          {
            class: normalizeClass(["w-4 h-full flex items-center justify-center mr-1.5", slotScoped.show ? "bg-color-bg-2 text-color-text hover:bg-color-info-secondary-subtle hover:text-color-brand" : "text-color-text-disabled bg-color-bg-2"])
          },
          [createVNode(_component_IconChevronLeft, {
            class: "h-3 w-3 fill-current"
          })],
          2
          /* CLASS */
        )];
      }),
      "right-button": withCtx(function(slotScoped) {
        return [createElementVNode(
          "div",
          {
            class: normalizeClass(["w-4 h-full flex items-center justify-center ml-1.5", slotScoped.show ? "bg-color-bg-2 text-color-text hover:bg-color-info-secondary-subtle hover:text-color-brand" : "text-color-text-disabled bg-color-bg-2"])
          },
          [createVNode(_component_IconChevronRight, {
            class: "h-3 w-3 fill-current"
          })],
          2
          /* CLASS */
        )];
      }),
      block: withCtx(function(_ref) {
        var _ref$slotScoped = _ref.slotScoped, node = _ref$slotScoped.node, index = _ref$slotScoped.index;
        return [createElementVNode("div", {
          "data-tag": "tiny-steps-block",
          class: normalizeClass(_ctx.m(_ctx.gcls("steps-block"), _ctx.gcls("steps-block-" + node[_ctx.statusField]), {
            "bg-color-brand hover:bg-color-brand": index === _ctx.active
          }, index === _ctx.active && _ctx.gcls("steps-block-" + node[_ctx.statusField] + "-active"), {
            "min-w-[theme(spacing.28)] max-w-[theme(spacing.56)]": !_ctx.flex
          })),
          onClick: function onClick($event) {
            return _ctx.$emit("click", index, node, $event);
          }
        }, [createElementVNode(
          "div",
          {
            "data-tag": "tiny-steps-content",
            class: normalizeClass(_ctx.gcls("steps-content"))
          },
          [renderSlot(_ctx.$slots, "item", {
            slotScope: node,
            index
          }, function() {
            return [createElementVNode(
              "div",
              {
                "data-tag": "tiny-steps-icon",
                class: normalizeClass(_ctx.m("w-4 h-4 flex flex-none items-center justify-center rounded-full mr-2 text-color-text-inverse", node[_ctx.statusField] === "done" ? "bg-color-brand" : "bg-gray-200", {
                  " bg-color-brand group-hover:text-color-brand group-hover:bg-color-bg-1 ": node[_ctx.statusField] === "doing"
                }, {
                  " text-color-brand bg-color-bg-1": index === _ctx.active && node[_ctx.statusField] !== "disabled"
                }, {
                  "text-color-text-inverse bg-color-error": node[_ctx.statusField] === "error"
                }, {
                  "text-color-error bg-color-bg-1": node[_ctx.statusField] === "error" && index === _ctx.active
                }))
              },
              [node[_ctx.statusField] === "done" ? (openBlock(), createBlock(_component_icon_finish, {
                key: 0,
                "custom-class": "w-2.5 h-2.5  fill-current"
              })) : node[_ctx.statusField] === "error" ? (openBlock(), createBlock(_component_icon_warn, {
                key: 1,
                "custom-class": "w-2.5  h-2.5  fill-current"
              })) : (openBlock(), createElementBlock(
                "span",
                _hoisted_2,
                toDisplayString(index + 1),
                1
                /* TEXT */
              ))],
              2
              /* CLASS */
            ), createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "prefix", {
              slotScope: {
                node,
                index
              }
            }, function() {
              return [_hoisted_4];
            })]), createElementVNode(
              "div",
              {
                "data-tag": "tiny-steps-text",
                class: normalizeClass(_ctx.m(_ctx.gcls("steps-text"), _ctx.gcls("steps-text-" + node[_ctx.statusField]), {
                  "text-color-text-inverse": index === _ctx.active && node[_ctx.statusField] !== "disabled"
                }))
              },
              toDisplayString(node[_ctx.nameField]),
              3
              /* TEXT, CLASS */
            )];
          })],
          2
          /* CLASS */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass(["h-full absolute left-0 border-t-transparent border-b-transparent border-white", {
              "hidden": index === 0
            }]),
            style: {
              "border-top-width": "calc(var(--tiny-spacing-8, 2rem) / 2)",
              "border-bottom-width": "calc(var(--tiny-spacing-8, 2rem) / 2)",
              "border-left-width": "var(--tiny-spacing-1_5, 0.375rem)"
            }
          },
          null,
          2
          /* CLASS */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass(["h-full absolute right-0 border-l-transparent border-white", {
              "hidden": index === _ctx.data.length - 1
            }]),
            style: {
              "border-top-width": "calc(var(--tiny-spacing-8, 2rem) / 2)",
              "border-bottom-width": "calc(var(--tiny-spacing-8, 2rem) / 2)",
              "border-left-width": "var(--tiny-spacing-1_5, 0.375rem)"
            }
          },
          null,
          2
          /* CLASS */
        )], 10, _hoisted_1)];
      }),
      "block-bottom": withCtx(function(_ref2) {
        var node = _ref2.node, index = _ref2.index;
        return [renderSlot(_ctx.$slots, "block-bottom", {
          node,
          index
        })];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["data", "duration", "no-arrow", "flex"])],
    2
    /* CLASS */
  );
}
var MobileFirstAdvancedTemplate = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render]]);
var template = function template2(mode2, props2) {
  return props2.advanced ? MobileFirstAdvancedTemplate : MobileFirstTemplate;
};
var _sfc_main = defineComponent({
  name: $prefix + "Steps",
  props: _extends({}, $props, {
    nameField: {
      type: String,
      default: "name"
    },
    countField: {
      type: String,
      default: "count"
    },
    statusField: {
      type: String,
      default: "status"
    },
    data: [Object, Array],
    space: Number,
    active: {
      type: Number,
      default: -1
    },
    visibleNum: {
      type: Number,
      default: 3
    },
    descriptionField: {
      type: String,
      default: "description"
    },
    size: {
      type: String,
      default: "medium"
    },
    vertical: Boolean,
    advanced: Boolean,
    line: Boolean,
    duration: {
      type: Number,
      default: 300
    },
    noArrow: Boolean,
    flex: Boolean
  }),
  setup: function setup4(props2, context) {
    return $setup({
      props: props2,
      context,
      template
    });
  }
});
export {
  _sfc_main as default
};
