// @opentiny/vue-radio/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/radio/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme/radio/index.css";
import { openBlock, createElementBlock, mergeProps, withKeys, withModifiers, createElementVNode, normalizeClass, withDirectives, vModelRadio, renderSlot, createTextVNode, toDisplayString, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["modelValue", "events", "label", "text", "disabled", "name", "border", "size", "tabindex", "displayOnly"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-checked", "aria-disabled"];
var _hoisted_2 = ["tabindex"];
var _hoisted_3 = ["value", "name", "disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return !_ctx.state.isDisplayOnly || _ctx.state.model === _ctx.label ? (openBlock(), createElementBlock("label", mergeProps({
    key: 0,
    class: ["tiny-radio", [_ctx.border && _ctx.state.radioSize ? "tiny-radio--" + _ctx.state.radioSize : "", {
      "is-disabled": _ctx.state.isDisabled
    }, {
      "is-focus": _ctx.state.focus
    }, {
      "is-bordered": _ctx.border
    }, {
      "is-checked": _ctx.state.model === _ctx.label
    }, {
      "is-display-only": _ctx.state.isDisplayOnly
    }]],
    role: "radio",
    "aria-checked": _ctx.state.model === _ctx.label,
    "aria-disabled": _ctx.state.isDisabled,
    tabindex: "-1",
    onKeydown: _cache[6] || (_cache[6] = withKeys(withModifiers(function($event) {
      return _ctx.state.model = _ctx.state.isDisabled ? _ctx.state.model : _ctx.label;
    }, ["stop", "prevent"]), ["space"]))
  }, _ctx.a(_ctx.$attrs, ["class", "style", "onClick"], true)), [createElementVNode(
    "span",
    {
      class: normalizeClass(["tiny-radio__input", {
        "is-disabled": _ctx.state.isDisabled,
        "is-checked": _ctx.state.model === _ctx.label
      }])
    },
    [createElementVNode("span", {
      class: "tiny-radio__inner",
      tabindex: _ctx.tabindex
    }, null, 8, _hoisted_2), withDirectives(createElementVNode("input", {
      ref: "radio",
      class: "tiny-radio__original",
      value: _ctx.label,
      type: "radio",
      "aria-hidden": "true",
      "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.model = $event;
      }),
      onFocus: _cache[1] || (_cache[1] = function($event) {
        return _ctx.state.focus = true;
      }),
      onBlur: _cache[2] || (_cache[2] = function($event) {
        return _ctx.state.focus = false;
      }),
      onChange: _cache[3] || (_cache[3] = function() {
        return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
      }),
      name: _ctx.name,
      disabled: _ctx.state.isDisabled,
      onClick: _cache[4] || (_cache[4] = withModifiers(function() {
      }, ["stop"]))
    }, null, 40, _hoisted_3), [[vModelRadio, _ctx.state.model]])],
    2
    /* CLASS */
  ), createElementVNode(
    "span",
    {
      class: "tiny-radio__label",
      onKeydown: _cache[5] || (_cache[5] = withModifiers(function() {
      }, ["stop"]))
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.text || _ctx.label),
        1
        /* TEXT */
      )];
    })],
    32
    /* NEED_HYDRATION */
  )], 16, _hoisted_1)) : createCommentVNode("v-if", true);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
