// @opentiny/vue-popover/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/popover/vue";
import { iconClose } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createVNode, Transition, withCtx, withDirectives, createElementVNode, normalizeStyle, renderSlot, toDisplayString, createCommentVNode, createTextVNode, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "title": "text-color-icon-primary font-bold text-xs",
  "popper": "sm:leading-5 sm:absolute sm:px-4 sm:py-3 sm:z-[2000] sm:text-justify sm:text-xs break-all sm:max-w-[theme(spacing.96)] sm:shadow-md sm:bg-color-bg-1 sm:rounded sm:[&[x-placement^=top]]:mb-2.5 sm:[&[x-placement^=bottom]]:mt-2.5 sm:[&[x-placement^=right]]:ml-2.5 sm:[&[x-placement^=left]]:mr-2.5",
  "arrow": 'absolute block w-0 h-0 border-[0.375rem] border-transparent border-solid after:absolute after:block after:w-0 after:h-0 after:border-[0.3125rem] after:border-transparent after:border-solid after:content-[""]',
  "placement-top": "-bottom-3 border-t-color-bg-1 drop-shadow-[0_2px_2px_rgba(0,0,0,0.08)] border-b-w-0 after:-bottom-[0.3125rem] after:-ml-[0.3125rem] after:border-t-color-bg-1 after:border-b-w-0",
  "placement-bottom": "-top-3 border-t-w-0 border-b-color-bg-1 drop-shadow-[0_-2px_2px_rgba(0,0,0,0.08)] after:-top-[0.3125rem] after:-ml-[0.3125rem] after:border-t-w-0 after:border-b-color-bg-1",
  "placement-right": "-left-3 border-r-color-bg-1 drop-shadow-[2px_0_2px_rgba(0,0,0,0.08)] border-l-w-0 after:-bottom-[0.3125rem] after:-left-[0.3125rem] after:border-r-color-bg-1 after:border-l-w-0",
  "placement-left": "-right-3 border-l-color-bg-1 drop-shadow-[-2px_0px_2px_rgba(0,0,0,0.08)] border-r-w-0 after:-bottom-[0.375rem] after:-right-[0.3125rem] after:-ml-[0.3125rem] after:border-l-color-bg-1 after:border-r-w-0"
};
var _sfc_main = defineComponent({
  inheritAttrs: false,
  name: $prefix + "Popover",
  components: {
    IconClose: iconClose()
  },
  emits: ["update:modelValue", "hide", "show", "after-enter", "after-leave", "created"],
  props: _extends({}, $props, {
    appendToBody: {
      type: Boolean,
      default: true
    },
    arrowOffset: {
      type: Number,
      default: 0
    },
    boundariesPadding: {
      type: Number,
      default: 5
    },
    closeDelay: {
      type: Number,
      default: 200
    },
    content: String,
    disabled: Boolean,
    modelValue: Boolean,
    offset: {
      default: 0
    },
    openDelay: {
      type: Number,
      default: 0
    },
    placement: {
      type: String,
      default: "bottom"
    },
    popper: {},
    popperClass: String,
    popperOptions: {
      type: Object,
      default: function _default() {
        return {
          gpuAcceleration: false
        };
      }
    },
    reference: {},
    tabindex: {
      type: Number,
      default: 0
    },
    title: String,
    transformOrigin: {
      type: [Boolean, String],
      default: true
    },
    transition: {
      type: String,
      default: "fade-in-linear"
    },
    trigger: {
      type: String,
      default: "click",
      validator: function validator(value) {
        return ~["click", "focus", "hover", "manual"].indexOf(value);
      }
    },
    visibleArrow: {
      default: true
    },
    genArrowByHtml: {
      type: Boolean,
      default: function _default2() {
        return true;
      }
    },
    width: {}
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["id", "aria-hidden"];
var _hoisted_2 = {
  class: "hidden sm:block",
  "data-tag": "tiny-popover-body"
};
var _hoisted_3 = ["textContent"];
var _hoisted_4 = {
  "data-tag": "tiny-popover-body",
  class: /* @__PURE__ */ normalizeClass(["sm:hidden block fixed inset-0 text-sm text-color-text-primary transition-top ease-out duration-300", "before:content-[''] before:fixed before:inset-0 before:w-full before:h-full before:opacity-30 before:bg-black"])
};
var _hoisted_5 = {
  class: /* @__PURE__ */ normalizeClass(["absolute bottom-0 top-auto left-1/2 -translate-x-1/2 w-full max-h-full p-3 pointer-events-auto"])
};
var _hoisted_6 = {
  class: "flex flex-col w-full h-full rounded-lg overflow-hidden shadow-xl bg-color-bg-1"
};
var _hoisted_7 = {
  "data-tag": "tiny-popover-header",
  class: "flex items-center leading-6 text-sm px-6 pt-5"
};
var _hoisted_8 = {
  class: "flex-auto text-base truncate font-semibold"
};
var _hoisted_9 = {
  "data-tag": "tiny-popover-content",
  class: "flex flex-auto leading-5 overflow-auto mx-6 mt-4 mb-5"
};
var _hoisted_10 = {
  key: 0,
  "data-tag": "tiny-popover-footer",
  class: "w-full py-2.5 px-6 border-t-0.5 border-color-border-separator text-center"
};
var _hoisted_11 = {
  "data-tag": "tiny-popover-reference",
  ref: "wrapper"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createElementBlock(
    "span",
    {
      "data-tag": "tiny-popover",
      ref: "root",
      class: normalizeClass(_ctx.$attrs.class)
    },
    [createVNode(Transition, {
      name: _ctx.transition,
      onAfterEnter: _ctx.handleAfterEnter,
      onAfterLeave: _ctx.handleAfterLeave,
      persisted: ""
    }, {
      default: withCtx(function() {
        return [withDirectives(createElementVNode("div", {
          "data-tag": "tiny-popover-content",
          class: normalizeClass([_ctx.gcls("popper"), _ctx.popperClass]),
          ref: "popper",
          style: normalizeStyle({
            width: _ctx.width + "px"
          }),
          role: "tooltip",
          id: _ctx.state.tooltipId,
          "aria-hidden": _ctx.disabled || !_ctx.state.showPopper ? "true" : "false"
        }, [createElementVNode("div", _hoisted_2, [_ctx.slots.header ? renderSlot(_ctx.$slots, "header", {
          key: 0
        }) : _ctx.title ? (openBlock(), createElementBlock("div", {
          key: 1,
          class: normalizeClass([_ctx.gcls("title")]),
          textContent: toDisplayString(_ctx.title)
        }, null, 10, _hoisted_3)) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default", {}, function() {
          return [createTextVNode(
            toDisplayString(_ctx.content),
            1
            /* TEXT */
          )];
        }), renderSlot(_ctx.$slots, "footer"), _ctx.visibleArrow ? (openBlock(), createElementBlock(
          "div",
          {
            key: 2,
            "data-tag": "tiny-popover-visible",
            "x-arrow": "",
            class: normalizeClass([_ctx.gcls("arrow"), _ctx.gcls("placement-" + _ctx.state.xPlacement.split("-")[0])])
          },
          null,
          2
          /* CLASS */
        )) : createCommentVNode("v-if", true)]), createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [createElementVNode(
          "span",
          _hoisted_8,
          toDisplayString(_ctx.title || _ctx.t("ui.alert.title")),
          1
          /* TEXT */
        ), createVNode(_component_icon_close, {
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.state.showPopper = false;
          })
        })]), createElementVNode("div", _hoisted_9, [renderSlot(_ctx.$slots, "default", {}, function() {
          return [createTextVNode(
            toDisplayString(_ctx.content),
            1
            /* TEXT */
          )];
        })]), _ctx.slots.footer ? (openBlock(), createElementBlock("div", _hoisted_10, [renderSlot(_ctx.$slots, "footer")])) : createCommentVNode("v-if", true)])])])], 14, _hoisted_1), [[vShow, !_ctx.disabled && _ctx.state.showPopper]])];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["name", "onAfterEnter", "onAfterLeave"]), createElementVNode(
      "span",
      _hoisted_11,
      [renderSlot(_ctx.$slots, "reference")],
      512
      /* NEED_PATCH */
    )],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
