// @opentiny/vue-floatbar/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/floatbar/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { openBlock, createElementBlock, normalizeStyle, renderSlot, createElementVNode, Fragment, renderList, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["position", "data"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "navs"
};
var _hoisted_2 = ["href", "target"];
var _hoisted_3 = {
  key: 1,
  class: "nav-text"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-float-bar tiny-toolbar",
      style: normalizeStyle(_ctx.position)
    },
    [renderSlot(_ctx.$slots, "default", {
      slotScope: {
        data: _ctx.state.data
      }
    }, function() {
      return [createElementVNode("ul", _hoisted_1, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.data, function(item, index) {
          return openBlock(), createElementBlock("li", {
            key: index,
            class: "nav-item"
          }, [item.url ? (openBlock(), createElementBlock("a", {
            key: 0,
            href: item.url,
            target: item.target,
            rel: "noopener noreferrer",
            class: "nav-link"
          }, toDisplayString(item.title), 9, _hoisted_2)) : (openBlock(), createElementBlock(
            "span",
            _hoisted_3,
            toDisplayString(item.title),
            1
            /* TEXT */
          ))]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])];
    })],
    4
    /* STYLE */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
