// @opentiny/vue-row/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/row/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Row = defineComponent({
  name: $prefix + "Row",
  componentName: "Row",
  props: _extends({}, $props, {
    flex: Boolean,
    gutter: {
      type: Number,
      default: 0,
      validator: function validator(value) {
        return value >= 0;
      }
    },
    justify: {
      type: String,
      default: "start",
      validator: function validator2(value) {
        return Boolean(~["start", "center", "end", "space-between", "space-around"].indexOf(value));
      }
    },
    align: {
      type: String,
      default: "top",
      validator: function validator3(value) {
        return Boolean(~["top", "middle", "bottom"].indexOf(value));
      }
    },
    order: {
      type: String,
      validator: function validator4(value) {
        return Boolean(~["asc", "des"].indexOf(value));
      }
    },
    tag: {
      type: String,
      default: "div"
    },
    noSpace: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Row.install = function(Vue) {
  Vue.component(Row.name, Row);
};
Row.version = version;
export {
  Row as default
};
