// @opentiny/vue-hrapprover/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/hrapprover/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Dept from "@opentiny/vue-dept";
import Loading from "@opentiny/vue-loading";
import { message } from "@opentiny/vue-modal";
import Radio from "@opentiny/vue-radio";
import { t } from "@opentiny/vue-locale";
import { resolveComponent, resolveDirective, openBlock, createBlock, withCtx, withDirectives, createElementBlock, createElementVNode, toDisplayString, Fragment, renderList, createVNode, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  inheritAttrs: false,
  props: [].concat(props, ["modelValue", "disabled", "approvalPerson", "category", "fetchDeptList", "fetchHrapprover", "fetchDeptByValue", "fetchDept", "title"]),
  components: {
    Dept,
    Radio
  },
  directives: {
    loading: Loading.directive
  },
  provide: function provide() {
    return {
      hrapprover: this
    };
  },
  setup: function setup(props2, context) {
    return Object.assign(_setup({
      props: props2,
      context,
      renderless,
      api
    }), {
      showModal: function showModal() {
        message({
          message: t("ui.hrapprover.noselected"),
          status: "error"
        });
      }
    });
  }
});
var _hoisted_1 = {
  class: "tiny-hrapprover__list"
};
var _hoisted_2 = {
  key: 0,
  width: "100%"
};
var _hoisted_3 = /* @__PURE__ */ createElementVNode(
  "th",
  {
    width: "50px"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_4 = {
  style: {
    "text-align": "center"
  }
};
var _hoisted_5 = {
  key: 1
};
var _hoisted_6 = {
  class: "tiny-hrapprover__error"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_radio = resolveComponent("radio");
  var _component_dept = resolveComponent("dept");
  var _directive_loading = resolveDirective("loading");
  return openBlock(), createBlock(_component_dept, {
    ref: "dept",
    class: "tiny-hrapprover",
    modelValue: _ctx.state.dept,
    "onUpdate:modelValue": _cache[1] || (_cache[1] = function($event) {
      return _ctx.state.dept = $event;
    }),
    "fetch-dept-list": _ctx.fetchDeptList,
    "fetch-dept-by-value": _ctx.fetchDeptByValue,
    "fetch-dept": _ctx.fetchDept,
    onClose: _ctx.resetPerson,
    onConfirm: _ctx.hrChange,
    disabled: _ctx.disabled,
    onSelected: _ctx.selectedDept,
    title: _ctx.title,
    onCancel: _ctx.cancel
  }, {
    hrapprover: withCtx(function() {
      return [withDirectives((openBlock(), createElementBlock("div", _hoisted_1, [_ctx.state.hrList.length ? (openBlock(), createElementBlock("table", _hoisted_2, [createElementVNode("thead", null, [createElementVNode("tr", null, [_hoisted_3, createElementVNode(
        "th",
        null,
        toDisplayString(_ctx.t("ui.hrapprover.approver")),
        1
        /* TEXT */
      ), createElementVNode(
        "th",
        null,
        toDisplayString(_ctx.t("ui.hrapprover.remark")),
        1
        /* TEXT */
      )])]), createElementVNode("tbody", null, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.hrList, function(hr, index) {
          return openBlock(), createElementBlock("tr", {
            key: index
          }, [createElementVNode("td", _hoisted_4, [createVNode(_component_radio, {
            modelValue: _ctx.state.approvalPerson,
            "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
              return _ctx.state.approvalPerson = $event;
            }),
            label: hr.approval_person
          }, {
            default: withCtx(function() {
              return [createTextVNode(" ")];
            }),
            _: 2
            /* DYNAMIC */
          }, 1032, ["modelValue", "label"])]), createElementVNode(
            "td",
            null,
            toDisplayString(hr.approval_person),
            1
            /* TEXT */
          ), createElementVNode(
            "td",
            null,
            toDisplayString(hr.remarks),
            1
            /* TEXT */
          )]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])])) : (openBlock(), createElementBlock("div", _hoisted_5, [createElementVNode(
        "div",
        _hoisted_6,
        toDisplayString(_ctx.t("ui.hrapprover.noapprover")),
        1
        /* TEXT */
      )]))])), [[_directive_loading, _ctx.state.loading]])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["modelValue", "fetch-dept-list", "fetch-dept-by-value", "fetch-dept", "onClose", "onConfirm", "disabled", "onSelected", "title", "onCancel"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
