// @opentiny/vue-split/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/split/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconLeft, IconRight } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, normalizeStyle, renderSlot, setBlockTracking, Fragment, renderList, createCommentVNode, createVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconSplitLeft: IconLeft(),
    // 暂时使用tiny-vue的图标代替。 备注：这个图标在简易模式也看不到
    IconSplitRight: IconRight()
  },
  emits: ["moving", "mousemove", "mouseup", "moveend", "movestart", "update:modelValue", "left-top-click", "right-bottom-click"],
  props: [].concat(props, ["modelValue", "mode", "leftTopMin", "rightBottomMin", "disabled", "triggerSimple", "collapseLeftTop", "collapseRightBottom", "threeAreas"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_split_left = resolveComponent("icon-split-left");
  var _component_icon_split_right = resolveComponent("icon-split-right");
  return openBlock(), createElementBlock(
    "div",
    {
      ref: "outerWrapper",
      class: normalizeClass(_ctx.state.wrapperClasses)
    },
    [_ctx.state.isHorizontal ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: normalizeClass(_ctx.state.prefix + "-horizontal")
      },
      [createElementVNode(
        "div",
        {
          style: normalizeStyle({
            right: _ctx.state.anotherOffset + "%",
            width: _ctx.state.anotherOffset != 100 && _ctx.state.isThreeAreas ? _ctx.state.leftTopPane + "px" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "left-pane"])
        },
        [renderSlot(_ctx.$slots, "left")],
        6
        /* CLASS, STYLE */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass([_ctx.state.prefix + "-trigger-con", _ctx.state.prefix + "-trigger-con-vertical", !_ctx.state.dragable ? _ctx.state.prefix + "-trigger-con-disabled" : "", _ctx.state.isMoving ? _ctx.state.prefix + "-trigger-con-drag" : "", _ctx.state.offset === 0 && _ctx.state.triggerSimple ? _ctx.state.prefix + "-trigger-con-left-active" : "", _ctx.state.triggerSimple ? _ctx.state.prefix + "-trigger-con-simple" : ""]),
          style: normalizeStyle({
            left: _ctx.state.isThreeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%"
          }),
          onMousedown: _cache[5] || (_cache[5] = function() {
            return _ctx.handleMousedown && _ctx.handleMousedown.apply(_ctx, arguments);
          })
        },
        [renderSlot(_ctx.$slots, "trigger", {}, function() {
          return [createElementVNode(
            "div",
            {
              class: normalizeClass([_ctx.state.prefix + "-trigger", _ctx.state.prefix + "-trigger-vertical"])
            },
            [!_ctx.state.triggerSimple ? (openBlock(), createElementBlock(
              "div",
              {
                key: 0,
                class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "vertical"])
              },
              [_cache[0] || (setBlockTracking(-1), _cache[0] = (openBlock(), createElementBlock(
                Fragment,
                null,
                renderList(3, function(i) {
                  return createElementVNode(
                    "i",
                    {
                      class: normalizeClass(_ctx.state.prefix + "-trigger-bar"),
                      key: "trigger-" + i
                    },
                    null,
                    2
                    /* CLASS */
                  );
                }),
                64
                /* STABLE_FRAGMENT */
              )), setBlockTracking(1), _cache[0])],
              2
              /* CLASS */
            )) : createCommentVNode("v-if", true)],
            2
            /* CLASS */
          )];
        }), _ctx.state.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-left-button", _ctx.state.offset === 100 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 100 || _ctx.state.collapseLeftTop ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[1] || (_cache[1] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[2] || (_cache[2] = function() {
              return _ctx.buttonLeftTopClick && _ctx.buttonLeftTopClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_split_left, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true), _ctx.state.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-right-button", _ctx.state.offset === 0 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 0 || _ctx.state.collapseRightBottom ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[3] || (_cache[3] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[4] || (_cache[4] = function() {
              return _ctx.buttonRightBottomClick && _ctx.buttonRightBottomClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_split_right, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true)],
        38
        /* CLASS, STYLE, NEED_HYDRATION */
      ), createElementVNode(
        "div",
        {
          style: normalizeStyle({
            left: _ctx.state.isThreeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%",
            width: _ctx.state.isThreeAreas ? "calc(100% - " + _ctx.state.leftTopPane + "px)" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "right-pane"])
        },
        [renderSlot(_ctx.$slots, "right")],
        6
        /* CLASS, STYLE */
      )],
      2
      /* CLASS */
    )) : (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        class: normalizeClass(_ctx.state.prefix + "-vertical")
      },
      [createElementVNode(
        "div",
        {
          style: normalizeStyle({
            bottom: _ctx.state.anotherOffset + "%",
            height: _ctx.state.anotherOffset != 100 ? _ctx.state.leftTopPane + "px" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "top-pane"])
        },
        [renderSlot(_ctx.$slots, "top")],
        6
        /* CLASS, STYLE */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass([_ctx.state.prefix + "-trigger-con", _ctx.state.prefix + "-trigger-con-horizontal", !_ctx.state.dragable ? _ctx.state.prefix + "-trigger-con-disabled" : "", _ctx.state.isMoving ? _ctx.state.prefix + "-trigger-con-drag" : "", _ctx.state.offset === 0 && _ctx.state.triggerSimple ? _ctx.state.prefix + "-trigger-con-top-active" : "", _ctx.state.triggerSimple ? _ctx.state.prefix + "-trigger-con-simple" : ""]),
          style: normalizeStyle({
            top: _ctx.state.isThreeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%"
          }),
          onMousedown: _cache[11] || (_cache[11] = function() {
            return _ctx.handleMousedown && _ctx.handleMousedown.apply(_ctx, arguments);
          })
        },
        [renderSlot(_ctx.$slots, "trigger", {}, function() {
          return [createElementVNode(
            "div",
            {
              class: normalizeClass([_ctx.state.prefix + "-trigger", _ctx.state.prefix + "-trigger-horizontal"])
            },
            [!_ctx.state.triggerSimple ? (openBlock(), createElementBlock(
              "div",
              {
                key: 0,
                class: normalizeClass([_ctx.state.prefix + "-trigger-bar-con", "horizontal"])
              },
              [_cache[6] || (setBlockTracking(-1), _cache[6] = (openBlock(), createElementBlock(
                Fragment,
                null,
                renderList(3, function(i) {
                  return createElementVNode(
                    "i",
                    {
                      class: normalizeClass(_ctx.state.prefix + "-trigger-bar"),
                      key: "trigger-" + i
                    },
                    null,
                    2
                    /* CLASS */
                  );
                }),
                64
                /* STABLE_FRAGMENT */
              )), setBlockTracking(1), _cache[6])],
              2
              /* CLASS */
            )) : createCommentVNode("v-if", true)],
            2
            /* CLASS */
          )];
        }), _ctx.state.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-top-button", _ctx.state.offset === 100 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 100 || _ctx.state.collapseLeftTop ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[7] || (_cache[7] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[8] || (_cache[8] = function() {
              return _ctx.buttonLeftTopClick && _ctx.buttonLeftTopClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_split_left, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true), _ctx.state.triggerSimple ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            class: normalizeClass([_ctx.state.prefix + "-trigger-button", _ctx.state.prefix + "-trigger-bottom-button", _ctx.state.offset === 0 ? _ctx.state.prefix + "-trigger-button-active" : "", _ctx.state.offset === 0 || _ctx.state.collapseRightBottom ? _ctx.state.prefix + "-trigger-button-show" : ""]),
            onMousedown: _cache[9] || (_cache[9] = function() {
              return _ctx.buttonMousedown && _ctx.buttonMousedown.apply(_ctx, arguments);
            }),
            onClick: _cache[10] || (_cache[10] = function() {
              return _ctx.buttonRightBottomClick && _ctx.buttonRightBottomClick.apply(_ctx, arguments);
            })
          },
          [createVNode(_component_icon_split_right, {
            class: "tiny-svg-size"
          })],
          34
          /* CLASS, NEED_HYDRATION */
        )) : createCommentVNode("v-if", true)],
        38
        /* CLASS, STYLE, NEED_HYDRATION */
      ), createElementVNode(
        "div",
        {
          style: normalizeStyle({
            top: _ctx.state.isThreeAreas ? _ctx.state.leftTopPane + "px" : _ctx.state.offset + "%",
            height: _ctx.state.isThreeAreas ? "calc(100% - " + _ctx.state.leftTopPane + "px)" : ""
          }),
          class: normalizeClass([_ctx.state.paneClasses, "bottom-pane"])
        },
        [renderSlot(_ctx.$slots, "bottom")],
        6
        /* CLASS, STYLE */
      )],
      2
      /* CLASS */
    ))],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
