// @opentiny/vue-time-picker-mobile/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/time-picker-mobile/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { resolveComponent, withDirectives, openBlock, createElementBlock, createVNode, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TimePickerMobile",
  components: {
    TinyCascaderSelect: CascaderSelect
  },
  props: _extends({}, $props, {
    modelValue: Array,
    visible: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-time-picker-mobile"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_cascader_select = resolveComponent("tiny-cascader-select");
  return withDirectives((openBlock(), createElementBlock(
    "div",
    _hoisted_1,
    [createVNode(_component_tiny_cascader_select, {
      modelValue: _ctx.state.dateArr,
      "onUpdate:modelValue": [_cache[0] || (_cache[0] = function($event) {
        return _ctx.state.dateArr = $event;
      }), _ctx.confirm],
      options: _ctx.state.optionList,
      visible: _ctx.state.visible,
      "onUpdate:visible": _ctx.updateVisible
    }, null, 8, ["modelValue", "options", "visible", "onUpdate:visible", "onUpdate:modelValue"])],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.visible]]);
}
var TimePickerMobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
TimePickerMobile.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
TimePickerMobile.install = function(Vue) {
  Vue.component(TimePickerMobile.name, TimePickerMobile);
};
TimePickerMobile.version = version;
export {
  TimePickerMobile as default
};
