// @opentiny/vue-dept/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/dept/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Dept = defineComponent({
  name: $prefix + "Dept",
  props: _extends({}, $props, {
    modelValue: String,
    disabled: {
      type: Boolean,
      default: false
    },
    autoSelect: {
      type: Boolean,
      default: true
    },
    size: String,
    fetchDeptList: Function,
    fetchDeptByValue: Function,
    fetchDept: Function,
    beforeConfirm: Function,
    title: {
      type: String,
      default: ""
    },
    displayOnly: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Dept.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Dept.install = function(Vue) {
  Vue.component(Dept.name, Dept);
};
Dept.version = version;
export {
  Dept as default
};
