// @opentiny/vue-multi-select/lib/mobile.js
import { defineComponent, $prefix, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/multi-select/vue";
import { IconChevronDown } from "@opentiny/vue-icon";
import Button from "@opentiny/vue-button";
import Wheel from "@opentiny/vue-wheel";
import { resolveComponent, openBlock, createElementBlock, createElementVNode, Fragment, renderList, normalizeClass, toDisplayString, normalizeStyle, createVNode, createCommentVNode, withCtx, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  name: $prefix + "MultiSelect",
  components: {
    IconChevronDown: IconChevronDown(),
    TinyButton: Button,
    TinyWheel: Wheel
  },
  props: {
    dataSource: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    defaultSelectedArray: {
      type: Array,
      default: function _default2() {
        return [];
      }
    }
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-multi-select"
};
var _hoisted_2 = {
  class: "tiny-mobile-multi-select__header"
};
var _hoisted_3 = {
  class: "tiny-mobile-multi-select__header__flexCenter",
  ref: "headerBox"
};
var _hoisted_4 = {
  class: "tiny-mobile-multi-select__header__calc"
};
var _hoisted_5 = {
  key: 0
};
var _hoisted_6 = {
  key: 1
};
var _hoisted_7 = ["onClick"];
var _hoisted_8 = {
  key: 0
};
var _hoisted_9 = {
  key: 1
};
var _hoisted_10 = {
  key: 1,
  class: "tiny-mobile-multi-select__footer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _ctx$dataSource$_ctx$, _ctx$dataSource$_ctx$2, _ctx$dataSource$_ctx$3;
  var _component_IconChevronDown = resolveComponent("IconChevronDown");
  var _component_tiny_wheel = resolveComponent("tiny-wheel");
  var _component_tiny_button = resolveComponent("tiny-button");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [createElementVNode(
    "div",
    _hoisted_3,
    [createElementVNode("div", _hoisted_4, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.dataSource, function(item, index) {
        var _ctx$state$headerInfo, _ctx$state$headerInfo2, _ctx$state$headerInfo3, _ctx$state$headerInfo4, _ctx$state$headerInfo5;
        return openBlock(), createElementBlock("div", {
          key: index,
          class: "tiny-mobile-multi-select__header__item",
          ref_for: true,
          ref: "label"
        }, [createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-multi-select__header__label", (_ctx$state$headerInfo = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo.isSelected ? "tiny-mobile-multi-select__header__active" : ""])
          },
          [!((_ctx$state$headerInfo2 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo2.isSelected) ? (openBlock(), createElementBlock(
            "span",
            _hoisted_5,
            toDisplayString(item.title),
            1
            /* TEXT */
          )) : (openBlock(), createElementBlock(
            "span",
            _hoisted_6,
            toDisplayString((_ctx$state$headerInfo3 = _ctx.state.headerInfo[index]) == null ? void 0 : _ctx$state$headerInfo3.title),
            1
            /* TEXT */
          ))],
          2
          /* CLASS */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-multi-select__header__icon", (_ctx$state$headerInfo4 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo4.isSelected ? "tiny-mobile-multi-select__header__active" : ""]),
            style: normalizeStyle({
              transform: (_ctx$state$headerInfo5 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo5.isUP ? "rotate(180deg)" : "none"
            })
          },
          [createVNode(_component_IconChevronDown)],
          6
          /* CLASS, STYLE */
        )]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))]), (openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.dataSource, function(item, index) {
        var _ctx$state$headerInfo6, _ctx$state$headerInfo7, _ctx$state$headerInfo8, _ctx$state$headerInfo9, _ctx$state$headerInfo10;
        return openBlock(), createElementBlock("div", {
          key: index,
          class: "tiny-mobile-multi-select__header__item",
          onClick: function onClick($event) {
            return _ctx.handleClick(index);
          },
          style: normalizeStyle(_ctx.state.labelsStyle[index])
        }, [createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-multi-select__header__label", (_ctx$state$headerInfo6 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo6.isSelected ? "tiny-mobile-multi-select__header__active" : ""])
          },
          [!((_ctx$state$headerInfo7 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo7.isSelected) ? (openBlock(), createElementBlock(
            "span",
            _hoisted_8,
            toDisplayString(item.title),
            1
            /* TEXT */
          )) : (openBlock(), createElementBlock(
            "span",
            _hoisted_9,
            toDisplayString((_ctx$state$headerInfo8 = _ctx.state.headerInfo[index]) == null ? void 0 : _ctx$state$headerInfo8.title),
            1
            /* TEXT */
          ))],
          2
          /* CLASS */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-multi-select__header__icon", (_ctx$state$headerInfo9 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo9.isSelected ? "tiny-mobile-multi-select__header__active" : ""]),
            style: normalizeStyle({
              transform: (_ctx$state$headerInfo10 = _ctx.state.headerInfo[index]) != null && _ctx$state$headerInfo10.isUP ? "rotate(180deg)" : "none"
            })
          },
          [createVNode(_component_IconChevronDown)],
          6
          /* CLASS, STYLE */
        )], 12, _hoisted_7);
      }),
      128
      /* KEYED_FRAGMENT */
    ))],
    512
    /* NEED_PATCH */
  )]), _ctx.state.showWheel ? (openBlock(), createElementBlock(
    "div",
    {
      key: 0,
      class: normalizeClass(["tiny-mobile-multi-select__content", !((_ctx$dataSource$_ctx$ = _ctx.dataSource[_ctx.state.headerIndex]) != null && _ctx$dataSource$_ctx$.hasFooter) ? "noFooter" : ""])
    },
    [createVNode(_component_tiny_wheel, {
      dataSource: _ctx.state.wheelData,
      defaultSelectedIndexs: _ctx.state.defaultSelectedIndexs,
      hasFooter: (_ctx$dataSource$_ctx$2 = _ctx.dataSource[_ctx.state.headerIndex]) == null ? void 0 : _ctx$dataSource$_ctx$2.hasFooter,
      onChange: _ctx.wheelChange,
      onClickWheelItem: _ctx.clickWheelItem
    }, null, 8, ["dataSource", "defaultSelectedIndexs", "hasFooter", "onChange", "onClickWheelItem"])],
    2
    /* CLASS */
  )) : createCommentVNode("v-if", true), _ctx.state.showWheel && (_ctx$dataSource$_ctx$3 = _ctx.dataSource[_ctx.state.headerIndex]) != null && _ctx$dataSource$_ctx$3.hasFooter ? (openBlock(), createElementBlock("div", _hoisted_10, [createVNode(_component_tiny_button, {
    round: "",
    size: "large",
    onClick: _ctx.reset
  }, {
    default: withCtx(function() {
      return [createTextVNode(
        toDisplayString(_ctx.t("ui.base.reset")),
        1
        /* TEXT */
      )];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onClick"]), createVNode(_component_tiny_button, {
    round: "",
    size: "large",
    type: "primary",
    onClick: _ctx.confirm
  }, {
    default: withCtx(function() {
      return [createTextVNode(
        toDisplayString(_ctx.t("ui.button.confirm")),
        1
        /* TEXT */
      )];
    }),
    _: 1
    /* STABLE */
  }, 8, ["onClick"])])) : createCommentVNode("v-if", true)]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
