// @opentiny/vue-detail-page/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/detail-page/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  LOCAL_TIPS: "ui.detailpage.localTips",
  LOCAL_TITLE: "ui.detailpage.localTitle",
  VALUE_TITLE: "ui.detailpage.valueTitle",
  LABEL_TITLE: "ui.detailpage.labelTitle",
  DIALOG_TITLE: "ui.detailpage.dialogTitle",
  SAVE_BUTTON_TEXT: "ui.detailpage.saveButtonText",
  CANCEL_BUTTTON_TEXT: "ui.detailpage.cancelButtonText"
};
var DetailPage = defineComponent({
  name: $prefix + "DetailPage",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: Array,
    title: {
      type: String,
      default: ""
    },
    textSplit: {
      type: String,
      default: "---"
    },
    dialogTitle: {
      type: String,
      default: ""
    },
    labelTitle: {
      type: String,
      default: ""
    },
    tips: {
      type: String,
      default: ""
    },
    valueTitle: {
      type: String,
      default: ""
    },
    saveButton: {
      type: String,
      default: ""
    },
    cancelButton: {
      type: String,
      default: ""
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
DetailPage.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
DetailPage.install = function(Vue) {
  Vue.component(DetailPage.name, DetailPage);
};
DetailPage.version = version;
export {
  DetailPage as default
};
