// @opentiny/vue-button-group/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/button-group/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var buttonGroupProps = _extends({}, $props, {
  size: String,
  data: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  plain: Boolean,
  modelValue: [String, Number],
  disabled: Boolean,
  valueField: {
    type: String,
    default: "value"
  },
  textField: {
    type: String,
    default: "text"
  },
  showMore: Number,
  showEdit: {
    type: Boolean,
    default: false
  },
  border: {
    type: Boolean,
    default: true
  }
});
var ButtonGroup = defineComponent({
  name: $prefix + "ButtonGroup",
  provide: function provide() {
    return {
      buttonGroup: this
    };
  },
  props: buttonGroupProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
ButtonGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ButtonGroup.install = function(Vue) {
  Vue.component(ButtonGroup.name, ButtonGroup);
};
ButtonGroup.version = version;
export {
  ButtonGroup as default
};
