// @opentiny/vue-toast/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props, createComponent } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
import "@opentiny/vue-theme-mobile/toast/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileTemplate;
};
var toastProps = _extends({}, $props, {
  type: {
    type: String,
    validator: function validator(value) {
      return Boolean(~["text", "correct", "error"].indexOf(value));
    }
  },
  zIndex: {
    type: Number,
    default: 100
  },
  text: {
    type: String,
    default: "text"
  },
  time: {
    type: [String, Number],
    default: 2e3
  },
  timeout: {
    type: Function,
    default: null
  }
});
var Toast$1 = defineComponent({
  name: $prefix + "Toast",
  emits: ["timeout"],
  props: toastProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var defaults = {
  type: "text"
};
var service = function service2(configs) {
  if (configs === void 0) {
    configs = {};
  }
  configs = _extends({}, defaults, configs);
  var parent = document.body;
  var instance = createComponent({
    component: Toast$1,
    propsData: {},
    el: document.createElement("div")
  });
  for (var key in configs) {
    if (Object.prototype.hasOwnProperty.call(configs, key)) {
      instance.state[key] = configs[key];
    }
  }
  parent.appendChild(instance.$el);
  setTimeout(function() {
    if (instance.state.timeout) {
      instance.state.timeout();
    }
    parent.removeChild(instance.$el);
  }, instance.state.time);
  return instance;
};
var version = "3.13.0";
var Toast = {
  install: function install() {
  },
  service,
  version
};
export {
  Toast as default
};
