// @opentiny/vue-toast/lib/mobile.js
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/toast/vue";
import "@opentiny/vue-theme-mobile/toast/index.css";
import { IconMobileErrorWhite, IconMobileSuccessWhite } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, createElementVNode, normalizeClass, createBlock, createCommentVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconMobileErrorWhite: IconMobileErrorWhite(),
    IconMobileSuccessWhite: IconMobileSuccessWhite()
  },
  props: [].concat(props, ["type", "zIndex", "text", "time", "timeout"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-toast"
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-mobile-toast-icon"
};
var _hoisted_3 = {
  class: "tiny-mobile-toast-text"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_mobile_error_white = resolveComponent("icon-mobile-error-white");
  var _component_icon_mobile_success_white = resolveComponent("icon-mobile-success-white");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      class: normalizeClass(["tiny-mobile-toast-content", "tiny-mobile-toast-content-" + _ctx.state.type])
    },
    [_ctx.state.type !== "text" ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.state.type === "error" ? (openBlock(), createBlock(_component_icon_mobile_error_white, {
      key: 0
    })) : createCommentVNode("v-if", true), _ctx.state.type === "correct" ? (openBlock(), createBlock(_component_icon_mobile_success_white, {
      key: 1
    })) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      _hoisted_3,
      toDisplayString(_ctx.state.text),
      1
      /* TEXT */
    )],
    2
    /* CLASS */
  )]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
