// @opentiny/vue-switch/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/switch/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { openBlock, createElementBlock, normalizeClass, withKeys, createElementVNode, renderSlot, createTextVNode, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "switch-default": "inline-block align-middle transition-all duration-200 ease-in-out rounded-full relative outline-0",
  "switch-small": "w-10 h-6 sm:w-9 sm:h-5",
  "switch-mini": "w-8 h-5 sm:w-7 sm:h-4",
  "switch-medium": "w-10 h-6",
  "switch-handle-default": 'before:content-[""] before:rounded-full before:bg-color-bg-1 before:absolute before:cursor-pointer before:transition-all before:duration-200 before:ease-in-out before:text-center before:top-0.5',
  "switch-handle-small": "before:w-5 before:h-5 before:sm:w-4 before:sm:h-4",
  "switch-handle-mini": "before:w-4 before:h-4 before:sm:w-3 before:sm:h-3",
  "switch-handle-medium": "before:w-5 before:h-5",
  "switch-true-disabled": "bg-color-brand-disabled cursor-not-allowed",
  "switch-false-disabled": "bg-color-border-separator cursor-not-allowed",
  "switch-true-not-disabled": "before:[&_span:nth-of-type(1)]:shadow-xsm bg-color-brand cursor-pointer",
  "switch-false-not-disabled": "before:[&_span:nth-of-type(1)]:shadow-xsm bg-color-icon-disabled cursor-pointer",
  "switch-handle-false-small": "before:left-0.5",
  "switch-handle-false-medium": "before:left-0.5",
  "switch-handle-false-mini": "before:left-0.5",
  "switch-handle-false-small-disabled": "before:left-0.5",
  "switch-handle-false-medium-disabled": "before:left-0.5",
  "switch-handle-false-mini-disabled": "before:left-0.5",
  "switch-handle-true-small": "before:left-4.5",
  "switch-handle-true-medium": "before:left-4.5",
  "switch-handle-true-mini": "before:left-3.5",
  "switch-handle-true-small-disabled": "before:left-4.5",
  "switch-handle-true-medium-disabled": "before:left-4.5",
  "switch-handle-true-mini-disabled": "before:left-3.5",
  "switch-inner-default": "absolute transition-all duration-200 ease-in-out text-color-bg-1 flex items-center justify-center top-0.5",
  "switch-inner-small": "w-5 h-5 sm:w-4 sm:h-4",
  "switch-inner-mini": "w-4 h-4 sm:w-3 sm:h-3",
  "switch-inner-medium": "w-5 h-5",
  "switch-inner-true-word": "left-px",
  "switch-inner-false-small-word": "left-4.5",
  "switch-inner-false-medium-word": "left-4.5",
  "switch-inner-false-mini-word": "left-3.5",
  "switch-inner-false-icon": "left-0.5",
  "switch-inner-true-small-icon": "left-4.5",
  "switch-inner-true-medium-icon": "left-4.5",
  "switch-inner-true-mini-icon": "left-3.5",
  "switch-icon-small": "w-4 h-4 sm:w-3 sm:h-3",
  "switch-icon-medium": "w-4 h-4",
  "switch-icon-mini": "w-3 h-3 sm:w-2 sm:h-2",
  "switch-icon-false-disabled": "fill-color-border-separator",
  "switch-icon-false-default": "fill-color-icon-disabled",
  "switch-icon-true-disabled": "fill-color-brand-disabled",
  "switch-icon-true-default": "fill-color-brand",
  "switch-loading-small": "w-5 h-5 sm:w-4 sm:h-4",
  "switch-loading-medium": "w-5 h-5",
  "switch-loading-mini": "w-4 h-4 sm:w-3 sm:h-3 "
};
var _sfc_main = defineComponent({
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["modelValue", "trueValue", "falseValue", "disabled", "size", "tabindex", "types"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["tabindex"];
var _hoisted_2 = {
  key: 0,
  "data-tag": "tiny-switch-word",
  class: "sm:text-xs text-sm"
};
var _hoisted_3 = {
  key: 1,
  "data-tag": "tiny-switch-icon"
};
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "path",
  {
    d: "M9.9 15.6l9.4-9.4c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4l-10 10.1c-.4.4-1 .4-1.4 0-.1 0-.1-.1-.2-.1l-5.8-5.8c-.4-.4-.4-1 0-1.4.4-.4 1-.4 1.4 0l5.2 5.2z"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_5 = [_hoisted_4];
var _hoisted_6 = /* @__PURE__ */ createElementVNode(
  "path",
  {
    d: "M13.4 12l6.3-6.3c.4-.4.4-1 0-1.4-.4-.4-1-.4-1.4 0L12 10.6 5.7 4.3c-.4-.4-1-.4-1.4 0-.4.4-.4 1 0 1.4l6.3 6.3-6.3 6.3c-.4.4-.4 1 0 1.4.2.2.4.3.7.3.3 0 .5-.1.7-.3l6.3-6.3 6.3 6.3c.2.2.4.3.7.3s.5-.1.7-.3c.4-.4.4-1 0-1.4L13.4 12z"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_7 = [_hoisted_6];
var _hoisted_8 = {
  key: 2,
  "data-tag": "tiny-switch-loading"
};
var _hoisted_9 = /* @__PURE__ */ createElementVNode(
  "path",
  {
    d: "M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z",
    transform: "rotate(275.098 75 75)"
  },
  [/* @__PURE__ */ createElementVNode("animateTransform", {
    attributeType: "xml",
    attributeName: "transform",
    type: "rotate",
    from: "0 25 25",
    to: "360 25 25",
    dur: "1s",
    repeatCount: "indefinite"
  })],
  -1
  /* HOISTED */
);
var _hoisted_10 = [_hoisted_9];
var _hoisted_11 = /* @__PURE__ */ createElementVNode(
  "path",
  {
    d: "M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z",
    transform: "rotate(275.098 75 75)"
  },
  [/* @__PURE__ */ createElementVNode("animateTransform", {
    attributeType: "xml",
    attributeName: "transform",
    type: "rotate",
    from: "0 25 25",
    to: "360 25 25",
    dur: "1s",
    repeatCount: "indefinite"
  })],
  -1
  /* HOISTED */
);
var _hoisted_12 = [_hoisted_11];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("span", {
    class: normalizeClass(_ctx.m(_ctx.gcls("switch-default"), _ctx.gcls("switch-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls((_ctx.state.currentValue === _ctx.trueValue ? "switch-true" : "switch-false") + (_ctx.disabled || _ctx.types === "loading" ? "-disabled" : "-not-disabled")))),
    "data-tag": "tiny-switch",
    tabindex: _ctx.tabindex,
    onClick: _cache[0] || (_cache[0] = function() {
      return _ctx.toggle && _ctx.toggle.apply(_ctx, arguments);
    }),
    onKeydown: _cache[1] || (_cache[1] = withKeys(function() {
      return _ctx.toggle && _ctx.toggle.apply(_ctx, arguments);
    }, ["space"]))
  }, [createElementVNode(
    "span",
    {
      "data-tag": "tiny-switch-handle",
      class: normalizeClass(_ctx.m(_ctx.gcls("switch-handle-default"), _ctx.gcls("switch-handle-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls(_ctx.state.currentValue === _ctx.trueValue ? "switch-handle-true-" + (_ctx.size ? _ctx.size : "small") + (_ctx.disabled || _ctx.types === "loading" ? "-disabled" : "") : "switch-handle-false-" + (_ctx.size ? _ctx.size : "small") + (_ctx.disabled || _ctx.types === "loading" ? "-disabled" : ""))))
    },
    null,
    2
    /* CLASS */
  ), createElementVNode(
    "span",
    {
      "data-tag": "tiny-switch-inner",
      class: normalizeClass(_ctx.m(_ctx.gcls("switch-inner-default"), _ctx.gcls("switch-inner-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls(_ctx.state.currentValue === _ctx.trueValue ? _ctx.types === "word" ? "switch-inner-true-word" : "switch-inner-true-" + (_ctx.size ? _ctx.size : "small") + "-icon" : _ctx.types === "word" ? "switch-inner-false-" + (_ctx.size ? _ctx.size : "small") + "-word" : "switch-inner-false-icon")))
    },
    [_ctx.types === "word" ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.state.currentValue === _ctx.trueValue ? renderSlot(_ctx.$slots, "open", {
      key: 0
    }, function() {
      return [createTextVNode("A")];
    }) : createCommentVNode("v-if", true), _ctx.state.currentValue === _ctx.falseValue ? renderSlot(_ctx.$slots, "close", {
      key: 1
    }, function() {
      return [createTextVNode("B")];
    }) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), _ctx.types === "icon" ? (openBlock(), createElementBlock("div", _hoisted_3, [_ctx.state.currentValue === _ctx.trueValue ? renderSlot(_ctx.$slots, "open", {
      key: 0
    }, function() {
      return [(openBlock(), createElementBlock(
        "svg",
        {
          viewBox: "0 0 24 24",
          class: normalizeClass([_ctx.gcls("switch-icon-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls("switch-icon-true-" + (_ctx.disabled ? "disabled" : "default"))])
        },
        _hoisted_5,
        2
        /* CLASS */
      ))];
    }) : createCommentVNode("v-if", true), _ctx.state.currentValue === _ctx.falseValue ? renderSlot(_ctx.$slots, "close", {
      key: 1
    }, function() {
      return [(openBlock(), createElementBlock(
        "svg",
        {
          viewBox: "0 0 24 24",
          class: normalizeClass([_ctx.gcls("switch-icon-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls("switch-icon-false-" + (_ctx.disabled ? "disabled" : "default"))])
        },
        _hoisted_7,
        2
        /* CLASS */
      ))];
    }) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), _ctx.types === "loading" ? (openBlock(), createElementBlock("div", _hoisted_8, [_ctx.state.currentValue === _ctx.trueValue ? renderSlot(_ctx.$slots, "open", {
      key: 0
    }, function() {
      return [(openBlock(), createElementBlock(
        "svg",
        {
          viewBox: "0 0 48 48",
          class: normalizeClass([_ctx.gcls("switch-loading-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls("switch-icon-true-disabled")])
        },
        _hoisted_10,
        2
        /* CLASS */
      ))];
    }) : createCommentVNode("v-if", true), _ctx.state.currentValue === _ctx.falseValue ? renderSlot(_ctx.$slots, "close", {
      key: 1,
      dataTag: "tiny-switch-close"
    }, function() {
      return [(openBlock(), createElementBlock(
        "svg",
        {
          viewBox: "0 0 48 48",
          class: normalizeClass([_ctx.gcls("switch-loading-" + (_ctx.size ? _ctx.size : "small")), _ctx.gcls("switch-icon-false-disabled")])
        },
        _hoisted_12,
        2
        /* CLASS */
      ))];
    }) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  )], 42, _hoisted_1);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
