// @opentiny/vue-switch/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/switch/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme/switch/index.css";
import { openBlock, createElementBlock, normalizeClass, withKeys, createElementVNode, renderSlot, createTextVNode, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["modelValue", "trueValue", "falseValue", "disabled", "mini", "tabindex", "showText", "beforeChange", "displayOnly"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["tabindex"];
var _hoisted_2 = {
  key: 0
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("span", {
    class: normalizeClass([_ctx.state.wrapClasses, _ctx.state.showText ? "tiny-switch__text" : ""]),
    tabindex: _ctx.tabindex,
    onClick: _cache[0] || (_cache[0] = function() {
      return _ctx.toggle && _ctx.toggle.apply(_ctx, arguments);
    }),
    onKeydown: _cache[1] || (_cache[1] = withKeys(function() {
      return _ctx.toggle && _ctx.toggle.apply(_ctx, arguments);
    }, ["space"]))
  }, [createElementVNode(
    "span",
    {
      class: normalizeClass(_ctx.state.innerClasses)
    },
    [!_ctx.mini && _ctx.state.showText ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.state.currentValue === _ctx.trueValue ? renderSlot(_ctx.$slots, "open", {
      key: 0
    }, function() {
      return [createTextVNode("ON")];
    }) : createCommentVNode("v-if", true), _ctx.state.currentValue === _ctx.falseValue ? renderSlot(_ctx.$slots, "close", {
      key: 1
    }, function() {
      return [createTextVNode("OFF")];
    }) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  )], 42, _hoisted_1);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
