// @opentiny/vue-collapse-item/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/collapse-item/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { IconArrowBottom } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, createElementVNode, normalizeClass, withKeys, withModifiers, renderSlot, createTextVNode, toDisplayString, createVNode, withCtx, withDirectives, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["title", "titleRight", "name", "disabled"]),
  components: {
    CollapseTransition,
    IconArrowBottom: IconArrowBottom()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "border-b-0.5 sm:border-b border-solid border-color-border-disabled last:border-0"
};
var _hoisted_2 = ["aria-expanded", "aria-controls", "aria-describedby"];
var _hoisted_3 = ["id", "tabindex"];
var _hoisted_4 = {
  class: "text-xs font-normal"
};
var _hoisted_5 = ["aria-hidden", "aria-labelledby", "id"];
var _hoisted_6 = {
  class: "pb-6 pt-0 px-4 sm:pt-0 sm:pr-0 sm:pl-4 sm:pb-4 text-xs text-color-text-primary leading-relaxed"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_arrow_bottom = resolveComponent("icon-arrow-bottom");
  var _component_collapse_transition = resolveComponent("collapse-transition");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode("div", {
    role: "tab",
    "aria-expanded": _ctx.state.isActive,
    "aria-controls": "tiny-collapse-content-" + _ctx.state.id,
    "aria-describedby": "tiny-collapse-content-" + _ctx.state.id
  }, [createElementVNode("div", {
    class: normalizeClass(["flex relative items-center h-12 sm:h-14 py-0 sm:pr-0 pl-4 sm:pl-4 pr-14 text-sm font-bold justify-between whitespace-nowrap", _ctx.disabled ? "text-color-text-disabled" : "text-color-text-primary [&:has(.peer:hover)_[role=title]]:text-color-brand"]),
    id: "tiny-collapse-head-" + _ctx.state.id,
    tabindex: _ctx.disabled ? void 0 : 0,
    onKeyup: _cache[2] || (_cache[2] = withKeys(withModifiers(function() {
      return _ctx.handleEnterClick && _ctx.handleEnterClick.apply(_ctx, arguments);
    }, ["stop"]), ["space", "enter"])),
    onFocus: _cache[3] || (_cache[3] = function() {
      return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
    }),
    onBlur: _cache[4] || (_cache[4] = function($event) {
      return _ctx.state.focusing = false;
    })
  }, [createElementVNode(
    "div",
    {
      class: normalizeClass(["whitespace-nowrap overflow-hidden overflow-ellipsis inline-block peer", [_ctx.disabled ? "cursor-not-allowed" : "cursor-pointer hover:text-color-brand"]]),
      role: "title",
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
      })
    },
    [renderSlot(_ctx.$slots, "title", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.title),
        1
        /* TEXT */
      )];
    })],
    2
    /* CLASS */
  ), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "title-right", {}, function() {
    return [createTextVNode(
      toDisplayString(_ctx.titleRight),
      1
      /* TEXT */
    )];
  })]), createElementVNode(
    "div",
    {
      class: normalizeClass(["absolute sm:left-0 right-2 text-xs mr-1 w-3 peer", [_ctx.disabled ? "fill-color-text-disabled  cursor-not-allowed" : "fill-color-icon-secondary cursor-pointer peer-hover:fill-color-brand hover:fill-color-brand"]]),
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
      })
    },
    [renderSlot(_ctx.$slots, "icon", {
      active: _ctx.state.isActive,
      disabled: _ctx.disabled
    }, function() {
      return [createVNode(_component_icon_arrow_bottom, {
        class: normalizeClass(["w-2.5 h-2.5 sm:w-3 sm:h-3 transition-transform duration-300", [_ctx.state.isActive ? "sm:rotate-0 rotate-180" : "sm:-rotate-90 rotate-0"]])
      }, null, 8, ["class"])];
    })],
    2
    /* CLASS */
  )], 42, _hoisted_3)], 8, _hoisted_2), createVNode(_component_collapse_transition, null, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode("div", {
        class: "will-change-[height] bg-color-bg-1 overflow-hidden box-border",
        role: "tabpanel",
        "aria-hidden": !_ctx.state.isActive,
        "aria-labelledby": "tiny-collapse-head-" + _ctx.state.id,
        id: "tiny-collapse-content-" + _ctx.state.id
      }, [createElementVNode("div", _hoisted_6, [renderSlot(_ctx.$slots, "default")])], 8, _hoisted_5), [[vShow, _ctx.state.isActive]])];
    }),
    _: 3
    /* FORWARDED */
  })]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
