// @opentiny/vue-dropdown-item/lib/pc.js
import { defineComponent, $prefix, props, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/dropdown-item/vue";
import { iconDeltaLeft } from "@opentiny/vue-icon";
import Tooltip from "@opentiny/vue-tooltip";
import "@opentiny/vue-theme/dropdown-item/index.css";
import { resolveComponent, openBlock, createBlock, withCtx, createElementVNode, normalizeClass, withModifiers, createElementBlock, resolveDynamicComponent, createCommentVNode, renderSlot, toDisplayString, Fragment, renderList } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "DropdownItem",
  componentName: $prefix + "DropdownItem",
  emits: ["item-click"],
  props: [].concat(props, ["disabled", "divided", "icon", "itemData", "title", "titleClass", "options", "type", "label", "appendToBody", "textField", "selected", "selectedField", "multiStage", "currentIndex", "tip", "tipPosition"]),
  components: {
    IconDeltaLeft: iconDeltaLeft(),
    TinyTooltip: Tooltip
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-disabled", "tabindex"];
var _hoisted_2 = {
  class: "tiny-dropdown-item__wrap"
};
var _hoisted_3 = {
  key: 0,
  class: "tiny-dropdown-item__expand tiny-dropdown-menu__item-expand"
};
var _hoisted_4 = {
  class: "tiny-dropdown-item__content tiny-dropdown-menu__item-content"
};
var _hoisted_5 = {
  class: "tiny-dropdown-item__label"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_dropdown_item = resolveComponent("tiny-dropdown-item");
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  return openBlock(), createBlock(_component_tiny_tooltip, {
    content: _ctx.getTip,
    placement: _ctx.tipPosition
  }, {
    default: withCtx(function() {
      return [createElementVNode("li", {
        class: normalizeClass(["tiny-dropdown-item tiny-dropdown-menu__item", {
          "is-disabled": _ctx.disabled,
          "tiny-dropdown-item--divided tiny-dropdown-menu__item--divided": _ctx.divided,
          "has-children": _ctx.itemData.children && _ctx.itemData.children.length
        }]),
        ref: "dropdownItem",
        onClick: _cache[0] || (_cache[0] = withModifiers(function() {
          return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
        }, ["stop"])),
        onMousedown: _cache[1] || (_cache[1] = withModifiers(function() {
        }, ["stop"])),
        "aria-disabled": _ctx.disabled,
        tabindex: _ctx.disabled ? null : -1
      }, [createElementVNode("div", _hoisted_2, [_ctx.itemData.children && _ctx.itemData.children.length ? (openBlock(), createElementBlock("span", _hoisted_3, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.getIcon)))])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_4, [_ctx.icon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
        key: 0,
        class: "tiny-svg-size"
      })) : createCommentVNode("v-if", true), createElementVNode("span", _hoisted_5, [renderSlot(_ctx.$slots, "default", {
        itemData: _ctx.itemData
      }, function() {
        return [createElementVNode(
          "span",
          null,
          toDisplayString(_ctx.label),
          1
          /* TEXT */
        )];
      })])]), _ctx.itemData.children && _ctx.itemData.children.length ? (openBlock(), createElementBlock(
        "ul",
        {
          key: 1,
          class: normalizeClass(["tiny-dropdown-menu", "tiny-dropdown-item--child tiny-dropdown-menu__item--child", _ctx.state.popperClass])
        },
        [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.itemData.children, function(item, index) {
            return openBlock(), createBlock(_component_tiny_dropdown_item, {
              key: index,
              label: item[_ctx.state.textField],
              "item-data": item,
              icon: item.icon,
              disabled: item.disabled,
              divided: item.divided,
              tip: item.tip,
              "tip-position": item.tipPosition
            }, {
              default: withCtx(function() {
                return [renderSlot(_ctx.$slots, "default", {
                  itemData: item
                })];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["label", "item-data", "icon", "disabled", "divided", "tip", "tip-position"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true)])], 42, _hoisted_1)];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["content", "placement"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
