// @opentiny/vue-list/lib/index.js
import { defineComponent, $prefix, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/list/vue";
import { openBlock, createElementBlock, normalizeClass, renderSlot, createCommentVNode, createElementVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-theme-mobile/list/index.css";
var _sfc_main = defineComponent({
  name: $prefix + "List",
  props: {
    content: {
      type: String,
      default: ""
    },
    subText: {
      type: String,
      default: ""
    },
    contentDes: {
      type: String,
      default: ""
    },
    list: {
      type: Boolean,
      default: false
    },
    id: {
      type: [Number, String],
      default: ""
    }
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  key: 0,
  class: "tiny-mobile-list__prefix"
};
var _hoisted_2 = {
  class: "tiny-mobile-list__content"
};
var _hoisted_3 = {
  class: "tiny-mobile-list__content-text"
};
var _hoisted_4 = {
  class: "tiny-mobile-list__main-text"
};
var _hoisted_5 = {
  key: 0,
  class: "tiny-mobile-list__sub-text"
};
var _hoisted_6 = {
  key: 0,
  class: "tiny-mobile-list__content-des"
};
var _hoisted_7 = {
  key: 1,
  class: "tiny-mobile-list__content-des"
};
var _hoisted_8 = {
  key: 1,
  class: "tiny-mobile-list__suffix"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass([{
        "is-show-prefix": _ctx.slots.prefix,
        "is-padding": _ctx.list
      }, "tiny-mobile-list"]),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.clickList && _ctx.clickList.apply(_ctx, arguments);
      })
    },
    [_ctx.slots.prefix ? (openBlock(), createElementBlock("div", _hoisted_1, [renderSlot(_ctx.$slots, "prefix")])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createElementVNode(
        "span",
        _hoisted_4,
        toDisplayString(_ctx.content),
        1
        /* TEXT */
      )];
    }), _ctx.subText ? (openBlock(), createElementBlock(
      "span",
      _hoisted_5,
      toDisplayString(_ctx.subText),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true)]), _ctx.slots.description ? (openBlock(), createElementBlock("div", _hoisted_6, [renderSlot(_ctx.$slots, "description")])) : createCommentVNode("v-if", true), _ctx.contentDes ? (openBlock(), createElementBlock(
      "p",
      _hoisted_7,
      toDisplayString(_ctx.contentDes),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true)]), _ctx.slots.suffix ? (openBlock(), createElementBlock("div", _hoisted_8, [renderSlot(_ctx.$slots, "suffix")])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  );
}
var List = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
List.install = function(Vue) {
  Vue.component(List.name, List);
};
List.version = version;
export {
  List as default
};
