// @opentiny/vue-multi-select/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileTemplate from "./mobile.js";
import "@opentiny/vue-theme-mobile/multi-select/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileTemplate;
};
var MultiSelect = defineComponent({
  name: $prefix + "MultiSelect",
  inject: {},
  props: _extends({}, $props),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
MultiSelect.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
MultiSelect.install = function(Vue) {
  Vue.component(MultiSelect.name, MultiSelect);
};
MultiSelect.version = version;
export {
  MultiSelect as default
};
