// @opentiny/vue/pc.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2 } from "@opentiny/vue-action-menu";
import Alert from "@opentiny/vue-alert";
import { default as default3 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default4 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default5 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default6 } from "@opentiny/vue-area";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default7 } from "@opentiny/vue-autocomplete";
import AutonaviMap from "@opentiny/vue-autonavi-map";
import { default as default8 } from "@opentiny/vue-autonavi-map";
import Badge from "@opentiny/vue-badge";
import { default as default9 } from "@opentiny/vue-badge";
import BaiduMap from "@opentiny/vue-baidu-map";
import { default as default10 } from "@opentiny/vue-baidu-map";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default11 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default12 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default13 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default14 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default15 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default16 } from "@opentiny/vue-calendar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default17 } from "@opentiny/vue-calendar-view";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default18 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default19 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default20 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default21 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default22 } from "@opentiny/vue-cascader-menu";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default23 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default24 } from "@opentiny/vue-cascader-panel";
import Chart from "@opentiny/vue-chart";
import { default as default25 } from "@opentiny/vue-chart";
import ChartBar from "@opentiny/vue-chart-bar";
import { default as default26 } from "@opentiny/vue-chart-bar";
import ChartBoxplot from "@opentiny/vue-chart-boxplot";
import { default as default28 } from "@opentiny/vue-chart-boxplot";
import ChartCandle from "@opentiny/vue-chart-candle";
import { default as default29 } from "@opentiny/vue-chart-candle";
import Core from "@opentiny/vue-chart-core";
import { default as default30 } from "@opentiny/vue-chart-core";
import ChartFunnel from "@opentiny/vue-chart-funnel";
import { default as default31 } from "@opentiny/vue-chart-funnel";
import ChartGauge from "@opentiny/vue-chart-gauge";
import { default as default32 } from "@opentiny/vue-chart-gauge";
import ChartGraph from "@opentiny/vue-chart-graph";
import { default as default33 } from "@opentiny/vue-chart-graph";
import ChartHeatmap from "@opentiny/vue-chart-heatmap";
import { default as default34 } from "@opentiny/vue-chart-heatmap";
import ChartHistogram from "@opentiny/vue-chart-histogram";
import { default as default35 } from "@opentiny/vue-chart-histogram";
import ChartLine from "@opentiny/vue-chart-line";
import { default as default36 } from "@opentiny/vue-chart-line";
import ChartLiquidfill from "@opentiny/vue-chart-liquidfill";
import { default as default37 } from "@opentiny/vue-chart-liquidfill";
import ChartMap from "@opentiny/vue-chart-map";
import { default as default38 } from "@opentiny/vue-chart-map";
import ChartPie from "@opentiny/vue-chart-pie";
import { default as default39 } from "@opentiny/vue-chart-pie";
import ChartRadar from "@opentiny/vue-chart-radar";
import { default as default40 } from "@opentiny/vue-chart-radar";
import ChartRing from "@opentiny/vue-chart-ring";
import { default as default41 } from "@opentiny/vue-chart-ring";
import ChartSankey from "@opentiny/vue-chart-sankey";
import { default as default42 } from "@opentiny/vue-chart-sankey";
import ChartScatter from "@opentiny/vue-chart-scatter";
import { default as default43 } from "@opentiny/vue-chart-scatter";
import ChartSunburst from "@opentiny/vue-chart-sunburst";
import { default as default44 } from "@opentiny/vue-chart-sunburst";
import ChartTree from "@opentiny/vue-chart-tree";
import { default as default45 } from "@opentiny/vue-chart-tree";
import ChartWaterfall from "@opentiny/vue-chart-waterfall";
import { default as default46 } from "@opentiny/vue-chart-waterfall";
import ChartWordcloud from "@opentiny/vue-chart-wordcloud";
import { default as default47 } from "@opentiny/vue-chart-wordcloud";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default48 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default49 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default50 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default51 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default52 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default53 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default54 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default55 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default56 } from "@opentiny/vue-color-select-panel";
import Company from "@opentiny/vue-company";
import { default as default57 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default58 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default59 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default60 } from "@opentiny/vue-country";
import CreditCard from "@opentiny/vue-credit-card";
import { default as default61 } from "@opentiny/vue-credit-card";
import CreditCardForm from "@opentiny/vue-credit-card-form";
import { default as default62 } from "@opentiny/vue-credit-card-form";
import Crop from "@opentiny/vue-crop";
import { default as default63 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default64 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default65 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default66 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default67 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default68 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default69 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default70 } from "@opentiny/vue-dept";
import DetailPage from "@opentiny/vue-detail-page";
import { default as default71 } from "@opentiny/vue-detail-page";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default72 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default73 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default74 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default75 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default76 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default77 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default78 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default79 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default80 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default81 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default82 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default83 } from "@opentiny/vue-espace";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default84 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default85 } from "@opentiny/vue-file-upload";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default86 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default87 } from "@opentiny/vue-filter-panel";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default88 } from "@opentiny/vue-floatbar";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default89 } from "@opentiny/vue-flowchart";
import Form from "@opentiny/vue-form";
import { default as default90 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default91 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default92 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default93 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default94 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default95 } from "@opentiny/vue-grid-manager";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default96 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default97 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default98 } from "@opentiny/vue-hrapprover";
import Image from "@opentiny/vue-image";
import { default as default99 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default100 } from "@opentiny/vue-image-viewer";
import IndexBarAnchor from "@opentiny/vue-index-bar-anchor";
import { default as default101 } from "@opentiny/vue-index-bar-anchor";
import Input from "@opentiny/vue-input";
import { default as default102 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default103 } from "@opentiny/vue-ip-address";
import Layout from "@opentiny/vue-layout";
import { default as default104 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default105 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default106 } from "@opentiny/vue-link-menu";
import Loading from "@opentiny/vue-loading";
import { default as default107 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default108 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default109 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default110 } from "@opentiny/vue-logout";
import Message from "@opentiny/vue-message";
import { default as default111 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default112 } from "@opentiny/vue-milestone";
import Modal from "@opentiny/vue-modal";
import { default as default113 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default114 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default115 } from "@opentiny/vue-month-table";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default116 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default117 } from "@opentiny/vue-notify";
import Numeric from "@opentiny/vue-numeric";
import { default as default118 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default119 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default120 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default121 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default122 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default123 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default124 } from "@opentiny/vue-picker";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default125 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default126 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default127 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default128 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default129 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default130 } from "@opentiny/vue-progress";
import QrCode from "@opentiny/vue-qr-code";
import { default as default131 } from "@opentiny/vue-qr-code";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default132 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default133 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default134 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default135 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default136 } from "@opentiny/vue-rate";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default137 } from "@opentiny/vue-recycle-scroller";
import RichTextEditor from "@opentiny/vue-rich-text-editor";
import { default as default138 } from "@opentiny/vue-rich-text-editor";
import River from "@opentiny/vue-river";
import { default as default139 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default140 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default141 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default142 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default143 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default144 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default145 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default146 } from "@opentiny/vue-select-dropdown";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default147 } from "@opentiny/vue-selected-box";
import SlideBar from "@opentiny/vue-slide-bar";
import { default as default148 } from "@opentiny/vue-slide-bar";
import Slider from "@opentiny/vue-slider";
import { default as default149 } from "@opentiny/vue-slider";
import Split from "@opentiny/vue-split";
import { default as default150 } from "@opentiny/vue-split";
import Steps from "@opentiny/vue-steps";
import { default as default151 } from "@opentiny/vue-steps";
import SvgIcon from "@opentiny/vue-svg-icon";
import { default as default152 } from "@opentiny/vue-svg-icon";
import Switch from "@opentiny/vue-switch";
import { default as default153 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default154 } from "@opentiny/vue-tab-item";
import Table from "@opentiny/vue-table";
import { default as default155 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default156 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default157 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default158 } from "@opentiny/vue-tag-group";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default159 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default160 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default161 } from "@opentiny/vue-time-line";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default162 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default163 } from "@opentiny/vue-time-picker";
import TimeRange from "@opentiny/vue-time-range";
import { default as default164 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default165 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default166 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default167 } from "@opentiny/vue-timeline-item";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default168 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default169 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default170 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default171 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default172 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default173 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default174 } from "@opentiny/vue-tree-menu";
import Upload from "@opentiny/vue-upload";
import { default as default175 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default176 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default177 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default178 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default179 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default180 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default181 } from "@opentiny/vue-user-head";
import UserLink from "@opentiny/vue-user-link";
import { default as default182 } from "@opentiny/vue-user-link";
import Watermark from "@opentiny/vue-watermark";
import { default as default183 } from "@opentiny/vue-watermark";
import Wizard from "@opentiny/vue-wizard";
import { default as default184 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default185 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default186 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, Alert, Amount, Anchor, Area, Autocomplete, AutonaviMap, Badge, BaiduMap, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarView, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderNode, CascaderPanel, Chart, ChartBar, ChartBoxplot, ChartCandle, Core, ChartFunnel, ChartGauge, ChartGraph, ChartHeatmap, ChartHistogram, ChartLine, ChartLiquidfill, ChartMap, ChartPie, ChartRadar, ChartRing, ChartSankey, ChartScatter, ChartSunburst, ChartTree, ChartWaterfall, ChartWordcloud, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, Company, ConfigProvider, Container, Country, CreditCard, CreditCardForm, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DetailPage, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, FallMenu, FileUpload, FilterBox, FilterPanel, Floatbar, Flowchart, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridToolbar, Guide, Hrapprover, Image, ImageViewer, IndexBarAnchor, Input, IpAddress, Layout, Link, LinkMenu, Loading, Locales, LogonUser, Logout, Message, Milestone, Modal, MonthRange, MonthTable, NavMenu, Notify, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, QrCode, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, RecycleScroller, RichTextEditor, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectedBox, SlideBar, Slider, Split, Steps, SvgIcon, Switch, TabItem, Table, Tabs, Tag, TagGroup, TextPopup, Time, TimeLine, TimePanel, TimePicker, TimeRange, TimeSelect, TimeSpinner, TimelineItem, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserLink, Watermark, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "pc"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.12.0";
var pc = {
  ActionMenu,
  Alert,
  Amount,
  Anchor,
  Area,
  Autocomplete,
  AutonaviMap,
  Badge,
  BaiduMap,
  Breadcrumb,
  BreadcrumbItem,
  BulletinBoard,
  Button,
  ButtonGroup,
  Calendar,
  CalendarView,
  CardTemplate,
  Carousel,
  CarouselItem,
  Cascader,
  CascaderMenu,
  CascaderNode,
  CascaderPanel,
  Chart,
  ChartBar,
  ChartBoxplot,
  ChartCandle,
  ChartCore: Core,
  ChartFunnel,
  ChartGauge,
  ChartGraph,
  ChartHeatmap,
  ChartHistogram,
  ChartLine,
  ChartLiquidfill,
  ChartMap,
  ChartPie,
  ChartRadar,
  ChartRing,
  ChartSankey,
  ChartScatter,
  ChartSunburst,
  ChartTree,
  ChartWaterfall,
  ChartWordcloud,
  Checkbox,
  CheckboxButton,
  CheckboxGroup,
  Col,
  Collapse,
  CollapseItem,
  CollapseTransition,
  ColorPicker,
  ColorSelectPanel,
  Company,
  ConfigProvider,
  Container,
  Country,
  CreditCard,
  CreditCardForm,
  Crop,
  Currency,
  DatePanel,
  DatePicker,
  DatePickerMobileFirst,
  DateRange,
  DateTable,
  Dept,
  DetailPage,
  DialogBox,
  DialogSelect,
  Divider,
  Drawer,
  DropRoles,
  DropTimes,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DynamicScroller,
  DynamicScrollerItem,
  Espace,
  FallMenu,
  FileUpload,
  FilterBox,
  FilterPanel,
  Floatbar,
  Flowchart,
  Form,
  FormItem,
  Fullscreen,
  Grid,
  GridColumn,
  GridManager,
  GridToolbar,
  Guide,
  Hrapprover,
  Image,
  ImageViewer,
  IndexBarAnchor,
  Input,
  IpAddress,
  Layout,
  Link,
  LinkMenu,
  Loading,
  Locales,
  LogonUser,
  Logout,
  Message,
  Milestone,
  Modal,
  MonthRange,
  MonthTable,
  NavMenu,
  Notify,
  Numeric,
  Option,
  OptionGroup,
  Pager,
  PagerItem,
  Panel,
  Picker,
  PopUpload,
  Popconfirm,
  Popeditor,
  Popover,
  Popup,
  Progress,
  QrCode,
  QueryBuilder,
  Radio,
  RadioButton,
  RadioGroup,
  Rate,
  RecycleScroller,
  RichTextEditor,
  River,
  Roles,
  Row,
  ScrollText,
  Scrollbar,
  Search,
  Select,
  SelectDropdown,
  SelectedBox,
  SlideBar,
  Slider,
  Split,
  Steps,
  SvgIcon,
  Switch,
  TabItem,
  Table,
  Tabs,
  Tag,
  TagGroup,
  TextPopup,
  Time,
  TimeLine,
  TimePanel,
  TimePicker,
  TimeRange,
  TimeSelect,
  TimeSpinner,
  TimelineItem,
  ToggleMenu,
  Tooltip,
  TopBox,
  Transfer,
  TransferPanel,
  Tree,
  TreeMenu,
  Upload,
  UploadDragger,
  UploadList,
  User,
  UserAccount,
  UserContact,
  UserHead,
  UserLink,
  Watermark,
  Wizard,
  YearRange,
  YearTable,
  install
};
export {
  default2 as ActionMenu,
  default3 as Alert,
  default4 as Amount,
  default5 as Anchor,
  default6 as Area,
  default7 as Autocomplete,
  default8 as AutonaviMap,
  default9 as Badge,
  default10 as BaiduMap,
  default11 as Breadcrumb,
  default12 as BreadcrumbItem,
  default13 as BulletinBoard,
  default14 as Button,
  default15 as ButtonGroup,
  default16 as Calendar,
  default17 as CalendarView,
  default18 as CardTemplate,
  default19 as Carousel,
  default20 as CarouselItem,
  default21 as Cascader,
  default22 as CascaderMenu,
  default23 as CascaderNode,
  default24 as CascaderPanel,
  default25 as Chart,
  default26 as ChartBar,
  default28 as ChartBoxplot,
  default29 as ChartCandle,
  default30 as ChartCore,
  default31 as ChartFunnel,
  default32 as ChartGauge,
  default33 as ChartGraph,
  default34 as ChartHeatmap,
  default35 as ChartHistogram,
  default36 as ChartLine,
  default37 as ChartLiquidfill,
  default38 as ChartMap,
  default39 as ChartPie,
  default40 as ChartRadar,
  default41 as ChartRing,
  default42 as ChartSankey,
  default43 as ChartScatter,
  default44 as ChartSunburst,
  default45 as ChartTree,
  default46 as ChartWaterfall,
  default47 as ChartWordcloud,
  default48 as Checkbox,
  default49 as CheckboxButton,
  default50 as CheckboxGroup,
  default51 as Col,
  default52 as Collapse,
  default53 as CollapseItem,
  default54 as CollapseTransition,
  default55 as ColorPicker,
  default56 as ColorSelectPanel,
  default57 as Company,
  default58 as ConfigProvider,
  default59 as Container,
  default60 as Country,
  default61 as CreditCard,
  default62 as CreditCardForm,
  default63 as Crop,
  default64 as Currency,
  default65 as DatePanel,
  default66 as DatePicker,
  default67 as DatePickerMobileFirst,
  default68 as DateRange,
  default69 as DateTable,
  default70 as Dept,
  default71 as DetailPage,
  default72 as DialogBox,
  default73 as DialogSelect,
  default74 as Divider,
  default75 as Drawer,
  default76 as DropRoles,
  default77 as DropTimes,
  default78 as Dropdown,
  default79 as DropdownItem,
  default80 as DropdownMenu,
  default81 as DynamicScroller,
  default82 as DynamicScrollerItem,
  default83 as Espace,
  default84 as FallMenu,
  default85 as FileUpload,
  default86 as FilterBox,
  default87 as FilterPanel,
  default88 as Floatbar,
  default89 as Flowchart,
  default90 as Form,
  default91 as FormItem,
  default92 as Fullscreen,
  default93 as Grid,
  default94 as GridColumn,
  default95 as GridManager,
  default96 as GridToolbar,
  default97 as Guide,
  default98 as Hrapprover,
  default99 as Image,
  default100 as ImageViewer,
  default101 as IndexBarAnchor,
  default102 as Input,
  default103 as IpAddress,
  default104 as Layout,
  default105 as Link,
  default106 as LinkMenu,
  default107 as Loading,
  default108 as Locales,
  default109 as LogonUser,
  default110 as Logout,
  default111 as Message,
  default112 as Milestone,
  default113 as Modal,
  default114 as MonthRange,
  default115 as MonthTable,
  default116 as NavMenu,
  default117 as Notify,
  default118 as Numeric,
  default119 as Option,
  default120 as OptionGroup,
  default121 as Pager,
  default122 as PagerItem,
  default123 as Panel,
  default124 as Picker,
  default125 as PopUpload,
  default126 as Popconfirm,
  default127 as Popeditor,
  default128 as Popover,
  default129 as Popup,
  default130 as Progress,
  default131 as QrCode,
  default132 as QueryBuilder,
  default133 as Radio,
  default134 as RadioButton,
  default135 as RadioGroup,
  default136 as Rate,
  default137 as RecycleScroller,
  default138 as RichTextEditor,
  default139 as River,
  default140 as Roles,
  default141 as Row,
  default142 as ScrollText,
  default143 as Scrollbar,
  default144 as Search,
  default145 as Select,
  default146 as SelectDropdown,
  default147 as SelectedBox,
  default148 as SlideBar,
  default149 as Slider,
  default150 as Split,
  default151 as Steps,
  default152 as SvgIcon,
  default153 as Switch,
  default154 as TabItem,
  default155 as Table,
  default156 as Tabs,
  default157 as Tag,
  default158 as TagGroup,
  default159 as TextPopup,
  default160 as Time,
  default161 as TimeLine,
  default162 as TimePanel,
  default163 as TimePicker,
  default164 as TimeRange,
  default165 as TimeSelect,
  default166 as TimeSpinner,
  default167 as TimelineItem,
  default168 as ToggleMenu,
  default169 as Tooltip,
  default170 as TopBox,
  default171 as Transfer,
  default172 as TransferPanel,
  default173 as Tree,
  default174 as TreeMenu,
  default175 as Upload,
  default176 as UploadDragger,
  default177 as UploadList,
  default178 as User,
  default179 as UserAccount,
  default180 as UserContact,
  default181 as UserHead,
  default182 as UserLink,
  default183 as Watermark,
  default184 as Wizard,
  default185 as YearRange,
  default186 as YearTable,
  pc as default,
  install,
  version
};
