// @opentiny/vue-checkbox-button/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/checkbox-button/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { openBlock, createElementBlock, normalizeClass, withDirectives, vModelCheckbox, normalizeStyle, renderSlot, createTextVNode, toDisplayString, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["modelValue", "label", "text", "events", "disabled", "checked", "name", "trueLabel", "falseLabel"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-checked", "aria-disabled"];
var _hoisted_2 = ["name", "disabled", "true-value", "false-value"];
var _hoisted_3 = ["name", "disabled", "value"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("label", {
    class: normalizeClass(["tiny-checkbox-button", [_ctx.state.size ? "tiny-checkbox-button--" + _ctx.state.size : "", {
      "is-disabled": _ctx.state.isDisabled
    }, {
      "is-checked": _ctx.state.isChecked
    }, {
      "is-focus": _ctx.state.focus
    }]]),
    role: "checkbox",
    "aria-checked": _ctx.state.isChecked,
    "aria-disabled": _ctx.state.isDisabled
  }, [_ctx.trueLabel || _ctx.falseLabel ? withDirectives((openBlock(), createElementBlock("input", {
    key: 0,
    class: "tiny-checkbox-button__original",
    type: "checkbox",
    name: _ctx.name,
    disabled: _ctx.state.isDisabled,
    "true-value": _ctx.trueLabel,
    "false-value": _ctx.falseLabel,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[1] || (_cache[1] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    onFocus: _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[3] || (_cache[3] = function($event) {
      return _ctx.state.focus = false;
    })
  }, null, 40, _hoisted_2)), [[vModelCheckbox, _ctx.state.model]]) : withDirectives((openBlock(), createElementBlock("input", {
    key: 1,
    class: "tiny-checkbox-button__original",
    type: "checkbox",
    name: _ctx.name,
    disabled: _ctx.state.isDisabled,
    value: _ctx.label,
    "onUpdate:modelValue": _cache[4] || (_cache[4] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[5] || (_cache[5] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    onFocus: _cache[6] || (_cache[6] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[7] || (_cache[7] = function($event) {
      return _ctx.state.focus = false;
    })
  }, null, 40, _hoisted_3)), [[vModelCheckbox, _ctx.state.model]]), _ctx.slots.default || _ctx.state.isShowText ? (openBlock(), createElementBlock(
    "span",
    {
      key: 2,
      class: "tiny-checkbox-button__inner",
      style: normalizeStyle(_ctx.state.isChecked ? _ctx.state.activeStyle : null)
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.state.showText),
        1
        /* TEXT */
      )];
    })],
    4
    /* STYLE */
  )) : createCommentVNode("v-if", true)], 10, _hoisted_1);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
