// @opentiny/vue-carousel/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/carousel/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconChevronLeft, IconChevronRight, IconChevronUp, IconChevronDown } from "@opentiny/vue-icon";
import { openBlock, createElementBlock, withModifiers, createElementVNode, normalizeStyle, withDirectives, normalizeClass, createBlock, resolveDynamicComponent, vShow, createCommentVNode, renderSlot, Fragment, renderList, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "complete"],
  props: [].concat(props, ["initialIndex", "height", "trigger", "autoplay", "interval", "indicatorPosition", "indicatorStyle", "arrow", "type", "showTitle", "loop", "aspectRatio"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  },
  components: {
    IconChevronLeft: IconChevronLeft(),
    IconChevronRight: IconChevronRight(),
    IconChevronUp: IconChevronUp(),
    IconChevronDown: IconChevronDown()
  }
});
var _hoisted_1 = ["onMouseenter", "onClick"];
var _hoisted_2 = {
  key: 0,
  "data-tag": "tiny-carousel-label"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-carousel",
      class: "relative overflow-hidden rounded",
      onMouseenter: _cache[9] || (_cache[9] = withModifiers(function() {
        return _ctx.handleMouseEnter && _ctx.handleMouseEnter.apply(_ctx, arguments);
      }, ["stop"])),
      onMouseleave: _cache[10] || (_cache[10] = withModifiers(function() {
        return _ctx.handleMouseLeave && _ctx.handleMouseLeave.apply(_ctx, arguments);
      }, ["stop"]))
    },
    [createElementVNode(
      "div",
      {
        "data-tag": "tiny-carousel-block",
        onTouchstart: _cache[6] || (_cache[6] = function() {
          return _ctx.touchstart && _ctx.touchstart.apply(_ctx, arguments);
        }),
        onTouchmove: _cache[7] || (_cache[7] = function() {
          return _ctx.touchmove && _ctx.touchmove.apply(_ctx, arguments);
        }),
        onTouchend: _cache[8] || (_cache[8] = function() {
          return _ctx.touchend && _ctx.touchend.apply(_ctx, arguments);
        }),
        ref: "carousel",
        class: "relative overflow-hidden rounded",
        style: normalizeStyle(_ctx.state.style)
      },
      [_ctx.arrow !== "never" && _ctx.state.items.length > 1 ? withDirectives((openBlock(), createElementBlock(
        "button",
        {
          key: 0,
          type: "button",
          class: normalizeClass(["absolute group border-none outline-0 p-0 m-0 h-8 w-8 cursor-pointer z-[3] rounded-full bg-color-bg-3 active:bg-color-bg-3 text-color-text-inverse text-center text-xs flex items-center justify-center hover:bg-color-text-primary", [_ctx.type === "vertical" ? "top-4 left-1/2" : "left-4 top-1/2 -translate-y-4"]]),
          onMouseenter: _cache[0] || (_cache[0] = function($event) {
            return _ctx.handleButtonEnter("left");
          }),
          onMouseleave: _cache[1] || (_cache[1] = function() {
            return _ctx.handleButtonLeave && _ctx.handleButtonLeave.apply(_ctx, arguments);
          }),
          onClick: _cache[2] || (_cache[2] = withModifiers(function($event) {
            return _ctx.throttledArrowClick(_ctx.state.activeIndex - 1);
          }, ["stop"]))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.type === "vertical" ? "icon-chevron-up" : "icon-chevron-left"), {
          class: "w-4.5 h-4.5 opacity-50 fill-color-bg-2 group-active:fill-color-bg-2 group-hover:fill-color-bg-1"
        }))],
        34
        /* CLASS, NEED_HYDRATION */
      )), [[vShow, (_ctx.arrow === "always" || _ctx.state.hover) && (_ctx.loop || _ctx.state.activeIndex > 0)]]) : createCommentVNode("v-if", true), _ctx.arrow !== "never" && _ctx.state.items.length > 1 ? withDirectives((openBlock(), createElementBlock(
        "button",
        {
          key: 1,
          type: "button",
          class: normalizeClass(["absolute group border-none outline-0 p-0 m-0 h-8 w-8 cursor-pointer z-[3] rounded-full bg-color-bg-3 active:bg-color-bg-3 text-color-text-inverse text-center text-xs flex items-center justify-center hover:bg-color-text-primary", [_ctx.type === "vertical" ? "top-auto left-1/2 bottom-4" : "right-4 top-1/2 -translate-y-4"]]),
          onMouseenter: _cache[3] || (_cache[3] = function($event) {
            return _ctx.handleButtonEnter("right");
          }),
          onMouseleave: _cache[4] || (_cache[4] = function() {
            return _ctx.handleButtonLeave && _ctx.handleButtonLeave.apply(_ctx, arguments);
          }),
          onClick: _cache[5] || (_cache[5] = withModifiers(function($event) {
            return _ctx.throttledArrowClick(_ctx.state.activeIndex + 1);
          }, ["stop"]))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.type === "vertical" ? "icon-chevron-down" : "icon-chevron-right"), {
          class: "w-4.5 h-4.5 opacity-50 fill-color-bg-2 group-active:fill-color-bg-2 group-hover:fill-color-bg-1"
        }))],
        34
        /* CLASS, NEED_HYDRATION */
      )), [[vShow, (_ctx.arrow === "always" || _ctx.state.hover) && (_ctx.loop || _ctx.state.activeIndex < _ctx.state.items.length - 1)]]) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default")],
      36
      /* STYLE, NEED_HYDRATION */
    ), _ctx.indicatorPosition !== "none" && _ctx.state.items.length > 1 ? (openBlock(), createElementBlock(
      "ul",
      {
        key: 0,
        "data-tag": "tiny-carousel-body",
        class: normalizeClass(["list-none m-0 py-0 px-1 z-[2] flex justify-center items-center", _ctx.type === "vertical" && _ctx.indicatorPosition !== "outside" ? "flex-col left-4 bottom-0 transform-none w-1 h-full py-1 px-0" : "w-full h-1 left-0 bottom-4", _ctx.indicatorPosition === "outside" ? "static mt-2" : "absolute", _ctx.state.hasLabel ? "left-0 right-0 text-center transform-none" : "", _ctx.showTitle ? "left-auto right-3 bottom-1 transform-none bg-transparent" : ""])
      },
      [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.items, function(item, index) {
          return openBlock(), createElementBlock("li", {
            "data-tag": "tiny-carousel-list",
            key: index,
            class: normalizeClass(["group cursor-pointer", _ctx.type === "vertical" ? "block h-auto mb-1.5 mx-0 p-0 first:pl-0 last:pr-0" : "inline-block h-1 p-0 my-0 mr-1.5", _ctx.state.hasLabel ? "py-1.5 px-1" : ""]),
            onMouseenter: function onMouseenter($event) {
              return _ctx.throttledIndicatorHover(index);
            },
            onClick: withModifiers(function($event) {
              return _ctx.handleIndicatorClick(index);
            }, ["stop"])
          }, [createElementVNode(
            "button",
            {
              type: "button",
              class: normalizeClass(_ctx.m(["block border-none outline-0 p-0 m-0 cursor-pointer rounded-sm", _ctx.indicatorPosition === "outside" || _ctx.indicatorStyle === "light" ? "bg-color-bg-3 hover:bg-color-brand" : "bg-color-bg-6 hover:bg-color-bg-5", _ctx.state.hasLabel ? "py-0.5 px-4.5 text-xs" : "", _ctx.type === "vertical" ? "h-1 w-1" : "w-1 h-1", index === _ctx.state.activeIndex ? _ctx.type === "vertical" ? "h-4 w-1" : "w-4 h-1" : "", index === _ctx.state.activeIndex ? _ctx.indicatorPosition === "outside" || _ctx.indicatorStyle === "light" ? "bg-color-brand" : "bg-color-bg-1" : ""]))
            },
            [_ctx.state.hasLabel ? (openBlock(), createElementBlock(
              "span",
              _hoisted_2,
              toDisplayString(item.label),
              1
              /* TEXT */
            )) : createCommentVNode("v-if", true)],
            2
            /* CLASS */
          )], 42, _hoisted_1);
        }),
        128
        /* KEYED_FRAGMENT */
      ))],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true)],
    32
    /* NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
