// @opentiny/vue-checkbox/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/checkbox/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme/checkbox/index.css";
import { iconHalfselect, iconCheckedSur, iconCheck } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, mergeProps, createElementVNode, normalizeClass, createBlock, withDirectives, withModifiers, vModelCheckbox, renderSlot, createTextVNode, toDisplayString, createCommentVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "update:modelValue", "complete"],
  props: [].concat(props, ["modelValue", "text", "events", "label", "indeterminate", "disabled", "checked", "name", "trueLabel", "falseLabel", "id", "controls", "size", "border", "validateEvent", "displayOnly"]),
  components: {
    IconHalfselect: iconHalfselect(),
    IconCheckedSur: iconCheckedSur(),
    IconCheck: iconCheck()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["id"];
var _hoisted_2 = ["role", "aria-checked"];
var _hoisted_3 = {
  class: "tiny-checkbox__inner",
  tabindex: "1"
};
var _hoisted_4 = ["aria-hidden", "name", "disabled", "true-value", "false-value"];
var _hoisted_5 = ["aria-hidden", "disabled", "value", "name"];
var _hoisted_6 = {
  key: 0,
  class: "tiny-checkbox__label tiny-checkbox-display-only"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_halfselect = resolveComponent("icon-halfselect");
  var _component_icon_checked_sur = resolveComponent("icon-checked-sur");
  var _component_icon_check = resolveComponent("icon-check");
  return openBlock(), createElementBlock("label", mergeProps({
    class: ["tiny-checkbox", [_ctx.border && _ctx.state.checkboxSize ? "tiny-checkbox--" + _ctx.state.checkboxSize : "", {
      "is-disabled": _ctx.state.isDisabled || _ctx.state.isDisplayOnly
    }, {
      "is-bordered": _ctx.border
    }, {
      "is-checked": _ctx.state.isChecked
    }, {
      "is-group-display-only": _ctx.state.isGroupDisplayOnly
    }, {
      "is-display-only": _ctx.state.isDisplayOnly
    }]],
    id: _ctx.id,
    tabindex: "-1"
  }, _ctx.a(_ctx.$attrs, ["class", "style", "onClick"], true)), [createElementVNode("span", {
    class: normalizeClass(["tiny-checkbox__input", {
      "is-disabled": _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
      "is-checked": _ctx.state.isChecked,
      "is-indeterminate": _ctx.indeterminate,
      "is-focus": _ctx.state.focus
    }]),
    role: _ctx.indeterminate ? "checkbox" : void 0,
    "aria-checked": _ctx.indeterminate ? "mixed" : false
  }, [createElementVNode("span", _hoisted_3, [_ctx.indeterminate ? (openBlock(), createBlock(_component_icon_halfselect, {
    key: 0,
    class: "tiny-svg-size icon-halfselect"
  })) : _ctx.state.isChecked ? (openBlock(), createBlock(_component_icon_checked_sur, {
    key: 1,
    class: "tiny-svg-size icon-checked-sur"
  })) : (openBlock(), createBlock(_component_icon_check, {
    key: 2,
    class: "tiny-svg-size icon-check"
  }))]), _ctx.trueLabel || _ctx.falseLabel ? withDirectives((openBlock(), createElementBlock("input", {
    key: 0,
    class: "tiny-checkbox__original",
    type: "checkbox",
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    name: _ctx.name,
    disabled: _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
    "true-value": _ctx.trueLabel,
    "false-value": _ctx.falseLabel,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[1] || (_cache[1] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    onFocus: _cache[2] || (_cache[2] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[3] || (_cache[3] = function($event) {
      return _ctx.state.focus = false;
    }),
    onClick: _cache[4] || (_cache[4] = withModifiers(function() {
    }, ["stop"]))
  }, null, 40, _hoisted_4)), [[vModelCheckbox, _ctx.state.model]]) : withDirectives((openBlock(), createElementBlock("input", {
    key: 1,
    class: "tiny-checkbox__original",
    type: "checkbox",
    "aria-hidden": _ctx.indeterminate ? "true" : "false",
    disabled: _ctx.state.isDisabled || _ctx.state.isDisplayOnly,
    value: _ctx.label,
    name: _ctx.name,
    "onUpdate:modelValue": _cache[5] || (_cache[5] = function($event) {
      return _ctx.state.model = $event;
    }),
    onChange: _cache[6] || (_cache[6] = function() {
      return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
    }),
    onFocus: _cache[7] || (_cache[7] = function($event) {
      return _ctx.state.focus = true;
    }),
    onBlur: _cache[8] || (_cache[8] = function($event) {
      return _ctx.state.focus = false;
    }),
    onClick: _cache[9] || (_cache[9] = withModifiers(function() {
    }, ["stop"]))
  }, null, 40, _hoisted_5)), [[vModelCheckbox, _ctx.state.model]])], 10, _hoisted_2), _ctx.slots.default && _ctx.slots.default() || _ctx.state.isShowText ? (openBlock(), createElementBlock("span", _hoisted_6, [renderSlot(_ctx.$slots, "default", {}, function() {
    return [createTextVNode(
      toDisplayString(_ctx.state.showText),
      1
      /* TEXT */
    )];
  })])) : createCommentVNode("v-if", true)], 16, _hoisted_1);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
