// @opentiny/vue-tag/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileTemplate from "./mobile.js";
import MobileFirstTemplate from "./mobile-first.js";
import "@opentiny/vue-theme/tag/index.css";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile" === (tinyMode || mode)) {
    return MobileTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var tagProps = _extends({}, $props, {
  hit: Boolean,
  text: String,
  type: String,
  theme: String,
  size: String,
  color: String,
  closable: Boolean,
  operable: Boolean,
  disabled: Boolean,
  selectable: Boolean,
  customClass: {
    type: String,
    default: ""
  },
  effect: {
    type: String,
    default: "light",
    validator: function validator(value) {
      return Boolean(~["dark", "light", "plain"].indexOf(value));
    }
  },
  beforeDelete: Function,
  value: [Number, String],
  // mobile
  mini: {
    type: Boolean,
    default: false
  },
  maxWidth: {
    type: [String, Number],
    default: null
  }
});
var Tag = defineComponent({
  name: $prefix + "Tag",
  props: tagProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Tag.install = function(Vue) {
  Vue.component(Tag.name, Tag);
};
Tag.version = version;
export {
  Tag as default
};
