// @opentiny/vue-slider-button/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var $constants = {
  SLIDER_BUTTON_GROUP: "SliderButtonGroup"
};
var SliderButton = defineComponent({
  name: $prefix + "SliderButton",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    text: [Number, String],
    events: {
      type: Object,
      default: function _default2() {
        return {};
      }
    },
    label: [Number, String],
    disabled: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
SliderButton.install = function(Vue) {
  Vue.component(SliderButton.name, SliderButton);
};
SliderButton.version = version;
export {
  SliderButton as default
};
