// @opentiny/vue-pager-item/lib/pc.js
import { defineComponent, $prefix, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/pager-item/vue";
import { iconPopup, iconDoubleLeft, iconDoubleRight } from "@opentiny/vue-icon";
import { openBlock, createElementBlock, createElementVNode, normalizeClass, createCommentVNode, createBlock, resolveDynamicComponent, Fragment, renderList, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "PagerItem",
  components: {
    IconPopup: iconPopup(),
    IconDoubleLeft: iconDoubleLeft(),
    IconDoubleRight: iconDoubleRight()
  },
  props: {
    disabled: {
      type: Boolean,
      default: false
    },
    currentPage: Number,
    pageCount: Number,
    pagerCount: Number,
    popupIcon: {
      type: Object,
      default: function _default() {
        return iconPopup();
      }
    },
    doubleLeftIcon: {
      type: Object,
      default: function _default2() {
        return iconDoubleLeft();
      }
    },
    doubleRightIcon: {
      type: Object,
      default: function _default3() {
        return iconDoubleRight();
      }
    },
    isBeforePageChange: Boolean
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-pager__group tiny-unselect"
};
var _hoisted_2 = ["textContent"];
var _hoisted_3 = ["textContent"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode("ul", {
    class: "tiny-pager__pages",
    onClick: _cache[4] || (_cache[4] = function() {
      return _ctx.onPagerClick && _ctx.onPagerClick.apply(_ctx, arguments);
    })
  }, [_ctx.pageCount > 0 ? (openBlock(), createElementBlock(
    "li",
    {
      key: 0,
      class: normalizeClass({
        "is-active": _ctx.currentPage === 1
      }),
      textContent: "1"
    },
    null,
    2
    /* CLASS */
  )) : createCommentVNode("v-if", true), _ctx.state.showPrevMore ? (openBlock(), createElementBlock(
    "li",
    {
      key: 1,
      class: normalizeClass(["dot quickprev", [_ctx.state.quickprevIconClass === _ctx.popupIcon && "is-popup"]]),
      onMouseenter: _cache[0] || (_cache[0] = function($event) {
        return _ctx.onMouseenter("left");
      }),
      onMouseleave: _cache[1] || (_cache[1] = function($event) {
        return _ctx.state.quickprevIconClass = _ctx.popupIcon;
      })
    },
    [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.quickprevIconClass)))],
    34
    /* CLASS, NEED_HYDRATION */
  )) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.pagers, function(pager) {
      return openBlock(), createElementBlock("li", {
        key: pager,
        class: normalizeClass({
          "is-active": _ctx.currentPage === pager
        }),
        textContent: toDisplayString("" + pager)
      }, null, 10, _hoisted_2);
    }),
    128
    /* KEYED_FRAGMENT */
  )), _ctx.state.showNextMore ? (openBlock(), createElementBlock(
    "li",
    {
      key: 2,
      class: normalizeClass(["dot quicknext", [_ctx.state.quicknextIconClass === _ctx.popupIcon && "is-popup"]]),
      onMouseenter: _cache[2] || (_cache[2] = function($event) {
        return _ctx.onMouseenter("right");
      }),
      onMouseleave: _cache[3] || (_cache[3] = function($event) {
        return _ctx.state.quicknextIconClass = _ctx.popupIcon;
      })
    },
    [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.quicknextIconClass)))],
    34
    /* CLASS, NEED_HYDRATION */
  )) : createCommentVNode("v-if", true), _ctx.pageCount > 1 ? (openBlock(), createElementBlock("li", {
    key: 3,
    class: normalizeClass({
      "is-active": _ctx.pageCount === _ctx.currentPage
    }),
    textContent: toDisplayString("" + _ctx.pageCount)
  }, null, 10, _hoisted_3)) : createCommentVNode("v-if", true)])]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
