// @opentiny/vue-select-mobile/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/select-mobile/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { iconCheck, iconYes, iconCheckedSur, iconHalfselect, IconCheck, IconCheckedSur, IconHalfselect, IconSearch, IconClose } from "@opentiny/vue-icon";
import Button from "@opentiny/vue-button";
import ActionSheet from "@opentiny/vue-action-sheet";
import Exception from "@opentiny/vue-exception";
import Input from "@opentiny/vue-input";
import Cell from "@opentiny/vue-cell";
import { resolveComponent, openBlock, createElementBlock, createBlock, resolveDynamicComponent, normalizeClass, createCommentVNode, createElementVNode, renderSlot, toDisplayString, withDirectives, createVNode, vShow, withCtx, createTextVNode, Fragment, renderList, vModelText } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main$1 = defineComponent({
  emits: ["click"],
  components: {
    IconCheck: iconCheck(),
    IconYes: iconYes(),
    IconCheckedSur: iconCheckedSur(),
    IconHalfselect: iconHalfselect()
  },
  props: ["selected", "ellipsis", "multiple", "option", "textField", "textField2", "textField3"]
});
var _hoisted_1$1 = {
  class: "flex"
};
var _hoisted_2$1 = {
  class: "flex flex-none w-5 items-center justify-center ml-2"
};
function _sfc_render$1(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconYes = resolveComponent("IconYes");
  return openBlock(), createElementBlock("div", {
    class: "flex items-start leading-6 py-3 cursor-pointer select-none",
    onClick: _cache[0] || (_cache[0] = function($event) {
      return _ctx.$emit("click", $event);
    })
  }, [_ctx.multiple ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.selected ? "icon-checked-sur" : "icon-check"), {
    key: 0,
    class: normalizeClass(["flex-none mt-1 mr-2", _ctx.selected ? "fill-color-brand" : "fill-color-icon-disabled"]),
    "custom-class": "w-4.5 h-4.5"
  }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      class: normalizeClass(["flex-auto flex w-full text-sm", {
        "truncate": _ctx.ellipsis
      }, {
        "text-color-brand": !_ctx.multiple && _ctx.selected
      }])
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createElementVNode(
        "div",
        {
          class: normalizeClass(["flex-auto", {
            "truncate": _ctx.ellipsis
          }])
        },
        [createElementVNode("div", _hoisted_1$1, [createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "min-w-[3.75rem]": !_ctx.textField3
            }, {
              "truncate": _ctx.ellipsis
            }])
          },
          toDisplayString(_ctx.option[_ctx.textField]),
          3
          /* TEXT, CLASS */
        ), _ctx.textField2 ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass(["ml-5 text-color-text-placeholder", {
              "truncate": _ctx.ellipsis
            }])
          },
          toDisplayString(_ctx.option[_ctx.textField2]),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true)]), _ctx.textField3 ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass(["text-color-text-placeholder", {
              "truncate": _ctx.ellipsis
            }])
          },
          toDisplayString(_ctx.option[_ctx.textField3]),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      ), withDirectives(createElementVNode(
        "div",
        _hoisted_2$1,
        [createVNode(_component_IconYes, {
          "custom-class": "w-5 h-5",
          class: "fill-color-brand"
        })],
        512
        /* NEED_PATCH */
      ), [[vShow, !_ctx.multiple && _ctx.selected]])];
    })],
    2
    /* CLASS */
  )]);
}
var Option = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var _sfc_main = defineComponent({
  name: $prefix + "SelectMobile",
  emits: ["click", "search-click", "update:visible", "update:text"],
  components: {
    TinyActionSheet: ActionSheet,
    TinyOption: Option,
    TinyButton: Button,
    TinyException: Exception,
    IconCheck: IconCheck(),
    IconCheckedSur: IconCheckedSur(),
    IconHalfselect: IconHalfselect(),
    IconSearch: IconSearch(),
    IconClose: IconClose(),
    TinyInput: Input,
    TinyCell: Cell
  },
  props: _extends({}, $props, {
    menus: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    modelValue: [Number, String, Array],
    visible: {
      type: Boolean,
      default: false
    },
    ellipsis: {
      type: Boolean,
      default: false
    },
    valueField: {
      type: String,
      default: "id"
    },
    textField: {
      type: String,
      default: "label"
    },
    textField2: {
      type: String,
      default: ""
    },
    textField3: {
      type: String,
      default: ""
    },
    title: String,
    textSplit: {
      type: String,
      default: "; "
    },
    showHeader: {
      type: Boolean,
      default: true
    },
    showFooter: {
      type: Boolean,
      default: false
    },
    multiple: Boolean,
    searchConfig: {
      type: Object,
      default: function _default2() {
        return {
          options: [],
          searchMethod: null,
          openSearchSlot: false
        };
      }
    },
    lockScroll: {
      type: Boolean,
      default: true
    },
    mode: {
      type: String,
      default: "",
      validator: function validator(val) {
        return ["", "form"].includes(val);
      }
    },
    placeholder: String
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-select-mobile"
};
var _hoisted_2 = {
  "data-tag": "tiny-select-mobile-content",
  class: /* @__PURE__ */ normalizeClass(["flex flex-col px-4"])
};
var _hoisted_3 = {
  "data-tag": "tiny-select-mobile-header",
  class: "flex leading-6 pt-4 pb-4 px-4 text-base items-center"
};
var _hoisted_4 = {
  class: "flex-auto flex items-center h-8 py-1 px-3 bg-color-bg-4 rounded"
};
var _hoisted_5 = ["placeholder"];
var _hoisted_6 = {
  class: "flex items-center pl-3 cursor-pointer"
};
var _hoisted_7 = {
  "data-tag": "tiny-select-mobile-body",
  class: "flex-auto overflow-auto flex flex-col"
};
var _hoisted_8 = {
  "data-tag": "tiny-select-mobile-filter",
  class: /* @__PURE__ */ normalizeClass(["flex flex-col px-4"])
};
var _hoisted_9 = {
  "data-tag": "tiny-select-mobile-nodata",
  class: "w-full flex justify-center items-center text-center flex-auto"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_cell = resolveComponent("tiny-cell");
  var _component_tiny_option = resolveComponent("tiny-option");
  var _component_IconSearch = resolveComponent("IconSearch");
  var _component_tiny_exception = resolveComponent("tiny-exception");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  return openBlock(), createElementBlock("div", _hoisted_1, [_ctx.mode === "form" ? (openBlock(), createBlock(_component_tiny_cell, {
    key: 0,
    placeholder: _ctx.placeholder,
    data: _ctx.state.selectedLabel,
    onClick: _cache[0] || (_cache[0] = function($event) {
      return _ctx.$emit("update:visible", true);
    })
  }, null, 8, ["placeholder", "data"])) : createCommentVNode("v-if", true), withDirectives(createVNode(_component_tiny_action_sheet, {
    tiny_mode: "mobile-first",
    title: _ctx.title,
    "lock-scroll": _ctx.lockScroll,
    visible: _ctx.state.toggle,
    "onUpdate:visible": _cache[6] || (_cache[6] = function($event) {
      return _ctx.$emit("update:visible", $event);
    }),
    onClose: _ctx.hide,
    "show-header": !_ctx.state.search.show,
    "show-footer": _ctx.multiple,
    "custom-class": [{
      "min-h-[90%]": _ctx.state.search.show
    }, {
      "pb-4": _ctx.state.search.show && _ctx.multiple
    }]
  }, {
    "header-left": withCtx(function() {
      return [createVNode(_component_IconSearch, {
        "custom-class": "h-5 w-5",
        class: "cursor-pointer",
        onClick: _cache[5] || (_cache[5] = function($event) {
          return _ctx.searchBoxToggle(true);
        })
      })];
    }),
    footer: withCtx(function() {
      return [_ctx.multiple ? (openBlock(), createBlock(_component_tiny_button, {
        key: 0,
        tiny_mode: "mobile-first",
        class: "flex-1 mx-4 sm:mx-0",
        type: "primary",
        "reset-time": 0,
        onClick: _ctx.confirm
      }, {
        default: withCtx(function() {
          return [createTextVNode(
            toDisplayString(_ctx.t("ui.button.confirm")),
            1
            /* TEXT */
          )];
        }),
        _: 1
        /* STABLE */
      }, 8, ["onClick"])) : createCommentVNode("v-if", true)];
    }),
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        _hoisted_2,
        [_ctx.multiple ? (openBlock(), createElementBlock("div", {
          key: 0,
          "data-tag": "tiny-select-mobile-multiple",
          class: normalizeClass(["flex items-start leading-6 py-3 cursor-pointer select-none"]),
          onClick: _cache[1] || (_cache[1] = function() {
            return _ctx.allCheckHandler && _ctx.allCheckHandler.apply(_ctx, arguments);
          })
        }, [_ctx.multiple ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.checkList.length === 0 ? "icon-check" : _ctx.state.checkList.length === _ctx.menus.length ? "icon-checked-sur" : "icon-halfselect"), {
          key: 0,
          class: normalizeClass(["flex-none mt-1 mr-2", _ctx.state.checkList.length ? "fill-color-brand" : "fill-color-icon-disabled"]),
          "custom-class": "w-4.5 h-4.5"
        }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
          "div",
          {
            "data-tag": "tiny-select-mobile-ellipsis",
            class: normalizeClass(["flex-auto", {
              "truncate": _ctx.ellipsis
            }])
          },
          toDisplayString(_ctx.t("ui.base.all")),
          3
          /* TEXT, CLASS */
        )])) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.menus, function(item, index) {
            return openBlock(), createBlock(_component_tiny_option, {
              key: item[_ctx.valueField],
              multiple: _ctx.multiple,
              ellipsis: _ctx.ellipsis,
              textField: _ctx.textField,
              textField2: _ctx.textField2,
              textField3: _ctx.textField3,
              option: item,
              selected: _ctx.isSelected(item),
              onClick: function onClick($event) {
                return _ctx.selectOption(item, index);
              }
            }, {
              default: withCtx(function() {
                return [renderSlot(_ctx.$slots, "default", {
                  item,
                  index
                })];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["multiple", "ellipsis", "textField", "textField2", "textField3", "option", "selected", "onClick"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        512
        /* NEED_PATCH */
      ), [[vShow, !_ctx.state.search.show]]), createElementVNode(
        "div",
        {
          "data-tag": "tiny-select-mobile-box",
          class: normalizeClass([_ctx.state.search.show ? "flex flex-col flex-auto" : "hidden"])
        },
        [createElementVNode("div", _hoisted_3, [createElementVNode("div", _hoisted_4, [createVNode(_component_IconSearch, {
          "custom-class": "h-4 w-4",
          class: "mr-1 fill-color-icon-disabled"
        }), withDirectives(createElementVNode("input", {
          "onUpdate:modelValue": _cache[2] || (_cache[2] = function($event) {
            return _ctx.state.search.input = $event;
          }),
          class: "h-5 flex-auto text-xs bg-transparent outline-0",
          placeholder: _ctx.t("ui.select.pleaseSearch"),
          onInput: _cache[3] || (_cache[3] = function() {
            return _ctx.searchMethod && _ctx.searchMethod.apply(_ctx, arguments);
          })
        }, null, 40, _hoisted_5), [[vModelText, _ctx.state.search.input]])]), createElementVNode("div", _hoisted_6, [createElementVNode(
          "div",
          {
            onClick: _cache[4] || (_cache[4] = function($event) {
              return _ctx.searchBoxToggle(false);
            })
          },
          toDisplayString(_ctx.t("ui.base.cancel")),
          1
          /* TEXT */
        )])]), createElementVNode("div", _hoisted_7, [withDirectives(createElementVNode(
          "div",
          _hoisted_8,
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.search.filterOptions, function(item, index) {
              return openBlock(), createBlock(_component_tiny_option, {
                key: item[_ctx.valueField],
                multiple: _ctx.multiple,
                ellipsis: _ctx.ellipsis,
                textField: _ctx.textField,
                textField2: _ctx.textField2,
                textField3: _ctx.textField3,
                option: item,
                selected: _ctx.isSelected(item),
                onClick: function onClick($event) {
                  return _ctx.searchSelectHandler(item, index);
                }
              }, {
                default: withCtx(function() {
                  return [_ctx.searchConfig.openSearchSlot ? renderSlot(_ctx.$slots, "search-item", {
                    key: 0,
                    item,
                    index
                  }) : renderSlot(_ctx.$slots, "default", {
                    key: 1,
                    item,
                    index
                  })];
                }),
                _: 2
                /* DYNAMIC */
              }, 1032, ["multiple", "ellipsis", "textField", "textField2", "textField3", "option", "selected", "onClick"]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          512
          /* NEED_PATCH */
        ), [[vShow, _ctx.state.search.filterOptions.length]]), withDirectives(createElementVNode(
          "div",
          _hoisted_9,
          [createVNode(_component_tiny_exception, {
            "component-page": "",
            type: "nodata"
          }, {
            content: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.select.noSearchData")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          })],
          512
          /* NEED_PATCH */
        ), [[vShow, !_ctx.state.search.filterOptions.length]])])],
        2
        /* CLASS */
      )];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["title", "lock-scroll", "visible", "onClose", "show-header", "show-footer", "custom-class"]), [[vShow, _ctx.visible]])]);
}
var SelectMobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
SelectMobile.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
SelectMobile.install = function(Vue) {
  Vue.component(SelectMobile.name, SelectMobile);
};
SelectMobile.version = version;
export {
  SelectMobile as default
};
