// @opentiny/vue-guide/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { $prefix, $props, $setup } from "@opentiny/vue-common";
import browser from "@opentiny/vue-renderless/common/browser";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Guide = {
  name: $prefix + "Guide",
  props: _extends({}, $props, {
    showStep: {
      type: Boolean,
      default: false
    },
    domData: Array,
    mainAxis: {
      type: Number,
      default: 18
    },
    crossAxis: {
      type: Number,
      default: null
    },
    alignmentAxis: {
      type: Number,
      default: null
    },
    popPosition: {
      type: String,
      default: "bottom"
    },
    arrow: {
      type: Boolean,
      default: true
    },
    modalOverlayOpeningPadding: {
      type: Number,
      default: 0
    },
    modalOverlayOpeningRadius: {
      type: Number,
      default: 0
    },
    lightClass: {
      type: String,
      default: ""
    },
    width: {
      type: String,
      default: browser.isMobile ? "350" : "510"
    },
    height: {
      type: String,
      default: ""
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
};
var version = "3.13.0";
Guide.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Guide.install = function(Vue) {
  Vue.component(Guide.name, Guide);
};
Guide.version = version;
export {
  Guide as default
};
