// @opentiny/vue-renderless/area/vue.js
import "../chunk-G2ADBYYC.js";
import { getRegion, changeOffice, getRep, getOffice, fetchDefaultData, beforeMount } from "./index";
const api = ["state", "getRep", "getRegion", "getOffice", "changeOffice"];
const renderless = (props, { onBeforeMount, reactive }, { emit, service, refs, nextTick }) => {
  const api2 = {};
  const { fetchArea } = service || {};
  const state = reactive({
    jcr: "",
    region: "",
    rep: "",
    office: "",
    jcrData: [],
    regionData: [],
    repData: [],
    officeData: []
  });
  Object.assign(api2, {
    state,
    changeOffice: changeOffice({ emit, state }),
    beforeMount: beforeMount({ api: api2, props }),
    getRep: getRep({ emit, fetchArea, nextTick, props, refs, state }),
    getRegion: getRegion({ emit, fetchArea, nextTick, props, refs, state }),
    getOffice: getOffice({ emit, fetchArea, nextTick, props, refs, state }),
    fetchDefaultData: fetchDefaultData({ emit, fetchArea, nextTick, props, refs, state })
  });
  onBeforeMount(api2.beforeMount);
  return api2;
};
export {
  api,
  renderless
};
