// @opentiny/vue-upload/lib/pc.js
import { createVNode, isVNode } from "vue";
import { defineComponent, $prefix, props, setup as _setup, h } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/upload/vue";
import UploadDragger from "@opentiny/vue-upload-dragger";
import Modal from "@opentiny/vue-modal";
function _isSlot(s) {
  return typeof s === "function" || Object.prototype.toString.call(s) === "[object Object]" && !isVNode(s);
}
var _sfc_main = /* @__PURE__ */ defineComponent({
  inheritAttrs: false,
  name: $prefix + "Upload",
  props: [].concat(props, ["accept", "action", "autoUpload", "beforeUpload", "pasteUpload", "data", "disabled", "drag", "edmToken", "fileList", "headers", "httpRequest", "isFolder", "limit", "listType", "multiple", "name", "onError", "onExceed", "onPreview", "onProgress", "onRemove", "onStart", "onSuccess", "type", "withCredentials", "isHidden", "handleTriggerClick"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      h,
      extendOptions: {
        Modal
      }
    });
  },
  render: function render() {
    var accept = this.accept, disabled = this.disabled, drag = this.drag, handleChange = this.handleChange, handleClick = this.handleClick, handlePaste = this.handlePaste, handleKeydown = this.handleKeydown, isFolder = this.isFolder, listType = this.listType, multiple = this.multiple, name = this.name, uploadFiles = this.uploadFiles, fileList = this.fileList, limit = this.limit, isHidden = this.isHidden, type = this.type;
    var defaultSlot = this.slots.default && this.slots.default() || [];
    var hidden = isHidden && fileList.length >= limit;
    return createVNode("div", {
      "class": ["tiny-upload", "tiny-upload--" + listType, disabled ? "is-disabled" : "", hidden ? "is-hidden" : ""],
      "onClick": function onClick($event) {
        return handleClick($event, type);
      },
      "onPaste": handlePaste,
      "onKeydown": handleKeydown,
      "tabindex": "0"
    }, [drag ? createVNode(UploadDragger, {
      "disabled": disabled,
      "onFile": uploadFiles
    }, _isSlot(defaultSlot) ? defaultSlot : {
      default: function _default() {
        return [defaultSlot];
      }
    }) : defaultSlot, createVNode("input", {
      "class": "tiny-upload__input",
      "type": "file",
      "webkitdirectory": isFolder,
      "ref": "input",
      "name": name,
      "onChange": handleChange,
      "multiple": isFolder ? true : multiple,
      "accept": accept
    }, null)]);
  }
});
export {
  _sfc_main as default
};
