// @opentiny/vue-slider/lib/mobile.js
import { renderless, api } from "@opentiny/vue-renderless/slider/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme-mobile/slider/index.css";
import { openBlock, createElementBlock, createElementVNode, normalizeClass, normalizeStyle, createCommentVNode, withDirectives, toDisplayString, vShow, renderSlot, vModelText } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["update:modelValue", "change", "start", "stop"],
  props: [].concat(props, ["modelValue", "disabled", "max", "min", "vertical", "step", "numPages", "showTip", "showInput", "height", "range", "formatTooltip"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  style: {
    "position": "relative"
  }
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-mobile-slider__input"
};
var _hoisted_3 = ["disabled"];
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "span",
  {
    class: "tiny-mobile-slider__per"
  },
  "%",
  -1
  /* HOISTED */
);
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      class: normalizeClass(["tiny-mobile-slider", {
        "tiny-mobile-slider__vertical": _ctx.vertical,
        disabled: _ctx.disabled
      }]),
      style: normalizeStyle({
        height: _ctx.vertical ? _ctx.height : ""
      }),
      onTouchstart: _cache[6] || (_cache[6] = function() {
        return _ctx.bindMouseDown && _ctx.bindMouseDown.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-mobile-slider__range", _ctx.vertical ? "tiny-mobile-slider__down" : "tiny-mobile-slider__left"]),
        style: normalizeStyle(_ctx.state.barStyle)
      },
      null,
      6
      /* CLASS, STYLE */
    ), _ctx.state.leftBtnShow ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: normalizeClass(["tiny-mobile-slider__handle", {
          "handle-focus": _ctx.state.activeIndex === 0
        }]),
        tabindex: "0",
        style: normalizeStyle(_ctx.state.leftBtnStyle),
        onMouseenter: _cache[0] || (_cache[0] = function() {
          return _ctx.displayTip && _ctx.displayTip.apply(_ctx, arguments);
        }),
        onMouseleave: _cache[1] || (_cache[1] = function() {
          return _ctx.hideTip && _ctx.hideTip.apply(_ctx, arguments);
        }),
        onKeydown: _cache[2] || (_cache[2] = function() {
          return _ctx.bindKeyDown && _ctx.bindKeyDown.apply(_ctx, arguments);
        })
      },
      null,
      38
      /* CLASS, STYLE, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), _ctx.state.rightBtnShow ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        class: normalizeClass(["tiny-mobile-slider__handle", {
          "handle-focus": _ctx.state.activeIndex === 1
        }]),
        tabindex: "0",
        style: normalizeStyle(_ctx.state.rightBtnStyle),
        onMouseenter: _cache[3] || (_cache[3] = function() {
          return _ctx.displayTip && _ctx.displayTip.apply(_ctx, arguments);
        }),
        onMouseleave: _cache[4] || (_cache[4] = function() {
          return _ctx.hideTip && _ctx.hideTip.apply(_ctx, arguments);
        }),
        onKeydown: _cache[5] || (_cache[5] = function() {
          return _ctx.bindKeyDown && _ctx.bindKeyDown.apply(_ctx, arguments);
        })
      },
      null,
      38
      /* CLASS, STYLE, NEED_HYDRATION */
    )) : createCommentVNode("v-if", true), withDirectives(createElementVNode(
      "div",
      {
        class: "tiny-mobile-slider__tips",
        style: normalizeStyle(_ctx.state.tipStyle)
      },
      toDisplayString(_ctx.state.tipValue),
      5
      /* TEXT, STYLE */
    ), [[vShow, _ctx.showTip && _ctx.state.showTip]])],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  ), _ctx.showInput && !_ctx.state.isDouble ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "default", {
    slotScope: _ctx.state.activeValue
  }, function() {
    return [withDirectives(createElementVNode("input", {
      type: "text",
      "onUpdate:modelValue": _cache[7] || (_cache[7] = function($event) {
        return _ctx.state.activeValue = $event;
      }),
      disabled: _ctx.disabled
    }, null, 8, _hoisted_3), [[vModelText, _ctx.state.activeValue]]), _hoisted_4];
  })])) : createCommentVNode("v-if", true)]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
