// @opentiny/vue-layout/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/layout/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Layout = defineComponent({
  name: $prefix + "Layout",
  provide: function provide() {
    return {
      layout: this
    };
  },
  props: _extends({}, $props, {
    tag: {
      type: String,
      default: "div"
    },
    cols: {
      type: Number,
      default: 12,
      validator: function validator(value) {
        return Boolean(~[12, 24].indexOf(value));
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Layout.install = function(Vue) {
  Vue.component(Layout.name, Layout);
};
Layout.version = version;
export {
  Layout as default
};
