// @opentiny/vue/index.js
import ActionMenu from "@opentiny/vue-action-menu";
import { default as default2 } from "@opentiny/vue-action-menu";
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default3 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default4 } from "@opentiny/vue-alert";
import Amount from "@opentiny/vue-amount";
import { default as default5 } from "@opentiny/vue-amount";
import Anchor from "@opentiny/vue-anchor";
import { default as default6 } from "@opentiny/vue-anchor";
import Area from "@opentiny/vue-area";
import { default as default7 } from "@opentiny/vue-area";
import AsyncFlowchart from "@opentiny/vue-async-flowchart";
import { default as default8 } from "@opentiny/vue-async-flowchart";
import Autocomplete from "@opentiny/vue-autocomplete";
import { default as default9 } from "@opentiny/vue-autocomplete";
import AutonaviMap from "@opentiny/vue-autonavi-map";
import { default as default10 } from "@opentiny/vue-autonavi-map";
import Avatar from "@opentiny/vue-avatar";
import { default as default11 } from "@opentiny/vue-avatar";
import Badge from "@opentiny/vue-badge";
import { default as default12 } from "@opentiny/vue-badge";
import BaiduMap from "@opentiny/vue-baidu-map";
import { default as default13 } from "@opentiny/vue-baidu-map";
import Breadcrumb from "@opentiny/vue-breadcrumb";
import { default as default14 } from "@opentiny/vue-breadcrumb";
import BreadcrumbItem from "@opentiny/vue-breadcrumb-item";
import { default as default15 } from "@opentiny/vue-breadcrumb-item";
import BulletinBoard from "@opentiny/vue-bulletin-board";
import { default as default16 } from "@opentiny/vue-bulletin-board";
import Button from "@opentiny/vue-button";
import { default as default17 } from "@opentiny/vue-button";
import ButtonGroup from "@opentiny/vue-button-group";
import { default as default18 } from "@opentiny/vue-button-group";
import Calendar from "@opentiny/vue-calendar";
import { default as default19 } from "@opentiny/vue-calendar";
import CalendarBar from "@opentiny/vue-calendar-bar";
import { default as default20 } from "@opentiny/vue-calendar-bar";
import CalendarView from "@opentiny/vue-calendar-view";
import { default as default21 } from "@opentiny/vue-calendar-view";
import Card from "@opentiny/vue-card";
import { default as default22 } from "@opentiny/vue-card";
import CardGroup from "@opentiny/vue-card-group";
import { default as default23 } from "@opentiny/vue-card-group";
import CardTemplate from "@opentiny/vue-card-template";
import { default as default24 } from "@opentiny/vue-card-template";
import Carousel from "@opentiny/vue-carousel";
import { default as default25 } from "@opentiny/vue-carousel";
import CarouselItem from "@opentiny/vue-carousel-item";
import { default as default26 } from "@opentiny/vue-carousel-item";
import Cascader from "@opentiny/vue-cascader";
import { default as default27 } from "@opentiny/vue-cascader";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { default as default28 } from "@opentiny/vue-cascader-menu";
import CascaderMobile from "@opentiny/vue-cascader-mobile";
import { default as default29 } from "@opentiny/vue-cascader-mobile";
import CascaderNode from "@opentiny/vue-cascader-node";
import { default as default30 } from "@opentiny/vue-cascader-node";
import CascaderPanel from "@opentiny/vue-cascader-panel";
import { default as default31 } from "@opentiny/vue-cascader-panel";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { default as default32 } from "@opentiny/vue-cascader-select";
import CascaderView from "@opentiny/vue-cascader-view";
import { default as default33 } from "@opentiny/vue-cascader-view";
import Cell from "@opentiny/vue-cell";
import { default as default34 } from "@opentiny/vue-cell";
import Chart from "@opentiny/vue-chart";
import { default as default35 } from "@opentiny/vue-chart";
import ChartBar from "@opentiny/vue-chart-bar";
import { default as default36 } from "@opentiny/vue-chart-bar";
import ChartBoxplot from "@opentiny/vue-chart-boxplot";
import { default as default38 } from "@opentiny/vue-chart-boxplot";
import ChartCandle from "@opentiny/vue-chart-candle";
import { default as default39 } from "@opentiny/vue-chart-candle";
import Core from "@opentiny/vue-chart-core";
import { default as default40 } from "@opentiny/vue-chart-core";
import ChartFunnel from "@opentiny/vue-chart-funnel";
import { default as default41 } from "@opentiny/vue-chart-funnel";
import ChartGauge from "@opentiny/vue-chart-gauge";
import { default as default42 } from "@opentiny/vue-chart-gauge";
import ChartGraph from "@opentiny/vue-chart-graph";
import { default as default43 } from "@opentiny/vue-chart-graph";
import ChartHeatmap from "@opentiny/vue-chart-heatmap";
import { default as default44 } from "@opentiny/vue-chart-heatmap";
import ChartHistogram from "@opentiny/vue-chart-histogram";
import { default as default45 } from "@opentiny/vue-chart-histogram";
import ChartLine from "@opentiny/vue-chart-line";
import { default as default46 } from "@opentiny/vue-chart-line";
import ChartLiquidfill from "@opentiny/vue-chart-liquidfill";
import { default as default47 } from "@opentiny/vue-chart-liquidfill";
import ChartMap from "@opentiny/vue-chart-map";
import { default as default48 } from "@opentiny/vue-chart-map";
import ChartPie from "@opentiny/vue-chart-pie";
import { default as default49 } from "@opentiny/vue-chart-pie";
import ChartRadar from "@opentiny/vue-chart-radar";
import { default as default50 } from "@opentiny/vue-chart-radar";
import ChartRing from "@opentiny/vue-chart-ring";
import { default as default51 } from "@opentiny/vue-chart-ring";
import ChartSankey from "@opentiny/vue-chart-sankey";
import { default as default52 } from "@opentiny/vue-chart-sankey";
import ChartScatter from "@opentiny/vue-chart-scatter";
import { default as default53 } from "@opentiny/vue-chart-scatter";
import ChartSunburst from "@opentiny/vue-chart-sunburst";
import { default as default54 } from "@opentiny/vue-chart-sunburst";
import ChartTree from "@opentiny/vue-chart-tree";
import { default as default55 } from "@opentiny/vue-chart-tree";
import ChartWaterfall from "@opentiny/vue-chart-waterfall";
import { default as default56 } from "@opentiny/vue-chart-waterfall";
import ChartWordcloud from "@opentiny/vue-chart-wordcloud";
import { default as default57 } from "@opentiny/vue-chart-wordcloud";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default58 } from "@opentiny/vue-checkbox";
import CheckboxButton from "@opentiny/vue-checkbox-button";
import { default as default59 } from "@opentiny/vue-checkbox-button";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default60 } from "@opentiny/vue-checkbox-group";
import Col from "@opentiny/vue-col";
import { default as default61 } from "@opentiny/vue-col";
import Collapse from "@opentiny/vue-collapse";
import { default as default62 } from "@opentiny/vue-collapse";
import CollapseItem from "@opentiny/vue-collapse-item";
import { default as default63 } from "@opentiny/vue-collapse-item";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { default as default64 } from "@opentiny/vue-collapse-transition";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default65 } from "@opentiny/vue-color-picker";
import ColorSelectPanel from "@opentiny/vue-color-select-panel";
import { default as default66 } from "@opentiny/vue-color-select-panel";
import ColumnListGroup from "@opentiny/vue-column-list-group";
import { default as default67 } from "@opentiny/vue-column-list-group";
import ColumnListItem from "@opentiny/vue-column-list-item";
import { default as default68 } from "@opentiny/vue-column-list-item";
import Company from "@opentiny/vue-company";
import { default as default69 } from "@opentiny/vue-company";
import ConfigProvider from "@opentiny/vue-config-provider";
import { default as default70 } from "@opentiny/vue-config-provider";
import Container from "@opentiny/vue-container";
import { default as default71 } from "@opentiny/vue-container";
import Country from "@opentiny/vue-country";
import { default as default72 } from "@opentiny/vue-country";
import CreditCard from "@opentiny/vue-credit-card";
import { default as default73 } from "@opentiny/vue-credit-card";
import CreditCardForm from "@opentiny/vue-credit-card-form";
import { default as default74 } from "@opentiny/vue-credit-card-form";
import Crop from "@opentiny/vue-crop";
import { default as default75 } from "@opentiny/vue-crop";
import Currency from "@opentiny/vue-currency";
import { default as default76 } from "@opentiny/vue-currency";
import DatePanel from "@opentiny/vue-date-panel";
import { default as default77 } from "@opentiny/vue-date-panel";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default78 } from "@opentiny/vue-date-picker";
import DatePickerMobileFirst from "@opentiny/vue-date-picker-mobile-first";
import { default as default79 } from "@opentiny/vue-date-picker-mobile-first";
import DateRange from "@opentiny/vue-date-range";
import { default as default80 } from "@opentiny/vue-date-range";
import DateTable from "@opentiny/vue-date-table";
import { default as default81 } from "@opentiny/vue-date-table";
import Dept from "@opentiny/vue-dept";
import { default as default82 } from "@opentiny/vue-dept";
import DetailPage from "@opentiny/vue-detail-page";
import { default as default83 } from "@opentiny/vue-detail-page";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default84 } from "@opentiny/vue-dialog-box";
import DialogSelect from "@opentiny/vue-dialog-select";
import { default as default85 } from "@opentiny/vue-dialog-select";
import Divider from "@opentiny/vue-divider";
import { default as default86 } from "@opentiny/vue-divider";
import Drawer from "@opentiny/vue-drawer";
import { default as default87 } from "@opentiny/vue-drawer";
import DropRoles from "@opentiny/vue-drop-roles";
import { default as default88 } from "@opentiny/vue-drop-roles";
import DropTimes from "@opentiny/vue-drop-times";
import { default as default89 } from "@opentiny/vue-drop-times";
import Dropdown from "@opentiny/vue-dropdown";
import { default as default90 } from "@opentiny/vue-dropdown";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default91 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default92 } from "@opentiny/vue-dropdown-menu";
import DynamicScroller from "@opentiny/vue-dynamic-scroller";
import { default as default93 } from "@opentiny/vue-dynamic-scroller";
import DynamicScrollerItem from "@opentiny/vue-dynamic-scroller-item";
import { default as default94 } from "@opentiny/vue-dynamic-scroller-item";
import Espace from "@opentiny/vue-espace";
import { default as default95 } from "@opentiny/vue-espace";
import Exception from "@opentiny/vue-exception";
import { default as default96 } from "@opentiny/vue-exception";
import FallMenu from "@opentiny/vue-fall-menu";
import { default as default97 } from "@opentiny/vue-fall-menu";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default98 } from "@opentiny/vue-file-upload";
import Filter from "@opentiny/vue-filter";
import { default as default99 } from "@opentiny/vue-filter";
import FilterBar from "@opentiny/vue-filter-bar";
import { default as default100 } from "@opentiny/vue-filter-bar";
import FilterBox from "@opentiny/vue-filter-box";
import { default as default101 } from "@opentiny/vue-filter-box";
import FilterPanel from "@opentiny/vue-filter-panel";
import { default as default102 } from "@opentiny/vue-filter-panel";
import Floatbar from "@opentiny/vue-floatbar";
import { default as default103 } from "@opentiny/vue-floatbar";
import FloatingButton from "@opentiny/vue-floating-button";
import { default as default104 } from "@opentiny/vue-floating-button";
import Flowchart from "@opentiny/vue-flowchart";
import { default as default105 } from "@opentiny/vue-flowchart";
import Form from "@opentiny/vue-form";
import { default as default106 } from "@opentiny/vue-form";
import FormItem from "@opentiny/vue-form-item";
import { default as default107 } from "@opentiny/vue-form-item";
import Fullscreen from "@opentiny/vue-fullscreen";
import { default as default108 } from "@opentiny/vue-fullscreen";
import Grid from "@opentiny/vue-grid";
import { default as default109 } from "@opentiny/vue-grid";
import GridColumn from "@opentiny/vue-grid-column";
import { default as default110 } from "@opentiny/vue-grid-column";
import GridManager from "@opentiny/vue-grid-manager";
import { default as default111 } from "@opentiny/vue-grid-manager";
import GridToolbar from "@opentiny/vue-grid-toolbar";
import { default as default112 } from "@opentiny/vue-grid-toolbar";
import Guide from "@opentiny/vue-guide";
import { default as default113 } from "@opentiny/vue-guide";
import Hrapprover from "@opentiny/vue-hrapprover";
import { default as default114 } from "@opentiny/vue-hrapprover";
import Image from "@opentiny/vue-image";
import { default as default115 } from "@opentiny/vue-image";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default116 } from "@opentiny/vue-image-viewer";
import IndexBar from "@opentiny/vue-index-bar";
import { default as default117 } from "@opentiny/vue-index-bar";
import IndexBarAnchor from "@opentiny/vue-index-bar-anchor";
import { default as default118 } from "@opentiny/vue-index-bar-anchor";
import Input from "@opentiny/vue-input";
import { default as default119 } from "@opentiny/vue-input";
import IpAddress from "@opentiny/vue-ip-address";
import { default as default120 } from "@opentiny/vue-ip-address";
import Label from "@opentiny/vue-label";
import { default as default121 } from "@opentiny/vue-label";
import Layout from "@opentiny/vue-layout";
import { default as default122 } from "@opentiny/vue-layout";
import Link from "@opentiny/vue-link";
import { default as default123 } from "@opentiny/vue-link";
import LinkMenu from "@opentiny/vue-link-menu";
import { default as default124 } from "@opentiny/vue-link-menu";
import List from "@opentiny/vue-list";
import { default as default125 } from "@opentiny/vue-list";
import Loading from "@opentiny/vue-loading";
import { default as default126 } from "@opentiny/vue-loading";
import Locales from "@opentiny/vue-locales";
import { default as default127 } from "@opentiny/vue-locales";
import LogonUser from "@opentiny/vue-logon-user";
import { default as default128 } from "@opentiny/vue-logon-user";
import Logout from "@opentiny/vue-logout";
import { default as default129 } from "@opentiny/vue-logout";
import Mask from "@opentiny/vue-mask";
import { default as default130 } from "@opentiny/vue-mask";
import Menu from "@opentiny/vue-menu";
import { default as default131 } from "@opentiny/vue-menu";
import Message from "@opentiny/vue-message";
import { default as default132 } from "@opentiny/vue-message";
import Milestone from "@opentiny/vue-milestone";
import { default as default133 } from "@opentiny/vue-milestone";
import MiniPicker from "@opentiny/vue-mini-picker";
import { default as default134 } from "@opentiny/vue-mini-picker";
import Modal from "@opentiny/vue-modal";
import { default as default135 } from "@opentiny/vue-modal";
import MonthRange from "@opentiny/vue-month-range";
import { default as default136 } from "@opentiny/vue-month-range";
import MonthTable from "@opentiny/vue-month-table";
import { default as default137 } from "@opentiny/vue-month-table";
import MultiSelect from "@opentiny/vue-multi-select";
import { default as default138 } from "@opentiny/vue-multi-select";
import NavBar from "@opentiny/vue-nav-bar";
import { default as default139 } from "@opentiny/vue-nav-bar";
import NavMenu from "@opentiny/vue-nav-menu";
import { default as default140 } from "@opentiny/vue-nav-menu";
import Notify from "@opentiny/vue-notify";
import { default as default141 } from "@opentiny/vue-notify";
import Numeric from "@opentiny/vue-numeric";
import { default as default142 } from "@opentiny/vue-numeric";
import Option from "@opentiny/vue-option";
import { default as default143 } from "@opentiny/vue-option";
import OptionGroup from "@opentiny/vue-option-group";
import { default as default144 } from "@opentiny/vue-option-group";
import Pager from "@opentiny/vue-pager";
import { default as default145 } from "@opentiny/vue-pager";
import PagerItem from "@opentiny/vue-pager-item";
import { default as default146 } from "@opentiny/vue-pager-item";
import Panel from "@opentiny/vue-panel";
import { default as default147 } from "@opentiny/vue-panel";
import Picker from "@opentiny/vue-picker";
import { default as default148 } from "@opentiny/vue-picker";
import PickerColumn from "@opentiny/vue-picker-column";
import { default as default149 } from "@opentiny/vue-picker-column";
import PopUpload from "@opentiny/vue-pop-upload";
import { default as default150 } from "@opentiny/vue-pop-upload";
import Popconfirm from "@opentiny/vue-popconfirm";
import { default as default151 } from "@opentiny/vue-popconfirm";
import Popeditor from "@opentiny/vue-popeditor";
import { default as default152 } from "@opentiny/vue-popeditor";
import Popover from "@opentiny/vue-popover";
import { default as default153 } from "@opentiny/vue-popover";
import Popup from "@opentiny/vue-popup";
import { default as default154 } from "@opentiny/vue-popup";
import Progress from "@opentiny/vue-progress";
import { default as default155 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default156 } from "@opentiny/vue-pull-refresh";
import QrCode from "@opentiny/vue-qr-code";
import { default as default157 } from "@opentiny/vue-qr-code";
import QueryBuilder from "@opentiny/vue-query-builder";
import { default as default158 } from "@opentiny/vue-query-builder";
import Radio from "@opentiny/vue-radio";
import { default as default159 } from "@opentiny/vue-radio";
import RadioButton from "@opentiny/vue-radio-button";
import { default as default160 } from "@opentiny/vue-radio-button";
import RadioGroup from "@opentiny/vue-radio-group";
import { default as default161 } from "@opentiny/vue-radio-group";
import Rate from "@opentiny/vue-rate";
import { default as default162 } from "@opentiny/vue-rate";
import Record from "@opentiny/vue-record";
import { default as default163 } from "@opentiny/vue-record";
import RecycleScroller from "@opentiny/vue-recycle-scroller";
import { default as default164 } from "@opentiny/vue-recycle-scroller";
import RichTextEditor from "@opentiny/vue-rich-text-editor";
import { default as default165 } from "@opentiny/vue-rich-text-editor";
import River from "@opentiny/vue-river";
import { default as default166 } from "@opentiny/vue-river";
import Roles from "@opentiny/vue-roles";
import { default as default167 } from "@opentiny/vue-roles";
import Row from "@opentiny/vue-row";
import { default as default168 } from "@opentiny/vue-row";
import ScrollText from "@opentiny/vue-scroll-text";
import { default as default169 } from "@opentiny/vue-scroll-text";
import Scrollbar from "@opentiny/vue-scrollbar";
import { default as default170 } from "@opentiny/vue-scrollbar";
import Search from "@opentiny/vue-search";
import { default as default171 } from "@opentiny/vue-search";
import Select from "@opentiny/vue-select";
import { default as default172 } from "@opentiny/vue-select";
import SelectDropdown from "@opentiny/vue-select-dropdown";
import { default as default173 } from "@opentiny/vue-select-dropdown";
import SelectMobile from "@opentiny/vue-select-mobile";
import { default as default174 } from "@opentiny/vue-select-mobile";
import SelectView from "@opentiny/vue-select-view";
import { default as default175 } from "@opentiny/vue-select-view";
import SelectedBox from "@opentiny/vue-selected-box";
import { default as default176 } from "@opentiny/vue-selected-box";
import SlideBar from "@opentiny/vue-slide-bar";
import { default as default177 } from "@opentiny/vue-slide-bar";
import Slider from "@opentiny/vue-slider";
import { default as default178 } from "@opentiny/vue-slider";
import SliderButton from "@opentiny/vue-slider-button";
import { default as default179 } from "@opentiny/vue-slider-button";
import SliderButtonGroup from "@opentiny/vue-slider-button-group";
import { default as default180 } from "@opentiny/vue-slider-button-group";
import Split from "@opentiny/vue-split";
import { default as default181 } from "@opentiny/vue-split";
import StandardListItem from "@opentiny/vue-standard-list-item";
import { default as default182 } from "@opentiny/vue-standard-list-item";
import Steps from "@opentiny/vue-steps";
import { default as default183 } from "@opentiny/vue-steps";
import SvgIcon from "@opentiny/vue-svg-icon";
import { default as default184 } from "@opentiny/vue-svg-icon";
import Switch from "@opentiny/vue-switch";
import { default as default185 } from "@opentiny/vue-switch";
import TabItem from "@opentiny/vue-tab-item";
import { default as default186 } from "@opentiny/vue-tab-item";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default187 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default188 } from "@opentiny/vue-tabbar-item";
import Table from "@opentiny/vue-table";
import { default as default189 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default190 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default191 } from "@opentiny/vue-tag";
import TagGroup from "@opentiny/vue-tag-group";
import { default as default192 } from "@opentiny/vue-tag-group";
import TextPopup from "@opentiny/vue-text-popup";
import { default as default193 } from "@opentiny/vue-text-popup";
import Time from "@opentiny/vue-time";
import { default as default194 } from "@opentiny/vue-time";
import TimeLine from "@opentiny/vue-time-line";
import { default as default195 } from "@opentiny/vue-time-line";
import TimePanel from "@opentiny/vue-time-panel";
import { default as default196 } from "@opentiny/vue-time-panel";
import TimePicker from "@opentiny/vue-time-picker";
import { default as default197 } from "@opentiny/vue-time-picker";
import TimePickerMobile from "@opentiny/vue-time-picker-mobile";
import { default as default198 } from "@opentiny/vue-time-picker-mobile";
import TimeRange from "@opentiny/vue-time-range";
import { default as default199 } from "@opentiny/vue-time-range";
import TimeSelect from "@opentiny/vue-time-select";
import { default as default200 } from "@opentiny/vue-time-select";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { default as default201 } from "@opentiny/vue-time-spinner";
import TimelineItem from "@opentiny/vue-timeline-item";
import { default as default202 } from "@opentiny/vue-timeline-item";
import Toast from "@opentiny/vue-toast";
import { default as default203 } from "@opentiny/vue-toast";
import ToggleMenu from "@opentiny/vue-toggle-menu";
import { default as default204 } from "@opentiny/vue-toggle-menu";
import Tooltip from "@opentiny/vue-tooltip";
import { default as default205 } from "@opentiny/vue-tooltip";
import TopBox from "@opentiny/vue-top-box";
import { default as default206 } from "@opentiny/vue-top-box";
import Transfer from "@opentiny/vue-transfer";
import { default as default207 } from "@opentiny/vue-transfer";
import TransferPanel from "@opentiny/vue-transfer-panel";
import { default as default208 } from "@opentiny/vue-transfer-panel";
import Tree from "@opentiny/vue-tree";
import { default as default209 } from "@opentiny/vue-tree";
import TreeMenu from "@opentiny/vue-tree-menu";
import { default as default210 } from "@opentiny/vue-tree-menu";
import Upload from "@opentiny/vue-upload";
import { default as default211 } from "@opentiny/vue-upload";
import UploadDragger from "@opentiny/vue-upload-dragger";
import { default as default212 } from "@opentiny/vue-upload-dragger";
import UploadList from "@opentiny/vue-upload-list";
import { default as default213 } from "@opentiny/vue-upload-list";
import User from "@opentiny/vue-user";
import { default as default214 } from "@opentiny/vue-user";
import UserAccount from "@opentiny/vue-user-account";
import { default as default215 } from "@opentiny/vue-user-account";
import UserContact from "@opentiny/vue-user-contact";
import { default as default216 } from "@opentiny/vue-user-contact";
import UserHead from "@opentiny/vue-user-head";
import { default as default217 } from "@opentiny/vue-user-head";
import UserHeadGroup from "@opentiny/vue-user-head-group";
import { default as default218 } from "@opentiny/vue-user-head-group";
import UserLink from "@opentiny/vue-user-link";
import { default as default219 } from "@opentiny/vue-user-link";
import Watermark from "@opentiny/vue-watermark";
import { default as default220 } from "@opentiny/vue-watermark";
import Wheel from "@opentiny/vue-wheel";
import { default as default221 } from "@opentiny/vue-wheel";
import Wizard from "@opentiny/vue-wizard";
import { default as default222 } from "@opentiny/vue-wizard";
import YearRange from "@opentiny/vue-year-range";
import { default as default223 } from "@opentiny/vue-year-range";
import YearTable from "@opentiny/vue-year-table";
import { default as default224 } from "@opentiny/vue-year-table";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionMenu, ActionSheet, Alert, Amount, Anchor, Area, AsyncFlowchart, Autocomplete, AutonaviMap, Avatar, Badge, BaiduMap, Breadcrumb, BreadcrumbItem, BulletinBoard, Button, ButtonGroup, Calendar, CalendarBar, CalendarView, Card, CardGroup, CardTemplate, Carousel, CarouselItem, Cascader, CascaderMenu, CascaderMobile, CascaderNode, CascaderPanel, CascaderSelect, CascaderView, Cell, Chart, ChartBar, ChartBoxplot, ChartCandle, Core, ChartFunnel, ChartGauge, ChartGraph, ChartHeatmap, ChartHistogram, ChartLine, ChartLiquidfill, ChartMap, ChartPie, ChartRadar, ChartRing, ChartSankey, ChartScatter, ChartSunburst, ChartTree, ChartWaterfall, ChartWordcloud, Checkbox, CheckboxButton, CheckboxGroup, Col, Collapse, CollapseItem, CollapseTransition, ColorPicker, ColorSelectPanel, ColumnListGroup, ColumnListItem, Company, ConfigProvider, Container, Country, CreditCard, CreditCardForm, Crop, Currency, DatePanel, DatePicker, DatePickerMobileFirst, DateRange, DateTable, Dept, DetailPage, DialogBox, DialogSelect, Divider, Drawer, DropRoles, DropTimes, Dropdown, DropdownItem, DropdownMenu, DynamicScroller, DynamicScrollerItem, Espace, Exception, FallMenu, FileUpload, Filter, FilterBar, FilterBox, FilterPanel, Floatbar, FloatingButton, Flowchart, Form, FormItem, Fullscreen, Grid, GridColumn, GridManager, GridToolbar, Guide, Hrapprover, Image, ImageViewer, IndexBar, IndexBarAnchor, Input, IpAddress, Label, Layout, Link, LinkMenu, List, Loading, Locales, LogonUser, Logout, Mask, Menu, Message, Milestone, MiniPicker, Modal, MonthRange, MonthTable, MultiSelect, NavBar, NavMenu, Notify, Numeric, Option, OptionGroup, Pager, PagerItem, Panel, Picker, PickerColumn, PopUpload, Popconfirm, Popeditor, Popover, Popup, Progress, PullRefresh, QrCode, QueryBuilder, Radio, RadioButton, RadioGroup, Rate, Record, RecycleScroller, RichTextEditor, River, Roles, Row, ScrollText, Scrollbar, Search, Select, SelectDropdown, SelectMobile, SelectView, SelectedBox, SlideBar, Slider, SliderButton, SliderButtonGroup, Split, StandardListItem, Steps, SvgIcon, Switch, TabItem, Tabbar, TabbarItem, Table, Tabs, Tag, TagGroup, TextPopup, Time, TimeLine, TimePanel, TimePicker, TimePickerMobile, TimeRange, TimeSelect, TimeSpinner, TimelineItem, Toast, ToggleMenu, Tooltip, TopBox, Transfer, TransferPanel, Tree, TreeMenu, Upload, UploadDragger, UploadList, User, UserAccount, UserContact, UserHead, UserHeadGroup, UserLink, Watermark, Wheel, Wizard, YearRange, YearTable];
var install = function install2(app, opts) {
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.13.0";
var index = {
  ActionMenu,
  ActionSheet,
  Alert,
  Amount,
  Anchor,
  Area,
  AsyncFlowchart,
  Autocomplete,
  AutonaviMap,
  Avatar,
  Badge,
  BaiduMap,
  Breadcrumb,
  BreadcrumbItem,
  BulletinBoard,
  Button,
  ButtonGroup,
  Calendar,
  CalendarBar,
  CalendarView,
  Card,
  CardGroup,
  CardTemplate,
  Carousel,
  CarouselItem,
  Cascader,
  CascaderMenu,
  CascaderMobile,
  CascaderNode,
  CascaderPanel,
  CascaderSelect,
  CascaderView,
  Cell,
  Chart,
  ChartBar,
  ChartBoxplot,
  ChartCandle,
  ChartCore: Core,
  ChartFunnel,
  ChartGauge,
  ChartGraph,
  ChartHeatmap,
  ChartHistogram,
  ChartLine,
  ChartLiquidfill,
  ChartMap,
  ChartPie,
  ChartRadar,
  ChartRing,
  ChartSankey,
  ChartScatter,
  ChartSunburst,
  ChartTree,
  ChartWaterfall,
  ChartWordcloud,
  Checkbox,
  CheckboxButton,
  CheckboxGroup,
  Col,
  Collapse,
  CollapseItem,
  CollapseTransition,
  ColorPicker,
  ColorSelectPanel,
  ColumnListGroup,
  ColumnListItem,
  Company,
  ConfigProvider,
  Container,
  Country,
  CreditCard,
  CreditCardForm,
  Crop,
  Currency,
  DatePanel,
  DatePicker,
  DatePickerMobileFirst,
  DateRange,
  DateTable,
  Dept,
  DetailPage,
  DialogBox,
  DialogSelect,
  Divider,
  Drawer,
  DropRoles,
  DropTimes,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DynamicScroller,
  DynamicScrollerItem,
  Espace,
  Exception,
  FallMenu,
  FileUpload,
  Filter,
  FilterBar,
  FilterBox,
  FilterPanel,
  Floatbar,
  FloatingButton,
  Flowchart,
  Form,
  FormItem,
  Fullscreen,
  Grid,
  GridColumn,
  GridManager,
  GridToolbar,
  Guide,
  Hrapprover,
  Image,
  ImageViewer,
  IndexBar,
  IndexBarAnchor,
  Input,
  IpAddress,
  Label,
  Layout,
  Link,
  LinkMenu,
  List,
  Loading,
  Locales,
  LogonUser,
  Logout,
  Mask,
  Menu,
  Message,
  Milestone,
  MiniPicker,
  Modal,
  MonthRange,
  MonthTable,
  MultiSelect,
  NavBar,
  NavMenu,
  Notify,
  Numeric,
  Option,
  OptionGroup,
  Pager,
  PagerItem,
  Panel,
  Picker,
  PickerColumn,
  PopUpload,
  Popconfirm,
  Popeditor,
  Popover,
  Popup,
  Progress,
  PullRefresh,
  QrCode,
  QueryBuilder,
  Radio,
  RadioButton,
  RadioGroup,
  Rate,
  Record,
  RecycleScroller,
  RichTextEditor,
  River,
  Roles,
  Row,
  ScrollText,
  Scrollbar,
  Search,
  Select,
  SelectDropdown,
  SelectMobile,
  SelectView,
  SelectedBox,
  SlideBar,
  Slider,
  SliderButton,
  SliderButtonGroup,
  Split,
  StandardListItem,
  Steps,
  SvgIcon,
  Switch,
  TabItem,
  Tabbar,
  TabbarItem,
  Table,
  Tabs,
  Tag,
  TagGroup,
  TextPopup,
  Time,
  TimeLine,
  TimePanel,
  TimePicker,
  TimePickerMobile,
  TimeRange,
  TimeSelect,
  TimeSpinner,
  TimelineItem,
  Toast,
  ToggleMenu,
  Tooltip,
  TopBox,
  Transfer,
  TransferPanel,
  Tree,
  TreeMenu,
  Upload,
  UploadDragger,
  UploadList,
  User,
  UserAccount,
  UserContact,
  UserHead,
  UserHeadGroup,
  UserLink,
  Watermark,
  Wheel,
  Wizard,
  YearRange,
  YearTable,
  install
};
export {
  default2 as ActionMenu,
  default3 as ActionSheet,
  default4 as Alert,
  default5 as Amount,
  default6 as Anchor,
  default7 as Area,
  default8 as AsyncFlowchart,
  default9 as Autocomplete,
  default10 as AutonaviMap,
  default11 as Avatar,
  default12 as Badge,
  default13 as BaiduMap,
  default14 as Breadcrumb,
  default15 as BreadcrumbItem,
  default16 as BulletinBoard,
  default17 as Button,
  default18 as ButtonGroup,
  default19 as Calendar,
  default20 as CalendarBar,
  default21 as CalendarView,
  default22 as Card,
  default23 as CardGroup,
  default24 as CardTemplate,
  default25 as Carousel,
  default26 as CarouselItem,
  default27 as Cascader,
  default28 as CascaderMenu,
  default29 as CascaderMobile,
  default30 as CascaderNode,
  default31 as CascaderPanel,
  default32 as CascaderSelect,
  default33 as CascaderView,
  default34 as Cell,
  default35 as Chart,
  default36 as ChartBar,
  default38 as ChartBoxplot,
  default39 as ChartCandle,
  default40 as ChartCore,
  default41 as ChartFunnel,
  default42 as ChartGauge,
  default43 as ChartGraph,
  default44 as ChartHeatmap,
  default45 as ChartHistogram,
  default46 as ChartLine,
  default47 as ChartLiquidfill,
  default48 as ChartMap,
  default49 as ChartPie,
  default50 as ChartRadar,
  default51 as ChartRing,
  default52 as ChartSankey,
  default53 as ChartScatter,
  default54 as ChartSunburst,
  default55 as ChartTree,
  default56 as ChartWaterfall,
  default57 as ChartWordcloud,
  default58 as Checkbox,
  default59 as CheckboxButton,
  default60 as CheckboxGroup,
  default61 as Col,
  default62 as Collapse,
  default63 as CollapseItem,
  default64 as CollapseTransition,
  default65 as ColorPicker,
  default66 as ColorSelectPanel,
  default67 as ColumnListGroup,
  default68 as ColumnListItem,
  default69 as Company,
  default70 as ConfigProvider,
  default71 as Container,
  default72 as Country,
  default73 as CreditCard,
  default74 as CreditCardForm,
  default75 as Crop,
  default76 as Currency,
  default77 as DatePanel,
  default78 as DatePicker,
  default79 as DatePickerMobileFirst,
  default80 as DateRange,
  default81 as DateTable,
  default82 as Dept,
  default83 as DetailPage,
  default84 as DialogBox,
  default85 as DialogSelect,
  default86 as Divider,
  default87 as Drawer,
  default88 as DropRoles,
  default89 as DropTimes,
  default90 as Dropdown,
  default91 as DropdownItem,
  default92 as DropdownMenu,
  default93 as DynamicScroller,
  default94 as DynamicScrollerItem,
  default95 as Espace,
  default96 as Exception,
  default97 as FallMenu,
  default98 as FileUpload,
  default99 as Filter,
  default100 as FilterBar,
  default101 as FilterBox,
  default102 as FilterPanel,
  default103 as Floatbar,
  default104 as FloatingButton,
  default105 as Flowchart,
  default106 as Form,
  default107 as FormItem,
  default108 as Fullscreen,
  default109 as Grid,
  default110 as GridColumn,
  default111 as GridManager,
  default112 as GridToolbar,
  default113 as Guide,
  default114 as Hrapprover,
  default115 as Image,
  default116 as ImageViewer,
  default117 as IndexBar,
  default118 as IndexBarAnchor,
  default119 as Input,
  default120 as IpAddress,
  default121 as Label,
  default122 as Layout,
  default123 as Link,
  default124 as LinkMenu,
  default125 as List,
  default126 as Loading,
  default127 as Locales,
  default128 as LogonUser,
  default129 as Logout,
  default130 as Mask,
  default131 as Menu,
  default132 as Message,
  default133 as Milestone,
  default134 as MiniPicker,
  default135 as Modal,
  default136 as MonthRange,
  default137 as MonthTable,
  default138 as MultiSelect,
  default139 as NavBar,
  default140 as NavMenu,
  default141 as Notify,
  default142 as Numeric,
  default143 as Option,
  default144 as OptionGroup,
  default145 as Pager,
  default146 as PagerItem,
  default147 as Panel,
  default148 as Picker,
  default149 as PickerColumn,
  default150 as PopUpload,
  default151 as Popconfirm,
  default152 as Popeditor,
  default153 as Popover,
  default154 as Popup,
  default155 as Progress,
  default156 as PullRefresh,
  default157 as QrCode,
  default158 as QueryBuilder,
  default159 as Radio,
  default160 as RadioButton,
  default161 as RadioGroup,
  default162 as Rate,
  default163 as Record,
  default164 as RecycleScroller,
  default165 as RichTextEditor,
  default166 as River,
  default167 as Roles,
  default168 as Row,
  default169 as ScrollText,
  default170 as Scrollbar,
  default171 as Search,
  default172 as Select,
  default173 as SelectDropdown,
  default174 as SelectMobile,
  default175 as SelectView,
  default176 as SelectedBox,
  default177 as SlideBar,
  default178 as Slider,
  default179 as SliderButton,
  default180 as SliderButtonGroup,
  default181 as Split,
  default182 as StandardListItem,
  default183 as Steps,
  default184 as SvgIcon,
  default185 as Switch,
  default186 as TabItem,
  default187 as Tabbar,
  default188 as TabbarItem,
  default189 as Table,
  default190 as Tabs,
  default191 as Tag,
  default192 as TagGroup,
  default193 as TextPopup,
  default194 as Time,
  default195 as TimeLine,
  default196 as TimePanel,
  default197 as TimePicker,
  default198 as TimePickerMobile,
  default199 as TimeRange,
  default200 as TimeSelect,
  default201 as TimeSpinner,
  default202 as TimelineItem,
  default203 as Toast,
  default204 as ToggleMenu,
  default205 as Tooltip,
  default206 as TopBox,
  default207 as Transfer,
  default208 as TransferPanel,
  default209 as Tree,
  default210 as TreeMenu,
  default211 as Upload,
  default212 as UploadDragger,
  default213 as UploadList,
  default214 as User,
  default215 as UserAccount,
  default216 as UserContact,
  default217 as UserHead,
  default218 as UserHeadGroup,
  default219 as UserLink,
  default220 as Watermark,
  default221 as Wheel,
  default222 as Wizard,
  default223 as YearRange,
  default224 as YearTable,
  index as default,
  install,
  version
};
