// @opentiny/vue/mobile.js
import ActionSheet from "@opentiny/vue-action-sheet";
import { default as default2 } from "@opentiny/vue-action-sheet";
import Alert from "@opentiny/vue-alert";
import { default as default3 } from "@opentiny/vue-alert";
import Avatar from "@opentiny/vue-avatar";
import { default as default4 } from "@opentiny/vue-avatar";
import Badge from "@opentiny/vue-badge";
import { default as default5 } from "@opentiny/vue-badge";
import Button from "@opentiny/vue-button";
import { default as default6 } from "@opentiny/vue-button";
import CascaderSelect from "@opentiny/vue-cascader-select";
import { default as default7 } from "@opentiny/vue-cascader-select";
import Checkbox from "@opentiny/vue-checkbox";
import { default as default8 } from "@opentiny/vue-checkbox";
import CheckboxGroup from "@opentiny/vue-checkbox-group";
import { default as default9 } from "@opentiny/vue-checkbox-group";
import ColorPicker from "@opentiny/vue-color-picker";
import { default as default10 } from "@opentiny/vue-color-picker";
import Container from "@opentiny/vue-container";
import { default as default11 } from "@opentiny/vue-container";
import DatePicker from "@opentiny/vue-date-picker";
import { default as default12 } from "@opentiny/vue-date-picker";
import DialogBox from "@opentiny/vue-dialog-box";
import { default as default13 } from "@opentiny/vue-dialog-box";
import DropdownItem from "@opentiny/vue-dropdown-item";
import { default as default14 } from "@opentiny/vue-dropdown-item";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import { default as default15 } from "@opentiny/vue-dropdown-menu";
import Exception from "@opentiny/vue-exception";
import { default as default16 } from "@opentiny/vue-exception";
import FileUpload from "@opentiny/vue-file-upload";
import { default as default17 } from "@opentiny/vue-file-upload";
import Form from "@opentiny/vue-form";
import { default as default18 } from "@opentiny/vue-form";
import ImageViewer from "@opentiny/vue-image-viewer";
import { default as default19 } from "@opentiny/vue-image-viewer";
import IndexBar from "@opentiny/vue-index-bar";
import { default as default20 } from "@opentiny/vue-index-bar";
import Input from "@opentiny/vue-input";
import { default as default21 } from "@opentiny/vue-input";
import Label from "@opentiny/vue-label";
import { default as default22 } from "@opentiny/vue-label";
import List from "@opentiny/vue-list";
import { default as default23 } from "@opentiny/vue-list";
import Loading from "@opentiny/vue-loading";
import { default as default24 } from "@opentiny/vue-loading";
import Mask from "@opentiny/vue-mask";
import { default as default25 } from "@opentiny/vue-mask";
import MiniPicker from "@opentiny/vue-mini-picker";
import { default as default26 } from "@opentiny/vue-mini-picker";
import Modal from "@opentiny/vue-modal";
import { default as default27 } from "@opentiny/vue-modal";
import MultiSelect from "@opentiny/vue-multi-select";
import { default as default28 } from "@opentiny/vue-multi-select";
import NavBar from "@opentiny/vue-nav-bar";
import { default as default29 } from "@opentiny/vue-nav-bar";
import Numeric from "@opentiny/vue-numeric";
import { default as default30 } from "@opentiny/vue-numeric";
import PickerColumn from "@opentiny/vue-picker-column";
import { default as default31 } from "@opentiny/vue-picker-column";
import Popover from "@opentiny/vue-popover";
import { default as default32 } from "@opentiny/vue-popover";
import Progress from "@opentiny/vue-progress";
import { default as default33 } from "@opentiny/vue-progress";
import PullRefresh from "@opentiny/vue-pull-refresh";
import { default as default34 } from "@opentiny/vue-pull-refresh";
import Radio from "@opentiny/vue-radio";
import { default as default35 } from "@opentiny/vue-radio";
import Search from "@opentiny/vue-search";
import { default as default36 } from "@opentiny/vue-search";
import Slider from "@opentiny/vue-slider";
import { default as default37 } from "@opentiny/vue-slider";
import Switch from "@opentiny/vue-switch";
import { default as default38 } from "@opentiny/vue-switch";
import Tabbar from "@opentiny/vue-tabbar";
import { default as default39 } from "@opentiny/vue-tabbar";
import TabbarItem from "@opentiny/vue-tabbar-item";
import { default as default40 } from "@opentiny/vue-tabbar-item";
import Table from "@opentiny/vue-table";
import { default as default41 } from "@opentiny/vue-table";
import Tabs from "@opentiny/vue-tabs";
import { default as default42 } from "@opentiny/vue-tabs";
import Tag from "@opentiny/vue-tag";
import { default as default43 } from "@opentiny/vue-tag";
import TimeLine from "@opentiny/vue-time-line";
import { default as default44 } from "@opentiny/vue-time-line";
import Toast from "@opentiny/vue-toast";
import { default as default45 } from "@opentiny/vue-toast";
import UserHead from "@opentiny/vue-user-head";
import { default as default46 } from "@opentiny/vue-user-head";
import Wheel from "@opentiny/vue-wheel";
import { default as default47 } from "@opentiny/vue-wheel";
import { $prefix } from "@opentiny/vue-common";
var components = [ActionSheet, Alert, Avatar, Badge, Button, CascaderSelect, Checkbox, CheckboxGroup, ColorPicker, Container, DatePicker, DialogBox, DropdownItem, DropdownMenu, Exception, FileUpload, Form, ImageViewer, IndexBar, Input, Label, List, Loading, Mask, MiniPicker, Modal, MultiSelect, NavBar, Numeric, PickerColumn, Popover, Progress, PullRefresh, Radio, Search, Slider, Switch, Tabbar, TabbarItem, Table, Tabs, Tag, TimeLine, Toast, UserHead, Wheel];
var install = function install2(app, opts) {
  var _app$config;
  if (opts === void 0) {
    opts = {};
  }
  var regex = new RegExp("^" + $prefix);
  var appProperties = ((_app$config = app.config) == null ? void 0 : _app$config.globalProperties) || app.prototype;
  appProperties.tiny_mode = {
    value: "mobile"
  };
  components.forEach(function(component) {
    var name = component.name;
    var alias = opts.alias || opts.prefix;
    if (typeof component.install !== "function")
      return;
    if (name && alias) {
      app.component(name.replace(regex, alias), component);
    } else {
      component.install(app);
    }
  });
};
var version = "3.12.0";
var mobile = {
  ActionSheet,
  Alert,
  Avatar,
  Badge,
  Button,
  CascaderSelect,
  Checkbox,
  CheckboxGroup,
  ColorPicker,
  Container,
  DatePicker,
  DialogBox,
  DropdownItem,
  DropdownMenu,
  Exception,
  FileUpload,
  Form,
  ImageViewer,
  IndexBar,
  Input,
  Label,
  List,
  Loading,
  Mask,
  MiniPicker,
  Modal,
  MultiSelect,
  NavBar,
  Numeric,
  PickerColumn,
  Popover,
  Progress,
  PullRefresh,
  Radio,
  Search,
  Slider,
  Switch,
  Tabbar,
  TabbarItem,
  Table,
  Tabs,
  Tag,
  TimeLine,
  Toast,
  UserHead,
  Wheel,
  install
};
export {
  default2 as ActionSheet,
  default3 as Alert,
  default4 as Avatar,
  default5 as Badge,
  default6 as Button,
  default7 as CascaderSelect,
  default8 as Checkbox,
  default9 as CheckboxGroup,
  default10 as ColorPicker,
  default11 as Container,
  default12 as DatePicker,
  default13 as DialogBox,
  default14 as DropdownItem,
  default15 as DropdownMenu,
  default16 as Exception,
  default17 as FileUpload,
  default18 as Form,
  default19 as ImageViewer,
  default20 as IndexBar,
  default21 as Input,
  default22 as Label,
  default23 as List,
  default24 as Loading,
  default25 as Mask,
  default26 as MiniPicker,
  default27 as Modal,
  default28 as MultiSelect,
  default29 as NavBar,
  default30 as Numeric,
  default31 as PickerColumn,
  default32 as Popover,
  default33 as Progress,
  default34 as PullRefresh,
  default35 as Radio,
  default36 as Search,
  default37 as Slider,
  default38 as Switch,
  default39 as Tabbar,
  default40 as TabbarItem,
  default41 as Table,
  default42 as Tabs,
  default43 as Tag,
  default44 as TimeLine,
  default45 as Toast,
  default46 as UserHead,
  default47 as Wheel,
  mobile as default,
  install,
  version
};
