// @opentiny/vue-renderless/timeline-item/vue.js
import "../chunk-G2ADBYYC.js";
import {
  handleClick,
  getStatusCls,
  computedItemCls,
  getStatus,
  getDate,
  computedCurrent,
  computedIsReverse,
  computedWidth,
  computedItemStyle
} from "./index";
const api = ["state", "handleClick", "getStatusCls", "getStatus", "getDate", "rootProps"];
const renderless = (props, { computed, reactive, inject }, { t, emit, constants }) => {
  const api2 = {};
  const { timelineItems, nodes, props: rootProps } = inject("nodesInject");
  timelineItems.push(props.node);
  const state = reactive({
    nodesLength: computed(() => nodes.length),
    current: computed(() => api2.computedCurrent()),
    isReverse: computed(() => api2.computedIsReverse()),
    computedSpace: computed(() => api2.computedWidth(props.space || api2.rootProps.space)),
    computedItemCls: computed(() => api2.computedItemCls()),
    computedItemStyle: computed(() => api2.computedItemStyle()),
    computedLineWidth: computed(() => api2.computedWidth(props.lineWidth || api2.rootProps.lineWidth))
  });
  Object.assign(api2, {
    state,
    rootProps,
    getDate,
    computedCurrent: computedCurrent({ state, api: api2 }),
    computedIsReverse: computedIsReverse(api2),
    computedItemCls: computedItemCls({ props, api: api2, state }),
    computedItemStyle: computedItemStyle({ props, state, api: api2 }),
    computedWidth: computedWidth(),
    getStatus: getStatus({ state, t }),
    handleClick: handleClick({ emit, state }),
    getStatusCls: getStatusCls({ constants, state })
  });
  return api2;
};
export {
  api,
  renderless
};
