// @opentiny/vue-user-head/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/user-head/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconUser } from "@opentiny/vue-icon";
import { openBlock, createElementBlock, createElementVNode, normalizeStyle, normalizeClass, renderSlot, createBlock, resolveDynamicComponent, createCommentVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconUser: IconUser()
  },
  emits: ["click", "mouseenter"],
  props: [].concat(props, ["min", "round", "color", "backgroundColor", "type", "modelValue", "messageTotal", "messageType", "messageUpperLimit", "size"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["title"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "inline-block relative",
      "data-tag": "tiny-user-head",
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
      }),
      onMouseenter: _cache[1] || (_cache[1] = function() {
        return _ctx.mouseEnter && _ctx.mouseEnter.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      {
        "data-tag": "tiny-user-head-body",
        style: normalizeStyle([_ctx.state.style, {
          height: _ctx.state.size + "px",
          width: _ctx.state.size + "px",
          lineHeight: _ctx.state.size + "px"
        }]),
        class: normalizeClass(["overflow-hidden w-10 h-10  text-center rounded  bg-center bg-no-repeat bg-cover", {
          "h-8 w-8 leading-8": _ctx.min
        }, {
          "[&_span]:overflow-hidden [&_span]:flex [&_span]:justify-center [&_span]:h-full px-0.5": _ctx.type === "label"
        }, {
          "rounded-full": _ctx.round
        }, {
          "[&_svg]:top-1/2  [&_svg]:absolute [&_svg]:left-1/2  [&_svg]:-translate-y-1/2 [&_svg]:-translate-x-1/2": _ctx.type === "icon"
        }, {
          "text-sm": _ctx.type === "icon" && _ctx.min
        }])
      },
      [renderSlot(_ctx.$slots, "default", {}, function() {
        return [_ctx.type === "icon" ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.internalValue), {
          key: 0,
          class: "inline-block w-6 h-6",
          style: normalizeStyle([{
            height: _ctx.state.size / 1.5 + "px",
            width: _ctx.state.size / 1.5 + "px",
            lineHeight: _ctx.state.size / 1.5 + "px",
            top: "" + _ctx.state.size / 2
          }])
        }, null, 8, ["style"])) : createCommentVNode("v-if", true), _ctx.type === "label" ? (openBlock(), createElementBlock("span", {
          key: 1,
          style: normalizeStyle(_ctx.state.fontSize),
          title: _ctx.state.internalValue
        }, toDisplayString(_ctx.state.label), 13, _hoisted_1)) : createCommentVNode("v-if", true)];
      })],
      6
      /* CLASS, STYLE */
    ), _ctx.messageTotal ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        "data-tag": "tiny-user-head-message",
        class: normalizeClass(["absolute rounded-lg text-color-text-primary text-center box-content bg-color-error", _ctx.messageType === "basic" ? "w-0 h-1 px-0.5 border-2 -right-1.5 border-color-bg-1 border-solid -top-1 left-9" : "min-w-[0.625rem] w-auto left-7 flex items-center -top-2.5 px-0.5 text-xs h-3.5 border-2 border-color-bg-1 border-solid", {
          "-top-1 h-2 min-w-2 leading-none rounded": _ctx.messageType === "icon"
        }, {
          "top-0 -right-1.5": (_ctx.messageType === "basic" || _ctx.messageType === "icon") && _ctx.round
        }, {
          "-top-1.5 left-8 -right-1.5 w-3 h-3": (_ctx.messageType === "basic" || _ctx.messageType === "icon") && _ctx.min
        }, {
          "top-0": (_ctx.messageType === "basic" || _ctx.messageType === "icon") && _ctx.round && _ctx.min
        }])
      },
      toDisplayString(_ctx.state.message),
      3
      /* TEXT, CLASS */
    )) : createCommentVNode("v-if", true)],
    32
    /* NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
