// @opentiny/vue-pager-item/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { $prefix, $props, $setup } from "@opentiny/vue-common";
import { IconPopup, IconDoubleLeft, IconDoubleRight } from "@opentiny/vue-icon";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var PagerItem = {
  name: $prefix + "PagerItem",
  props: _extends({}, $props, {
    disabled: {
      type: Boolean,
      default: false
    },
    currentPage: Number,
    pageCount: Number,
    pagerCount: Number,
    popupIcon: {
      type: Object,
      default: IconPopup
    },
    doubleLeftIcon: {
      type: Object,
      default: IconDoubleLeft
    },
    doubleRightIcon: {
      type: Object,
      default: IconDoubleRight
    },
    isBeforePageChange: Boolean
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
};
var version = "3.13.0";
PagerItem.install = function(Vue) {
  Vue.component(PagerItem.name, PagerItem);
};
PagerItem.version = version;
export {
  PagerItem as default
};
