// @opentiny/vue-button/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/button/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconLoading } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/button/index.css";
import { resolveComponent, openBlock, createElementBlock, mergeProps, createBlock, createCommentVNode, resolveDynamicComponent, normalizeClass, renderSlot, createElementVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["click", "hook-updated"],
  props: [].concat(props, ["type", "text", "size", "icon", "resetTime", "nativeType", "loading", "disabled", "plain", "autofocus", "round", "circle", "tabindex", "customClass", "ghost"]),
  components: {
    IconLoading: iconLoading()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["disabled", "autofocus", "type", "tabindex"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_loading = resolveComponent("icon-loading");
  return openBlock(), createElementBlock("button", mergeProps({
    class: ["tiny-button", [_ctx.type ? "tiny-button--" + _ctx.type : "", _ctx.size ? "tiny-button--" + _ctx.size : "", {
      "is-disabled": _ctx.state.buttonDisabled,
      "is-loading": _ctx.loading,
      "is-plain": _ctx.state.plain,
      "is-ghost": _ctx.ghost,
      "is-round": _ctx.round,
      "is-circle": _ctx.circle,
      "is-icon": _ctx.icon && !_ctx.loading && (_ctx.text || _ctx.slots.default),
      "is-only-icon": _ctx.icon && !_ctx.loading && !(_ctx.text || _ctx.slots.default)
    }]],
    onClick: _cache[0] || (_cache[0] = function() {
      return _ctx.handleClick && _ctx.handleClick.apply(_ctx, arguments);
    }),
    disabled: _ctx.state.buttonDisabled || _ctx.loading,
    autofocus: _ctx.autofocus,
    type: _ctx.nativeType,
    tabindex: _ctx.tabindex
  }, _ctx.a(_ctx.$attrs, ["class", "style"], true)), [_ctx.loading ? (openBlock(), createBlock(_component_icon_loading, {
    key: 0,
    class: "tiny-icon-loading tiny-svg-size"
  })) : createCommentVNode("v-if", true), _ctx.icon && !_ctx.loading ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.icon), {
    key: 1,
    class: normalizeClass({
      "is-text": _ctx.text || _ctx.slots.default
    })
  }, null, 8, ["class"])) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default", {}, function() {
    return [createElementVNode(
      "span",
      null,
      toDisplayString(_ctx.text),
      1
      /* TEXT */
    )];
  })], 16, _hoisted_1);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
