// @opentiny/vue-filter-panel/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/filter-panel/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import FilterBox from "@opentiny/vue-filter-box";
import Popover from "@opentiny/vue-popover";
import { resolveComponent, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, renderSlot } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-theme/filter-panel/index.css";
var _sfc_main = defineComponent({
  name: $prefix + "FilterPanel",
  emits: ["handle-clear", "visible-change"],
  components: {
    TinyPopover: Popover,
    TinyFilterBox: FilterBox
  },
  props: _extends({}, $props, {
    label: String,
    value: String,
    tip: String,
    popperClass: {
      type: String,
      default: ""
    },
    clearable: {
      type: Boolean,
      default: true
    },
    disabled: {
      type: Boolean,
      default: false
    },
    placement: {
      type: String,
      default: "bottom-start"
    },
    popperAppendToBody: {
      type: Boolean,
      default: true
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "tiny-filter-panel"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_filter_box = resolveComponent("tiny-filter-box");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_popover, {
    ref: "popover",
    modelValue: _ctx.state.visible,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.visible = $event;
    }),
    placement: _ctx.placement,
    "visible-arrow": false,
    "popper-class": _ctx.popperClass + " tiny-filter-panel__popover",
    "append-to-body": _ctx.popperAppendToBody,
    trigger: "manual",
    onShow: _ctx.popoverShow,
    onHide: _ctx.popoverHide
  }, {
    reference: withCtx(function() {
      return [createElementVNode("div", null, [createVNode(_component_tiny_filter_box, {
        ref: "filterBox",
        onClick: _ctx.togglePanel,
        onHandleClear: _ctx.handleClear,
        "show-close": _ctx.clearable,
        disabled: _ctx.disabled,
        label: _ctx.label,
        tip: _ctx.tip,
        value: _ctx.value,
        "drop-down-visible": _ctx.state.visible
      }, null, 8, ["onClick", "onHandleClear", "show-close", "disabled", "label", "tip", "value", "drop-down-visible"])])];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", null, [renderSlot(_ctx.$slots, "default")])];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["modelValue", "placement", "popper-class", "append-to-body", "onShow", "onHide"])]);
}
var FilterPanel = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
FilterPanel.install = function(Vue) {
  Vue.component(FilterPanel.name, FilterPanel);
};
FilterPanel.version = version;
export {
  FilterPanel as default
};
