// @opentiny/vue-top-box/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/top-box/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Button from "@opentiny/vue-button";
import { iconClose, iconSuccess, iconError, iconHelp, iconWarning, iconInfo } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, normalizeStyle, resolveDynamicComponent, renderSlot, createElementBlock, toDisplayString, Fragment, renderList, createTextVNode, createCommentVNode, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  components: {
    TinyButton: Button,
    IconClose: iconClose(),
    IconSuccess: iconSuccess(),
    IconError: iconError(),
    IconHelp: iconHelp(),
    IconWarning: iconWarning(),
    IconInfo: iconInfo()
  },
  props: [].concat(props, ["emitter"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  class: "tiny-top-box__content"
};
var _hoisted_2 = ["innerHTML"];
var _hoisted_3 = {
  key: 2,
  class: "tiny-top-box__toolbar"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createBlock(Transition, {
    name: "tiny-top-box-fade",
    onAfterLeave: _ctx.handleAfterLeave,
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-top-box", _ctx.state.type && !_ctx.state.iconClass ? "tiny-top-box--" + _ctx.state.type : "", _ctx.state.center ? "is-center" : "", _ctx.state.showClose ? "is-closable" : "", _ctx.state.customClass]),
          style: normalizeStyle(_ctx.state.positionStyle),
          onMouseenter: _cache[0] || (_cache[0] = function() {
            return _ctx.clearTimer && _ctx.clearTimer.apply(_ctx, arguments);
          }),
          onMouseleave: _cache[1] || (_cache[1] = function() {
            return _ctx.startTimer && _ctx.startTimer.apply(_ctx, arguments);
          }),
          role: "alert"
        },
        [_ctx.state.iconClass ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.iconClass), {
          key: 0,
          class: "tiny-svg-size"
        })) : (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.typeClass), {
          key: 1,
          class: "tiny-svg-size tiny-top-box__icon"
        })), renderSlot(_ctx.$slots, "default", {}, function() {
          return [!_ctx.state.dangerouslyUseHTMLString ? (openBlock(), createElementBlock(
            "p",
            _hoisted_1,
            toDisplayString(_ctx.state.message),
            1
            /* TEXT */
          )) : (openBlock(), createElementBlock("p", {
            key: 1,
            innerHTML: _ctx.state.message,
            class: "tiny-top-box__content"
          }, null, 8, _hoisted_2))];
        }), _ctx.state.buttons && _ctx.state.buttons.length ? (openBlock(), createElementBlock("div", _hoisted_3, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.buttons, function(item, key) {
            return openBlock(), createBlock(_component_tiny_button, {
              key,
              type: item.type || "",
              onClick: item.onClick
            }, {
              default: withCtx(function() {
                return [createTextVNode(
                  toDisplayString(item.text),
                  1
                  /* TEXT */
                )];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["type", "onClick"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])) : createCommentVNode("v-if", true), _ctx.state.showClose ? (openBlock(), createBlock(_component_icon_close, {
          key: 3,
          onClick: _ctx.close,
          class: "tiny-svg-size tiny-top-box__closeBtn"
        }, null, 8, ["onClick"])) : createCommentVNode("v-if", true)],
        38
        /* CLASS, STYLE, NEED_HYDRATION */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["onAfterLeave"]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
