// @opentiny/vue-roles/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/roles/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  AUTH: "authenticated",
  ANONYMOUS: "anonymous",
  GLOBAL: "global",
  ROLE: "role"
};
var Roles = defineComponent({
  name: $prefix + "Roles",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    fetchRole: Function,
    getEnvInfoSync: Function,
    getChangeRoleUrl: Function,
    getCustomized: Function,
    pushCustomized: Function
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Roles.install = function(Vue) {
  Vue.component(Roles.name, Roles);
};
Roles.version = version;
export {
  Roles as default
};
