// @opentiny/vue-image-viewer/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/image-viewer/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { directive } from "@opentiny/vue-common";
import { isObject } from "@opentiny/vue-renderless/common/type";
var TinyTouch = /* @__PURE__ */ function() {
  function TinyTouch2(element, tinyBinding, type) {
    var that = this;
    that.element = element;
    that.tinyBinding = tinyBinding;
    that.touchType = type;
    that.tinyVueTouches = {
      x: 0,
      y: 0
    };
    that.tinyVueMoves = true;
    that.tinyVueLeave = true;
    that.tinyLongTouch = true;
    that.tinyVueCallBack = isObject(tinyBinding.value) ? tinyBinding.value.fn : tinyBinding.value;
    that.element.addEventListener("touchstart", function(e) {
      that.start(e);
    });
    that.element.addEventListener("touchend", function(e) {
      that.end(e);
    });
    that.element.addEventListener("touchmove", function(e) {
      that.move(e);
    });
  }
  var _proto = TinyTouch2.prototype;
  _proto.start = function start(e) {
    var _this = this;
    if (e.touches >= 2) {
      return;
    }
    this.tinyVueMoves = true;
    this.tinyVueLeave = true;
    this.tinyLongTouch = true;
    this.tinyVueTouches = {
      x: e.changedTouches[0].pageX,
      y: e.changedTouches[0].pageY
    };
    this.time = setTimeout(function() {
      if (_this.tinyVueLeave && _this.tinyVueMoves) {
        _this.touchType == "longtap" && _this.tinyVueCallBack(_this.tinyBinding.value, e);
        _this.tinyLongTouch = false;
      }
    }, 1e3);
  };
  _proto.end = function end(e) {
    if (e.touches >= 2) {
      return;
    }
    var disX = e.changedTouches[0].pageX - this.tinyVueTouches.x;
    var disY = e.changedTouches[0].pageY - this.tinyVueTouches.y;
    clearTimeout(this.time);
    if (Math.abs(disX) > 10 || Math.abs(disY) > 100) {
      this.touchType == "swipe" && this.tinyVueCallBack(this.tinyBinding.value, e);
      if (Math.abs(disX) > Math.abs(disY)) {
        if (disX > 10) {
          this.touchType == "swiperight" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
        if (disX < -10) {
          this.touchType == "swipeleft" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
      } else {
        if (disY > 10) {
          this.touchType == "swipedown" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
        if (disY < -10) {
          this.touchType == "swipeup" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
      }
    } else {
      if (this.tinyLongTouch && this.tinyVueMoves) {
        this.touchType == "tap" && this.tinyVueCallBack(this.tinyBinding.value, e);
        this.tinyVueLeave = false;
      }
    }
  };
  _proto.move = function move() {
    this.tinyVueMoves = false;
  };
  return TinyTouch2;
}();
var mapDirective = function mapDirective2() {
  var deactives = {};
  var names = ["tap", "swipe", "swipeleft", "swiperight", "swipedown", "swipeup", "longtap"];
  names.forEach(function(name) {
    deactives[name] = directive({
      vTouch: {
        bind: function bind(el, tinyBinding) {
          new TinyTouch(el, tinyBinding, name);
        }
      }
    }).vTouch;
  });
  return deactives;
};
var touchDeactives = mapDirective();

import Dropdown from "@opentiny/vue-dropdown";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import DropdownItem from "@opentiny/vue-dropdown-item";
import ActionSheet from "@opentiny/vue-action-sheet";
import { iconClose, iconChevronLeft, iconChevronRight, iconZoomOut, iconZoomIn, iconFullscreen, iconMinscreen, iconRepeat, iconRefres, iconDel, iconDownload, iconArrowLeft, iconArrowRight, iconEditorMenuRight, iconPicture } from "@opentiny/vue-icon";
import { resolveComponent, resolveDirective, openBlock, createElementBlock, createVNode, Transition, withCtx, createElementVNode, normalizeClass, normalizeStyle, Fragment, renderList, toDisplayString, createCommentVNode, createBlock, resolveDynamicComponent, createTextVNode, withDirectives } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-renderless/common/type";
var _sfc_main = defineComponent({
  emits: ["close", "update:preview-visible", "delete", "change", "newImageList"],
  props: [].concat(props, ["urlList", "zIndex", "onSwitch", "onClose", "showIndex", "isThumbnail", "isMenuView", "modalView", "modalHeight", "bgColor"]),
  components: {
    TinyActionSheet: ActionSheet,
    TinyDropdown: Dropdown,
    TinyDropdownMenu: DropdownMenu,
    TinyDropdownItem: DropdownItem,
    IconClose: iconClose(),
    IconChevronLeft: iconChevronLeft(),
    IconChevronRight: iconChevronRight(),
    IconZoomOut: iconZoomOut(),
    IconZoomIn: iconZoomIn(),
    IconFullscreen: iconFullscreen(),
    IconMinscreen: iconMinscreen(),
    IconRepeat: iconRepeat(),
    IconRefres: iconRefres(),
    IconDel: iconDel(),
    IconDownload: iconDownload(),
    IconArrowLeft: iconArrowLeft(),
    IconArrowRight: iconArrowRight(),
    IconEditorMenuRight: iconEditorMenuRight(),
    IconPicture: iconPicture()
  },
  directives: _extends({}, touchDeactives),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  "data-tag": "tiny-image-viewer"
};
var _hoisted_2 = ["onClick"];
var _hoisted_3 = ["src"];
var _hoisted_4 = ["title"];
var _hoisted_5 = ["onClick"];
var _hoisted_6 = ["title"];
var _hoisted_7 = {
  class: "flex-auto h-full relative",
  "data-tag": "tiny-image-tools"
};
var _hoisted_8 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    "data-tag": "tiny-image-pc-mask",
    class: "absolute w-full h-full top-0 left-0 bg-color-bg-7"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_9 = {
  "data-tag": "tiny-image-pc-actions",
  class: "z-[1] absolute inline-flex items-center justify-center box-border select-none left-1/2 bottom-12 -translate-x-1/2 w-80 h-10 px-4 bg-color-bg-7 rounded cursor-pointer"
};
var _hoisted_10 = {
  class: "w-full h-full text-justify cursor-default text-2xl flex items-center justify-between fill-color-bg-1"
};
var _hoisted_11 = ["href"];
var _hoisted_12 = {
  key: 1,
  "data-tag": "tiny-image-pc-index",
  class: "z-10 absolute flex items-center justify-center box-border left-1/2 -translate-x-1/2 px-4 bottom-5"
};
var _hoisted_13 = {
  class: "h-4.5 inline-flex items-center text-color-text-primary"
};
var _hoisted_14 = {
  "data-tag": "tiny-image-pc-canvas",
  class: "z-10 flex w-full h-full items-center justify-center"
};
var _hoisted_15 = ["src"];
var _hoisted_16 = {
  class: "bg-color-bg-1 w-screen h-screen overflow-auto"
};
var _hoisted_17 = ["onTouchstart"];
var _hoisted_18 = ["src"];
var _hoisted_19 = ["title"];
var _hoisted_20 = {
  key: 1,
  "data-tag": "tiny-image-mobile-common",
  class: "w-screen h-screen fixed top-0 left-0 z-50 sm:hidden"
};
var _hoisted_21 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    "data-tag": "tiny-image-mobile-mask",
    class: "absolute w-full h-full top-0 left-0 bg-black"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_22 = {
  "data-tag": "tiny-image-mobile-index",
  class: "left-0 right-0 top-0 bottom-0 fixed"
};
var _hoisted_23 = {
  key: 0,
  class: "absolute flex items-center justify-center box-border left-1/2 -translate-x-1/2 px-4 bottom-12"
};
var _hoisted_24 = {
  class: "h-4.5 inline-flex items-center text-color-bg-1"
};
var _hoisted_25 = {
  "data-tag": "tiny-image-mobile-canvas",
  ref: "canvasBox",
  class: "w-full h-full"
};
var _hoisted_26 = {
  class: "absolute top-0 left-0 right-0 bottom-0 text-center transition-transform inline-block duration-300"
};
var _hoisted_27 = ["src"];
var _hoisted_28 = {
  key: 0,
  "data-tag": "tiny-image-isthumbnail-preview",
  class: "w-screen h-screen fixed top-0 left-0 z-50 sm:hidden"
};
var _hoisted_29 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    "data-tag": "tiny-image-mobile-mask",
    class: "absolute w-full h-full top-0 left-0 bg-black"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_30 = {
  "data-tag": "tiny-image-isthumbnail-index",
  class: "left-0 right-0 top-0 bottom-0 fixed"
};
var _hoisted_31 = {
  key: 0,
  class: "absolute flex items-center justify-center box-border left-1/2 -translate-x-1/2 px-4 bottom-12"
};
var _hoisted_32 = {
  class: "h-4.5 inline-flex items-center text-color-bg-1"
};
var _hoisted_33 = {
  "data-tag": "tiny-image-isthumbnail-canvas",
  ref: "thumbnailCanvasBox",
  class: "w-full h-full"
};
var _hoisted_34 = {
  class: "absolute top-0 left-0 right-0 bottom-0 text-center transition-transform inline-block duration-300"
};
var _hoisted_35 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_picture = resolveComponent("icon-picture");
  var _component_icon_close = resolveComponent("icon-close");
  var _component_icon_zoom_out = resolveComponent("icon-zoom-out");
  var _component_icon_zoom_in = resolveComponent("icon-zoom-in");
  var _component_icon_refres = resolveComponent("icon-refres");
  var _component_icon_repeat = resolveComponent("icon-repeat");
  var _component_icon_del = resolveComponent("icon-del");
  var _component_icon_download = resolveComponent("icon-download");
  var _component_icon_editor_menu_right = resolveComponent("icon-editor-menu-right");
  var _component_tiny_dropdown_item = resolveComponent("tiny-dropdown-item");
  var _component_tiny_dropdown_menu = resolveComponent("tiny-dropdown-menu");
  var _component_tiny_dropdown = resolveComponent("tiny-dropdown");
  var _component_icon_arrow_left = resolveComponent("icon-arrow-left");
  var _component_icon_arrow_right = resolveComponent("icon-arrow-right");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  var _directive_swipeleft = resolveDirective("swipeleft");
  var _directive_swiperight = resolveDirective("swiperight");
  return _ctx.state.showImageViewer ? (openBlock(), createElementBlock("div", _hoisted_1, [createVNode(Transition, {
    name: "viewer-fade"
  }, {
    default: withCtx(function() {
      return [createElementVNode(
        "div",
        {
          "data-tag": "tiny-image-body",
          class: normalizeClass(["left-0 right-0 top-0 bottom-0 hidden sm:flex z-10", _ctx.modalView ? "" : "fixed"]),
          style: normalizeStyle({
            "z-index": _ctx.state.zIndex,
            "height": _ctx.modalView && _ctx.modalHeight + "px"
          })
        },
        [_ctx.state.isThumbnail ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            "data-tag": "tiny-isthumbnail",
            ref: "isThumbnailContent",
            class: normalizeClass(["flex-none w-[12%] h-full flex flex-col overflow-y-scroll items-center scrollbar-size-0", _ctx.bgColor !== "white" ? "bg-color-icon-primary" : ""]),
            style: normalizeStyle({
              backgroundColor: _ctx.bgColor === "white" && _ctx.bgColor
            })
          },
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.urlList, function(item, i) {
              return openBlock(), createElementBlock("div", {
                key: i,
                class: "w-[88%] [&:not(:last-child)]:mb-2 [&:first-of-type]:mt-2",
                ref_for: true,
                ref: "isThumbnail_" + i
              }, [createElementVNode("div", {
                onClick: function onClick($event) {
                  return _ctx.activeItems(i);
                },
                class: normalizeClass(["rounded cursor-pointer p-1", {
                  "bg-color-brand": _ctx.state.currentIndex === i
                }])
              }, [createElementVNode("img", {
                class: "max-h-full max-w-full aspect-square bg-current rounded",
                ref_for: true,
                ref: "img_" + i,
                src: item.url
              }, null, 8, _hoisted_3), createElementVNode("div", {
                title: item.name || item.url,
                class: normalizeClass(["px-1 mt-1 sm:text-xs text-sm leading-5 overflow-hidden whitespace-nowrap text-ellipsis", _ctx.bgColor === "white" ? _ctx.state.currentIndex !== i ? "text-color-text-primary" : "text-color-bg-1" : "text-color-bg-1"])
              }, toDisplayString(item.name), 11, _hoisted_4)], 10, _hoisted_2)]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          6
          /* CLASS, STYLE */
        )) : createCommentVNode("v-if", true), _ctx.state.isMenuView ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            "data-tag": "tiny-is-menu-view",
            ref: "isMenuViewContent",
            class: normalizeClass(["flex-none w-[12%] h-full flex flex-col overflow-y-scroll items-center scrollbar-size-0", _ctx.bgColor !== "white" ? "bg-color-icon-primary" : "bg-color-bg-1"])
          },
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.urlList, function(items, i) {
              return openBlock(), createElementBlock("div", {
                key: i,
                class: "w-[88%]"
              }, [createElementVNode("div", {
                onClick: function onClick($event) {
                  return _ctx.activeItems(i);
                },
                ref_for: true,
                ref: "isMenuView_" + i,
                class: normalizeClass(["[&:first-of-type]:mt-2.5 rounded cursor-pointer flex mx-1 h-9 items-center", {
                  "bg-color-brand": _ctx.state.currentIndex === i
                }])
              }, [createVNode(_component_icon_picture, {
                class: normalizeClass(["max-h-full max-w-full w-[9%] h-auto my-2.5 mx-1.5", _ctx.bgColor === "white" ? _ctx.state.menuItemIndex !== i ? "fill-color-text-primary" : "fill-color-bg-1" : "fill-color-bg-1"]),
                ref_for: true,
                ref: "img_" + i,
                src: items.url
              }, null, 8, ["class", "src"]), createElementVNode("div", {
                title: items.name || items.url,
                class: normalizeClass(["w-[67%] sm:text-xs text-sm leading-5 overflow-hidden whitespace-nowrap text-ellipsis", _ctx.bgColor === "white" ? _ctx.state.menuItemIndex !== i ? "text-color-text-primary" : "text-color-bg-1" : "text-color-bg-1"])
              }, toDisplayString(items.name), 11, _hoisted_6)], 10, _hoisted_5)]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          2
          /* CLASS */
        )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_7, [_hoisted_8, _ctx.modalView || !_ctx.state.isThumbnail && !_ctx.state.isMenuView ? (openBlock(), createElementBlock("span", {
          key: 0,
          "data-tag": "tiny-image-pc-close",
          class: "z-[1] flex absolute items-center justify-center rounded-full box-border select-none top-20 right-20 bottom-10 text-xl cursor-pointer w-8 h-8 hover:bg-color-text-primary active:bg-color-border fill-color-bg-1 bg-color-border",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.hide && _ctx.hide.apply(_ctx, arguments);
          })
        }, [createVNode(_component_icon_close, {
          class: "fill-color-bg-1 cursor-pointer"
        })])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_9, [createElementVNode("div", _hoisted_10, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.mode.icon), {
          class: "cursor-pointer",
          onClick: _ctx.toggleMode
        }, null, 8, ["onClick"])), createVNode(_component_icon_zoom_out, {
          class: "cursor-pointer",
          onClick: _cache[1] || (_cache[1] = function($event) {
            return _ctx.handleActions("zoomOut");
          })
        }), createVNode(_component_icon_zoom_in, {
          class: "cursor-pointer",
          onClick: _cache[2] || (_cache[2] = function($event) {
            return _ctx.handleActions("zoomIn");
          })
        }), createVNode(_component_icon_refres, {
          class: "cursor-pointer",
          onClick: _cache[3] || (_cache[3] = function($event) {
            return _ctx.handleActions("clocelise");
          })
        }), createVNode(_component_icon_repeat, {
          class: "cursor-pointer",
          onClick: _cache[4] || (_cache[4] = function($event) {
            return _ctx.handleActions("anticlocelise");
          })
        }), createVNode(_component_icon_del, {
          class: "cursor-pointer",
          onClick: _cache[5] || (_cache[5] = function($event) {
            return _ctx.handleActions("delImage");
          })
        }), createElementVNode("a", {
          href: "" + _ctx.state.currentImg,
          target: "_blank",
          download: "",
          class: "cursor-pointer inline-flex fill-color-bg-1"
        }, [createVNode(_component_icon_download)], 8, _hoisted_11), createVNode(_component_tiny_dropdown, {
          "show-self-icon": "",
          class: "flex",
          onItemClick: _ctx.itemClick
        }, {
          dropdown: withCtx(function() {
            return [createVNode(_component_tiny_dropdown_menu, {
              "checked-status": "",
              placement: "top"
            }, {
              default: withCtx(function(_ref) {
                var selectedIndex = _ref.selectedIndex;
                return [createVNode(_component_tiny_dropdown_item, {
                  "current-index": 0,
                  selected: selectedIndex === "0" && !_ctx.state.isThumbnail && !_ctx.state.isMenuView,
                  "item-data": !_ctx.state.isThumbnail && !_ctx.state.isMenuView ? "4" : "1"
                }, {
                  default: withCtx(function() {
                    return [createTextVNode(
                      toDisplayString(_ctx.state.isThumbnail || _ctx.state.isMenuView ? _ctx.t("ui.imageViewer.hide") : _ctx.t("ui.imageViewer.show")),
                      1
                      /* TEXT */
                    )];
                  }),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["selected", "item-data"]), createVNode(_component_tiny_dropdown_item, {
                  "current-index": 1,
                  selected: selectedIndex === "1" || _ctx.state.isThumbnail,
                  "item-data": "2"
                }, {
                  default: withCtx(function() {
                    return [createTextVNode(
                      toDisplayString(_ctx.t("ui.imageViewer.thumbnail")),
                      1
                      /* TEXT */
                    )];
                  }),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["selected"]), createVNode(_component_tiny_dropdown_item, {
                  "current-index": 2,
                  selected: selectedIndex === "2" || _ctx.state.isMenuView,
                  "item-data": "3"
                }, {
                  default: withCtx(function() {
                    return [createTextVNode(
                      toDisplayString(_ctx.t("ui.imageViewer.menu")),
                      1
                      /* TEXT */
                    )];
                  }),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["selected"])];
              }),
              _: 1
              /* STABLE */
            })];
          }),
          default: withCtx(function() {
            return [createElementVNode("span", null, [createVNode(_component_icon_editor_menu_right)])];
          }),
          _: 1
          /* STABLE */
        }, 8, ["onItemClick"]), createVNode(_component_icon_arrow_left, {
          onClick: _ctx.prev,
          class: normalizeClass(["cursor-pointer", {
            "is-disabled": !_ctx.state.infinite && _ctx.state.isFirst
          }])
        }, null, 8, ["onClick", "class"]), createVNode(_component_icon_arrow_right, {
          onClick: _ctx.next,
          class: normalizeClass(["cursor-pointer", {
            "is-disabled": !_ctx.state.infinite && _ctx.state.isLast
          }])
        }, null, 8, ["onClick", "class"])])]), _ctx.showIndex ? (openBlock(), createElementBlock("div", _hoisted_12, [createElementVNode(
          "span",
          _hoisted_13,
          toDisplayString(_ctx.state.index + 1 + "/" + _ctx.state.urlList.length),
          1
          /* TEXT */
        )])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_14, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.urlList, function(url, i) {
            return openBlock(), createElementBlock(
              Fragment,
              null,
              [i === _ctx.state.index ? (openBlock(), createElementBlock("img", {
                ref_for: true,
                ref: "img_" + i,
                key: i,
                src: _ctx.state.currentImg,
                style: normalizeStyle(_ctx.state.imgStyle),
                onError: _cache[6] || (_cache[6] = function() {
                  return _ctx.handleImgError && _ctx.handleImgError.apply(_ctx, arguments);
                }),
                onMousedown: _cache[7] || (_cache[7] = function() {
                  return _ctx.handleMouseDown && _ctx.handleMouseDown.apply(_ctx, arguments);
                }),
                onLoad: _cache[8] || (_cache[8] = function() {
                  return _ctx.handleImgLoad && _ctx.handleImgLoad.apply(_ctx, arguments);
                })
              }, null, 44, _hoisted_15)) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            );
          }),
          256
          /* UNKEYED_FRAGMENT */
        ))])])],
        6
        /* CLASS, STYLE */
      )];
    }),
    _: 1
    /* STABLE */
  }), createVNode(Transition, {
    name: "viewer-fade"
  }, {
    default: withCtx(function() {
      return [_ctx.state.isThumbnail ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          "data-tag": "tiny-image-mobile-thumbnail-list",
          class: normalizeClass(["w-screen h-screen fixed top-0 left-0 sm:hidden z-40", {
            "hidden": _ctx.state.hiddenThumbnail
          }])
        },
        [createElementVNode("div", _hoisted_16, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.urlList, function(item, i) {
            return openBlock(), createElementBlock("div", {
              class: "px-4 pt-4 [&:last-of-type]:mb-5 cursor-pointer",
              key: i,
              onTouchstart: function onTouchstart($event) {
                return _ctx.imagePreview(i);
              }
            }, [createElementVNode("img", {
              class: "bg-current w-full h-auto",
              ref_for: true,
              ref: "img_" + i,
              src: item.url
            }, null, 8, _hoisted_18), createElementVNode("div", {
              title: item.name || item.url,
              class: "color-text-primary mt-2 sm:text-xs text-sm leading-5 overflow-hidden whitespace-nowrap text-ellipsis text-color-text-primary"
            }, toDisplayString(item.name), 9, _hoisted_19)], 40, _hoisted_17);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])],
        2
        /* CLASS */
      )) : !_ctx.state.isMenuView ? (openBlock(), createElementBlock("div", _hoisted_20, [_hoisted_21, withDirectives((openBlock(), createElementBlock("div", _hoisted_22, [_ctx.showIndex ? (openBlock(), createElementBlock("div", _hoisted_23, [createElementVNode(
        "span",
        _hoisted_24,
        toDisplayString(_ctx.state.index + 1 + "/" + _ctx.state.urlList.length),
        1
        /* TEXT */
      )])) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        _hoisted_25,
        [createElementVNode(
          "div",
          {
            class: "h-full flex items-center justify-center",
            style: normalizeStyle({
              "width": _ctx.state.imageAllWidth + "px",
              "transition-duration": _ctx.state.imageTransition + "ms",
              transform: "translateX(" + _ctx.state.imageTransformSize + "px)"
            })
          },
          [createElementVNode("template", null, [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.urlList, function(url, i) {
              return openBlock(), createElementBlock(
                "div",
                {
                  "data-tag": "tiny-image-mobile-viewer-item",
                  class: "relative shrink h-full",
                  ref_for: true,
                  ref: "viewerItem",
                  key: i,
                  style: normalizeStyle(Object.assign({
                    width: _ctx.state.imageItemWidth + "px"
                  }, i === _ctx.state.index ? _ctx.state.imgStyle : ""))
                },
                [createElementVNode("div", _hoisted_26, [i === _ctx.state.index ? (openBlock(), createElementBlock("img", {
                  key: i,
                  class: normalizeClass(["block w-full h-full object-contain", {
                    "object-fill": _ctx.state.fullScreen
                  }]),
                  ref_for: true,
                  ref: "img_" + i,
                  src: _ctx.state.currentImg,
                  style: normalizeStyle(_ctx.state.imgStyle),
                  onMousedown: _cache[9] || (_cache[9] = function() {
                    return _ctx.handleMouseDown && _ctx.handleMouseDown.apply(_ctx, arguments);
                  }),
                  onError: _cache[10] || (_cache[10] = function() {
                    return _ctx.handleImgError && _ctx.handleImgError.apply(_ctx, arguments);
                  }),
                  onLoad: _cache[11] || (_cache[11] = function() {
                    return _ctx.handleImgLoad && _ctx.handleImgLoad.apply(_ctx, arguments);
                  }),
                  onTouchstart: _cache[12] || (_cache[12] = function() {
                    return _ctx.touchstart && _ctx.touchstart.apply(_ctx, arguments);
                  }),
                  onTouchmove: _cache[13] || (_cache[13] = function() {
                    return _ctx.touchmove && _ctx.touchmove.apply(_ctx, arguments);
                  }),
                  onTouchend: _cache[14] || (_cache[14] = function() {
                    return _ctx.touchend && _ctx.touchend.apply(_ctx, arguments);
                  })
                }, null, 46, _hoisted_27)) : createCommentVNode("v-if", true)])],
                4
                /* STYLE */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))])],
          4
          /* STYLE */
        )],
        512
        /* NEED_PATCH */
      )])), [[_directive_swipeleft, _ctx.swipeLeft], [_directive_swiperight, _ctx.swipeRight]])])) : createCommentVNode("v-if", true)];
    }),
    _: 1
    /* STABLE */
  }), createVNode(Transition, {
    name: "viewer-fade"
  }, {
    default: withCtx(function() {
      return [_ctx.state.isImagePreview ? (openBlock(), createElementBlock("div", _hoisted_28, [_hoisted_29, withDirectives((openBlock(), createElementBlock("div", _hoisted_30, [_ctx.showIndex ? (openBlock(), createElementBlock("div", _hoisted_31, [createElementVNode(
        "span",
        _hoisted_32,
        toDisplayString(_ctx.state.index + 1 + "/" + _ctx.state.urlList.length),
        1
        /* TEXT */
      )])) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        _hoisted_33,
        [createElementVNode(
          "div",
          {
            class: "h-full flex items-center justify-center",
            style: normalizeStyle({
              "width": _ctx.state.imageAllWidth + "px",
              "transition-duration": _ctx.state.imageTransition + "ms",
              transform: "translateX(" + _ctx.state.imageTransformSize + "px)"
            })
          },
          [createElementVNode("template", null, [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.urlList, function(url, i) {
              return openBlock(), createElementBlock(
                "div",
                {
                  "data-tag": "tiny-image-isthumbnail-viewer-item",
                  class: "relative shrink h-full",
                  ref_for: true,
                  ref: "viewerItem",
                  key: i,
                  style: normalizeStyle(Object.assign({
                    width: _ctx.state.imageItemWidth + "px"
                  }, i === _ctx.state.index ? _ctx.state.imgStyle : ""))
                },
                [createElementVNode("div", _hoisted_34, [(openBlock(), createElementBlock("img", {
                  key: i,
                  class: normalizeClass(["block w-full h-full object-contain", {
                    "object-fill": _ctx.state.fullScreen
                  }]),
                  ref_for: true,
                  ref: "img_" + i,
                  src: _ctx.state.currentImg,
                  style: normalizeStyle(_ctx.state.imgStyle),
                  onMousedown: _cache[15] || (_cache[15] = function() {
                    return _ctx.handleMouseDown && _ctx.handleMouseDown.apply(_ctx, arguments);
                  }),
                  onError: _cache[16] || (_cache[16] = function() {
                    return _ctx.handleImgError && _ctx.handleImgError.apply(_ctx, arguments);
                  }),
                  onLoad: _cache[17] || (_cache[17] = function() {
                    return _ctx.handleImgLoad && _ctx.handleImgLoad.apply(_ctx, arguments);
                  }),
                  onTouchstart: _cache[18] || (_cache[18] = function() {
                    return _ctx.touchstart && _ctx.touchstart.apply(_ctx, arguments);
                  }),
                  onTouchmove: _cache[19] || (_cache[19] = function() {
                    return _ctx.touchmove && _ctx.touchmove.apply(_ctx, arguments);
                  }),
                  onTouchend: _cache[20] || (_cache[20] = function() {
                    return _ctx.touchend && _ctx.touchend.apply(_ctx, arguments);
                  })
                }, null, 46, _hoisted_35))])],
                4
                /* STYLE */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))])],
          4
          /* STYLE */
        )],
        512
        /* NEED_PATCH */
      )])), [[_directive_swipeleft, _ctx.swipeLeft], [_directive_swiperight, _ctx.swipeRight]])])) : createCommentVNode("v-if", true)];
    }),
    _: 1
    /* STABLE */
  }), createVNode(_component_tiny_action_sheet, {
    type: "action",
    menus: [{
      type: "save",
      label: _ctx.t("ui.imageViewer.save")
    }, {
      type: "thumbnail",
      label: _ctx.t("ui.imageViewer.thumbnail")
    }, {
      type: "del",
      label: _ctx.t("ui.imageViewer.del")
    }],
    visible: _ctx.state.boxVisibility,
    "onUpdate:visible": _cache[21] || (_cache[21] = function($event) {
      return _ctx.state.boxVisibility = $event;
    }),
    onClick: _ctx.selectOption
  }, {
    default: withCtx(function(data) {
      return [createElementVNode(
        "div",
        {
          class: normalizeClass([{
            "text-color-error": data.data.type === "del"
          }])
        },
        toDisplayString(data.data.label),
        3
        /* TEXT, CLASS */
      )];
    }),
    _: 1
    /* STABLE */
  }, 8, ["menus", "visible", "onClick"])])) : createCommentVNode("v-if", true);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
