// @opentiny/vue-image-viewer/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/image-viewer/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconClose, iconChevronLeft, iconChevronRight, iconZoomOut, iconZoomIn, iconFullscreen, iconMinscreen, iconRepeat, iconRefres } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/image-viewer/index.css";
import { resolveComponent, openBlock, createBlock, Transition, withCtx, createElementVNode, normalizeStyle, withModifiers, createVNode, createElementBlock, Fragment, normalizeClass, createCommentVNode, resolveDynamicComponent, renderList } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  emits: ["close", "update:preview-visible", "delete", "change", "newImageList"],
  props: [].concat(props, ["urlList", "zIndex", "onSwitch", "onClose"]),
  components: {
    IconClose: iconClose(),
    IconChevronLeft: iconChevronLeft(),
    IconChevronRight: iconChevronRight(),
    IconZoomOut: iconZoomOut(),
    IconZoomIn: iconZoomIn(),
    IconFullscreen: iconFullscreen(),
    IconMinscreen: iconMinscreen(),
    IconRepeat: iconRepeat(),
    IconRefres: iconRefres()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-image-viewer__mask"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_2 = {
  class: "tiny-image-viewer__btn tiny-image-viewer__actions"
};
var _hoisted_3 = {
  class: "tiny-image-viewer__actions-inner"
};
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "i",
  {
    class: "tiny-image-viewer__actions-divider"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_5 = /* @__PURE__ */ createElementVNode(
  "i",
  {
    class: "tiny-image-viewer__actions-divider"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_6 = {
  class: "tiny-image-viewer__canvas"
};
var _hoisted_7 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  var _component_icon_chevron_left = resolveComponent("icon-chevron-left");
  var _component_icon_chevron_right = resolveComponent("icon-chevron-right");
  var _component_icon_zoom_out = resolveComponent("icon-zoom-out");
  var _component_icon_zoom_in = resolveComponent("icon-zoom-in");
  var _component_icon_repeat = resolveComponent("icon-repeat");
  var _component_icon_refres = resolveComponent("icon-refres");
  return openBlock(), createBlock(Transition, {
    name: "viewer-fade"
  }, {
    default: withCtx(function() {
      return [createElementVNode(
        "div",
        {
          class: "tiny-image-viewer tiny-image-viewer__wrapper",
          style: normalizeStyle({
            "z-index": _ctx.state.zIndex
          }),
          onMousewheel: _cache[10] || (_cache[10] = withModifiers(function() {
          }, ["prevent"]))
        },
        [_hoisted_1, createElementVNode("span", {
          class: "tiny-image-viewer__btn tiny-image-viewer__close",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.hide && _ctx.hide.apply(_ctx, arguments);
          })
        }, [createVNode(_component_icon_close, {
          class: "tiny-svg-size"
        })]), !_ctx.state.isSingle ? (openBlock(), createElementBlock(
          Fragment,
          {
            key: 0
          },
          [createElementVNode(
            "span",
            {
              class: normalizeClass(["tiny-image-viewer__btn tiny-image-viewer__prev", {
                "is-disabled": !_ctx.state.infinite && _ctx.state.isFirst
              }]),
              onClick: _cache[1] || (_cache[1] = function() {
                return _ctx.prev && _ctx.prev.apply(_ctx, arguments);
              })
            },
            [createVNode(_component_icon_chevron_left, {
              class: "tiny-svg-size"
            })],
            2
            /* CLASS */
          ), createElementVNode(
            "span",
            {
              class: normalizeClass(["tiny-image-viewer__btn tiny-image-viewer__next", {
                "is-disabled": !_ctx.state.infinite && _ctx.state.isLast
              }]),
              onClick: _cache[2] || (_cache[2] = function() {
                return _ctx.next && _ctx.next.apply(_ctx, arguments);
              })
            },
            [createVNode(_component_icon_chevron_right, {
              class: "tiny-svg-size"
            })],
            2
            /* CLASS */
          )],
          64
          /* STABLE_FRAGMENT */
        )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [createVNode(_component_icon_zoom_out, {
          class: "tiny-svg-size",
          onClick: _cache[3] || (_cache[3] = function($event) {
            return _ctx.handleActions("zoomOut");
          })
        }), createVNode(_component_icon_zoom_in, {
          class: "tiny-svg-size",
          onClick: _cache[4] || (_cache[4] = function($event) {
            return _ctx.handleActions("zoomIn");
          })
        }), _hoisted_4, (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.mode.icon), {
          class: "tiny-svg-size",
          onClick: _ctx.toggleMode
        }, null, 8, ["onClick"])), _hoisted_5, createVNode(_component_icon_repeat, {
          class: "tiny-svg-size",
          onClick: _cache[5] || (_cache[5] = function($event) {
            return _ctx.handleActions("anticlocelise");
          })
        }), createVNode(_component_icon_refres, {
          class: "tiny-svg-size",
          onClick: _cache[6] || (_cache[6] = function($event) {
            return _ctx.handleActions("clocelise");
          })
        })])]), createElementVNode("div", _hoisted_6, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.urlList, function(url, i) {
            return openBlock(), createElementBlock(
              Fragment,
              null,
              [i === _ctx.state.index ? (openBlock(), createElementBlock("img", {
                ref_for: true,
                ref: "img_" + i,
                class: "tiny-image-viewer__img",
                key: url,
                src: _ctx.state.currentImg,
                style: normalizeStyle(_ctx.state.imgStyle),
                onError: _cache[7] || (_cache[7] = function() {
                  return _ctx.handleImgError && _ctx.handleImgError.apply(_ctx, arguments);
                }),
                onMousedown: _cache[8] || (_cache[8] = function() {
                  return _ctx.handleMouseDown && _ctx.handleMouseDown.apply(_ctx, arguments);
                }),
                onLoad: _cache[9] || (_cache[9] = function() {
                  return _ctx.handleImgLoad && _ctx.handleImgLoad.apply(_ctx, arguments);
                })
              }, null, 44, _hoisted_7)) : createCommentVNode("v-if", true)],
              64
              /* STABLE_FRAGMENT */
            );
          }),
          256
          /* UNKEYED_FRAGMENT */
        ))])],
        36
        /* STYLE, NEED_HYDRATION */
      )];
    }),
    _: 1
    /* STABLE */
  });
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
