// @opentiny/vue-carousel/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/carousel/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconChevronLeft, iconChevronRight, iconChevronUp, iconChevronDown } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/carousel/index.css";
import { openBlock, createElementBlock, normalizeClass, withModifiers, createElementVNode, normalizeStyle, createVNode, Transition, withCtx, withDirectives, createBlock, resolveDynamicComponent, vShow, createCommentVNode, renderSlot, Fragment, renderList, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "complete"],
  props: [].concat(props, ["initialIndex", "height", "trigger", "autoplay", "interval", "indicatorPosition", "arrow", "type", "showTitle", "loop"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  },
  components: {
    IconChevronLeft: iconChevronLeft(),
    IconChevronRight: iconChevronRight(),
    IconChevronUp: iconChevronUp(),
    IconChevronDown: iconChevronDown()
  }
});
var _hoisted_1 = ["onMouseenter", "onClick"];
var _hoisted_2 = {
  type: "button",
  class: "tiny-carousel__button"
};
var _hoisted_3 = {
  key: 0
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass([{
        "tiny-carousel--card": _ctx.type === "card"
      }, "tiny-carousel"]),
      onMouseenter: _cache[6] || (_cache[6] = withModifiers(function() {
        return _ctx.handleMouseEnter && _ctx.handleMouseEnter.apply(_ctx, arguments);
      }, ["stop"])),
      onMouseleave: _cache[7] || (_cache[7] = withModifiers(function() {
        return _ctx.handleMouseLeave && _ctx.handleMouseLeave.apply(_ctx, arguments);
      }, ["stop"]))
    },
    [createElementVNode(
      "div",
      {
        style: normalizeStyle({
          height: _ctx.height
        }),
        class: "tiny-carousel__container"
      },
      [createVNode(Transition, {
        name: _ctx.type === "vertical" ? "tiny-transition-carousel-arrow-top" : "tiny-transition-carousel-arrow-left"
      }, {
        default: withCtx(function() {
          return [_ctx.arrow !== "never" ? withDirectives((openBlock(), createElementBlock(
            "button",
            {
              key: 0,
              type: "button",
              class: normalizeClass(["tiny-carousel__arrow", _ctx.type === "vertical" ? "tiny-carousel__arrow-top" : "tiny-carousel__arrow-left"]),
              onMouseenter: _cache[0] || (_cache[0] = function($event) {
                return _ctx.handleButtonEnter("left");
              }),
              onMouseleave: _cache[1] || (_cache[1] = function() {
                return _ctx.handleButtonLeave && _ctx.handleButtonLeave.apply(_ctx, arguments);
              }),
              onClick: _cache[2] || (_cache[2] = withModifiers(function($event) {
                return _ctx.throttledArrowClick(_ctx.state.activeIndex - 1);
              }, ["stop"]))
            },
            [(openBlock(), createBlock(resolveDynamicComponent(_ctx.type === "vertical" ? "icon-chevron-up" : "icon-chevron-left"), {
              class: "tiny-svg-size"
            }))],
            34
            /* CLASS, NEED_HYDRATION */
          )), [[vShow, (_ctx.arrow === "always" || _ctx.state.hover) && (_ctx.loop || _ctx.state.activeIndex > 0)]]) : createCommentVNode("v-if", true)];
        }),
        _: 1
        /* STABLE */
      }, 8, ["name"]), createVNode(Transition, {
        name: _ctx.type === "vertical" ? "tiny-transition-carousel-arrow-bottom" : "tiny-transition-carousel-arrow-right"
      }, {
        default: withCtx(function() {
          return [_ctx.arrow !== "never" ? withDirectives((openBlock(), createElementBlock(
            "button",
            {
              key: 0,
              type: "button",
              class: normalizeClass(["tiny-carousel__arrow", _ctx.type === "vertical" ? "tiny-carousel__arrow-bottom" : "tiny-carousel__arrow-right"]),
              onMouseenter: _cache[3] || (_cache[3] = function($event) {
                return _ctx.handleButtonEnter("right");
              }),
              onMouseleave: _cache[4] || (_cache[4] = function() {
                return _ctx.handleButtonLeave && _ctx.handleButtonLeave.apply(_ctx, arguments);
              }),
              onClick: _cache[5] || (_cache[5] = withModifiers(function($event) {
                return _ctx.throttledArrowClick(_ctx.state.activeIndex + 1);
              }, ["stop"]))
            },
            [(openBlock(), createBlock(resolveDynamicComponent(_ctx.type === "vertical" ? "icon-chevron-down" : "icon-chevron-right"), {
              class: "tiny-svg-size"
            }))],
            34
            /* CLASS, NEED_HYDRATION */
          )), [[vShow, (_ctx.arrow === "always" || _ctx.state.hover) && (_ctx.loop || _ctx.state.activeIndex < _ctx.state.items.length - 1)]]) : createCommentVNode("v-if", true)];
        }),
        _: 1
        /* STABLE */
      }, 8, ["name"]), renderSlot(_ctx.$slots, "default")],
      4
      /* STYLE */
    ), _ctx.indicatorPosition !== "none" ? (openBlock(), createElementBlock(
      "ul",
      {
        key: 0,
        class: normalizeClass([{
          "tiny-carousel__indicators-labels": _ctx.state.hasLabel,
          "tiny-carousel__indicators-title": _ctx.showTitle,
          "tiny-carousel__indicators-vertical": _ctx.type === "vertical",
          "tiny-carousel__indicators-outside": _ctx.indicatorPosition === "outside" || _ctx.type === "card"
        }, "tiny-carousel__indicators"])
      },
      [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.items, function(item, index) {
          return openBlock(), createElementBlock("li", {
            key: index,
            class: normalizeClass([{
              "is-active": index === _ctx.state.activeIndex
            }, "tiny-carousel__indicator"]),
            onMouseenter: function onMouseenter($event) {
              return _ctx.throttledIndicatorHover(index);
            },
            onClick: withModifiers(function($event) {
              return _ctx.handleIndicatorClick(index);
            }, ["stop"])
          }, [createElementVNode("button", _hoisted_2, [_ctx.state.hasLabel ? (openBlock(), createElementBlock(
            "span",
            _hoisted_3,
            toDisplayString(item.label),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)])], 42, _hoisted_1);
        }),
        128
        /* KEYED_FRAGMENT */
      ))],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true)],
    34
    /* CLASS, NEED_HYDRATION */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
