// @opentiny/vue-breadcrumb-item/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/breadcrumb-item/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var breadcrumbItemProps = _extends({}, $props, {
  to: Object,
  replace: {
    type: Boolean,
    default: false
  },
  option: {
    type: Object,
    default: function _default() {
      return {};
    }
  }
});
var BreadcrumbItem = defineComponent({
  name: $prefix + "BreadcrumbItem",
  props: breadcrumbItemProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
BreadcrumbItem.install = function(Vue) {
  Vue.component(BreadcrumbItem.name, BreadcrumbItem);
};
BreadcrumbItem.version = version;
export {
  BreadcrumbItem as default
};
