// @opentiny/vue-anchor/lib/index.js
import { defineComponent, $prefix, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/anchor/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var anchorProps = {
  isAffix: {
    type: Boolean,
    default: false
  },
  links: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  containerId: {
    type: String,
    default: ""
  },
  markClass: {
    type: String,
    default: ""
  },
  type: {
    type: String,
    default: "line"
  }
};
var Anchor = defineComponent({
  name: $prefix + "Anchor",
  props: anchorProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Anchor.install = function(Vue) {
  Vue.component(Anchor.name, Anchor);
};
Anchor.version = version;
export {
  Anchor as default
};
