// @opentiny/vue-tag-group/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  var tinyMode = typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  if ("pc" === (tinyMode || mode)) {
    return PcTemplate;
  }
  if ("mobile-first" === (tinyMode || mode)) {
    return MobileFirstTemplate;
  }
  return PcTemplate;
};
var TagGroup = defineComponent({
  name: $prefix + "TagGroup",
  props: _extends({}, $props, {
    size: {
      type: String,
      default: "medium",
      validator: function validator(value) {
        return ["medium", "small", "mini"].includes(value);
      }
    },
    data: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    effect: {
      type: String,
      default: "light",
      validator: function validator2(value) {
        return ["dark", "light", "plain"].includes(value);
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
TagGroup.install = function(Vue) {
  Vue.component(TagGroup.name, TagGroup);
};
TagGroup.version = version;
export {
  TagGroup as default
};
