// @opentiny/vue-tag-group/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/tag-group/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { IconEllipsis } from "@opentiny/vue-icon";
import Popover from "@opentiny/vue-popover";
import Tag from "@opentiny/vue-tag";
import { resolveComponent, openBlock, createElementBlock, Fragment, renderList, createBlock, withCtx, createTextVNode, toDisplayString, createVNode, createElementVNode, normalizeClass } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "TagGroup",
  emits: ["item-click"],
  props: _extends({}, $props, {
    size: {
      type: String,
      default: "medium",
      validator: function validator(value) {
        return ~["medium", "small", "mini"].indexOf(value);
      }
    },
    data: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    effect: {
      type: String,
      default: "light",
      validator: function validator2(value) {
        return ~["dark", "light", "plain"].indexOf(value);
      }
    }
  }),
  components: {
    TinyTag: Tag,
    TinyPopover: Popover,
    IconEllipsis: IconEllipsis()
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  ref: "tagGroup",
  "data-tag": "tiny-tag-group",
  class: "overflow-hidden relative pr-8"
};
var _hoisted_2 = {
  "data-tag": "tiny-tag-group-popover"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_tag = resolveComponent("tiny-tag");
  var _component_icon_ellipsis = resolveComponent("icon-ellipsis");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock(
    "div",
    _hoisted_1,
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.data, function(item, i) {
        return openBlock(), createBlock(_component_tiny_tag, {
          key: item.name + i,
          size: _ctx.size,
          type: item.type || "info",
          effect: _ctx.effect,
          onClick: function onClick($event) {
            return _ctx.handelItemClick(item, i, $event);
          }
        }, {
          default: withCtx(function() {
            return [createTextVNode(
              toDisplayString(item.name),
              1
              /* TEXT */
            )];
          }),
          _: 2
          /* DYNAMIC */
        }, 1032, ["size", "type", "effect", "onClick"]);
      }),
      128
      /* KEYED_FRAGMENT */
    )), createVNode(_component_tiny_popover, {
      placement: "top-start",
      trigger: "hover",
      class: "absolute top-0 right-0"
    }, {
      reference: withCtx(function() {
        return [createElementVNode(
          "div",
          {
            class: normalizeClass(["absolute top-0 right-0 pointer-events-none sm:pointer-events-auto", [_ctx.state.showMore ? "block" : "invisible"]])
          },
          [createVNode(_component_tiny_tag, {
            ref: "more",
            size: _ctx.size,
            type: "info",
            customClass: "px-1.5 mr-0"
          }, {
            default: withCtx(function() {
              return [createVNode(_component_icon_ellipsis, {
                class: "fill-color-icon-secondary"
              })];
            }),
            _: 1
            /* STABLE */
          }, 8, ["size"])],
          2
          /* CLASS */
        )];
      }),
      default: withCtx(function() {
        return [createElementVNode("div", _hoisted_2, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.hiddenTags, function(tag, i) {
            return openBlock(), createBlock(_component_tiny_tag, {
              key: tag.name + i,
              size: _ctx.size,
              type: tag.type || "info",
              customClass: "mb-2",
              onClick: function onClick($event) {
                return _ctx.handelItemClick(tag, _ctx.data.length - _ctx.state.hiddenTags.length + i, $event);
              }
            }, {
              default: withCtx(function() {
                return [createTextVNode(
                  toDisplayString(tag.name),
                  1
                  /* TEXT */
                )];
              }),
              _: 2
              /* DYNAMIC */
            }, 1032, ["size", "type", "onClick"]);
          }),
          128
          /* KEYED_FRAGMENT */
        ))])];
      }),
      _: 1
      /* STABLE */
    })],
    512
    /* NEED_PATCH */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
