// @opentiny/vue-floating-button/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var FloatingButton = {
  name: $prefix + "FloatingButton",
  props: _extends({}, $props, {
    position: {
      type: String,
      default: ""
    },
    backgroundColor: {
      type: String,
      default: ""
    },
    icon: {
      type: [Object, String],
      default: "icon-plus"
    },
    resetTime: {
      type: [String, Number],
      default: 0
    },
    animated: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
};
var version = "3.13.0";
FloatingButton.install = function(Vue) {
  Vue.component(FloatingButton.name, FloatingButton);
};
FloatingButton.version = version;
export {
  FloatingButton as default
};
