// @opentiny/vue-pull-refresh/lib/mobile.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, props, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/pull-refresh/vue";
import "@opentiny/vue-theme-mobile/pull-refresh/index.css";
import { openBlock, createElementBlock, normalizeStyle, createElementVNode, toDisplayString, createCommentVNode, renderSlot, normalizeClass } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  name: $prefix + "PullRefresh",
  props: _extends({}, props, {
    modelValue: Boolean,
    loosingText: String,
    animationDuration: {
      type: [Number, String],
      default: 300
    },
    hasMore: {
      type: Boolean,
      default: true
    },
    disabledPullDown: {
      type: Boolean,
      default: false
    },
    disabledPullUp: {
      type: Boolean,
      default: false
    },
    pullUpDistance: {
      type: [Number, String],
      default: 18
    },
    pullUpLoadingText: {
      type: String,
      default: null
    },
    pullDownLoadingText: {
      type: String,
      default: null
    }
  }),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-mobile-pull-refresh__tips tiny-mobile-pull-refresh__head"
};
var _hoisted_2 = {
  key: 0
};
var _hoisted_3 = {
  key: 0,
  class: "tiny-mobile-pull-refresh-loading-content"
};
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-mobile-pull-refresh__loading-inner"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_5 = [_hoisted_4];
var _hoisted_6 = {
  key: 0,
  class: "tiny-mobile-pull-refresh__text"
};
var _hoisted_7 = {
  class: "tiny-mobile-pull-refresh__content",
  ref: "content"
};
var _hoisted_8 = {
  key: 0,
  class: "tiny-mobile-pull-refresh__foot",
  ref: "foot"
};
var _hoisted_9 = {
  key: 0,
  class: "tiny-mobile-pull-refresh-loading-content"
};
var _hoisted_10 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-mobile-pull-refresh__loading tiny-mobile-pull-refresh__loading-animation"
  },
  [/* @__PURE__ */ createElementVNode("div", {
    class: "tiny-mobile-pull-refresh__loading-inner"
  })],
  -1
  /* HOISTED */
);
var _hoisted_11 = {
  key: 0,
  class: "tiny-mobile-pull-refresh__text"
};
var _hoisted_12 = {
  key: 1
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-mobile-pull-refresh",
      style: normalizeStyle(_ctx.state.refreshStyle)
    },
    [createElementVNode(
      "div",
      {
        class: "tiny-mobile-pull-refresh__track",
        ref: "track",
        style: normalizeStyle({
          "transition-duration": _ctx.state.animationDuration + "ms",
          transform: "translate3d(0px," + _ctx.state.translate3d + "px,0px)"
        })
      },
      [createElementVNode("div", _hoisted_1, [!_ctx.state.pullDownLoading ? (openBlock(), createElementBlock(
        "span",
        _hoisted_2,
        toDisplayString(_ctx.state.pullDownReplaces),
        1
        /* TEXT */
      )) : createCommentVNode("v-if", true), _ctx.state.pullDownLoading ? renderSlot(_ctx.$slots, "header", {
        key: 1
      }, function() {
        return [_ctx.state.pullDownLoading ? (openBlock(), createElementBlock("div", _hoisted_3, [createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-mobile-pull-refresh__loading", _ctx.state.pullDownLoading ? "tiny-mobile-pull-refresh__loading-animation" : null])
          },
          _hoisted_5,
          2
          /* CLASS */
        ), _ctx.state.pullDownLoadingText ? (openBlock(), createElementBlock(
          "div",
          _hoisted_6,
          toDisplayString(_ctx.state.pullDownLoadingText),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true)];
      }) : createCommentVNode("v-if", true)]), createElementVNode(
        "div",
        _hoisted_7,
        [renderSlot(_ctx.$slots, "default"), !_ctx.state.disabledPullUp ? (openBlock(), createElementBlock(
          "div",
          _hoisted_8,
          [renderSlot(_ctx.$slots, "footer", {}, function() {
            return [_ctx.state.hasMore ? (openBlock(), createElementBlock("div", _hoisted_9, [_hoisted_10, _ctx.state.pullUpLoadingText ? (openBlock(), createElementBlock(
              "div",
              _hoisted_11,
              toDisplayString(_ctx.state.pullUpLoadingText),
              1
              /* TEXT */
            )) : createCommentVNode("v-if", true)])) : (openBlock(), createElementBlock(
              "div",
              _hoisted_12,
              toDisplayString(_ctx.state.noMoreText),
              1
              /* TEXT */
            ))];
          })],
          512
          /* NEED_PATCH */
        )) : createCommentVNode("v-if", true)],
        512
        /* NEED_PATCH */
      )],
      4
      /* STYLE */
    )],
    4
    /* STYLE */
  );
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
