// @opentiny/vue-selected-box/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/selected-box/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconClose } from "@opentiny/vue-icon";
import Popover from "@opentiny/vue-popover";
import "@opentiny/vue-theme/selected-box/index.css";
import { resolveComponent, openBlock, createElementBlock, createElementVNode, renderSlot, toDisplayString, normalizeClass, Fragment, renderList, createCommentVNode, createVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["change", "clear", "delete", "drag"],
  props: [].concat(props, ["inverse", "total", "select", "config"]),
  components: {
    IconClose: IconClose(),
    TinyPopover: Popover
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-selected-box"
};
var _hoisted_2 = {
  class: "tiny-selected-box__header"
};
var _hoisted_3 = {
  class: "tiny-selected-box__header-select"
};
var _hoisted_4 = {
  class: "tiny-selected-box__header-select-label"
};
var _hoisted_5 = {
  class: "tiny-selected-box__header-right"
};
var _hoisted_6 = ["data-key"];
var _hoisted_7 = {
  class: "tiny-selected-box__item-grid"
};
var _hoisted_8 = {
  class: "tiny-selected-box__item-grid-left"
};
var _hoisted_9 = {
  class: "tiny-selected-box__item-grid-left-label"
};
var _hoisted_10 = {
  key: 0,
  class: "tiny-selected-box__item-grid-left-text"
};
var _hoisted_11 = {
  class: "tiny-selected-box__item-grid-right"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "select", {
    params: _ctx.state.slotParams
  }, function() {
    return [createElementVNode(
      "span",
      _hoisted_4,
      toDisplayString(_ctx.i18nSelected()),
      1
      /* TEXT */
    )];
  })]), createElementVNode("div", _hoisted_5, [renderSlot(_ctx.$slots, "button", {
    params: _ctx.state.slotParams
  }), renderSlot(_ctx.$slots, "clear", {
    params: _ctx.state.slotParams,
    clear: _ctx.handleClear
  }, function() {
    return [createElementVNode(
      "div",
      {
        class: "tiny-selected-box__header-right-label",
        onClick: _cache[0] || (_cache[0] = function() {
          return _ctx.handleClear && _ctx.handleClear.apply(_ctx, arguments);
        })
      },
      toDisplayString(_ctx.t("ui.selectedBox.clear")),
      1
      /* TEXT */
    )];
  })])]), createElementVNode(
    "div",
    {
      ref: "list",
      class: normalizeClass(["tiny-selected-box__list", {
        "tiny-selected-box__list--inverse": _ctx.state.inverse
      }])
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.select, function(option, i) {
        return openBlock(), createElementBlock("div", {
          key: "o" + i + "-" + _ctx.state.refresh,
          "data-key": _ctx.keyOption(option),
          "data-tag": "tiny-selected-box-item",
          class: "tiny-selected-box__item"
        }, [createElementVNode("div", _hoisted_7, [createElementVNode("div", _hoisted_8, [renderSlot(_ctx.$slots, "option", {
          params: _ctx.state.slotParams,
          option,
          mouseenter: _ctx.handleMouseenter,
          mouseleave: _ctx.handleMouseleave
        }, function() {
          return [createElementVNode("div", _hoisted_9, [createElementVNode(
            "div",
            {
              class: "tiny-selected-box__item-grid-left-label-pri",
              onMouseenter: _cache[1] || (_cache[1] = function() {
                return _ctx.handleMouseenter && _ctx.handleMouseenter.apply(_ctx, arguments);
              }),
              onMouseleave: _cache[2] || (_cache[2] = function() {
                return _ctx.handleMouseleave && _ctx.handleMouseleave.apply(_ctx, arguments);
              })
            },
            toDisplayString(_ctx.textPrimary(option)),
            33
            /* TEXT, NEED_HYDRATION */
          )]), _ctx.state.showAuxi ? (openBlock(), createElementBlock("div", _hoisted_10, [createElementVNode(
            "div",
            {
              class: "tiny-selected-box__item-grid-left-text-aux",
              onMouseenter: _cache[3] || (_cache[3] = function() {
                return _ctx.handleMouseenter && _ctx.handleMouseenter.apply(_ctx, arguments);
              }),
              onMouseleave: _cache[4] || (_cache[4] = function() {
                return _ctx.handleMouseleave && _ctx.handleMouseleave.apply(_ctx, arguments);
              })
            },
            toDisplayString(_ctx.textAuxi(option)),
            33
            /* TEXT, NEED_HYDRATION */
          )])) : createCommentVNode("v-if", true)];
        })]), createElementVNode("div", null, [renderSlot(_ctx.$slots, "close", {
          params: _ctx.state.slotParams,
          option,
          close: _ctx.handleDelete
        }, function() {
          return [createElementVNode("div", _hoisted_11, [createVNode(_component_icon_close, {
            onClick: function onClick($event) {
              return _ctx.handleDelete(option);
            }
          }, null, 8, ["onClick"])])];
        })])])], 8, _hoisted_6);
      }),
      128
      /* KEYED_FRAGMENT */
    ))],
    2
    /* CLASS */
  ), createVNode(_component_tiny_popover, {
    ref: "popover",
    trigger: "manual",
    class: "tiny-selected-box__popover",
    content: _ctx.state.popoverContent,
    modelValue: _ctx.state.popoverVisible,
    "onUpdate:modelValue": _cache[5] || (_cache[5] = function($event) {
      return _ctx.state.popoverVisible = $event;
    })
  }, null, 8, ["content", "modelValue"])]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
