// @opentiny/vue-alert/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/alert/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconClose, iconSuccess, iconError, iconHelp, iconWarning } from "@opentiny/vue-icon";
import "@opentiny/vue-theme/alert/index.css";
import { resolveComponent, openBlock, createBlock, Transition, withCtx, createElementBlock, normalizeClass, resolveDynamicComponent, createCommentVNode, createElementVNode, renderSlot, createTextVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["icon", "type", "size", "description", "title", "closable", "center", "showIcon", "closeText", "customClass"]),
  components: {
    IconClose: iconClose(),
    IconSuccess: iconSuccess(),
    IconError: iconError(),
    IconHelp: iconHelp(),
    IconWarning: iconWarning()
  },
  emits: ["close"],
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-alert__content"
};
var _hoisted_2 = {
  key: 0,
  class: "tiny-alert__title"
};
var _hoisted_3 = {
  key: 1,
  class: "tiny-alert__opration"
};
var _hoisted_4 = {
  key: 2,
  class: "is-custom"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createBlock(Transition, {
    name: "tiny-transition-alert-fade"
  }, {
    default: withCtx(function() {
      return [_ctx.state.show ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          class: normalizeClass(["tiny-alert", "tiny-alert--" + _ctx.type, "tiny-alert--" + _ctx.size, {
            "is-center": _ctx.center
          }, _ctx.customClass])
        },
        [_ctx.showIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.getIcon), {
          key: 0,
          class: "tiny-svg-size tiny-alert__icon"
        })) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_1, [_ctx.size === "large" ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "title", {}, function() {
          return [createTextVNode(
            toDisplayString(_ctx.state.getTitle),
            1
            /* TEXT */
          )];
        })])) : createCommentVNode("v-if", true), createElementVNode(
          "div",
          {
            class: normalizeClass(["tiny-alert__description", {
              "is-hide": _ctx.size === "large" && !_ctx.description && !_ctx.slots.description
            }])
          },
          [renderSlot(_ctx.$slots, "description", {}, function() {
            return [createTextVNode(
              toDisplayString(_ctx.description),
              1
              /* TEXT */
            )];
          })],
          2
          /* CLASS */
        ), _ctx.size === "large" ? (openBlock(), createElementBlock("div", _hoisted_3, [renderSlot(_ctx.$slots, "default")])) : createCommentVNode("v-if", true)]), !_ctx.closeText && _ctx.closable ? (openBlock(), createBlock(_component_icon_close, {
          key: 1,
          onClick: _ctx.handleClose,
          class: "tiny-svg-size tiny-alert__icon tiny-alert__close"
        }, null, 8, ["onClick"])) : !_ctx.closeText && !_ctx.closable ? (openBlock(), createElementBlock("span", _hoisted_4, [renderSlot(_ctx.$slots, "close")])) : _ctx.closeText && _ctx.closable ? (openBlock(), createElementBlock(
          "span",
          {
            key: 3,
            onClick: _cache[0] || (_cache[0] = function() {
              return _ctx.handleClose && _ctx.handleClose.apply(_ctx, arguments);
            }),
            class: "is-custom"
          },
          toDisplayString(_ctx.closeText),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true)];
    }),
    _: 3
    /* FORWARDED */
  });
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
