// @opentiny/vue-renderless/option/index.js
import "../chunk-G2ADBYYC.js";
import { getObj } from "../common/object";
const escapeRegexpString = (value = "") => String(value).replace(/[|\\{}()[\]^$+*?.]/g, "\\$&");
const isEqual = ({ select, state }) => (a, b) => {
  if (!state.isObject) {
    return a === b;
  } else {
    const valueKey = select.valueKey;
    return getObj(a, valueKey) === getObj(b, valueKey);
  }
};
const contains = ({ select, state }) => (arr = [], target = null) => {
  if (!state.isObject) {
    return arr && arr.includes(target);
  } else {
    const valueKey = select.valueKey;
    return arr && arr.some((item) => getObj(item, valueKey) === getObj(target, valueKey));
  }
};
const handleGroupDisabled = (state) => (val) => {
  state.groupDisabled = val;
};
const hoverItem = ({ select, props, state, vm }) => () => {
  if (!props.disabled && !state.groupDisabled) {
    select.state.hoverIndex = select.state.options.indexOf(vm);
  }
};
const selectOptionClick = ({ props, state, select, constants, vm }) => () => {
  if (props.disabled !== true && state.groupDisabled !== true) {
    if (select.multiple && props.required === true)
      return;
    select.state.selectEmitter.emit(constants.EVENT_NAME.handleOptionClick, vm, true);
  }
};
const queryChange = ({ props, state }) => (query) => {
  state.visible = new RegExp(escapeRegexpString(query), "i").test(state.currentLabel) || props.created;
};
const toggleEvent = ({ props, vm, type }) => {
  const optionEl = vm.$refs.option;
  Object.keys(props.events).forEach((ev) => {
    optionEl[type + "EventListener"](ev, props.events[ev]);
  });
};
const initValue = ({ select, props, constants, vm }) => () => {
  if (select.multiple && props.required) {
    select.state.selectEmitter.emit(constants.EVENT_NAME.initValue, vm);
  }
};
export {
  contains,
  handleGroupDisabled,
  hoverItem,
  initValue,
  isEqual,
  queryChange,
  selectOptionClick,
  toggleEvent
};
