// @opentiny/vue-collapse-item/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/collapse-item/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import CollapseTransition from "@opentiny/vue-collapse-transition";
import { iconChevronRight } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, withKeys, withModifiers, renderSlot, createVNode, createTextVNode, toDisplayString, withCtx, withDirectives, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = /* @__PURE__ */ defineComponent({
  props: [].concat(props, ["title", "titleRight", "name", "disabled"]),
  components: {
    CollapseTransition,
    IconChevronRight: iconChevronRight()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["aria-expanded", "aria-controls", "aria-describedby"];
var _hoisted_2 = ["id", "tabindex"];
var _hoisted_3 = {
  class: "tiny-collapse-item__title__right"
};
var _hoisted_4 = ["aria-hidden", "aria-labelledby", "id"];
var _hoisted_5 = {
  class: "tiny-collapse-item__content"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_chevron_right = resolveComponent("icon-chevron-right");
  var _component_collapse_transition = resolveComponent("collapse-transition");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-collapse-item", {
        "is-active": _ctx.state.isActive,
        "is-disabled": _ctx.disabled
      }])
    },
    [createElementVNode("div", {
      role: "tab",
      "aria-expanded": _ctx.state.isActive,
      "aria-controls": "tiny-collapse-content-" + _ctx.state.id,
      "aria-describedby": "tiny-collapse-content-" + _ctx.state.id
    }, [createElementVNode("div", {
      class: normalizeClass(["tiny-collapse-item__header", {
        focusing: _ctx.state.focusing,
        "is-active": _ctx.state.isActive
      }]),
      role: "button",
      id: "tiny-collapse-head-" + _ctx.state.id,
      tabindex: _ctx.disabled ? void 0 : 0,
      onKeyup: _cache[2] || (_cache[2] = withKeys(withModifiers(function() {
        return _ctx.handleEnterClick && _ctx.handleEnterClick.apply(_ctx, arguments);
      }, ["stop"]), ["space", "enter"])),
      onFocus: _cache[3] || (_cache[3] = function() {
        return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
      }),
      onBlur: _cache[4] || (_cache[4] = function($event) {
        return _ctx.state.focusing = false;
      }),
      onClick: _cache[5] || (_cache[5] = function() {
        return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
      })
    }, [createElementVNode("div", {
      class: "tiny-collapse-item__arrow",
      onClick: _cache[0] || (_cache[0] = withModifiers(function() {
        return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
      }, ["stop"]))
    }, [renderSlot(_ctx.$slots, "icon", {}, function() {
      return [createVNode(_component_icon_chevron_right, {
        class: normalizeClass(["tiny-svg-size", {
          "is-active": _ctx.state.isActive
        }])
      }, null, 8, ["class"])];
    })]), createElementVNode("div", {
      class: "tiny-collapse-item__word-overflow",
      onClick: _cache[1] || (_cache[1] = withModifiers(function() {
        return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
      }, ["stop"]))
    }, [renderSlot(_ctx.$slots, "title", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.title),
        1
        /* TEXT */
      )];
    })]), createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "title-right", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.titleRight),
        1
        /* TEXT */
      )];
    })])], 42, _hoisted_2), createVNode(_component_collapse_transition, null, {
      default: withCtx(function() {
        return [withDirectives(createElementVNode("div", {
          class: "tiny-collapse-item__wrap",
          role: "tabpanel",
          "aria-hidden": !_ctx.state.isActive,
          "aria-labelledby": "tiny-collapse-head-" + _ctx.state.id,
          id: "tiny-collapse-content-" + _ctx.state.id
        }, [createElementVNode("div", _hoisted_5, [renderSlot(_ctx.$slots, "default")])], 8, _hoisted_4), [[vShow, _ctx.state.isActive]])];
      }),
      _: 3
      /* FORWARDED */
    })], 8, _hoisted_1)],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
