// @opentiny/vue-filter-bar/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var FilterBar = defineComponent({
  name: $prefix + "FilterBar",
  props: _extends({}, $props, {
    modelValue: Number,
    data: Array
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
FilterBar.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
FilterBar.install = function(Vue) {
  Vue.component(FilterBar.name, FilterBar);
};
FilterBar.version = version;
export {
  FilterBar as default
};
