// @opentiny/vue-filter-bar/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/filter-bar/vue";
import { defineComponent, $props, setup as _setup } from "@opentiny/vue-common";
import { IconDown } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, Fragment, renderList, normalizeClass, withModifiers, renderSlot, createElementVNode, toDisplayString, createVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    IconDown: IconDown()
  },
  emits: ["click", "update:modelValue"],
  props: _extends({}, $props, {
    modelValue: Number,
    data: Array
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-filter-bar",
  class: "relative text-sm text-color-text-primary w-full h-11 flex items-center justify-center px-4 border-b-0.5 border-color-border-separator"
};
var _hoisted_2 = ["onClick"];
var _hoisted_3 = ["title"];
var _hoisted_4 = {
  class: "h-full flex items-center justify-center truncate",
  "data-tag": "tiny-filter-bar-icon"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconDown = resolveComponent("IconDown");
  return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.data, function(item, index) {
      return openBlock(), createElementBlock("div", {
        "data-tag": "tiny-filter-bar-body",
        key: index,
        class: normalizeClass(["flex-1 flex items-center justify-center cursor-pointer overflow-hidden mx-3", {
          "text-color-brand": _ctx.modelValue === index
        }]),
        onClick: withModifiers(function($event) {
          return _ctx.click(index);
        }, ["stop"])
      }, [renderSlot(_ctx.$slots, "default", {
        data: item,
        index,
        active: _ctx.modelValue === index
      }, function() {
        return [createElementVNode("span", {
          "data-tag": "tiny-filter-bar-item",
          class: "truncate",
          title: item
        }, toDisplayString(item), 9, _hoisted_3), createVNode(_component_IconDown, {
          "custom-class": "flex-none w-3 h-3 ml-2 fill-current transition-transform duration-300",
          class: normalizeClass(_ctx.modelValue !== index ? "text-color-icon-placeholder rotate-0" : "rotate-180")
        }, null, 8, ["class"])];
      })], 10, _hoisted_2);
    }),
    128
    /* KEYED_FRAGMENT */
  )), createElementVNode("div", _hoisted_4, [renderSlot(_ctx.$slots, "icon")])]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
