// @opentiny/vue-timeline-item/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  PROCESS_DONE_CLS: "process-done",
  PROCESS_CUR_CLS: "process-current",
  PROCESS_WAIT_CLS: "process-wait",
  PROCESS_DISABLED_CLS: "process-disabled",
  PROCESS_ERROR_CLS: "process-error",
  STACK_NODES_MAX: 7,
  LIMITED_STACK_NODES: 3
};
var timelineItemProps = _extends({}, $props, {
  node: {
    type: Object,
    default: {}
  },
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  space: {
    type: [String, Number],
    default: ""
  },
  lineWidth: {
    type: [String, Number],
    default: ""
  }
});
var TimelineItem = defineComponent({
  name: $prefix + "TimelineItem",
  props: timelineItemProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
TimelineItem.install = function(Vue) {
  Vue.component(TimelineItem.name, TimelineItem);
};
TimelineItem.version = version;
export {
  TimelineItem as default
};
