// @opentiny/vue-currency/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/currency/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Currency = defineComponent({
  name: $prefix + "Currency",
  props: _extends({}, $props, {
    clearable: {
      type: Boolean,
      default: true
    },
    currency: {
      type: String,
      default: "CNY"
    },
    disabled: {
      type: Boolean,
      default: false
    },
    fetchCurrency: Function,
    fields: Object,
    modelValue: {},
    placeholder: {
      type: String,
      default: ""
    },
    popperAppendToBody: {
      type: Boolean,
      default: true
    },
    popperClass: String,
    size: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Currency.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Currency.install = function(Vue) {
  Vue.component(Currency.name, Currency);
};
Currency.version = version;
export {
  Currency as default
};
