// @opentiny/vue-image-viewer/lib/mobile.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/image-viewer/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconClose, iconChevronLeft, iconChevronRight, iconZoomOut, iconZoomIn, iconRepeat, iconRefres, iconDel } from "@opentiny/vue-icon";
import { directive } from "@opentiny/vue-common";
import { isObject } from "@opentiny/vue-renderless/common/type";
var TinyTouch = /* @__PURE__ */ function() {
  function TinyTouch2(element, tinyBinding, type) {
    var that = this;
    that.element = element;
    that.tinyBinding = tinyBinding;
    that.touchType = type;
    that.tinyVueTouches = {
      x: 0,
      y: 0
    };
    that.tinyVueMoves = true;
    that.tinyVueLeave = true;
    that.tinyLongTouch = true;
    that.tinyVueCallBack = isObject(tinyBinding.value) ? tinyBinding.value.fn : tinyBinding.value;
    that.element.addEventListener("touchstart", function(e) {
      that.start(e);
    });
    that.element.addEventListener("touchend", function(e) {
      that.end(e);
    });
    that.element.addEventListener("touchmove", function(e) {
      that.move(e);
    });
  }
  var _proto = TinyTouch2.prototype;
  _proto.start = function start(e) {
    var _this = this;
    if (e.touches >= 2) {
      return;
    }
    this.tinyVueMoves = true;
    this.tinyVueLeave = true;
    this.tinyLongTouch = true;
    this.tinyVueTouches = {
      x: e.changedTouches[0].pageX,
      y: e.changedTouches[0].pageY
    };
    this.time = setTimeout(function() {
      if (_this.tinyVueLeave && _this.tinyVueMoves) {
        _this.touchType == "longtap" && _this.tinyVueCallBack(_this.tinyBinding.value, e);
        _this.tinyLongTouch = false;
      }
    }, 1e3);
  };
  _proto.end = function end(e) {
    if (e.touches >= 2) {
      return;
    }
    var disX = e.changedTouches[0].pageX - this.tinyVueTouches.x;
    var disY = e.changedTouches[0].pageY - this.tinyVueTouches.y;
    clearTimeout(this.time);
    if (Math.abs(disX) > 10 || Math.abs(disY) > 100) {
      this.touchType == "swipe" && this.tinyVueCallBack(this.tinyBinding.value, e);
      if (Math.abs(disX) > Math.abs(disY)) {
        if (disX > 10) {
          this.touchType == "swiperight" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
        if (disX < -10) {
          this.touchType == "swipeleft" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
      } else {
        if (disY > 10) {
          this.touchType == "swipedown" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
        if (disY < -10) {
          this.touchType == "swipeup" && this.tinyVueCallBack(this.tinyBinding.value, e);
        }
      }
    } else {
      if (this.tinyLongTouch && this.tinyVueMoves) {
        this.touchType == "tap" && this.tinyVueCallBack(this.tinyBinding.value, e);
        this.tinyVueLeave = false;
      }
    }
  };
  _proto.move = function move() {
    this.tinyVueMoves = false;
  };
  return TinyTouch2;
}();
var mapDirective = function mapDirective2() {
  var deactives = {};
  var names = ["tap", "swipe", "swipeleft", "swiperight", "swipedown", "swipeup", "longtap"];
  names.forEach(function(name) {
    deactives[name] = directive({
      vTouch: {
        bind: function bind(el, tinyBinding) {
          new TinyTouch(el, tinyBinding, name);
        }
      }
    }).vTouch;
  });
  return deactives;
};
var touchDeactives = mapDirective();

import "@opentiny/vue-theme-mobile/image-viewer/index.css";
import { resolveComponent, resolveDirective, openBlock, createBlock, Transition, withCtx, withDirectives, createElementBlock, normalizeStyle, withModifiers, createVNode, createCommentVNode, renderSlot, createElementVNode, toDisplayString, Fragment, normalizeClass, renderList, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-renderless/common/type";
var _sfc_main = /* @__PURE__ */ defineComponent({
  props: [].concat(props, ["urlList", "zIndex", "onSwitch", "onClose", "previewVisible", "closeShow", "arrowShow", "toolShow", "showIndex", "imageFullCurrent", "startPosition", "asyncClose", "deleteButton"]),
  components: {
    IconClose: iconClose(),
    IconChevronLeft: iconChevronLeft(),
    IconChevronRight: iconChevronRight(),
    IconZoomOut: iconZoomOut(),
    IconZoomIn: iconZoomIn(),
    IconRepeat: iconRepeat(),
    IconRefres: iconRefres(),
    IconDel: iconDel()
  },
  directives: _extends({}, touchDeactives),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "tiny-mobile-image-viewer__mask"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_2 = {
  key: 1,
  class: "tiny-mobile-image-viewer__btn tiny-mobile-image-viewer__index"
};
var _hoisted_3 = {
  class: "tiny-mobile-image-viewer__index--curren"
};
var _hoisted_4 = /* @__PURE__ */ createElementVNode(
  "span",
  null,
  "/",
  -1
  /* HOISTED */
);
var _hoisted_5 = {
  class: "tiny-mobile-image-viewer__index--total"
};
var _hoisted_6 = {
  key: 3,
  class: "tiny-mobile-image-viewer__btn tiny-mobile-image-viewer__actions"
};
var _hoisted_7 = {
  class: "tiny-mobile-image-viewer__actions-inner"
};
var _hoisted_8 = {
  class: "tiny-mobile-image-viewer__canvas"
};
var _hoisted_9 = {
  class: "tiny-mobile-image-viewer__detail",
  style: {
    "transition-duration": "0.3s"
  }
};
var _hoisted_10 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_close = resolveComponent("icon-close");
  var _component_icon_chevron_left = resolveComponent("icon-chevron-left");
  var _component_icon_chevron_right = resolveComponent("icon-chevron-right");
  var _component_icon_zoom_out = resolveComponent("icon-zoom-out");
  var _component_icon_zoom_in = resolveComponent("icon-zoom-in");
  var _component_icon_repeat = resolveComponent("icon-repeat");
  var _component_icon_refres = resolveComponent("icon-refres");
  var _component_icon_del = resolveComponent("icon-del");
  var _directive_swipeleft = resolveDirective("swipeleft");
  var _directive_swiperight = resolveDirective("swiperight");
  return openBlock(), createBlock(Transition, {
    name: "viewer-fade",
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives((openBlock(), createElementBlock(
        "div",
        {
          class: "tiny-mobile-image-viewer__wrapper",
          ref: "imagePreview",
          style: normalizeStyle({
            "z-index": _ctx.zIndex
          }),
          onClick: _cache[14] || (_cache[14] = function() {
            return _ctx.handleVisible && _ctx.handleVisible.apply(_ctx, arguments);
          })
        },
        [_hoisted_1, _ctx.closeShow ? (openBlock(), createElementBlock("span", {
          key: 0,
          class: "tiny-mobile-image-viewer__btn tiny-mobile-image-viewer__close",
          onClick: _cache[0] || (_cache[0] = withModifiers(function() {
            return _ctx.handleVisible && _ctx.handleVisible.apply(_ctx, arguments);
          }, ["stop"]))
        }, [createVNode(_component_icon_close, {
          class: "tiny-svg-size"
        })])) : createCommentVNode("v-if", true), _ctx.showIndex ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "index", {
          value: _ctx.state.index + 1
        }, function() {
          return [createElementVNode(
            "span",
            _hoisted_3,
            toDisplayString(_ctx.state.index + 1),
            1
            /* TEXT */
          ), _hoisted_4, createElementVNode(
            "span",
            _hoisted_5,
            toDisplayString(_ctx.urlList.length),
            1
            /* TEXT */
          )];
        })])) : createCommentVNode("v-if", true), !_ctx.state.isSingle && _ctx.arrowShow ? (openBlock(), createElementBlock(
          Fragment,
          {
            key: 2
          },
          [createElementVNode(
            "span",
            {
              class: normalizeClass(["tiny-mobile-image-viewer__btn tiny-mobile-image-viewer__prev", {
                "is-disabled": !_ctx.state.infinite && _ctx.state.isFirst,
                "is-arrow-disabled": _ctx.state.arrowStyle === "N"
              }]),
              onClick: _cache[1] || (_cache[1] = withModifiers(function() {
                return _ctx.swipeRight && _ctx.swipeRight.apply(_ctx, arguments);
              }, ["stop"]))
            },
            [createVNode(_component_icon_chevron_left, {
              class: "tiny-svg-size"
            })],
            2
            /* CLASS */
          ), createElementVNode(
            "span",
            {
              class: normalizeClass(["tiny-mobile-image-viewer__btn tiny-mobile-image-viewer__next", {
                "is-disabled": !_ctx.state.infinite && _ctx.state.isLast,
                "is-arrow-disabled": _ctx.state.arrowStyle === "Y"
              }]),
              onClick: _cache[2] || (_cache[2] = withModifiers(function() {
                return _ctx.swipeLeft && _ctx.swipeLeft.apply(_ctx, arguments);
              }, ["stop"]))
            },
            [createVNode(_component_icon_chevron_right, {
              class: "tiny-svg-size"
            })],
            2
            /* CLASS */
          )],
          64
          /* STABLE_FRAGMENT */
        )) : createCommentVNode("v-if", true), _ctx.toolShow ? (openBlock(), createElementBlock("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [renderSlot(_ctx.$slots, "tool", {}, function() {
          return [createVNode(_component_icon_zoom_out, {
            class: "tiny-svg-size",
            onClick: _cache[3] || (_cache[3] = withModifiers(function($event) {
              return _ctx.handleActions("zoomOut");
            }, ["stop"]))
          }), createVNode(_component_icon_zoom_in, {
            class: "tiny-svg-size",
            onClick: _cache[4] || (_cache[4] = withModifiers(function($event) {
              return _ctx.handleActions("zoomIn");
            }, ["stop"]))
          }), createVNode(_component_icon_repeat, {
            class: "tiny-svg-size",
            onClick: _cache[5] || (_cache[5] = withModifiers(function($event) {
              return _ctx.handleActions("anticlocelise");
            }, ["stop"]))
          }), createVNode(_component_icon_refres, {
            class: "tiny-svg-size",
            onClick: _cache[6] || (_cache[6] = withModifiers(function($event) {
              return _ctx.handleActions("clocelise");
            }, ["stop"]))
          }), _ctx.deleteButton ? (openBlock(), createBlock(_component_icon_del, {
            key: 0,
            class: "tiny-svg-size",
            onClick: _cache[7] || (_cache[7] = withModifiers(function($event) {
              return _ctx.handleDelete();
            }, ["stop"]))
          })) : createCommentVNode("v-if", true)];
        })])])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_8, [createElementVNode(
          "div",
          {
            class: "tiny-mobile-image-viewer__wrap",
            style: normalizeStyle({
              width: _ctx.state.iamgeAllWidth + "px",
              "transition-duration": _ctx.state.imageTransition + "ms",
              transform: "translateX(" + _ctx.state.imageTransformSize + "px)"
            })
          },
          [(openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(_ctx.state.urlList, function(url, i) {
              return openBlock(), createElementBlock(
                "div",
                {
                  class: "tiny-mobile-image-viewer__item",
                  key: i,
                  style: normalizeStyle(Object.assign({
                    width: _ctx.state.imageItemWidth + "px"
                  }, i === _ctx.state.index ? _ctx.state.imgStyle : ""))
                },
                [createElementVNode("div", _hoisted_9, [(openBlock(), createElementBlock("img", {
                  ref_for: true,
                  ref: "img_" + i,
                  class: normalizeClass(["tiny-mobile-image-viewer__img", {
                    "is-full-screen": _ctx.state.fullScreen
                  }]),
                  key: url,
                  src: url,
                  onLoad: _cache[8] || (_cache[8] = function() {
                    return _ctx.handleImgLoad && _ctx.handleImgLoad.apply(_ctx, arguments);
                  }),
                  onError: _cache[9] || (_cache[9] = function() {
                    return _ctx.handleImgError && _ctx.handleImgError.apply(_ctx, arguments);
                  }),
                  onMousedown: _cache[10] || (_cache[10] = function() {
                    return _ctx.handleMouseDown && _ctx.handleMouseDown.apply(_ctx, arguments);
                  }),
                  onTouchstart: _cache[11] || (_cache[11] = function() {
                    return _ctx.touchstart && _ctx.touchstart.apply(_ctx, arguments);
                  }),
                  onTouchmove: _cache[12] || (_cache[12] = function() {
                    return _ctx.touchmove && _ctx.touchmove.apply(_ctx, arguments);
                  }),
                  onTouchend: _cache[13] || (_cache[13] = function() {
                    return _ctx.touchend && _ctx.touchend.apply(_ctx, arguments);
                  })
                }, null, 42, _hoisted_10))])],
                4
                /* STYLE */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))],
          4
          /* STYLE */
        )])],
        4
        /* STYLE */
      )), [[vShow, _ctx.state.previewVisible], [_directive_swipeleft, _ctx.swipeLeft], [_directive_swiperight, _ctx.swipeRight]])];
    }),
    _: 3
    /* FORWARDED */
  });
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
