// @opentiny/vue-radio/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/radio/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconRadio, iconRadioselected, iconMobileRadio, iconMobileRadioSelected } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, mergeProps, withKeys, withModifiers, createElementVNode, normalizeClass, createVNode, withDirectives, vModelRadio, renderSlot, createTextVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var classes = {
  "radio-default": "radio inline-flex items-center align-middle leading-4 cursor-pointer sm:flex-row py-px sm:py-0 sm:h-fit h-11",
  "radio-label-common": "relative text-center w-11 h-11",
  "radio-label-size-common": "sm:w-4 sm:h-4",
  "radio-label-size-medium": "sm:w-6 sm:h-6",
  "radio-label-circle": "inline-flex p-3 sm:p-0",
  "icon-common": "absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 z-0 transition-all",
  "icon-show": "opacity-100 scale-100 z-10",
  "icon-hidden": "opacity-0 scale-0 z-0",
  "icon-radioselected-common": "w-5 h-5 fill-color-bg-1 flex-1 [&_circle:nth-of-type(1)]:fill-color-bg-1",
  "icon-radioselected-size-common": "sm:w-4 sm:h-4",
  "icon-radioselected-size-medium": "sm:w-6 sm:h-6",
  "icon-radioselected-disabled": "fill-color-brand-disabled cursor-not-allowed [&_circle:nth-of-type(1)]:fill-color-bg-1",
  "icon-radioselected-default": "fill-color-brand [&_circle:nth-of-type(1)]:shadow-xsm",
  "icon-radio-common": "w-5 h-5 flex-1",
  "icon-radio-size-common": "sm:w-4 sm:h-4",
  "icon-radio-size-medium": "sm:w-6 sm:h-6",
  "icon-radio-default": "fill-color-bg-1 [&_path:nth-of-type(1)]:fill-color-icon-secondary",
  "icon-radio-disabled": "[&_path:nth-of-type(1)]:fill-color-icon-disabled [&_path:nth-of-type(2)]:fill-color-bg-3 cursor-not-allowed",
  "mobile-icon-radioselected-disabled": "fill-color-bg-1 [&_path:nth-of-type(1)]:fill-color-brand-disabled cursor-not-allowed",
  "mobile-icon-radioselected-default": "fill-color-bg-1 [&_path:nth-of-type(1)]:fill-color-brand [&_path:nth-of-type(1)]:shadow-xsm",
  "mobile-icon-radio-default": "[&_path:nth-of-type(2)]:fill-color-none-hover [&_path:nth-of-type(1)]:fill-color-bg-1",
  "mobile-icon-radio-disabled": "[&_path:nth-of-type(2)]:fill-color-icon-disabled [&_path:nth-of-type(1)]:fill-color-bg-3 cursor-not-allowed",
  "radio-input": "absolute left-0 right-0 top-0 bottom-0 w-0 h-0 -z-10 opacity-0",
  "radio-text-common": "sm:h-4 text-center text-color-text-primary align-middle text-sm",
  "radio-text-size-common": "sm:text-xs",
  "radio-text-size-medium": "sm:text-sm",
  "radio-hover": "sm:[&_path:nth-of-type(1)]:hover:fill-color-brand-hover",
  "radio-hover-selected": "sm:[&_path:nth-of-type(1)]:hover:fill-color-brand-hover",
  "radio-active": "sm:[&_path:nth-of-type(1)]:active:fill-color-brand-active",
  "radio-active-selected": "sm:[&_path:nth-of-type(1)]:active:fill-color-brand-active",
  "label-disabled": "text-color-text-disabled cursor-not-allowed",
  "pc-show": "hidden sm:block",
  "mobile-show": "block sm:hidden",
  "readonly-is-checked": "sm:m-0",
  "readonly-is-not-checked": "sm:hidden",
  "not-readonly-common": "sm:mr-5 sm:[&:last-child]:mr-0",
  "hidden-radio": "sm:hidden",
  "not-readly-common-label": "sm:py-0 sm:pl-2 -ml-0.5 sm:ml-0",
  "readonly-checked-label": "sm:pl-0",
  "verticalStyle": "sm:mb-2 sm:[&:last-child]:mb-0 sm:mr-0"
};
var _sfc_main = defineComponent({
  emits: ["change", "update:modelValue"],
  props: [].concat(props, ["modelValue", "events", "label", "text", "disabled", "name", "border", "size", "tabindex", "displayOnly"]),
  inheritAttrs: false,
  components: {
    IconRadio: iconRadio(),
    IconRadioselected: iconRadioselected(),
    IconMobileRadio: iconMobileRadio(),
    IconMobileRadioSelected: iconMobileRadioSelected()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      classes
    });
  }
});
var _hoisted_1 = ["aria-checked", "aria-disabled"];
var _hoisted_2 = ["tabindex"];
var _hoisted_3 = ["value", "name", "disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_radio = resolveComponent("icon-radio");
  var _component_icon_radioselected = resolveComponent("icon-radioselected");
  var _component_icon_mobile_radio = resolveComponent("icon-mobile-radio");
  var _component_icon_mobile_radio_selected = resolveComponent("icon-mobile-radio-selected");
  return openBlock(), createElementBlock("label", mergeProps({
    "data-tag": "tiny-radio",
    class: _ctx.m(_ctx.gcls("radio-default"), _ctx.gcls(_ctx.state.isDisplayOnly ? _ctx.state.model === _ctx.label ? "readonly-is-checked" : "readonly-is-not-checked" : "not-readonly-common"), _ctx.gcls({
      "verticalStyle": _ctx.state.vertical
    })),
    role: "radio",
    "aria-checked": _ctx.state.model === _ctx.label,
    "aria-disabled": _ctx.state.isDisabled,
    tabindex: "-1",
    onKeydown: _cache[6] || (_cache[6] = withKeys(withModifiers(function($event) {
      return _ctx.state.model = _ctx.state.isDisabled ? _ctx.state.model : _ctx.label;
    }, ["stop", "prevent"]), ["space"]))
  }, _ctx.a(_ctx.$attrs, ["class", "style", "onClick"], true)), [createElementVNode(
    "span",
    {
      "data-tag": "tiny-radio-content",
      class: normalizeClass(_ctx.m(_ctx.gcls("radio-label-common"), _ctx.gcls(_ctx.state.size === "medium" ? "radio-label-size-medium" : "radio-label-size-common"), _ctx.gcls(_ctx.state.isDisplayOnly && _ctx.state.model === _ctx.label && "hidden-radio")))
    },
    [createElementVNode("span", {
      "data-tag": "tiny-radio-icon",
      class: normalizeClass(_ctx.m(_ctx.gcls("radio-label-circle"))),
      tabindex: _ctx.tabindex
    }, [createVNode(_component_icon_radio, {
      class: normalizeClass(_ctx.m(_ctx.gcls("icon-common"), _ctx.gcls("pc-show"), _ctx.gcls({
        "radio-active": !_ctx.state.isDisabled
      }), _ctx.gcls({
        "radio-hover": !_ctx.state.isDisabled
      }), _ctx.gcls({
        "icon-radio-disabled": _ctx.state.isDisabled
      }), _ctx.gcls({
        "icon-radio-default": !_ctx.state.isDisabled
      }), _ctx.gcls("icon-radio-common"), _ctx.gcls(_ctx.state.size === "medium" ? "icon-radio-size-medium" : "icon-radio-size-common")))
    }, null, 8, ["class"]), createVNode(_component_icon_radioselected, {
      class: normalizeClass(_ctx.m(_ctx.gcls("icon-common"), _ctx.gcls("pc-show"), _ctx.state.model === _ctx.label ? _ctx.gcls("icon-show") : _ctx.gcls("icon-hidden"), _ctx.gcls({
        "radio-hover-selected": !_ctx.state.isDisabled
      }), _ctx.gcls({
        "radio-active-selected": !_ctx.state.isDisabled
      }), _ctx.gcls("icon-radioselected-common"), _ctx.gcls({
        "icon-radioselected-disabled": _ctx.state.isDisabled
      }), _ctx.gcls({
        "icon-radioselected-default": !_ctx.state.isDisabled
      }), _ctx.gcls(_ctx.state.size === "medium" ? "icon-radioselected-size-medium" : "icon-radioselected-size-common")))
    }, null, 8, ["class"]), createVNode(_component_icon_mobile_radio, {
      class: normalizeClass(_ctx.m(_ctx.gcls("icon-common"), _ctx.gcls("mobile-show"), _ctx.gcls({
        "mobile-icon-radio-disabled": _ctx.state.isDisabled
      }), _ctx.gcls({
        "mobile-icon-radio-default": !_ctx.state.isDisabled
      }), _ctx.gcls("icon-radio-common")))
    }, null, 8, ["class"]), createVNode(_component_icon_mobile_radio_selected, {
      class: normalizeClass(_ctx.m(_ctx.gcls("icon-common"), _ctx.gcls("mobile-show"), _ctx.state.model === _ctx.label ? _ctx.gcls("icon-show") : _ctx.gcls("icon-hidden"), _ctx.gcls("icon-radioselected-common"), _ctx.gcls({
        "mobile-icon-radioselected-disabled": _ctx.state.isDisabled
      }), _ctx.gcls({
        "mobile-icon-radioselected-default": !_ctx.state.isDisabled
      })))
    }, null, 8, ["class"])], 10, _hoisted_2), withDirectives(createElementVNode("input", {
      ref: "radio",
      class: normalizeClass(_ctx.m(_ctx.gcls("radio-input"))),
      value: _ctx.label,
      type: "radio",
      "aria-hidden": "true",
      "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.model = $event;
      }),
      onFocus: _cache[1] || (_cache[1] = function($event) {
        return _ctx.state.focus = true;
      }),
      onBlur: _cache[2] || (_cache[2] = function($event) {
        return _ctx.state.focus = false;
      }),
      onChange: _cache[3] || (_cache[3] = function() {
        return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
      }),
      name: _ctx.name,
      disabled: _ctx.state.isDisabled,
      onClick: _cache[4] || (_cache[4] = withModifiers(function() {
      }, ["stop"]))
    }, null, 42, _hoisted_3), [[vModelRadio, _ctx.state.model]])],
    2
    /* CLASS */
  ), createElementVNode(
    "span",
    {
      "data-tag": "tiny-radio-label",
      ref: "label",
      class: normalizeClass(_ctx.m(_ctx.gcls("radio-text-common"), _ctx.gcls(_ctx.state.size === "medium" ? "radio-text-size-medium" : "radio-text-size-common"), _ctx.gcls({
        "label-disabled": _ctx.state.isDisabled
      }), _ctx.gcls(_ctx.state.isDisplayOnly ? _ctx.state.model === _ctx.label ? "readonly-checked-label" : "" : "not-readly-common-label"))),
      onKeydown: _cache[5] || (_cache[5] = withModifiers(function() {
      }, ["stop"]))
    },
    [renderSlot(_ctx.$slots, "default", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.text || _ctx.label),
        1
        /* TEXT */
      )];
    })],
    34
    /* CLASS, NEED_HYDRATION */
  )], 16, _hoisted_1);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
