// @opentiny/vue-rate/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/rate/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconStarActive, iconStarDisable, iconSmile, iconFrown, iconMeh } from "@opentiny/vue-icon";
import { openBlock, createElementBlock, Fragment, renderList, normalizeStyle, createBlock, resolveDynamicComponent, normalizeClass, createCommentVNode, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["modelValue", "radio", "lowThreshold", "highThreshold", "max", "colors", "voidColor", "disabledVoidColor", "iconClasses", "voidIconClass", "disabledVoidIconClass", "disabled", "allowHalf", "showText", "showScore", "textColor", "texts", "textOnBottom", "space", "size", "scoreTemplate"]),
  components: {
    IconStarActive: iconStarActive(),
    IconStarDisable: iconStarDisable(),
    IconSmile: iconSmile(),
    IconFrown: iconFrown(),
    IconMeh: iconMeh()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = ["onMousemove", "onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-rate",
      tabindex: "0",
      onKeydown: _cache[1] || (_cache[1] = function() {
        return _ctx.handelKey && _ctx.handelKey.apply(_ctx, arguments);
      })
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.max, function(item) {
        return openBlock(), createElementBlock("span", {
          key: item,
          style: normalizeStyle({
            width: _ctx.space,
            cursor: _ctx.disabled ? "auto" : "pointer"
          }),
          class: "tiny-rate__star",
          onMousemove: function onMousemove($event) {
            return _ctx.setCurrentValue({
              item,
              event: $event
            });
          },
          onMouseleave: _cache[0] || (_cache[0] = function() {
            return _ctx.resetCurrentValue && _ctx.resetCurrentValue.apply(_ctx, arguments);
          }),
          onClick: function onClick($event) {
            return _ctx.selectValue(item);
          }
        }, [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.classes[item - 1]), {
          style: normalizeStyle(_ctx.getIconStyle(item)),
          class: normalizeClass(["tiny-svg-size", "tiny-rate__icon", {
            hover: _ctx.state.hoverIndex === item
          }, _ctx.disabled ? "disabled" : "", _ctx.showDecimalIcon(item) ? "ratehalf" : ""])
        }, null, 8, ["style", "class"])), _ctx.showDecimalIcon(item) ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.decimalIconClass), {
          key: 0,
          style: normalizeStyle(_ctx.state.decimalStyle),
          class: "tiny-svg-size tiny-rate__decimal"
        }, null, 8, ["style"])) : createCommentVNode("v-if", true), _ctx.textOnBottom && !_ctx.radio ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            style: normalizeStyle({
              color: _ctx.textColor
            }),
            class: "tiny-rate__bottom-text"
          },
          toDisplayString(_ctx.texts[item - 1]),
          5
          /* TEXT, STYLE */
        )) : createCommentVNode("v-if", true), _ctx.textOnBottom && _ctx.radio ? (openBlock(), createElementBlock(
          "div",
          {
            key: 2,
            style: normalizeStyle(_ctx.getTextStyle()),
            class: "tiny-rate__bottom-text"
          },
          toDisplayString(_ctx.texts[item - 1]),
          5
          /* TEXT, STYLE */
        )) : createCommentVNode("v-if", true)], 44, _hoisted_1);
      }),
      128
      /* KEYED_FRAGMENT */
    )), !_ctx.textOnBottom && _ctx.showText || _ctx.showScore ? (openBlock(), createElementBlock(
      "span",
      {
        key: 0,
        style: normalizeStyle({
          color: _ctx.textColor
        }),
        class: "tiny-rate__text"
      },
      toDisplayString(_ctx.state.text),
      5
      /* TEXT, STYLE */
    )) : createCommentVNode("v-if", true)],
    32
    /* NEED_HYDRATION */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
