// @opentiny/vue-renderless/year-table/vue.js
import "../chunk-G2ADBYYC.js";
import { getRows, getCellStyle, handleYearTableClick, watchDate, markRange, handleMouseMove } from "./index";
const api = ["state", "handleYearTableClick", "getCellStyle", "handleMouseMove"];
const renderless = (props, { computed, reactive, watch }, { emit, vm }) => {
  const api2 = {};
  const state = reactive({
    tableRows: [[], [], []],
    rows: computed(() => api2.getRows())
  });
  Object.assign(api2, {
    state,
    getCellStyle: getCellStyle({ props }),
    handleYearTableClick: handleYearTableClick({ emit, props, state }),
    markRange: markRange({ props, state }),
    watchDate: watchDate({ api: api2, props }),
    getRows: getRows({ props, state, vm }),
    handleMouseMove: handleMouseMove({ api: api2, emit, props, state })
  });
  watch(
    () => props.rangeState,
    (value, oldValue) => value !== oldValue && api2.markRange(props.minDate, value.endDate),
    { deep: true }
  );
  watch(() => props.minDate, api2.watchDate);
  watch(() => props.maxDate, api2.watchDate);
  return api2;
};
export {
  api,
  renderless
};
