// @opentiny/vue-wizard/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $setup, $props } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/wizard/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  DOING_STATUS: "doing",
  READY_STATUS: "ready",
  WAIT_STATUS: "wait"
};
var wizardProps = _extends({}, $props, {
  _constants: {
    type: Object,
    default: function _default() {
      return $constants;
    }
  },
  data: {
    type: Array,
    default: function _default2() {
      return [];
    }
  },
  pageGuide: {
    type: Boolean,
    default: false
  },
  vertical: {
    type: Boolean,
    default: false
  },
  timeLineFlow: {
    type: Boolean,
    default: false
  }
});
var Wizard = defineComponent({
  name: $prefix + "Wizard",
  props: wizardProps,
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Wizard.install = function(Vue) {
  Vue.component(Wizard.name, Wizard);
};
Wizard.version = version;
export {
  Wizard as default
};
