// @opentiny/vue-input/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/input/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import TinyTallStorage from "./tall-storage";
import { IconClose, IconError, IconEyeopen, IconEyeclose } from "@opentiny/vue-icon";
import Tooltip from "@opentiny/vue-tooltip";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, Fragment, renderSlot, createCommentVNode, createElementVNode, createBlock, withCtx, toDisplayString, mergeProps, resolveDynamicComponent, withModifiers } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  inheritAttrs: false,
  emits: ["update:modelValue", "change", "clear", "focus", "blur", "keyup", "keydown", "paste", "mouseenter", "mouseleave", "click"],
  components: {
    IconClose: IconClose(),
    IconError: IconError(),
    IconEyeopen: IconEyeopen(),
    IconEyeclose: IconEyeclose(),
    TinyTallStorage,
    TinyTooltip: Tooltip
  },
  props: [].concat(props, ["name", "size", "form", "type", "label", "modelValue", "height", "resize", "counter", "autosize", "disabled", "readonly", "tabindex", "clearable", "suffixIcon", "prefixIcon", "autocomplete", "showPassword", "validateEvent", "showWordLimit", "customClass", "displayOnly", "displayOnlyContent"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  "data-tag": "tiny-input-prepend",
  ref: "prepend",
  class: "border-r-0 rounded-tr-none rounded-br-none bg-color-bg-2 text-color-text-placeholder border border-solid border-color-border rounded px-3 py-0 w-px align-middle table-cell relative whitespace-nowrap"
};
var _hoisted_2 = {
  class: "relative",
  "data-tag": "tiny-input-display-only"
};
var _hoisted_3 = {
  key: 0,
  class: "absolute top-0 left-0 max-w-full overflow-hidden text-ellipsis whitespace-nowrap"
};
var _hoisted_4 = {
  key: 1,
  class: "absolute top-0 left-0 max-w-full overflow-hidden text-ellipsis whitespace-nowrap"
};
var _hoisted_5 = ["name", "tabindex", "type", "disabled", "readonly", "unselectable", "autocomplete", "aria-label"];
var _hoisted_6 = {
  key: 2,
  ref: "prefix",
  class: "left-2 transition-all duration-300 ease-in-out text-xs absolute top-1/2 -translate-y-1/2 text-center text-color-text-placeholder flex items-center"
};
var _hoisted_7 = {
  key: 3,
  ref: "suffix",
  class: "right-2 transition-all duration-300 ease-in-out pointer-events-none text-xs absolute top-1/2 -translate-y-1/2 text-center text-color-text-placeholder flex items-center"
};
var _hoisted_8 = {
  class: "pointer-events-auto text-xs flex justify-start items-center"
};
var _hoisted_9 = {
  key: 4,
  class: "h-full inline-flex items-center text-xs text-color-text-placeholder"
};
var _hoisted_10 = {
  class: "bg-white leading-none inline-block"
};
var _hoisted_11 = {
  key: 4,
  ref: "append",
  class: "border-l-0 rounded-tl-none rounded-bl-none bg-color-bg-2 text-color-text-placeholder border border-solid border-color-border rounded-tr rounded-br px-3 py-0 w-px align-middle table-cell relative whitespace-nowrap"
};
var _hoisted_12 = {
  key: 5,
  ref: "panel"
};
var _hoisted_13 = {
  key: 1,
  "data-tag": "tiny-input-textarea"
};
var _hoisted_14 = {
  class: "max-w-full text-ellipsis break-words line-clamp-5"
};
var _hoisted_15 = ["tabindex", "disabled", "readonly", "unselectable", "autocomplete", "aria-label"];
var _hoisted_16 = {
  key: 2,
  "data-tag": "tiny-input-limit",
  class: "bg-white text-color-text-placeholder text-xs absolute bottom-1 right-3"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  var _component_tiny_tall_storage = resolveComponent("tiny-tall-storage");
  var _component_icon_error = resolveComponent("icon-error");
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(_ctx.m("relative text-xs", _ctx.type !== "textarea" || _ctx.slots.prepend || _ctx.slots.append ? "inline-table" : "", _ctx.type !== "textarea" ? "w-full" : "w-max align-bottom", _ctx.slots.prepend || _ctx.slots.append ? "leading-normal border-separate border-spacing-0" : "", _ctx.$attrs.class)),
      "data-tag": "tiny-input",
      style: normalizeStyle(_ctx.$attrs.style),
      onMouseenter: _cache[19] || (_cache[19] = function($event) {
        _ctx.state.hovering = true;
        _ctx.$emit("mouseenter", $event);
      }),
      onMouseleave: _cache[20] || (_cache[20] = function($event) {
        _ctx.state.hovering = false;
        _ctx.$emit("mouseleave", $event);
      }),
      onClick: _cache[21] || (_cache[21] = function($event) {
        return _ctx.$emit("click", $event);
      })
    },
    [_ctx.type !== "textarea" ? (openBlock(), createElementBlock(
      Fragment,
      {
        key: 0
      },
      [_ctx.slots.prepend && !_ctx.state.isDisplayOnly ? (openBlock(), createElementBlock(
        "div",
        _hoisted_1,
        [renderSlot(_ctx.$slots, "prepend")],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), createElementVNode("span", _hoisted_2, [_ctx.state.isDisplayOnly ? (openBlock(), createBlock(_component_tiny_tooltip, {
        key: 0,
        effect: "light",
        content: _ctx.state.displayOnlyTooltip,
        display: _ctx.type === "password",
        placement: "top",
        onMouseenter: _ctx.handleEnterDisplayOnlyContent
      }, {
        default: withCtx(function() {
          return [_ctx.type === "password" ? (openBlock(), createElementBlock(
            "span",
            _hoisted_3,
            toDisplayString(_ctx.state.hiddenPassword),
            1
            /* TEXT */
          )) : (openBlock(), createElementBlock(
            "span",
            _hoisted_4,
            toDisplayString(_ctx.displayOnlyContent || _ctx.state.nativeInputValue),
            1
            /* TEXT */
          ))];
        }),
        _: 1
        /* STABLE */
      }, 8, ["content", "display", "onMouseenter"])) : createCommentVNode("v-if", true), _ctx.type !== "textarea" ? (openBlock(), createElementBlock("input", mergeProps({
        key: 1,
        ref: "input",
        name: _ctx.name
      }, _ctx.a(_ctx.$attrs, ["type", "class", "style", "^onw+"]), {
        class: _ctx.m("w-full border-0 px-3 sm:border-solid sm:border-color-border sm:hover:border-color-border-hover sm:focus:border-color-brand-focus sm:disabled:border-color-border placeholder:text-color-text-placeholder placeholder:text-sm text-sm sm:placeholder:text-xs sm:text-xs text-color-text-primary bg-color-bg-1 disabled:cursor-not-allowed disabled:text-color-text-disabled disabled:bg-color-bg-4 py-0 outline-0 transition-colors duration-200 ease-in-out ", _ctx.state.inputSizeMf === "medium" ? "h-8 leading-8" : _ctx.state.inputSizeMf === "mini" ? "h-6 leading-6" : "h-7 leading-7", _ctx.slots.prepend || _ctx.slots.append ? "align-middle table-cell" : "inline-block", _ctx.slots.prepend && _ctx.slots.append ? "rounded-none" : _ctx.slots.prepend ? "rounded-tl-none rounded-bl-none rounded-tr rounded-br" : _ctx.slots.append ? "rounded-tl rounded-bl rounded-tr-none rounded-br-none" : "rounded", _ctx.readonly ? "sm:border-0 text-ellipsis overflow-hidden whitespace-nowrap" : "sm:border", (_ctx.slots.prefix || _ctx.prefixIcon) && (_ctx.slots.suffix || _ctx.suffixIcon || _ctx.clearable || _ctx.showPassword) ? "pl-6 pr-6" : _ctx.slots.prefix || _ctx.prefixIcon ? "pl-6 pr-3" : _ctx.slots.suffix || _ctx.suffixIcon || _ctx.clearable || _ctx.showPassword ? "pl-3 pr-6" : "", _ctx.state.isDisplayOnly ? "invisible h-auto leading-none border-0" : "", _ctx.customClass),
        tabindex: _ctx.tabindex,
        type: _ctx.showPassword ? _ctx.state.passwordVisible ? "text" : "password" : _ctx.type,
        disabled: _ctx.state.inputDisabled,
        readonly: _ctx.readonly,
        unselectable: _ctx.readonly ? "on" : "off",
        autocomplete: _ctx.autocomplete,
        onCompositionstart: _cache[0] || (_cache[0] = function() {
          return _ctx.handleCompositionStart && _ctx.handleCompositionStart.apply(_ctx, arguments);
        }),
        onCompositionupdate: _cache[1] || (_cache[1] = function() {
          return _ctx.handleCompositionUpdate && _ctx.handleCompositionUpdate.apply(_ctx, arguments);
        }),
        onCompositionend: _cache[2] || (_cache[2] = function() {
          return _ctx.handleCompositionEnd && _ctx.handleCompositionEnd.apply(_ctx, arguments);
        }),
        onInput: _cache[3] || (_cache[3] = function() {
          return _ctx.handleInput && _ctx.handleInput.apply(_ctx, arguments);
        }),
        onFocus: _cache[4] || (_cache[4] = function() {
          return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
        }),
        onBlur: _cache[5] || (_cache[5] = function() {
          return _ctx.handleBlur && _ctx.handleBlur.apply(_ctx, arguments);
        }),
        onChange: _cache[6] || (_cache[6] = function() {
          return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
        }),
        "aria-label": _ctx.label,
        onKeyup: _cache[7] || (_cache[7] = function($event) {
          return _ctx.$emit("keyup", $event);
        }),
        onKeydown: _cache[8] || (_cache[8] = function($event) {
          return _ctx.$emit("keydown", $event);
        }),
        onPaste: _cache[9] || (_cache[9] = function($event) {
          return _ctx.$emit("paste", $event);
        })
      }), null, 16, _hoisted_5)) : createCommentVNode("v-if", true)]), _ctx.isMemoryStorage ? (openBlock(), createBlock(_component_tiny_tall_storage, {
        key: 1,
        name: _ctx.name,
        "localstorage-data": _ctx.storageData,
        "is-memory-storage": _ctx.isMemoryStorage,
        onSelected: _ctx.selectedMemory
      }, null, 8, ["name", "localstorage-data", "is-memory-storage", "onSelected"])) : createCommentVNode("v-if", true), (_ctx.slots.prefix || _ctx.prefixIcon) && !_ctx.state.isDisplayOnly ? (openBlock(), createElementBlock(
        "span",
        _hoisted_6,
        [renderSlot(_ctx.$slots, "prefix"), _ctx.prefixIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.prefixIcon), {
          key: 0,
          class: normalizeClass(_ctx.m("text-center transition-all duration-300 ease-in-out text-xs", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7"))
        }, null, 8, ["class"])) : createCommentVNode("v-if", true)],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), !_ctx.state.isDisplayOnly && _ctx.getSuffixVisible() ? (openBlock(), createElementBlock(
        "span",
        _hoisted_7,
        [createElementVNode("span", _hoisted_8, [!_ctx.state.showClear || !_ctx.state.showPwdVisible || !_ctx.state.isWordLimitVisible ? (openBlock(), createElementBlock(
          Fragment,
          {
            key: 0
          },
          [renderSlot(_ctx.$slots, "suffix"), _ctx.suffixIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.suffixIcon), {
            key: 0,
            class: normalizeClass(_ctx.m("text-center transition-all duration-300 ease-in-out text-xs", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7"))
          }, null, 8, ["class"])) : createCommentVNode("v-if", true)],
          64
          /* STABLE_FRAGMENT */
        )) : createCommentVNode("v-if", true), _ctx.state.showClear ? (openBlock(), createBlock(_component_icon_error, {
          key: 1,
          class: normalizeClass(_ctx.m("text-center transition-all duration-300 ease-in-out text-xs cursor-pointer fill-color-none-hover block sm:hidden", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7")),
          onMousedown: _cache[10] || (_cache[10] = withModifiers(function() {
          }, ["prevent"])),
          onClick: _ctx.clear
        }, null, 8, ["class", "onClick"])) : createCommentVNode("v-if", true), _ctx.state.showClear ? (openBlock(), createBlock(_component_icon_close, {
          key: 2,
          class: normalizeClass(_ctx.m("hidden sm:block text-center transition-all duration-300 ease-in-out text-xs cursor-pointer", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7")),
          onMousedown: _cache[11] || (_cache[11] = withModifiers(function() {
          }, ["prevent"])),
          onClick: _ctx.clear
        }, null, 8, ["class", "onClick"])) : createCommentVNode("v-if", true), _ctx.showPassword ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.passwordVisible ? "icon-eyeopen" : "icon-eyeclose"), {
          key: 3,
          class: normalizeClass(_ctx.m("text-center transition-all duration-300 ease-in-out text-xs", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7")),
          onClick: _ctx.handlePasswordVisible
        }, null, 8, ["class", "onClick"])) : createCommentVNode("v-if", true), _ctx.state.isWordLimitVisible ? (openBlock(), createElementBlock("span", _hoisted_9, [createElementVNode(
          "span",
          _hoisted_10,
          toDisplayString(_ctx.state.showWordLimit ? _ctx.state.textLength + "/" + _ctx.state.upperLimit : _ctx.state.textLength),
          1
          /* TEXT */
        )])) : createCommentVNode("v-if", true)]), _ctx.state.validateState ? (openBlock(), createElementBlock(
          "i",
          {
            key: 0,
            class: normalizeClass(_ctx.m("text-center transition-all duration-300 ease-in-out text-xs pointer-events-none", _ctx.state.inputSizeMf === "medium" ? "leading-8" : _ctx.state.inputSizeMf === "mini" ? "leading-6" : "leading-7", _ctx.state.validateIcon))
          },
          null,
          2
          /* CLASS */
        )) : createCommentVNode("v-if", true)],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), _ctx.slots.append && !_ctx.state.isDisplayOnly ? (openBlock(), createElementBlock(
        "div",
        _hoisted_11,
        [renderSlot(_ctx.$slots, "append")],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), _ctx.slots.panel && !_ctx.state.isDisplayOnly ? (openBlock(), createElementBlock(
        "div",
        _hoisted_12,
        [renderSlot(_ctx.$slots, "panel")],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true)],
      64
      /* STABLE_FRAGMENT */
    )) : (openBlock(), createElementBlock("span", _hoisted_13, [_ctx.state.isDisplayOnly ? (openBlock(), createBlock(_component_tiny_tooltip, {
      key: 0,
      effect: "light",
      content: _ctx.state.displayOnlyTooltip,
      placement: "top",
      onMouseenter: _ctx.handleEnterDisplayOnlyContent
    }, {
      default: withCtx(function() {
        return [createElementVNode(
          "span",
          _hoisted_14,
          toDisplayString(_ctx.displayOnlyContent || _ctx.state.nativeInputValue),
          1
          /* TEXT */
        )];
      }),
      _: 1
      /* STABLE */
    }, 8, ["content", "onMouseenter"])) : createCommentVNode("v-if", true), createElementVNode("textarea", mergeProps({
      ref: "textarea"
    }, _ctx.a(_ctx.$attrs, ["type", "class", "style", "^on[A-Z]"]), {
      tabindex: _ctx.tabindex,
      class: ["block w-full border-0 sm:border-solid sm:border-color-border sm:hover:border-color-border-hover sm:focus:border-color-brand-focus sm:disabled:border-color-border outline-0 rounded placeholder:text-color-text-placeholder placeholder:text-xs text-xs text-color-text-primary bg-white disabled:cursor-not-allowed disabled:text-color-text-disabled disabled:bg-color-bg-4 leading-normal", [_ctx.readonly ? "sm:border-0 px-0 py-0" : "sm:border px-3 py-2", _ctx.state.isDisplayOnly ? "hidden" : ""]],
      onCompositionstart: _cache[12] || (_cache[12] = function() {
        return _ctx.handleCompositionStart && _ctx.handleCompositionStart.apply(_ctx, arguments);
      }),
      onCompositionupdate: _cache[13] || (_cache[13] = function() {
        return _ctx.handleCompositionUpdate && _ctx.handleCompositionUpdate.apply(_ctx, arguments);
      }),
      onCompositionend: _cache[14] || (_cache[14] = function() {
        return _ctx.handleCompositionEnd && _ctx.handleCompositionEnd.apply(_ctx, arguments);
      }),
      onInput: _cache[15] || (_cache[15] = function() {
        return _ctx.handleInput && _ctx.handleInput.apply(_ctx, arguments);
      }),
      disabled: _ctx.state.inputDisabled,
      readonly: _ctx.readonly,
      unselectable: _ctx.readonly ? "on" : "off",
      autocomplete: _ctx.autocomplete,
      style: _ctx.state.textareaStyle,
      onFocus: _cache[16] || (_cache[16] = function() {
        return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
      }),
      onBlur: _cache[17] || (_cache[17] = function() {
        return _ctx.handleBlur && _ctx.handleBlur.apply(_ctx, arguments);
      }),
      onChange: _cache[18] || (_cache[18] = function() {
        return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
      }),
      "aria-label": _ctx.label
    }), "\n      ", 16, _hoisted_15)])), _ctx.state.isWordLimitVisible && _ctx.type === "textarea" ? (openBlock(), createElementBlock(
      "span",
      _hoisted_16,
      toDisplayString(_ctx.state.showWordLimit ? _ctx.state.textLength + "/" + _ctx.state.upperLimit : _ctx.state.textLength),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default")],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
