// @opentiny/vue-input/lib/tall-storage.js
import { $prefix, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/tall-storage/vue";
import "@opentiny/vue-theme/tall-storage/index.css";
import { openBlock, createElementBlock, createElementVNode, withKeys, Fragment, renderList, normalizeClass, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = {
  name: $prefix + "TallStorage",
  inheritAttrs: false,
  props: {
    isMemoryStorage: Boolean,
    localstorageData: Array
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api
    });
  }
};
var _hoisted_1 = {
  class: "tiny-storage-list-style"
};
var _hoisted_2 = ["onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-tall-storage",
      onMousedown: _cache[1] || (_cache[1] = function() {
        return _ctx.mousedown && _ctx.mousedown.apply(_ctx, arguments);
      })
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode(
      "ul",
      {
        onKeydown: _cache[0] || (_cache[0] = withKeys(function() {
          return _ctx.keydownEvent && _ctx.keydownEvent.apply(_ctx, arguments);
        }, ["enter"])),
        class: "tiny-storage-list"
      },
      [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList($props.localstorageData, function(item) {
          return openBlock(), createElementBlock("li", {
            key: item,
            class: normalizeClass(["tiny-storage-item", [_ctx.state.hoverValue === item ? "item-hover" : ""]]),
            onClick: function onClick($event) {
              return _ctx.selectItem(item);
            }
          }, toDisplayString(item), 11, _hoisted_2);
        }),
        128
        /* KEYED_FRAGMENT */
      ))],
      32
      /* NEED_HYDRATION */
    )])],
    32
    /* NEED_HYDRATION */
  );
}
var tallStorage = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  tallStorage as default
};
