// @opentiny/vue-renderless/common/runtime.js
import "../chunk-G2ADBYYC.js";
import * as array from "./array";
import browser from "./browser";
import * as date from "./date";
import * as decimal from "./decimal";
import * as object from "./object";
import * as string from "./string";
import * as type from "./type";
import * as dataset from "./dataset";
import afterLeave from "./deps/after-leave";
import clickoutside from "./deps/clickoutside";
import debounce from "./deps/debounce";
import * as dom from "./deps/dom";
import popper from "./deps/popper";
import popupManager from "./deps/popup-manager";
import * as resizeEvent from "./deps/resize-event";
import * as scrollbarWidth from "./deps/scrollbar-width";
import throttle from "./deps/throttle";
import vueEmitter from "./deps/vue-emitter";
import vuePopper from "./deps/vue-popper";
import vuePopup from "./deps/vue-popup";
import validate from "./validate";
import memorize from "./deps/memorize";
import * as common from ".";
const version = "3.13.0";
const Renderless = {
  browser,
  array,
  date,
  object,
  decimal,
  type,
  string,
  afterLeave,
  dataset,
  clickoutside,
  dom,
  debounce,
  popper,
  resizeEvent,
  popupManager,
  scrollbarWidth,
  vueEmitter,
  vuePopper,
  throttle,
  vuePopup,
  memorize,
  common,
  validate,
  version
};
var runtime_default = Renderless;
export {
  afterLeave,
  array,
  browser,
  clickoutside,
  common,
  dataset,
  date,
  debounce,
  decimal,
  runtime_default as default,
  dom,
  memorize,
  object,
  popper,
  popupManager,
  resizeEvent,
  scrollbarWidth,
  string,
  throttle,
  type,
  validate,
  version,
  vueEmitter,
  vuePopper,
  vuePopup
};
