// @opentiny/vue-select-view/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/select-view/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { iconCheck, iconCheckedSur, iconHalfselect, iconChevronUp, iconChevronDown, iconSearch, iconOperationfaild, iconChevronLeft } from "@opentiny/vue-icon";
import Button from "@opentiny/vue-button";
import ActionSheet from "@opentiny/vue-action-sheet";
import Exception from "@opentiny/vue-exception";
import { openBlock, createElementBlock, withModifiers, createBlock, resolveDynamicComponent, normalizeClass, createCommentVNode, createElementVNode, renderSlot, resolveComponent, resolveDirective, createVNode, withCtx, withDirectives, withKeys, vModelText, toDisplayString, Fragment, renderList, vShow, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import InfiniteScroll from "@opentiny/vue-renderless/common/deps/infinite-scroll";
var _sfc_main$1 = defineComponent({
  inheritAttrs: false,
  emits: ["click"],
  components: {
    IconCheck: iconCheck(),
    IconCheckedSur: iconCheckedSur(),
    IconHalfselect: iconHalfselect()
  },
  props: {
    selected: Boolean,
    multiple: Boolean,
    readonly: Boolean
  }
});
function _sfc_render$1(_ctx, _cache, $props2, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    class: "flex items-start leading-5.5 py-3 cursor-pointer select-none border-b-0.5 border-color-bg-4",
    onClick: _cache[0] || (_cache[0] = withModifiers(function($event) {
      return _ctx.$emit("click", $event);
    }, ["stop"]))
  }, [_ctx.multiple ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.selected ? "icon-checked-sur" : "icon-check"), {
    key: 0,
    class: normalizeClass(["flex-none w-4 h-4 mt-0.5 mr-3", _ctx.selected && !_ctx.readonly ? "fill-color-brand" : "fill-color-icon-disabled"])
  }, null, 8, ["class"])) : createCommentVNode("v-if", true), createElementVNode(
    "div",
    {
      class: normalizeClass(["flex-auto truncate", {
        "text-color-brand": !_ctx.multiple && _ctx.selected
      }])
    },
    [renderSlot(_ctx.$slots, "default")],
    2
    /* CLASS */
  )]);
}
var Option = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var _sfc_main = defineComponent({
  name: $prefix + "SelectView",
  emits: ["click", "update:text", "update:visible", "close", "update:modelValue", "search-click", "confirm"],
  directives: {
    InfiniteScroll
  },
  components: {
    TinyOption: Option,
    TinyActionSheet: ActionSheet,
    TinyButton: Button,
    TinyException: Exception,
    IconChevronUp: iconChevronUp(),
    IconChevronDown: iconChevronDown(),
    IconSearch: iconSearch(),
    IconOperationfaild: iconOperationfaild(),
    IconChevronLeft: iconChevronLeft()
  },
  props: _extends({}, $props, {
    menus: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    modelValue: [Number, String, Array],
    visible: {
      type: Boolean,
      default: false
    },
    valueField: {
      type: String,
      default: "id"
    },
    textField: {
      type: String,
      default: "label"
    },
    textField2: {
      type: String,
      default: ""
    },
    textField3: {
      type: String,
      default: ""
    },
    title: String,
    multiple: Boolean,
    searchConfig: {
      type: Object,
      default: function _default2() {
        return {
          searchMethod: null
        };
      }
    },
    topConfig: {
      type: Object,
      default: function _default3() {
        return {
          label: "",
          options: []
        };
      }
    },
    requiredSelectedList: {
      type: Array,
      default: function _default4() {
        return [];
      }
    },
    textSplit: {
      type: String,
      default: "; "
    },
    showHeader: {
      type: Boolean,
      default: true
    },
    customClass: [String, Object, Array],
    mask: {
      type: Boolean,
      default: true
    },
    infiniteScroll: {
      type: Object,
      default: function _default5() {
        return {
          load: null,
          delay: 200,
          disabled: true,
          distance: 0,
          immediate: true
        };
      }
    },
    lockScroll: {
      type: Boolean,
      default: true
    },
    placeholder: {
      type: String,
      default: ""
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-select-view"
};
var _hoisted_2 = /* @__PURE__ */ createElementVNode(
  "div",
  null,
  null,
  -1
  /* HOISTED */
);
var _hoisted_3 = {
  "data-tag": "tiny-select-view-inner text-color-text-primary",
  class: /* @__PURE__ */ normalizeClass(["flex flex-col flex-auto h-full"])
};
var _hoisted_4 = {
  "data-tag": "select-view-header",
  class: "flex flex-none leading-6 py-2.5 px-4 text-base items-center border-b-0.5 border-color-bg-4 shadow-sm"
};
var _hoisted_5 = {
  class: "flex-auto flex items-center h-7 py-1 px-3 bg-color-bg-4 rounded"
};
var _hoisted_6 = ["placeholder"];
var _hoisted_7 = {
  class: "flex items-center pl-3 cursor-pointer"
};
var _hoisted_8 = {
  "data-tag": "select-view-body",
  class: "flex-auto flex flex-col overflow-auto scrollbar-size-0"
};
var _hoisted_9 = {
  key: 0,
  "data-tag": "select-view-recommend",
  class: "leading-5.5 text-color-text-placeholder mx-4 border-b-0.5 border-color-bg-4"
};
var _hoisted_10 = {
  key: 0,
  class: "mt-4"
};
var _hoisted_11 = {
  class: "mt-3"
};
var _hoisted_12 = ["onClick"];
var _hoisted_13 = {
  key: 1,
  class: "ml-2.5 truncate"
};
var _hoisted_14 = ["infinite-scroll-delay", "infinite-scroll-disabled", "infinite-scroll-distance", "infinite-scroll-immediate"];
var _hoisted_15 = {
  class: "flex"
};
var _hoisted_16 = {
  key: 0,
  class: "ml-5 text-color-text-placeholder truncate"
};
var _hoisted_17 = {
  key: 0,
  class: "text-color-text-placeholder truncate"
};
var _hoisted_18 = {
  "data-tag": "select-view-empty",
  class: "flex-auto w-full flex justify-center items-center text-center py-2"
};
var _hoisted_19 = {
  "data-tag": "select-view-loading",
  class: "w-full flex justify-center items-center text-center py-2"
};
var _hoisted_20 = {
  key: 0,
  "data-tag": "select-view-footer",
  class: "flex flex-none items-center py-4 px-3 border-t border-color-bg-4 shadow-sm"
};
var _hoisted_21 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "flex-auto"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_22 = {
  class: "h-full overflow-auto scrollbar-size-0"
};
var _hoisted_23 = {
  class: "flex px-4 items-center"
};
var _hoisted_24 = {
  class: "flex-auto truncate"
};
var _hoisted_25 = {
  class: "flex"
};
var _hoisted_26 = {
  key: 0,
  class: "ml-5 text-color-text-placeholder truncate"
};
var _hoisted_27 = {
  key: 0,
  class: "text-color-text-placeholder truncate"
};
var _hoisted_28 = {
  key: 0,
  class: "flex-none pl-4 text-right"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconChevronLeft = resolveComponent("IconChevronLeft");
  var _component_IconSearch = resolveComponent("IconSearch");
  var _component_tiny_option = resolveComponent("tiny-option");
  var _component_tiny_exception = resolveComponent("tiny-exception");
  var _component_IconChevronUp = resolveComponent("IconChevronUp");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  var _component_IconOperationfaild = resolveComponent("IconOperationfaild");
  var _directive_infinite_scroll = resolveDirective("infinite-scroll");
  return openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_action_sheet, {
    ref: "actionSheet",
    title: _ctx.title,
    "lock-scroll": _ctx.lockScroll,
    mask: _ctx.mask,
    "show-header": _ctx.showHeader,
    "custom-class": _ctx.m("h-full max-h-[100%] rounded-none text-color-text-primary pb-0", _ctx.customClass),
    visible: _ctx.visible,
    "onUpdate:visible": _cache[4] || (_cache[4] = function($event) {
      return _ctx.$emit("update:visible", $event);
    })
  }, {
    "header-left": withCtx(function() {
      return [createVNode(_component_IconChevronLeft, {
        class: "h-5 w-5 fill-color-text-primary cursor-pointer",
        onClick: _ctx.close
      }, null, 8, ["onClick"])];
    }),
    "header-right": withCtx(function() {
      return [_hoisted_2];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", _hoisted_3, [createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createVNode(_component_IconSearch, {
        class: "h-4 w-4 mr-1 fill-color-icon-disabled"
      }), withDirectives(createElementVNode("input", {
        "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
          return _ctx.state.search.input = $event;
        }),
        onKeydown: _cache[1] || (_cache[1] = withKeys(function() {
          return _ctx.searchMethod && _ctx.searchMethod.apply(_ctx, arguments);
        }, ["enter"])),
        class: "h-5 flex-auto text-xs bg-transparent outline-0",
        placeholder: _ctx.placeholder || _ctx.t("ui.select.pleaseSearch")
      }, null, 40, _hoisted_6), [[vModelText, _ctx.state.search.input]])]), createElementVNode("div", _hoisted_7, [createElementVNode(
        "div",
        {
          onClick: _cache[2] || (_cache[2] = function() {
            return _ctx.searchMethod && _ctx.searchMethod.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.select.search")),
        1
        /* TEXT */
      )])]), createElementVNode("div", _hoisted_8, [_ctx.topConfig && _ctx.topConfig.options.length ? (openBlock(), createElementBlock("div", _hoisted_9, [_ctx.topConfig.label ? (openBlock(), createElementBlock(
        "div",
        _hoisted_10,
        toDisplayString(_ctx.topConfig.label),
        1
        /* TEXT */
      )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_11, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.topConfig.options, function(item, index) {
          return openBlock(), createElementBlock("div", {
            key: item[_ctx.valueField],
            class: normalizeClass(["max-w-full h-8 rounded-sm inline-flex truncate items-center px-2.5 cursor-pointer mb-3 mr-4 border-0.5", (_ctx.multiple ? _ctx.includeOptionIndex(_ctx.state.checkList, item) > -1 : item[_ctx.valueField] === _ctx.modelValue) ? "border-current text-color-brand bg-color-bg-1" : "border-transparent bg-color-bg-2"]),
            onClick: withModifiers(function($event) {
              return _ctx.searchSelectHandler(item, index);
            }, ["stop"])
          }, [createElementVNode(
            "span",
            {
              class: normalizeClass([(_ctx.multiple ? _ctx.includeOptionIndex(_ctx.state.checkList, item) > -1 : item[_ctx.valueField] === _ctx.modelValue) ? "text-color-brand" : "text-color-text-primary"])
            },
            toDisplayString(item[_ctx.textField]),
            3
            /* TEXT, CLASS */
          ), _ctx.textField2 ? (openBlock(), createElementBlock(
            "span",
            {
              key: 0,
              class: normalizeClass(["ml-2.5", {
                "truncate": !_ctx.textField3
              }])
            },
            toDisplayString(item[_ctx.textField2]),
            3
            /* TEXT, CLASS */
          )) : createCommentVNode("v-if", true), _ctx.textField3 ? (openBlock(), createElementBlock(
            "span",
            _hoisted_13,
            toDisplayString(item[_ctx.textField3]),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)], 10, _hoisted_12);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])])) : createCommentVNode("v-if", true), withDirectives((openBlock(), createElementBlock("div", {
        "data-tag": "select-view-infinite-ccroll",
        "infinite-scroll-delay": _ctx.infiniteScroll.delay,
        "infinite-scroll-disabled": _ctx.infiniteScroll.disabled,
        "infinite-scroll-distance": _ctx.infiniteScroll.distance,
        "infinite-scroll-immediate": _ctx.infiniteScroll.immediate,
        class: normalizeClass(["px-4"])
      }, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.search.filterOptions, function(item, index) {
          return openBlock(), createBlock(_component_tiny_option, {
            key: item[_ctx.valueField],
            multiple: _ctx.multiple,
            readonly: _ctx.multiple && _ctx.requiredSelectedList.indexOf(item[_ctx.valueField]) !== -1,
            selected: _ctx.multiple ? _ctx.includeOptionIndex(_ctx.state.checkList, item) > -1 : item[_ctx.valueField] === _ctx.modelValue,
            onClick: function onClick($event) {
              return _ctx.searchSelectHandler(item, index);
            }
          }, {
            default: withCtx(function() {
              return [renderSlot(_ctx.$slots, "default", {
                item,
                index
              }, function() {
                return [createElementVNode("div", _hoisted_15, [createElementVNode(
                  "div",
                  {
                    class: normalizeClass([{
                      "min-w-[theme(spacing.14)]": !_ctx.textField3
                    }])
                  },
                  toDisplayString(item[_ctx.textField]),
                  3
                  /* TEXT, CLASS */
                ), _ctx.textField2 ? (openBlock(), createElementBlock(
                  "div",
                  _hoisted_16,
                  toDisplayString(item[_ctx.textField2]),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)]), _ctx.textField3 ? (openBlock(), createElementBlock(
                  "div",
                  _hoisted_17,
                  toDisplayString(item[_ctx.textField3]),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)];
              })];
            }),
            _: 2
            /* DYNAMIC */
          }, 1032, ["multiple", "readonly", "selected", "onClick"]);
        }),
        128
        /* KEYED_FRAGMENT */
      ))], 8, _hoisted_14)), [[vShow, _ctx.state.search.filterOptions.length], [_directive_infinite_scroll, _ctx.load]]), withDirectives(createElementVNode(
        "div",
        _hoisted_18,
        [createVNode(_component_tiny_exception, {
          "component-page": "",
          type: "nodata"
        }, {
          content: withCtx(function() {
            return [createTextVNode(
              toDisplayString(_ctx.t("ui.select.noSearchData")),
              1
              /* TEXT */
            )];
          }),
          _: 1
          /* STABLE */
        })],
        512
        /* NEED_PATCH */
      ), [[vShow, !_ctx.state.search.filterOptions.length]]), withDirectives(createElementVNode(
        "div",
        _hoisted_19,
        [createElementVNode("div", null, [createElementVNode(
          "p",
          null,
          toDisplayString(_ctx.infiniteScroll.loadingText),
          1
          /* TEXT */
        )])],
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.search.loading && _ctx.infiniteScroll.loadingText]])]), _ctx.multiple ? (openBlock(), createElementBlock("div", _hoisted_20, [createElementVNode("div", {
        class: "cursor-pointer pr-2",
        onClick: _cache[3] || (_cache[3] = withModifiers(function($event) {
          return _ctx.toggleCheckList(true);
        }, ["stop"]))
      }, [createElementVNode(
        "span",
        null,
        toDisplayString(_ctx.t("ui.select.selectedNum", {
          num: _ctx.state.checkList.length
        })),
        1
        /* TEXT */
      ), withDirectives(createVNode(
        _component_IconChevronUp,
        {
          class: "flex-none w-4 h-4 ml-2 fill-current"
        },
        null,
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.checkList.length]])]), _hoisted_21, createVNode(_component_tiny_button, {
        tiny_mode: "mobile-first",
        class: "flex-none ml-3 w-28",
        type: "primary",
        size: "medium",
        "reset-time": 0,
        onClick: _ctx.confirm
      }, {
        default: withCtx(function() {
          return [createTextVNode(
            toDisplayString(_ctx.t("ui.button.confirm")),
            1
            /* TEXT */
          )];
        }),
        _: 1
        /* STABLE */
      }, 8, ["onClick"])])) : createCommentVNode("v-if", true)])];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["title", "lock-scroll", "mask", "show-header", "custom-class", "visible"]), createVNode(_component_tiny_action_sheet, {
    "custom-class": "h-3/4 text-color-text-primary",
    title: _ctx.t("ui.select.selected"),
    "show-footer": "",
    "lock-scroll": false,
    menus: _ctx.state.selected.options,
    visible: _ctx.state.selected.show,
    "onUpdate:visible": _cache[7] || (_cache[7] = function($event) {
      return _ctx.state.selected.show = $event;
    })
  }, {
    "header-left": withCtx(function() {
      return [createElementVNode(
        "div",
        {
          class: "cursor-pointer text-color-text-placeholder",
          onClick: _cache[5] || (_cache[5] = function() {
            return _ctx.clean && _ctx.clean.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.datepicker.clear")),
        1
        /* TEXT */
      )];
    }),
    footer: withCtx(function() {
      return [createVNode(_component_tiny_button, {
        tiny_mode: "mobile-first",
        "custom-class": "flex-1 w-full mx-4 sm:mx-0",
        type: "primary",
        size: "medium",
        "reset-time": 0,
        onClick: _cache[6] || (_cache[6] = function($event) {
          return _ctx.toggleCheckList(false);
        })
      }, {
        default: withCtx(function() {
          return [createTextVNode(
            toDisplayString(_ctx.t("ui.button.confirm")),
            1
            /* TEXT */
          )];
        }),
        _: 1
        /* STABLE */
      })];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", _hoisted_22, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.selected.options, function(item) {
          return openBlock(), createBlock(
            _component_tiny_option,
            {
              key: item[_ctx.valueField]
            },
            {
              default: withCtx(function() {
                return [createElementVNode("div", _hoisted_23, [createElementVNode("div", _hoisted_24, [createElementVNode("div", _hoisted_25, [createElementVNode(
                  "div",
                  {
                    class: normalizeClass([{
                      "min-w-[theme(spacing.14)]": !_ctx.textField3
                    }])
                  },
                  toDisplayString(item[_ctx.textField]),
                  3
                  /* TEXT, CLASS */
                ), _ctx.textField2 ? (openBlock(), createElementBlock(
                  "div",
                  _hoisted_26,
                  toDisplayString(item[_ctx.textField2]),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)]), _ctx.textField3 ? (openBlock(), createElementBlock(
                  "div",
                  _hoisted_27,
                  toDisplayString(item[_ctx.textField3]),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)]), _ctx.requiredSelectedList.indexOf(item[_ctx.valueField]) === -1 ? (openBlock(), createElementBlock("div", _hoisted_28, [createVNode(_component_IconOperationfaild, {
                  class: "w-5 h-5 fill-color-text-placeholder",
                  onClick: function onClick($event) {
                    return _ctx.selectedOptionHandler(item);
                  }
                }, null, 8, ["onClick"])])) : createCommentVNode("v-if", true)])];
              }),
              _: 2
              /* DYNAMIC */
            },
            1024
            /* DYNAMIC_SLOTS */
          );
        }),
        128
        /* KEYED_FRAGMENT */
      ))])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["title", "menus", "visible"])]);
}
var SelectView = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
SelectView.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
SelectView.install = function(Vue) {
  Vue.component(SelectView.name, SelectView);
};
SelectView.version = version;
export {
  SelectView as default
};
