// @opentiny/vue-renderless/collapse-item/vue.js
import "../chunk-G2ADBYYC.js";
import { handleFocus, handleEnterClick, handleHeaderClick } from "./index";
import { guid } from "../common/string";
const api = ["state", "isActive", "handleFocus", "handleEnterClick", "handleHeaderClick"];
const renderless = (props, { computed, reactive }, { parent, constants, dispatch }) => {
  const _constants = parent.collapse._constants;
  const componentName = _constants.COMPONENT_NAME.Collapse;
  const eventName = _constants.EVENT_NAME.CollapseItemClick;
  const state = reactive({
    id: guid(),
    isClick: false,
    focusing: false,
    contentHeight: 0,
    contentWrapStyle: { height: "auto", display: "block" },
    isActive: computed(() => parent.collapse.state.activeNames.includes(props.name))
  });
  const api2 = {
    state,
    handleFocus: handleFocus({ state, interval: constants.INTERVAL }),
    handleEnterClick: handleEnterClick({ componentName, dispatch, eventName, parent }),
    handleHeaderClick: handleHeaderClick({ componentName, dispatch, eventName, props, parent, state })
  };
  return api2;
};
export {
  api,
  renderless
};
