// @opentiny/vue-filter-box/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/filter-box/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import { IconArrowBottom, IconError, IconHelpCircle } from "@opentiny/vue-icon";
import Tooltip from "@opentiny/vue-tooltip";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createElementVNode, toDisplayString, createBlock, withCtx, createVNode, createCommentVNode, withModifiers } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-theme/filter-box/index.css";
var _sfc_main = defineComponent({
  name: $prefix + "FilterBox",
  emits: ["click", "handle-clear"],
  components: {
    IconArrowBottom: IconArrowBottom(),
    IconError: IconError(),
    IconHelpCircle: IconHelpCircle(),
    TinyTooltip: Tooltip
  },
  props: _extends({}, $props, {
    label: String,
    value: [String, Number],
    tip: String,
    showClose: {
      type: Boolean,
      default: true
    },
    disabled: {
      type: Boolean,
      default: false
    },
    dropDownVisible: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "value"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_icon_help_circle = resolveComponent("icon-help-circle");
  var _component_tiny_tooltip = resolveComponent("tiny-tooltip");
  var _component_icon_error = resolveComponent("icon-error");
  var _component_icon_arrow_bottom = resolveComponent("icon-arrow-bottom");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-filter-box", _ctx.disabled && "disabled"]),
      onClick: _cache[0] || (_cache[0] = function() {
        return _ctx.handeClick && _ctx.handeClick.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "p",
      {
        class: normalizeClass(["title", _ctx.dropDownVisible && "active"])
      },
      [createElementVNode(
        "label",
        null,
        toDisplayString(_ctx.label),
        1
        /* TEXT */
      ), _ctx.tip ? (openBlock(), createBlock(_component_tiny_tooltip, {
        key: 0,
        effect: "light",
        content: _ctx.tip,
        placement: "top"
      }, {
        default: withCtx(function() {
          return [createVNode(_component_icon_help_circle)];
        }),
        _: 1
        /* STABLE */
      }, 8, ["content"])) : createCommentVNode("v-if", true)],
      2
      /* CLASS */
    ), createElementVNode(
      "p",
      _hoisted_1,
      toDisplayString(_ctx.value),
      1
      /* TEXT */
    ), _ctx.value && _ctx.showClose && !_ctx.disabled ? (openBlock(), createBlock(_component_icon_error, {
      key: 0,
      class: "filter-icon-close",
      onClick: withModifiers(_ctx.handleClear, ["stop"])
    }, null, 8, ["onClick"])) : (openBlock(), createBlock(_component_icon_arrow_bottom, {
      key: 1,
      class: normalizeClass(["filter-box-icon", _ctx.dropDownVisible && "is-reverse"])
    }, null, 8, ["class"]))],
    2
    /* CLASS */
  );
}
var FilterBox = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
FilterBox.install = function(Vue) {
  Vue.component(FilterBox.name, FilterBox);
};
FilterBox.version = version;
export {
  FilterBox as default
};
