// @opentiny/vue-credit-card/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/credit-card/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import jcb from "@opentiny/vue-theme/images/credit-jcb.png";
import amex from "@opentiny/vue-theme/images/credit-amex.png";
import chip from "@opentiny/vue-theme/images/credit-chip.png";
import troy from "@opentiny/vue-theme/images/credit-troy.png";
import visa from "@opentiny/vue-theme/images/credit-visa.png";
import discover from "@opentiny/vue-theme/images/credit-discover.png";
import unionpay from "@opentiny/vue-theme/images/credit-unionpay.png";
import mastercard from "@opentiny/vue-theme/images/credit-mastercard.png";
import dinersclub from "@opentiny/vue-theme/images/credit-dinersclub.png";
import background from "@opentiny/vue-theme/images/credit-background.jpeg";
import { openBlock, createElementBlock, normalizeClass, createElementVNode, normalizeStyle, createVNode, Transition, withCtx, createCommentVNode, Fragment, renderList, toDisplayString, TransitionGroup } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["labels", "fields", "isCardNumberMasked", "backgroundImage"]),
  setup: function setup(props2, context) {
    return Object.assign(_setup({
      props: props2,
      context,
      renderless,
      api
    }), {
      images: {
        jcb,
        amex,
        chip,
        troy,
        visa,
        discover,
        unionpay,
        background,
        dinersclub,
        mastercard
      }
    });
  }
});
var _hoisted_1 = {
  class: "credit-card-item__side -front"
};
var _hoisted_2 = {
  class: "credit-card-item__cover"
};
var _hoisted_3 = ["src"];
var _hoisted_4 = {
  class: "credit-card-item__wrapper"
};
var _hoisted_5 = {
  class: "credit-card-item__top"
};
var _hoisted_6 = ["src"];
var _hoisted_7 = {
  class: "credit-card-item__type"
};
var _hoisted_8 = ["src"];
var _hoisted_9 = ["for"];
var _hoisted_10 = {
  key: 0,
  class: "credit-card-item__numberItem"
};
var _hoisted_11 = {
  class: "credit-card-item__content"
};
var _hoisted_12 = ["for"];
var _hoisted_13 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "credit-card-item__holder"
  },
  "Card Holder",
  -1
  /* HOISTED */
);
var _hoisted_14 = {
  class: "credit-card-item__name",
  key: "1"
};
var _hoisted_15 = {
  class: "credit-card-item__name",
  key: "2"
};
var _hoisted_16 = {
  class: "credit-card-item__date",
  ref: "cardDate"
};
var _hoisted_17 = ["for"];
var _hoisted_18 = ["for"];
var _hoisted_19 = ["textContent"];
var _hoisted_20 = {
  key: "2",
  textContent: "MM"
};
var _hoisted_21 = /* @__PURE__ */ createElementVNode(
  "span",
  null,
  "/",
  -1
  /* HOISTED */
);
var _hoisted_22 = {
  class: "credit-card-item__dateItem",
  for: "cardYear"
};
var _hoisted_23 = ["textContent"];
var _hoisted_24 = {
  key: "2",
  textContent: "YY"
};
var _hoisted_25 = {
  class: "credit-card-item__side -back"
};
var _hoisted_26 = {
  class: "credit-card-item__cover"
};
var _hoisted_27 = ["src"];
var _hoisted_28 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "credit-card-item__band"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_29 = {
  class: "credit-card-item__cvv"
};
var _hoisted_30 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "credit-card-item__cvvTitle"
  },
  "CVV",
  -1
  /* HOISTED */
);
var _hoisted_31 = {
  class: "credit-card-item__cvvBand"
};
var _hoisted_32 = {
  class: "credit-card-item__type"
};
var _hoisted_33 = ["src"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["credit-card-item", {
        "-active": _ctx.state.isCardFlipped
      }])
    },
    [createElementVNode("div", _hoisted_1, [createElementVNode(
      "div",
      {
        class: normalizeClass(["credit-card-item__focus", {
          "-active": _ctx.state.focusElementStyle
        }]),
        style: normalizeStyle(_ctx.state.focusElementStyle),
        ref: "focusElement"
      },
      null,
      6
      /* CLASS, STYLE */
    ), createElementVNode("div", _hoisted_2, [createElementVNode("img", {
      class: "credit-card-item__bg",
      src: _ctx.backgroundImage || _ctx.images.background
    }, null, 8, _hoisted_3)]), createElementVNode("div", _hoisted_4, [createElementVNode("div", _hoisted_5, [createElementVNode("img", {
      src: _ctx.images.chip,
      class: "credit-card-item__chip"
    }, null, 8, _hoisted_6), createElementVNode("div", _hoisted_7, [createVNode(Transition, {
      name: "slide-fade-up"
    }, {
      default: withCtx(function() {
        return [_ctx.state.cardType ? (openBlock(), createElementBlock("img", {
          class: "credit-card-item__typeImg",
          src: _ctx.images[_ctx.state.cardType],
          key: _ctx.state.cardType,
          alt: ""
        }, null, 8, _hoisted_8)) : createCommentVNode("v-if", true)];
      }),
      _: 1
      /* STABLE */
    })])]), createElementVNode("label", {
      for: _ctx.fields.cardNumber,
      class: "credit-card-item__number",
      ref: _ctx.fields.cardNumber
    }, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.currentPlaceholder, function(n, $index) {
        return openBlock(), createElementBlock("span", {
          key: $index
        }, [createVNode(
          Transition,
          {
            name: "slide-fade-up"
          },
          {
            default: withCtx(function() {
              return [_ctx.getIsNumberMasked($index, n) ? (openBlock(), createElementBlock("div", _hoisted_10, "*")) : _ctx.labels.cardNumber.length > $index ? (openBlock(), createElementBlock(
                "div",
                {
                  class: normalizeClass(["credit-card-item__numberItem", {
                    "-active": n.trim() === ""
                  }]),
                  key: _ctx.state.currentPlaceholder
                },
                toDisplayString(_ctx.labels.cardNumber[$index]),
                3
                /* TEXT, CLASS */
              )) : (openBlock(), createElementBlock(
                "div",
                {
                  class: normalizeClass(["credit-card-item__numberItem", {
                    "-active": n.trim() === ""
                  }]),
                  key: _ctx.state.currentPlaceholder + 1
                },
                toDisplayString(n),
                3
                /* TEXT, CLASS */
              ))];
            }),
            _: 2
            /* DYNAMIC */
          },
          1024
          /* DYNAMIC_SLOTS */
        )]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))], 8, _hoisted_9), createElementVNode("div", _hoisted_11, [createElementVNode("label", {
      for: _ctx.fields.cardName,
      class: "credit-card-item__info",
      ref: _ctx.fields.cardName
    }, [_hoisted_13, createVNode(Transition, {
      name: "slide-fade-up"
    }, {
      default: withCtx(function() {
        return [_ctx.labels.cardName.length ? (openBlock(), createElementBlock("div", _hoisted_14, [createVNode(TransitionGroup, {
          name: "slide-fade-right"
        }, {
          default: withCtx(function() {
            return [(openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(_ctx.labels.cardName.replace(/\s\s+/g, " "), function(n, $index) {
                return openBlock(), createElementBlock(
                  "span",
                  {
                    class: "credit-card-item__nameItem",
                    key: $index + 1
                  },
                  toDisplayString(n),
                  1
                  /* TEXT */
                );
              }),
              128
              /* KEYED_FRAGMENT */
            ))];
          }),
          _: 1
          /* STABLE */
        })])) : (openBlock(), createElementBlock("div", _hoisted_15, "Full Name"))];
      }),
      _: 1
      /* STABLE */
    })], 8, _hoisted_12), createElementVNode(
      "div",
      _hoisted_16,
      [createElementVNode("label", {
        class: "credit-card-item__dateTitle",
        for: _ctx.fields.cardMonth
      }, "Expires", 8, _hoisted_17), createElementVNode("label", {
        class: "credit-card-item__dateItem",
        for: _ctx.fields.cardMonth
      }, [createVNode(Transition, {
        name: "slide-fade-up"
      }, {
        default: withCtx(function() {
          return [_ctx.labels.cardMonth ? (openBlock(), createElementBlock("span", {
            key: _ctx.labels.cardMonth,
            textContent: toDisplayString(_ctx.labels.cardMonth)
          }, null, 8, _hoisted_19)) : (openBlock(), createElementBlock("span", _hoisted_20))];
        }),
        _: 1
        /* STABLE */
      })], 8, _hoisted_18), _hoisted_21, createElementVNode("label", _hoisted_22, [createVNode(Transition, {
        name: "slide-fade-up"
      }, {
        default: withCtx(function() {
          return [_ctx.labels.cardYear ? (openBlock(), createElementBlock("span", {
            key: _ctx.labels.cardYear,
            textContent: toDisplayString(String(_ctx.labels.cardYear).slice(2, 4))
          }, null, 8, _hoisted_23)) : (openBlock(), createElementBlock("span", _hoisted_24))];
        }),
        _: 1
        /* STABLE */
      })])],
      512
      /* NEED_PATCH */
    )])])]), createElementVNode("div", _hoisted_25, [createElementVNode("div", _hoisted_26, [createElementVNode("img", {
      src: _ctx.backgroundImage || _ctx.images.background,
      class: "credit-card-item__bg"
    }, null, 8, _hoisted_27)]), _hoisted_28, createElementVNode("div", _hoisted_29, [_hoisted_30, createElementVNode("div", _hoisted_31, [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.labels.cardCvv, function(n, $index) {
        return openBlock(), createElementBlock("span", {
          key: $index
        }, "*");
      }),
      128
      /* KEYED_FRAGMENT */
    ))]), createElementVNode("div", _hoisted_32, [_ctx.state.cardType ? (openBlock(), createElementBlock("img", {
      key: 0,
      src: _ctx.images[_ctx.state.cardType],
      class: "credit-card-item__typeImg"
    }, null, 8, _hoisted_33)) : createCommentVNode("v-if", true)])])])],
    2
    /* CLASS */
  );
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
