// @opentiny/vue-input/lib/mobile.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/input/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconClose, iconChevronRight, iconEyeopen, iconEyeclose } from "@opentiny/vue-icon";
import ActionSheet from "@opentiny/vue-action-sheet";
import "@opentiny/vue-theme-mobile/input/index.css";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, renderSlot, createTextVNode, toDisplayString, createCommentVNode, createElementVNode, mergeProps, createVNode, Fragment, createBlock, resolveDynamicComponent, withModifiers } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  inheritAttrs: false,
  emits: ["update:modelValue", "change", "clear", "focus", "blur", "keyup", "keydown", "paste", "mouseenter", "mouseleave"],
  components: {
    IconClose: iconClose(),
    IconChevronRight: iconChevronRight(),
    IconEyeopen: iconEyeopen(),
    IconEyeclose: iconEyeclose(),
    TinyActionSheet: ActionSheet
  },
  props: [].concat(props, ["name", "selectMenu", "ellipsis", "contentStyle", "labelWidth", "tips", "isSelect", "type", "label", "modelValue", "disabled", "readonly", "clearable", "suffixIcon", "prefixIcon", "autocomplete", "showPassword", "validateEvent", "showWordLimit", "title", "counter", "autosize", "tabindex", "width", "textAlign", "resize"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  key: 0,
  class: "tiny-mobile-input__title"
};
var _hoisted_2 = {
  key: 1,
  class: "tiny-mobile-input__wrapper"
};
var _hoisted_3 = ["value", "disabled", "name", "aria-label", "tabindex"];
var _hoisted_4 = ["name", "tabindex", "type", "disabled", "readonly", "autocomplete", "aria-label"];
var _hoisted_5 = {
  key: 1,
  class: "tiny-mobile-input-group__prepend"
};
var _hoisted_6 = {
  key: 2,
  class: "tiny-mobile-input__prefix"
};
var _hoisted_7 = {
  key: 3,
  class: "tiny-mobile-input__suffix"
};
var _hoisted_8 = {
  key: 3,
  class: "tiny-mobile-input__count"
};
var _hoisted_9 = {
  class: "tiny-mobile-input__count-inner"
};
var _hoisted_10 = {
  key: 4,
  class: "tiny-mobile-input-group__append"
};
var _hoisted_11 = ["name", "tabindex", "disabled", "readonly", "autocomplete", "aria-label"];
var _hoisted_12 = {
  key: 3,
  class: "tiny-mobile-textarea__count"
};
var _hoisted_13 = {
  key: 4,
  class: "tiny-mobile-input__tips"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconChevronRight = resolveComponent("IconChevronRight");
  var _component_icon_close = resolveComponent("icon-close");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass([_ctx.type === "textarea" ? "tiny-mobile-textarea" : "tiny-mobile-input", _ctx.state.inputSize ? "tiny-mobile-input-" + _ctx.state.inputSize : "", {
        "is-focus": _ctx.state.focused,
        "is-disabled": _ctx.state.inputDisabled,
        "is-exceed": _ctx.state.inputExceed,
        "is-showlimit": _ctx.state.isWordLimitVisible && _ctx.type === "textarea",
        "tiny-mobile-input-group": _ctx.slots.prepend || _ctx.slots.append,
        "tiny-mobile-input-group-append": _ctx.slots.append,
        "tiny-mobile-input-group-prepend": _ctx.slots.prepend
      }]),
      style: normalizeStyle(_ctx.$attrs.style),
      onMouseenter: _cache[20] || (_cache[20] = function($event) {
        return _ctx.state.hovering = true;
      }),
      onMouseleave: _cache[21] || (_cache[21] = function($event) {
        return _ctx.state.hovering = false;
      })
    },
    [_ctx.slots.title || _ctx.title ? (openBlock(), createElementBlock("div", _hoisted_1, [renderSlot(_ctx.$slots, "title", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.title),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true), _ctx.type !== "textarea" ? (openBlock(), createElementBlock("div", _hoisted_2, [_ctx.isSelect ? (openBlock(), createElementBlock("div", {
      key: 0,
      class: "tiny-mobile-input__select",
      onClick: _cache[2] || (_cache[2] = function() {
        return _ctx.showBox && _ctx.showBox.apply(_ctx, arguments);
      })
    }, [createElementVNode("input", mergeProps({
      type: "text",
      readonly: "",
      class: "tiny-mobile-input__inner"
    }, _ctx.a(_ctx.$attrs, ["size", "class", "style", "^on[A-Z]"]), {
      value: _ctx.state.checkedLabel,
      disabled: _ctx.state.inputDisabled,
      name: _ctx.name,
      "aria-label": _ctx.label,
      style: _ctx.state.inputStyle,
      tabindex: _ctx.tabindex,
      onInput: _cache[0] || (_cache[0] = function() {
        return _ctx.handleInput && _ctx.handleInput.apply(_ctx, arguments);
      }),
      onChange: _cache[1] || (_cache[1] = function() {
        return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
      })
    }), null, 16, _hoisted_3), createElementVNode(
      "div",
      {
        class: "tiny-mobile-input__select-icon",
        style: normalizeStyle({
          transform: _ctx.state.boxVisibility ? "rotate(90deg)" : "none"
        })
      },
      [createVNode(_component_IconChevronRight)],
      4
      /* STYLE */
    )])) : (openBlock(), createElementBlock(
      Fragment,
      {
        key: 1
      },
      [_ctx.type !== "textarea" ? (openBlock(), createElementBlock("input", mergeProps({
        key: 0,
        ref: "input",
        name: _ctx.name
      }, _ctx.a(_ctx.$attrs, ["size", "class", "style", "^onw+"]), {
        class: "tiny-mobile-input__inner",
        style: _ctx.state.inputStyle,
        tabindex: _ctx.tabindex,
        type: _ctx.showPassword ? _ctx.state.passwordVisible ? "text" : "password" : _ctx.type,
        disabled: _ctx.state.inputDisabled,
        readonly: _ctx.readonly,
        autocomplete: _ctx.autocomplete,
        onCompositionstart: _cache[3] || (_cache[3] = function() {
          return _ctx.handleCompositionStart && _ctx.handleCompositionStart.apply(_ctx, arguments);
        }),
        onCompositionupdate: _cache[4] || (_cache[4] = function() {
          return _ctx.handleCompositionUpdate && _ctx.handleCompositionUpdate.apply(_ctx, arguments);
        }),
        onCompositionend: _cache[5] || (_cache[5] = function() {
          return _ctx.handleCompositionEnd && _ctx.handleCompositionEnd.apply(_ctx, arguments);
        }),
        onInput: _cache[6] || (_cache[6] = function() {
          return _ctx.handleInput && _ctx.handleInput.apply(_ctx, arguments);
        }),
        onFocus: _cache[7] || (_cache[7] = function() {
          return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
        }),
        onBlur: _cache[8] || (_cache[8] = function() {
          return _ctx.handleBlur && _ctx.handleBlur.apply(_ctx, arguments);
        }),
        onChange: _cache[9] || (_cache[9] = function() {
          return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
        }),
        "aria-label": _ctx.label
      }), null, 16, _hoisted_4)) : createCommentVNode("v-if", true), _ctx.slots.prepend ? (openBlock(), createElementBlock("div", _hoisted_5, [createElementVNode("span", null, [renderSlot(_ctx.$slots, "prepend")])])) : createCommentVNode("v-if", true), _ctx.slots.prefix || _ctx.prefixIcon ? (openBlock(), createElementBlock("div", _hoisted_6, [renderSlot(_ctx.$slots, "prefix"), _ctx.prefixIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.prefixIcon), {
        key: 0,
        class: "tiny-mobile-input__icon"
      })) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), _ctx.getSuffixVisible() ? (openBlock(), createElementBlock("span", _hoisted_7, [!_ctx.state.showClear || !_ctx.state.showPwdVisible || !_ctx.state.isWordLimitVisible ? (openBlock(), createElementBlock(
        Fragment,
        {
          key: 0
        },
        [renderSlot(_ctx.$slots, "suffix"), _ctx.suffixIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.suffixIcon), {
          key: 0,
          class: "tiny-mobile-input__icon"
        })) : createCommentVNode("v-if", true)],
        64
        /* STABLE_FRAGMENT */
      )) : createCommentVNode("v-if", true), _ctx.state.showClear ? (openBlock(), createBlock(_component_icon_close, {
        key: 1,
        class: "tiny-mobile-input__icon tiny-mobile-input__clear",
        onMousedown: _cache[10] || (_cache[10] = withModifiers(function() {
        }, ["prevent"])),
        onClick: _ctx.clear
      }, null, 8, ["onClick"])) : createCommentVNode("v-if", true), _ctx.state.showPwdVisible ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.passwordVisible ? "icon-eyeopen" : "icon-eyeclose"), {
        key: 2,
        class: "tiny-mobile-input__icon",
        onClick: _ctx.handlePasswordVisible
      }, null, 8, ["onClick"])) : createCommentVNode("v-if", true), _ctx.state.isWordLimitVisible ? (openBlock(), createElementBlock("span", _hoisted_8, [createElementVNode(
        "span",
        _hoisted_9,
        toDisplayString(_ctx.state.showWordLimit ? _ctx.state.textLength + "/" + _ctx.state.upperLimit : _ctx.state.textLength),
        1
        /* TEXT */
      )])) : createCommentVNode("v-if", true), _ctx.state.validateState ? (openBlock(), createElementBlock(
        "i",
        {
          key: 4,
          class: normalizeClass(["tiny-mobile-input__icon", ["tiny-mobile-input__validateIcon", _ctx.validateIcon]])
        },
        null,
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true)])) : createCommentVNode("v-if", true), _ctx.slots.append ? (openBlock(), createElementBlock("div", _hoisted_10, [renderSlot(_ctx.$slots, "append")])) : createCommentVNode("v-if", true)],
      64
      /* STABLE_FRAGMENT */
    ))])) : (openBlock(), createElementBlock("textarea", mergeProps({
      key: 2,
      ref: "textarea",
      name: _ctx.name
    }, _ctx.a(_ctx.$attrs, ["type", "class", "style", "^on[A-Z]"]), {
      class: ["tiny-mobile-textarea__inner", {
        "is-focus": _ctx.state.focused
      }],
      tabindex: _ctx.tabindex,
      onCompositionstart: _cache[11] || (_cache[11] = function() {
        return _ctx.handleCompositionStart && _ctx.handleCompositionStart.apply(_ctx, arguments);
      }),
      onCompositionupdate: _cache[12] || (_cache[12] = function() {
        return _ctx.handleCompositionUpdate && _ctx.handleCompositionUpdate.apply(_ctx, arguments);
      }),
      onCompositionend: _cache[13] || (_cache[13] = function() {
        return _ctx.handleCompositionEnd && _ctx.handleCompositionEnd.apply(_ctx, arguments);
      }),
      onInput: _cache[14] || (_cache[14] = function() {
        return _ctx.handleInput && _ctx.handleInput.apply(_ctx, arguments);
      }),
      disabled: _ctx.state.inputDisabled,
      readonly: _ctx.readonly,
      autocomplete: _ctx.autocomplete,
      style: _extends({}, _ctx.state.textareaStyle, {
        width: _ctx.$attrs.cols ? "auto" : "100%"
      }),
      onFocus: _cache[15] || (_cache[15] = function() {
        return _ctx.handleFocus && _ctx.handleFocus.apply(_ctx, arguments);
      }),
      onBlur: _cache[16] || (_cache[16] = function() {
        return _ctx.handleBlur && _ctx.handleBlur.apply(_ctx, arguments);
      }),
      onChange: _cache[17] || (_cache[17] = function() {
        return _ctx.handleChange && _ctx.handleChange.apply(_ctx, arguments);
      }),
      "aria-label": _ctx.label
    }), "\n    ", 16, _hoisted_11)), _ctx.state.isWordLimitVisible && _ctx.type === "textarea" ? (openBlock(), createElementBlock(
      "span",
      _hoisted_12,
      toDisplayString(_ctx.state.showWordLimit ? _ctx.state.textLength + "/" + _ctx.state.upperLimit : _ctx.state.textLength),
      1
      /* TEXT */
    )) : createCommentVNode("v-if", true), _ctx.slots.tips || _ctx.tips ? (openBlock(), createElementBlock("div", _hoisted_13, [renderSlot(_ctx.$slots, "tips", {}, function() {
      return [createTextVNode(
        toDisplayString(_ctx.tips),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true), createVNode(_component_tiny_action_sheet, {
      modelValue: _ctx.state.sheetvalue,
      "onUpdate:modelValue": _cache[18] || (_cache[18] = function($event) {
        return _ctx.state.sheetvalue = $event;
      }),
      menus: _ctx.selectMenu,
      ellipsis: _ctx.ellipsis,
      "content-style": _ctx.contentStyle,
      visible: _ctx.state.boxVisibility,
      "onUpdate:visible": _cache[19] || (_cache[19] = function($event) {
        return _ctx.state.boxVisibility = $event;
      })
    }, null, 8, ["modelValue", "menus", "ellipsis", "content-style", "visible"])],
    38
    /* CLASS, STYLE, NEED_HYDRATION */
  );
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
