// @opentiny/vue-renderless/tree-menu/index.js
import {
  __spreadValues
} from "../chunk-G2ADBYYC.js";
const initData = ({ state, props, service, api }) => async () => {
  if (props.data) {
    state.data = props.data;
    return;
  }
  if (typeof service.getMenuDataSync === "function") {
    const menuData = await service.getMenuDataSync();
    state.data = api.setMenuKey({ newData: [], menuData });
  }
};
const setMenuKey = (api) => ({ newData, menuData }) => {
  Array.isArray(menuData) && menuData.forEach((data) => {
    const item = {};
    Object.keys(data).forEach((key) => {
      if (key === "name") {
        item.label = data[key];
      } else if (key === "siteNodeId") {
        item.id = data[key];
      } else if (key === "url" && data[key]) {
        if (!data[key].includes("#")) {
          item[key] = "#/" + data[key];
        } else {
          item[key] = data[key];
        }
      } else if (key === "children" && data[key]) {
        item.children = api.setMenuKey({
          newData: [],
          menuData: data.children
        });
      } else {
        item[key] = data[key];
      }
    });
    newData.push(__spreadValues(__spreadValues({}, data), item));
  });
  return newData;
};
const filterNode = () => (value, data) => {
  if (!value) {
    return true;
  }
  return data.label.includes(value);
};
const watchFilterText = ({ vm }) => (value) => {
  vm.$refs.tree.filter(value);
};
const nodeDragStart = (emit) => (node, event) => {
  emit("node-drag-start", node, event);
};
const nodeDragEnter = (emit) => (dragNode, dropNode, event) => {
  emit("node-drag-enter", dragNode, dropNode, event);
};
const nodeDragOver = (emit) => (dragNode, dropNode, event) => {
  emit("node-drag-over", dragNode, dropNode, event);
};
const nodeDragEnd = (emit) => (dragNode, dropNode, dropType, event) => {
  emit("node-drag-end", dragNode, dropNode, dropType, event);
};
const nodeDrop = (emit) => (dragNode, dropNode, dropType, event) => {
  emit("node-drop", dragNode, dropNode, dropType, event);
};
const nodeExpand = (emit) => (nodeData, node) => {
  emit("node-expand", nodeData, node);
};
const nodeCollapse = (emit) => (nodeData, node) => {
  emit("node-collapse", nodeData, node);
};
const nodeClick = (emit) => (nodeData, node) => {
  emit("node-click", nodeData, node);
};
const checkChange = (emit) => (data, checked, indeterminate) => {
  emit("check-change", data, checked, indeterminate);
};
const check = (emit) => (data, checkedStatus) => {
  emit("check", data, checkedStatus);
};
const currentChange = (emit) => (data, node) => {
  emit("current-change", data, node);
};
const getTitle = (props) => (label) => props.showTitle ? label : "";
const collapseChange = ({ state, props, emit }) => () => {
  if (props.menuCollapsible) {
    state.isCollapsed = !state.isCollapsed;
    emit("collapse-change", state.isCollapsed);
  }
};
const collapseMenu = ({ state, props, api }) => () => {
  if (props.menuCollapsible && !state.isCollapsed) {
    api.collapseChange();
  }
};
const expandMenu = ({ state, props, api }) => () => {
  if (props.menuCollapsible && state.isCollapsed) {
    api.collapseChange();
  }
};
const setCurrentKey = ({ vm }) => (key) => {
  vm.$refs.tree.setCurrentKey(key);
};
const getCurrentKey = ({ vm }) => () => {
  return vm.$refs.tree.getCurrentKey();
};
const setCurrentNode = ({ vm }) => (key) => {
  vm.$refs.tree.setCurrentKey(key);
};
const getCurrentNode = ({ vm }) => () => {
  return vm.$refs.tree.getCurrentNode();
};
export {
  check,
  checkChange,
  collapseChange,
  collapseMenu,
  currentChange,
  expandMenu,
  filterNode,
  getCurrentKey,
  getCurrentNode,
  getTitle,
  initData,
  nodeClick,
  nodeCollapse,
  nodeDragEnd,
  nodeDragEnter,
  nodeDragOver,
  nodeDragStart,
  nodeDrop,
  nodeExpand,
  setCurrentKey,
  setCurrentNode,
  setMenuKey,
  watchFilterText
};
