// @opentiny/vue-renderless/tree-menu/vue.js
import "../chunk-G2ADBYYC.js";
import {
  initData,
  setMenuKey,
  filterNode,
  watchFilterText,
  nodeDragStart,
  nodeDragEnter,
  nodeDragOver,
  nodeDragEnd,
  nodeDrop,
  nodeExpand,
  nodeCollapse,
  nodeClick,
  checkChange,
  check,
  currentChange,
  getTitle,
  collapseChange,
  collapseMenu,
  expandMenu,
  setCurrentKey,
  getCurrentKey,
  setCurrentNode,
  getCurrentNode
} from "./index";
const api = [
  "state",
  "t",
  "initData",
  "setMenuKey",
  "filterNode",
  "nodeDragStart",
  "nodeDragEnter",
  "nodeDragOver",
  "nodeDragEnd",
  "nodeDrop",
  "nodeExpand",
  "nodeCollapse",
  "nodeClick",
  "checkChange",
  "check",
  "currentChange",
  "getTitle",
  "collapseChange",
  "collapseMenu",
  "expandMenu",
  "setCurrentKey",
  "getCurrentKey",
  "setCurrentNode",
  "getCurrentNode"
];
const renderless = (props, { watch, reactive, onMounted }, { t, service, emit, vm }) => {
  service = service || { base: {} };
  service = { getMenuDataSync: props.getMenuDataSync || service.base.getMenuDataSync };
  const state = reactive({
    data: [],
    filterText: "",
    isCollapsed: false
  });
  Object.assign(api, {
    t,
    state,
    check: check(emit),
    filterNode: filterNode(),
    nodeDrop: nodeDrop(emit),
    nodeClick: nodeClick(emit),
    nodeExpand: nodeExpand(emit),
    nodeDragEnd: nodeDragEnd(emit),
    checkChange: checkChange(emit),
    nodeCollapse: nodeCollapse(emit),
    nodeDragOver: nodeDragOver(emit),
    nodeDragStart: nodeDragStart(emit),
    nodeDragEnter: nodeDragEnter(emit),
    currentChange: currentChange(emit),
    watchFilterText: watchFilterText({ vm }),
    getTitle: getTitle(props),
    setMenuKey: setMenuKey(api),
    initData: initData({ state, props, service, api }),
    collapseChange: collapseChange({ state, props, emit }),
    collapseMenu: collapseMenu({ state, props, api }),
    expandMenu: expandMenu({ state, props, api }),
    setCurrentKey: setCurrentKey({ vm }),
    getCurrentKey: getCurrentKey({ vm }),
    setCurrentNode: setCurrentNode({ vm }),
    getCurrentNode: getCurrentNode({ vm })
  });
  watch(
    () => props.data,
    (value) => state.data = value,
    { immediate: true }
  );
  watch(() => state.filterText, api.watchFilterText, { deep: true });
  onMounted(api.initData);
  return api;
};
export {
  api,
  renderless
};
