// @opentiny/vue-filter/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var Filter = defineComponent({
  name: $prefix + "Filter",
  props: _extends({}, $props, {
    data: Array,
    modelValue: Array,
    columnNum: {
      type: Number,
      default: 3
    },
    filterGroup: Array,
    filterValue: Array,
    panelClass: String,
    manual: Boolean,
    customClass: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Filter.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Filter.install = function(Vue) {
  Vue.component(Filter.name, Filter);
};
Filter.version = version;
export {
  Filter as default
};
