// @opentiny/vue-filter/lib/mobile-first.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless as renderless$1, api as api$1 } from "@opentiny/vue-renderless/filter/vue";
import { defineComponent, $props, setup as _setup } from "@opentiny/vue-common";
import { IconSubScript, IconUnfilter, IconChevronDown, IconChevronUp } from "@opentiny/vue-icon";
import { renderless, api } from "@opentiny/vue-renderless/filter/tag-group";
import { resolveComponent, openBlock, createElementBlock, toDisplayString, createCommentVNode, createElementVNode, withDirectives, normalizeClass, normalizeStyle, withModifiers, vShow, Fragment, renderList, createBlock, createVNode, createSlots, withCtx, renderSlot, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import FilterBar from "@opentiny/vue-filter-bar";
import Button from "@opentiny/vue-button";
var _sfc_main$1 = defineComponent({
  emits: ["click"],
  components: {
    IconSubScript: IconSubScript()
  },
  model: {
    prop: "modelValue",
    event: "update:modelValue"
  },
  props: _extends({}, $props, {
    modelValue: [String, Number, Array],
    data: Array,
    type: String,
    multiple: Boolean,
    columnNum: Number,
    label: String
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1$1 = {
  "data-tag": "tiny-filter-tag-group overflow-hidden",
  class: "clear-both"
};
var _hoisted_2$1 = {
  key: 0,
  "data-tag": "tag-group-label",
  class: "h-5 text-sm leading-5 overflow-hidden text-ellipsis whitespace-nowrap"
};
var _hoisted_3$1 = {
  "data-tag": "tag-group-tags",
  class: "text-left -mr-3 overflow-hidden clear-both"
};
var _hoisted_4$1 = ["title", "onClick"];
function _sfc_render$1(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconSubScript = resolveComponent("IconSubScript");
  return openBlock(), createElementBlock("div", _hoisted_1$1, [_ctx.label ? (openBlock(), createElementBlock(
    "div",
    _hoisted_2$1,
    toDisplayString(_ctx.label),
    1
    /* TEXT */
  )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_3$1, [withDirectives(createElementVNode(
    "div",
    {
      class: normalizeClass(["h-9 mt-3 float-left relative"]),
      style: normalizeStyle({
        width: _ctx.state.tagWidth
      })
    },
    [createElementVNode(
      "span",
      {
        class: normalizeClass(["h-full block p-2 text-center rounded-sm truncate cursor-pointer border-0.5 mr-3", _ctx.state.selected.length === 0 ? "border-current text-color-brand bg-color-info-primary-subtler" : "border-transparent bg-color-bg-2"]),
        onClick: _cache[0] || (_cache[0] = withModifiers(function() {
          return _ctx.selectOptionAll && _ctx.selectOptionAll.apply(_ctx, arguments);
        }, ["stop"]))
      },
      toDisplayString(_ctx.t("ui.base.all")),
      3
      /* TEXT, CLASS */
    )],
    4
    /* STYLE */
  ), [[vShow, _ctx.type === "tag" && _ctx.multiple]]), (openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.data, function(item, index) {
      return openBlock(), createElementBlock(
        "div",
        {
          key: index,
          class: normalizeClass(["h-9 mt-3 float-left relative"]),
          style: normalizeStyle({
            width: _ctx.state.tagWidth
          })
        },
        [_ctx.state.selected.indexOf(item.value) !== -1 && _ctx.multiple ? (openBlock(), createBlock(_component_IconSubScript, {
          key: 0,
          class: "absolute right-3 w-4 h-4 fill-color-icon-focus"
        })) : createCommentVNode("v-if", true), createElementVNode("span", {
          class: normalizeClass(["h-full block p-2 text-center rounded-sm truncate cursor-pointer border-0.5 mr-3", _ctx.state.selected.indexOf(item.value) !== -1 ? "border-current text-color-brand bg-color-info-primary-subtler" : "border-transparent bg-color-bg-2"]),
          title: item.label,
          onClick: withModifiers(function($event) {
            return _ctx.selectOption(item.value);
          }, ["stop"])
        }, toDisplayString(item.label), 11, _hoisted_4$1)],
        4
        /* STYLE */
      );
    }),
    128
    /* KEYED_FRAGMENT */
  ))])]);
}
var TagGroup = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["render", _sfc_render$1]]);
var _sfc_main = defineComponent({
  components: {
    TagGroup,
    TinyFilterBar: FilterBar,
    TinyButton: Button,
    IconUnfilter: IconUnfilter(),
    IconChevronDown: IconChevronDown(),
    IconChevronUp: IconChevronUp()
  },
  emits: ["update:modelValue", "cancel", "panel", "update:filter-value"],
  props: _extends({}, $props, {
    data: Array,
    modelValue: Array,
    columnNum: {
      type: Number,
      default: 3
    },
    filterGroup: Array,
    filterValue: Array,
    panelClass: String,
    manual: Boolean,
    customClass: String
  }),
  setup: function setup2(props, context) {
    return _setup({
      props,
      context,
      renderless: renderless$1,
      api: api$1
    });
  }
});
var _hoisted_1 = {
  class: "w-10 ml-2 flex-none flex justify-center cursor-pointer"
};
var _hoisted_2 = {
  ref: "maskPoint"
};
var _hoisted_3 = {
  "data-tag": "panel-scrollbar",
  class: "flex-auto m-5 overflow-x-visible overflow-y-auto scrollbar-size-0"
};
var _hoisted_4 = {
  "data-tag": "filter-list",
  class: "leading-[theme(spacing.12)] cursor-pointer h-full py-2 overflow-x-visible overflow-y-auto scrollbar-size-0"
};
var _hoisted_5 = ["title", "onClick"];
var _hoisted_6 = {
  "data-tag": "filter-footer",
  class: "flex items-center justify-end mx-5 mb-5"
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_IconUnfilter = resolveComponent("IconUnfilter");
  var _component_tiny_filter_bar = resolveComponent("tiny-filter-bar");
  var _component_TagGroup = resolveComponent("TagGroup");
  var _component_tiny_button = resolveComponent("tiny-button");
  return openBlock(), createElementBlock(
    "div",
    {
      ref: "root",
      "data-tag": "tiny-filter",
      class: normalizeClass(_ctx.m("text-sm text-color-text-primary relative", _ctx.customClass))
    },
    [createVNode(_component_tiny_filter_bar, {
      "model-value": _ctx.state.showPanelIndex,
      data: _ctx.state.labelList,
      onClick: _ctx.panelToggle
    }, createSlots({
      default: withCtx(function() {
        return [renderSlot(_ctx.$slots, "default", {
          active: _ctx.state.showPanelIndex,
          value: _ctx.modelValue,
          labels: _ctx.state.labelList
        })];
      }),
      _: 2
      /* DYNAMIC */
    }, [_ctx.filterGroup ? {
      name: "icon",
      fn: withCtx(function() {
        return [createElementVNode("div", {
          "data-tag": "filter-icon",
          class: "w-full",
          onClick: _cache[0] || (_cache[0] = function($event) {
            return _ctx.panelToggle("filter");
          })
        }, [renderSlot(_ctx.$slots, "icon", {
          active: _ctx.state.filterPanel.show || _ctx.state.hasFilterValue,
          show: _ctx.state.filterPanel.show
        }, function() {
          return [createElementVNode("div", _hoisted_1, [createVNode(_component_IconUnfilter, {
            "custom-class": "w-5 h-5 fill-current",
            class: normalizeClass(_ctx.state.filterPanel.show || _ctx.state.hasFilterValue ? " text-color-brand" : "")
          }, null, 8, ["class"])])];
        })])];
      }),
      key: "0"
    } : void 0]), 1032, ["model-value", "data", "onClick"]), createElementVNode(
      "div",
      _hoisted_2,
      null,
      512
      /* NEED_PATCH */
    ), withDirectives(createElementVNode(
      "div",
      {
        ref: "panelMask",
        "data-tag": "panel-mask",
        class: "fixed left-0 right-0 z-40 w-full h-full bg-color-bg-7",
        onClick: _cache[1] || (_cache[1] = function() {
          return _ctx.cancel && _ctx.cancel.apply(_ctx, arguments);
        })
      },
      null,
      512
      /* NEED_PATCH */
    ), [[vShow, !_ctx.manual && _ctx.state.showPanel]]), withDirectives(createElementVNode(
      "div",
      {
        ref: "panel",
        "data-tag": "tiny-filter-panel",
        class: normalizeClass(_ctx.m("absolute top-11 flex flex-col z-40 w-full bg-color-bg-1 overflow-hidden", _ctx.panelClass))
      },
      [withDirectives(createElementVNode(
        "div",
        _hoisted_3,
        [withDirectives(createVNode(_component_TagGroup, {
          "data-tag": "select-group",
          modelValue: _ctx.state.selectPanel.selected,
          "onUpdate:modelValue": _cache[2] || (_cache[2] = function($event) {
            return _ctx.state.selectPanel.selected = $event;
          }),
          data: _ctx.state.selectPanel.config.options,
          label: _ctx.state.selectPanel.config.label,
          type: _ctx.state.selectPanel.config.type,
          multiple: _ctx.state.selectPanel.config.multiple,
          "column-num": _ctx.columnNum,
          onClick: _ctx.selectOption
        }, null, 8, ["modelValue", "data", "label", "type", "multiple", "column-num", "onClick"]), [[vShow, _ctx.state.showPanelIndex !== -1]]), _ctx.filterGroup ? (openBlock(true), createElementBlock(
          Fragment,
          {
            key: 0
          },
          renderList(_ctx.filterGroup, function(item, index) {
            return withDirectives((openBlock(), createBlock(_component_TagGroup, {
              "data-tag": "filter-group",
              key: index,
              modelValue: _ctx.state.filterPanel.selected[index],
              "onUpdate:modelValue": function onUpdateModelValue($event) {
                return _ctx.state.filterPanel.selected[index] = $event;
              },
              data: item.options,
              label: item.label,
              type: item.type,
              multiple: item.multiple,
              "column-num": _ctx.columnNum,
              class: normalizeClass([{
                "mt-5": index !== 0
              }]),
              onClick: function onClick($event) {
                return _ctx.filterSelectOption($event, index);
              }
            }, null, 8, ["modelValue", "onUpdate:modelValue", "data", "label", "type", "multiple", "column-num", "class", "onClick"])), [[vShow, _ctx.state.filterPanel.show]]);
          }),
          128
          /* KEYED_FRAGMENT */
        )) : createCommentVNode("v-if", true)],
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.selectPanel.config.type !== "list"]]), withDirectives(createElementVNode(
        "div",
        _hoisted_4,
        [_ctx.state.selectPanel.config.showAll ? (openBlock(), createElementBlock(
          "div",
          {
            key: 0,
            class: normalizeClass(["px-5 truncate", {
              "text-color-brand": _ctx.state.selectPanel.selected.length === 0
            }]),
            onClick: _cache[3] || (_cache[3] = withModifiers(function($event) {
              return _ctx.selectOption();
            }, ["stop"]))
          },
          toDisplayString(_ctx.t("ui.base.all")),
          3
          /* TEXT, CLASS */
        )) : createCommentVNode("v-if", true), (openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.state.selectPanel.config.options, function(item, index) {
            return openBlock(), createElementBlock("div", {
              key: index,
              title: item.label,
              class: normalizeClass(["px-5 truncate", {
                "text-color-brand": _ctx.state.selectPanel.selected.indexOf(item.value) !== -1
              }]),
              onClick: withModifiers(function($event) {
                return _ctx.selectOption(item.value);
              }, ["stop"])
            }, toDisplayString(item.label), 11, _hoisted_5);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.selectPanel.config.type === "list"]]), withDirectives(createElementVNode(
        "div",
        _hoisted_6,
        [renderSlot(_ctx.$slots, "footer", {
          cancel: _ctx.cancel,
          reset: _ctx.reset,
          confirm: _ctx.confirm
        }, function() {
          return [withDirectives(createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            key: "btn1",
            size: "medium",
            "custom-class": "sm:max-w-full mr-4 flex-1",
            onClick: _ctx.cancel
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.cancel")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"]), [[vShow, !_ctx.state.selectPanel.config.multiple]]), createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            key: "btn2",
            size: "medium",
            "reset-time": 0,
            "custom-class": "sm:max-w-full mr-4 flex-1",
            onClick: _cache[4] || (_cache[4] = function($event) {
              return _ctx.reset(_ctx.state.filterPanel.show);
            })
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.reset")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }), createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            key: "btn3",
            type: "primary",
            size: "medium",
            "custom-class": "sm:max-w-full flex-1",
            onClick: _cache[5] || (_cache[5] = function($event) {
              return _ctx.confirm(_ctx.state.filterPanel.show);
            })
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.base.confirm")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          })];
        })],
        512
        /* NEED_PATCH */
      ), [[vShow, _ctx.state.selectPanel.config.multiple || _ctx.state.filterPanel.show]])],
      2
      /* CLASS */
    ), [[vShow, !_ctx.manual && _ctx.state.showPanel]])],
    2
    /* CLASS */
  );
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
