// @opentiny/vue-link-menu/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import { iconSearch } from "@opentiny/vue-icon";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/link-menu/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var $constants = {
  DIALOG_TITLE: "ui.linkMenu.title"
};
var LinkMenu = defineComponent({
  name: $prefix + "LinkMenu",
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    icon: Object,
    searchIcon: {
      type: Object,
      default: function _default2() {
        return iconSearch();
      }
    },
    data: {
      type: Array
    },
    maxItem: {
      type: Number,
      validator: function validator(value) {
        return value >= 0;
      }
    },
    title: {
      type: String,
      default: ""
    },
    defaultExpandAll: {
      type: Boolean,
      default: false
    },
    ellipsis: {
      type: Boolean,
      default: true
    },
    wrap: {
      type: Boolean,
      default: false
    },
    props: {
      default: function _default3() {
        return {
          children: "children",
          disabled: "disabled",
          label: "label"
        };
      }
    },
    getMenuDataSync: Function,
    keepSelectedNodes: {
      type: Boolean,
      default: true
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
LinkMenu.install = function(Vue) {
  Vue.component(LinkMenu.name, LinkMenu);
};
LinkMenu.version = version;
export {
  LinkMenu as default
};
