// @opentiny/vue-roles/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/roles/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Popover from "@opentiny/vue-popover";
import { IconChevronDown, IconChevronUp } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, createTextVNode, toDisplayString, withDirectives, createBlock, resolveDynamicComponent, vShow, Fragment, renderList, normalizeClass } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["fetchRole", "getEnvInfoSync", "getChangeRoleUrl", "getCustomized", "pushCustomized"]),
  components: {
    TinyPopover: Popover,
    IconChevronDown: IconChevronDown(),
    IconChevronUp: IconChevronUp()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-roles"
};
var _hoisted_2 = {
  class: "tiny-roles tiny-roles__selector-body"
};
var _hoisted_3 = {
  class: "tiny-roles__poplist"
};
var _hoisted_4 = ["data-value", "current-scope", "onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_tiny_popover = resolveComponent("tiny-popover");
  return openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_popover, {
    "popper-class": "tiny-roles__selector",
    disabled: _ctx.state.roleList.length === 1,
    modelValue: _ctx.state.showPopover,
    "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
      return _ctx.state.showPopover = $event;
    }),
    placement: "bottom-start",
    trigger: "hover",
    onShow: _ctx.show,
    onHide: _ctx.hide
  }, {
    reference: withCtx(function() {
      return [createElementVNode("span", null, [createTextVNode(
        toDisplayString(_ctx.state.currentRole) + " ",
        1
        /* TEXT */
      ), withDirectives((openBlock(), createBlock(
        resolveDynamicComponent(_ctx.state.showIcon ? "icon-chevron-down" : "icon-chevron-up"),
        null,
        null,
        512
        /* NEED_PATCH */
      )), [[vShow, _ctx.state.roleList.length > 1]])])];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", _hoisted_2, [createElementVNode("ul", _hoisted_3, [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.roleList, function(item, key) {
          return openBlock(), createElementBlock("li", {
            class: normalizeClass(["tiny-roles__poplist-item", [_ctx.state.currentId === item.roleId ? "is-selected" : ""]]),
            key,
            "data-value": item.roleId,
            "current-scope": item.currentScope,
            onClick: function onClick($event) {
              return _ctx.changeRole(item);
            }
          }, [createElementVNode(
            "span",
            null,
            toDisplayString(item.roleName),
            1
            /* TEXT */
          )], 10, _hoisted_4);
        }),
        128
        /* KEYED_FRAGMENT */
      ))])])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["disabled", "modelValue", "onShow", "onHide"])]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
