// @opentiny/vue-company/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/company/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var Company = defineComponent({
  name: $prefix + "Company",
  props: _extends({}, $props, {
    clearable: {
      type: Boolean,
      default: function _default() {
        return true;
      }
    },
    disabled: {
      type: Boolean,
      default: function _default2() {
        return false;
      }
    },
    fetchCompany: Function,
    fields: Object,
    max: {
      type: Number,
      default: 1e3
    },
    modelValue: {},
    placeholder: {
      type: String,
      default: function _default3() {
        return "";
      }
    },
    popperAppendToBody: {
      type: Boolean,
      default: function _default4() {
        return true;
      }
    },
    popperClass: String,
    size: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
Company.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
Company.install = function(Vue) {
  Vue.component(Company.name, Company);
};
Company.version = version;
export {
  Company as default
};
