// @opentiny/vue-scroll-text/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/scroll-text/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { openBlock, createElementBlock, createElementVNode, normalizeClass, normalizeStyle, renderSlot, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["text", "time", "styles", "direction", "hoverStop", "widthAdapt"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-scroll-text__wrapper"
};
var _hoisted_2 = {
  ref: "content",
  class: "tiny-scroll-text__content"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [createElementVNode(
    "div",
    {
      ref: "wrap",
      class: normalizeClass(["tiny-scroll-text", {
        "width-adapt": _ctx.widthAdapt
      }]),
      onMouseenter: _cache[0] || (_cache[0] = function() {
        return _ctx.stopAnimation && _ctx.stopAnimation.apply(_ctx, arguments);
      }),
      onMouseleave: _cache[1] || (_cache[1] = function() {
        return _ctx.startAnimation && _ctx.startAnimation.apply(_ctx, arguments);
      })
    },
    [createElementVNode(
      "div",
      _hoisted_2,
      [createElementVNode(
        "div",
        {
          class: normalizeClass([{
            stop: _ctx.state.isStop
          }, _ctx.direction]),
          style: normalizeStyle(_ctx.state.bindStyle)
        },
        [renderSlot(_ctx.$slots, "default", {}, function() {
          return [createElementVNode(
            "p",
            null,
            toDisplayString(_ctx.text),
            1
            /* TEXT */
          )];
        })],
        6
        /* CLASS, STYLE */
      )],
      512
      /* NEED_PATCH */
    )],
    34
    /* CLASS, NEED_HYDRATION */
  )]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
