// @opentiny/vue-cascader-panel/lib/index.js
import { defineComponent, $prefix, setup as _setup } from "@opentiny/vue-common";
import { renderless, api } from "@opentiny/vue-renderless/cascader-panel/vue";
import CascaderMenu from "@opentiny/vue-cascader-menu";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, Fragment, renderList, createBlock } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-theme/cascader-panel/index.css";
var _sfc_main = defineComponent({
  name: $prefix + "CascaderPanel",
  componentName: $prefix + "CascaderPanel",
  components: {
    CascaderMenu
  },
  props: {
    modelValue: {},
    options: Array,
    props: Object,
    border: {
      type: Boolean,
      default: true
    },
    renderLabel: Function
  },
  provide: function provide() {
    return {
      panel: this
    };
  },
  emits: ["update:modelValue", "change", "close", "active-item-change", "expand-change"],
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_cascader_menu = resolveComponent("cascader-menu");
  return openBlock(), createElementBlock(
    "div",
    {
      class: normalizeClass(["tiny-cascader-panel", _ctx.border && "is-bordered"]),
      onKeydown: _cache[0] || (_cache[0] = function() {
        return _ctx.handleKeyDown && _ctx.handleKeyDown.apply(_ctx, arguments);
      })
    },
    [(openBlock(true), createElementBlock(
      Fragment,
      null,
      renderList(_ctx.state.menus, function(menu, index) {
        return openBlock(), createBlock(_component_cascader_menu, {
          index,
          key: index,
          nodes: menu
        }, null, 8, ["index", "nodes"]);
      }),
      128
      /* KEYED_FRAGMENT */
    ))],
    34
    /* CLASS, NEED_HYDRATION */
  );
}
var CascaderPanel = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
CascaderPanel.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
CascaderPanel.install = function(Vue) {
  Vue.component(CascaderPanel.name, CascaderPanel);
};
CascaderPanel.version = version;
export {
  CascaderPanel as default
};
