// @opentiny/vue-action-sheet/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/action-sheet/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconClose } from "@opentiny/vue-icon";
import Button from "@opentiny/vue-button";
import Drawer from "@opentiny/vue-drawer";
import BScroll from "@better-scroll/core";
import { resolveComponent, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, renderSlot, toDisplayString, createCommentVNode, Fragment, renderList, withModifiers, createTextVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  components: {
    TinyButton: Button,
    TinyDrawer: Drawer,
    IconClose: IconClose()
  },
  props: [].concat(props, ["modelValue", "visible", "title", "showHeader", "showFooter", "customClass", "menus", "type", "mask", "maskClosable"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      extendOptions: {
        BScroll
      }
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-action-sheet",
  class: "text-sm"
};
var _hoisted_2 = {
  key: 0,
  "data-tag": "action-sheet-header",
  ref: "header",
  class: "flex leading-6 pb-4 px-4 text-base items-center"
};
var _hoisted_3 = {
  class: "flex-1 flex items-center text-left"
};
var _hoisted_4 = {
  class: "min-w-[50%] max-w-[80%] px-4 text-center truncate"
};
var _hoisted_5 = {
  class: "flex-1 flex items-center justify-end"
};
var _hoisted_6 = {
  "data-tag": "action-sheet-body",
  ref: "body",
  class: "flex-auto overflow-auto"
};
var _hoisted_7 = {
  key: 0,
  "data-tag": "action-box",
  class: "flex flex-col h-full text-center"
};
var _hoisted_8 = {
  key: 0,
  "data-tag": "action-title",
  class: "flex-none flex items-center justify-center py-2.5 leading-5 text-sm text-color-none-hover border-b-0.5 border-color-bg-2"
};
var _hoisted_9 = {
  class: "px-4"
};
var _hoisted_10 = {
  class: "flex-auto min-h-[3.5rem]"
};
var _hoisted_11 = ["onClick"];
var _hoisted_12 = {
  class: "truncate px-4"
};
var _hoisted_13 = {
  key: 1,
  "data-tag": "action-sheet-footer",
  ref: "footer",
  class: "flex px-3 pt-2 justify-center"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconClose = resolveComponent("IconClose");
  var _component_tiny_button = resolveComponent("tiny-button");
  var _component_tiny_drawer = resolveComponent("tiny-drawer");
  return openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_drawer, {
    ref: "drawer",
    title: _ctx.title,
    mask: _ctx.mask,
    "mask-closable": _ctx.maskClosable,
    "show-header": false,
    "custom-class": _ctx.m("max-h-[95%] rounded-t-lg", {
      "min-h-[15.625rem]": _ctx.type !== "action"
    }, {
      "py-4": _ctx.showHeader && _ctx.type !== "action"
    }, _ctx.customClass),
    placement: "bottom",
    visible: _ctx.visible,
    "onUpdate:visible": _cache[1] || (_cache[1] = function($event) {
      return _ctx.$emit("update:visible", $event);
    })
  }, {
    default: withCtx(function() {
      return [_ctx.showHeader && _ctx.type !== "action" ? (openBlock(), createElementBlock(
        "div",
        _hoisted_2,
        [createElementVNode("div", _hoisted_3, [renderSlot(_ctx.$slots, "header-left")]), createElementVNode(
          "div",
          _hoisted_4,
          toDisplayString(_ctx.title),
          1
          /* TEXT */
        ), createElementVNode("div", _hoisted_5, [renderSlot(_ctx.$slots, "header-right", {}, function() {
          return [createVNode(_component_IconClose, {
            "custom-class": "h-5 w-5 cursor-pointer",
            onClick: _ctx.close
          }, null, 8, ["onClick"])];
        })])],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true), createElementVNode(
        "div",
        _hoisted_6,
        [_ctx.type === "action" ? (openBlock(), createElementBlock("div", _hoisted_7, [_ctx.title ? (openBlock(), createElementBlock("div", _hoisted_8, [createElementVNode(
          "span",
          _hoisted_9,
          toDisplayString(_ctx.title),
          1
          /* TEXT */
        )])) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_10, [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.menus, function(item, index) {
            return openBlock(), createElementBlock("div", {
              key: index,
              class: "flex-none flex items-center justify-center h-14 text-base border-b-0.5 border-color-bg-2 cursor-pointer",
              onClick: withModifiers(function($event) {
                return _ctx.actionSelectOption(item, index);
              }, ["stop"])
            }, [renderSlot(_ctx.$slots, "default", {
              data: item,
              index
            }, function() {
              return [createElementVNode(
                "span",
                _hoisted_12,
                toDisplayString(item),
                1
                /* TEXT */
              )];
            })], 8, _hoisted_11);
          }),
          128
          /* KEYED_FRAGMENT */
        ))]), createElementVNode(
          "div",
          {
            class: "flex-none flex items-center justify-center h-16 text-base border-t-8 border-color-bg-2 px-4 cursor-pointer",
            onClick: _cache[0] || (_cache[0] = withModifiers(function() {
              return _ctx.close && _ctx.close.apply(_ctx, arguments);
            }, ["stop"]))
          },
          toDisplayString(_ctx.t("ui.base.cancel")),
          1
          /* TEXT */
        )])) : renderSlot(_ctx.$slots, "default", {
          key: 1
        })],
        512
        /* NEED_PATCH */
      ), _ctx.showFooter ? (openBlock(), createElementBlock(
        "div",
        _hoisted_13,
        [renderSlot(_ctx.$slots, "footer", {}, function() {
          return [createVNode(_component_tiny_button, {
            tiny_mode: "mobile-first",
            class: "flex-1",
            type: "primary",
            onClick: _ctx.confirm
          }, {
            default: withCtx(function() {
              return [createTextVNode(
                toDisplayString(_ctx.t("ui.button.confirm")),
                1
                /* TEXT */
              )];
            }),
            _: 1
            /* STABLE */
          }, 8, ["onClick"])];
        })],
        512
        /* NEED_PATCH */
      )) : createCommentVNode("v-if", true)];
    }),
    _: 3
    /* FORWARDED */
  }, 8, ["title", "mask", "mask-closable", "custom-class", "visible"])]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
