// @opentiny/vue-action-sheet/lib/mobile.js
import { renderless, api } from "@opentiny/vue-renderless/action-sheet/vue";
import { defineComponent, $prefix, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme-mobile/action-sheet/index.css";
import BScroll from "@better-scroll/core";
import { withDirectives, openBlock, createElementBlock, normalizeStyle, createCommentVNode, createElementVNode, normalizeClass, Fragment, renderList, renderSlot, createTextVNode, toDisplayString, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  name: $prefix + "ActionSheet",
  props: [].concat(props, ["menus", "modelValue", "visible", "ellipsis", "height", "contentPosition", "contentStyle"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api,
      extendOptions: {
        BScroll
      }
    });
  }
});
var _hoisted_1 = ["onClick"];
var _hoisted_2 = {
  key: 1,
  class: "tiny-mobile-action-sheet__action"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return withDirectives((openBlock(), createElementBlock(
    "div",
    {
      class: "tiny-mobile-action-sheet",
      onClick: _cache[1] || (_cache[1] = function() {
        return _ctx.visibleHandle && _ctx.visibleHandle.apply(_ctx, arguments);
      })
    },
    [!_ctx.contentPosition ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: "tiny-mobile-action-sheet__mask",
        style: normalizeStyle(_ctx.state.sheetMaskStyle)
      },
      null,
      4
      /* STYLE */
    )) : createCommentVNode("v-if", true), createElementVNode(
      "div",
      {
        class: normalizeClass(["tiny-mobile-action-sheet__content", _ctx.state.toggle ? "is-toggle" : "", _ctx.contentPosition ? "" : "is-not-content"]),
        style: normalizeStyle([_ctx.state.sheetContentStyle]),
        ref: "scrollMenu"
      },
      [createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-mobile-action-sheet__menu", _ctx.ellipsis ? "is-ellipsis" : ""])
        },
        [(openBlock(true), createElementBlock(
          Fragment,
          null,
          renderList(_ctx.menus, function(item, index) {
            return openBlock(), createElementBlock("div", {
              class: normalizeClass(["tiny-mobile-action-sheet__item", item.warn ? "is-warn" : "", item.id === _ctx.modelValue || item.id === _ctx.state.active ? "is-active" : ""]),
              style: normalizeStyle(_ctx.state.contentStyle),
              key: index,
              onClick: function onClick($event) {
                return _ctx.menuHandle(item, index);
              }
            }, [renderSlot(_ctx.$slots, "item", {
              item
            }, function() {
              return [createTextVNode(
                toDisplayString(item.label),
                1
                /* TEXT */
              )];
            })], 14, _hoisted_1);
          }),
          128
          /* KEYED_FRAGMENT */
        ))],
        2
        /* CLASS */
      )],
      6
      /* CLASS, STYLE */
    ), _ctx.contentPosition ? (openBlock(), createElementBlock("div", _hoisted_2, [renderSlot(_ctx.$slots, "action", {}, function() {
      return [createElementVNode(
        "div",
        {
          class: "tiny-mobile-action-sheet__cancel",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.visibleHandle && _ctx.visibleHandle.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.actionSheet.cancel")),
        1
        /* TEXT */
      )];
    })])) : createCommentVNode("v-if", true)],
    512
    /* NEED_PATCH */
  )), [[vShow, _ctx.visible]]);
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
