// @opentiny/vue-alert/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/alert/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { iconClose, iconSuccess, iconError, iconHelp, iconWarning, iconChevronDown } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createBlock, Transition, withCtx, createElementBlock, normalizeClass, resolveDynamicComponent, createCommentVNode, createElementVNode, renderSlot, createTextVNode, toDisplayString, createVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["close"],
  props: [].concat(props, ["icon", "type", "size", "description", "title", "closable", "center", "showIcon", "closeText", "singleLine", "scrolling", "showFoldable", "customClass"]),
  components: {
    IconClose: iconClose(),
    IconSuccess: iconSuccess(),
    IconError: iconError(),
    IconHelp: iconHelp(),
    IconWarning: iconWarning(),
    IconChevronDown: iconChevronDown()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  "data-tag": "tiny-alert-title"
};
var _hoisted_2 = {
  "data-tag": "tiny-alert-icon",
  class: "ml-2.5"
};
var _hoisted_3 = {
  key: 0,
  "data-tag": "tiny-alert-large",
  class: "font-medium"
};
var _hoisted_4 = {
  key: 1,
  class: "pt-2"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_chevron_down = resolveComponent("icon-chevron-down");
  var _component_icon_close = resolveComponent("icon-close");
  return openBlock(), createBlock(Transition, null, {
    default: withCtx(function() {
      return [_ctx.state.show ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          "data-tag": "tiny-alert",
          class: normalizeClass(_ctx.m("min-h-min flex py-2 sm:py-3 px-4 my-2 rounded box-border font-light sm:font-normal", {
            "bg-color-info-primary-subtler": _ctx.type === "info" || !_ctx.type
          }, {
            "bg-color-error-subtler": _ctx.type === "error"
          }, {
            "bg-color-warning-subtler": _ctx.type === "warning"
          }, {
            "bg-color-success-subtler": _ctx.type === "success"
          }, {
            "text-center": _ctx.center
          }, _ctx.customClass))
        },
        [_ctx.showIcon ? (openBlock(), createBlock(resolveDynamicComponent(_ctx.state.getIcon), {
          key: 0,
          "custom-class": "h-4.5 w-4.5 mt-1 sm:mt-0.5 sm:h-6 sm:w-5 sm:h-5 fill-current",
          class: normalizeClass([{
            "text-color-info-primary": _ctx.type === "info" || !_ctx.type
          }, {
            "text-color-error": _ctx.type === "error"
          }, {
            "text-color-warning": _ctx.type === "warning"
          }, {
            "text-color-success": _ctx.type === "success"
          }])
        }, null, 8, ["class"])) : createCommentVNode("v-if", true), _ctx.showFoldable ? (openBlock(), createElementBlock(
          "div",
          {
            key: 1,
            "data-tag": "tiny-alert-foldable",
            class: normalizeClass(["flex-1 leading-6 text-sm overflow-hidden", _ctx.showIcon ? "mx-2" : "mr-2"])
          },
          [_ctx.size === "large" ? (openBlock(), createElementBlock("div", {
            key: 0,
            "data-tag": "tiny-alert-large",
            onClick: _cache[0] || (_cache[0] = function() {
              return _ctx.handleHeaderClick && _ctx.handleHeaderClick.apply(_ctx, arguments);
            }),
            class: "inline-flex cursor-pointer font-medium"
          }, [createElementVNode("span", _hoisted_1, [renderSlot(_ctx.$slots, "title", {}, function() {
            return [createTextVNode(
              toDisplayString(_ctx.state.getTitle),
              1
              /* TEXT */
            )];
          })]), createElementVNode("span", _hoisted_2, [createVNode(_component_icon_chevron_down, {
            class: normalizeClass(["transition-transform duration-300 align-top my-1 fill-color-icon-placeholder", [_ctx.state.contentVisible ? "rotate-180" : "rotate-0"]])
          }, null, 8, ["class"])])])) : createCommentVNode("v-if", true), createElementVNode(
            "div",
            {
              "data-tag": "tiny-alert-content",
              ref: "ContentDescribe",
              class: normalizeClass(["transition-all duration-900 ease-linear", {
                "hidden": _ctx.size === "large" && !_ctx.description && !_ctx.slots.description
              }, {
                "overflow-y-auto": _ctx.state.scrollStatus
              }, _ctx.state.contentVisible ? "max-h-[theme(spacing.72)]" : "max-h-0"])
            },
            [createElementVNode(
              "div",
              {
                "data-tag": "tiny-alert-singline",
                class: normalizeClass([{
                  "truncate": _ctx.singleLine
                }, {
                  "hover:animate-[leftMove_10s_linear_infinite] hover:overflow-visible cursor-pointer": _ctx.singleLine && _ctx.scrolling
                }])
              },
              [renderSlot(_ctx.$slots, "description", {}, function() {
                return [createTextVNode(
                  toDisplayString(_ctx.description),
                  1
                  /* TEXT */
                )];
              })],
              2
              /* CLASS */
            )],
            2
            /* CLASS */
          ), _ctx.size === "large" && _ctx.slots.default ? (openBlock(), createElementBlock(
            "div",
            {
              key: 1,
              "data-tag": "tiny-alert-default",
              ref: "ContentDefault",
              class: normalizeClass(["transition-all duration-900 ease-linear", {
                "pt-2": _ctx.description && _ctx.state.contentVisible
              }, {
                "overflow-y-auto": _ctx.state.scrollStatus
              }, _ctx.state.contentVisible ? "max-h-[theme(spacing.72)]" : "max-h-0"])
            },
            [renderSlot(_ctx.$slots, "default")],
            2
            /* CLASS */
          )) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        )) : (openBlock(), createElementBlock(
          "div",
          {
            key: 2,
            "data-tag": "tiny-alert-notfoldable",
            class: normalizeClass(["flex-1 leading-6 text-sm overflow-hidden cursor-pointer", _ctx.showIcon ? "mx-2" : "mr-2"])
          },
          [_ctx.size === "large" ? (openBlock(), createElementBlock("div", _hoisted_3, [renderSlot(_ctx.$slots, "title", {}, function() {
            return [createTextVNode(
              toDisplayString(_ctx.state.getTitle),
              1
              /* TEXT */
            )];
          })])) : createCommentVNode("v-if", true), createElementVNode(
            "div",
            {
              "data-tag": "tiny-alert-singlelinebox",
              class: normalizeClass([{
                "hidden": _ctx.size === "large" && !_ctx.description && !_ctx.slots.description
              }])
            },
            [createElementVNode(
              "div",
              {
                "data-tag": "tiny-alert-singleline",
                class: normalizeClass([{
                  "truncate": _ctx.singleLine
                }, {
                  "hover:animate-[leftMove_10s_linear_infinite] hover:overflow-visible cursor-pointer": _ctx.singleLine && _ctx.scrolling
                }])
              },
              [renderSlot(_ctx.$slots, "description", {}, function() {
                return [createTextVNode(
                  toDisplayString(_ctx.description),
                  1
                  /* TEXT */
                )];
              })],
              2
              /* CLASS */
            )],
            2
            /* CLASS */
          ), _ctx.size === "large" && _ctx.slots.default ? (openBlock(), createElementBlock("div", _hoisted_4, [renderSlot(_ctx.$slots, "default")])) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        )), !_ctx.closeText && _ctx.closable ? (openBlock(), createBlock(_component_icon_close, {
          key: 3,
          onClick: _ctx.handleClose,
          class: "h-4 w-4 mt-1 cursor-pointer fill-color-text-primary opacity-70"
        }, null, 8, ["onClick"])) : _ctx.closeText && _ctx.closable ? (openBlock(), createElementBlock(
          "span",
          {
            key: 4,
            "data-tag": "tiny-alert-close-text",
            onClick: _cache[1] || (_cache[1] = function() {
              return _ctx.handleClose && _ctx.handleClose.apply(_ctx, arguments);
            }),
            class: "leading-6 text-sm cursor-pointer"
          },
          toDisplayString(_ctx.closeText),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true)];
    }),
    _: 3
    /* FORWARDED */
  });
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
