// @opentiny/vue-user-head-group/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { $prefix, $props, $setup } from "@opentiny/vue-common";
import MobileFirstTemplate from "./mobile-first.js";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return MobileFirstTemplate;
};
var UserHeadGroup = {
  name: $prefix + "UserHeadGroup",
  componentName: "TinyUserHeadGroup",
  props: _extends({}, $props, {
    groupSize: {
      type: Number,
      default: 40
    },
    list: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    showMax: {
      type: Boolean,
      default: false
    },
    horizontal: {
      type: Boolean,
      default: false
    },
    color: String,
    backgroundColor: String
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
};
var version = "3.13.0";
UserHeadGroup.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
UserHeadGroup.install = function(Vue) {
  Vue.component(UserHeadGroup.name, UserHeadGroup);
};
UserHeadGroup.version = version;
export {
  UserHeadGroup as default
};
