// @opentiny/vue-time-range/lib/index.js
import { renderless, api } from "@opentiny/vue-renderless/time-range/vue";
import { defineComponent, $prefix, setup as _setup } from "@opentiny/vue-common";
import TimeSpinner from "@opentiny/vue-time-spinner";
import { resolveComponent, openBlock, createBlock, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, toDisplayString, createVNode, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

import "@opentiny/vue-theme/time-range/index.css";
var _sfc_main = defineComponent({
  name: $prefix + "TimeRange",
  emits: ["dodestroy", "pick", "select-range"],
  components: {
    TimeSpinner
  },
  props: {
    emitter: Object
  },
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  class: "tiny-time-range-picker__content"
};
var _hoisted_2 = {
  class: "tiny-time-range-picker__cell"
};
var _hoisted_3 = {
  class: "tiny-time-range-picker__header"
};
var _hoisted_4 = {
  class: "tiny-time-range-picker__cell"
};
var _hoisted_5 = {
  class: "tiny-time-range-picker__header"
};
var _hoisted_6 = {
  class: "tiny-time-panel__footer"
};
var _hoisted_7 = ["disabled"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_time_spinner = resolveComponent("time-spinner");
  return openBlock(), createBlock(Transition, {
    name: "tiny-zoom-in-top",
    onAfterLeave: _cache[2] || (_cache[2] = function($event) {
      return _ctx.$emit("dodestroy");
    }),
    persisted: ""
  }, {
    default: withCtx(function() {
      return [withDirectives(createElementVNode(
        "div",
        {
          class: normalizeClass(["tiny-time-range-picker tiny-picker-panel tiny-popper", _ctx.state.popperClass])
        },
        [createElementVNode("div", _hoisted_1, [createElementVNode("div", _hoisted_2, [createElementVNode(
          "div",
          _hoisted_3,
          toDisplayString(_ctx.t("ui.datepicker.startTime")),
          1
          /* TEXT */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "has-seconds": _ctx.state.showSeconds,
              "is-arrow": _ctx.state.arrowControl
            }, "tiny-time-range-picker__body tiny-time-panel__content"])
          },
          [createVNode(_component_time_spinner, {
            ref: "minSpinner",
            "show-seconds": _ctx.state.showSeconds,
            "am-pm-mode": _ctx.state.amPmMode,
            onChange: _ctx.handleMinChange,
            "arrow-control": _ctx.state.arrowControl,
            onSelectRange: _ctx.setMinSelectionRange,
            date: _ctx.state.minDate
          }, null, 8, ["show-seconds", "am-pm-mode", "onChange", "arrow-control", "onSelectRange", "date"])],
          2
          /* CLASS */
        )]), createElementVNode("div", _hoisted_4, [createElementVNode(
          "div",
          _hoisted_5,
          toDisplayString(_ctx.t("ui.datepicker.endTime")),
          1
          /* TEXT */
        ), createElementVNode(
          "div",
          {
            class: normalizeClass([{
              "has-seconds": _ctx.state.showSeconds,
              "is-arrow": _ctx.state.arrowControl
            }, "tiny-time-range-picker__body tiny-time-panel__content"])
          },
          [createVNode(_component_time_spinner, {
            ref: "maxSpinner",
            "show-seconds": _ctx.state.showSeconds,
            "am-pm-mode": _ctx.state.amPmMode,
            onChange: _ctx.handleMaxChange,
            "arrow-control": _ctx.state.arrowControl,
            onSelectRange: _ctx.setMaxSelectionRange,
            date: _ctx.state.maxDate
          }, null, 8, ["show-seconds", "am-pm-mode", "onChange", "arrow-control", "onSelectRange", "date"])],
          2
          /* CLASS */
        )])]), createElementVNode("div", _hoisted_6, [createElementVNode(
          "button",
          {
            type: "button",
            class: "tiny-time-panel__btn cancel",
            onClick: _cache[0] || (_cache[0] = function($event) {
              return _ctx.handleCancel();
            })
          },
          toDisplayString(_ctx.t("ui.datepicker.cancel")),
          1
          /* TEXT */
        ), createElementVNode("button", {
          type: "button",
          class: "tiny-time-panel__btn confirm",
          onClick: _cache[1] || (_cache[1] = function($event) {
            return _ctx.handleConfirm();
          }),
          disabled: _ctx.state.btnDisabled
        }, toDisplayString(_ctx.t("ui.datepicker.confirm")), 9, _hoisted_7)])],
        2
        /* CLASS */
      ), [[vShow, _ctx.state.visible]])];
    }),
    _: 1
    /* STABLE */
  });
}
var TimeRange = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
TimeRange.install = function(Vue) {
  Vue.component(TimeRange.name, TimeRange);
};
TimeRange.version = version;
export {
  TimeRange as default
};
