// @opentiny/vue-popover/lib/mobile.js
import { renderless, api } from "@opentiny/vue-renderless/popover/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import "@opentiny/vue-theme-mobile/popover/index.css";
import { openBlock, createElementBlock, createVNode, Transition, withCtx, withDirectives, createElementVNode, normalizeClass, normalizeStyle, toDisplayString, createCommentVNode, renderSlot, Fragment, renderList, vShow } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["update:modelValue", "hide", "show", "after-enter", "after-leave", "created", "itemClick"],
  props: [].concat(props, ["appendToBody", "arrowOffset", "boundariesPadding", "closeDelay", "disabled", "modelValue", "offset", "openDelay", "placement", "popper", "popperClass", "popperOptions", "reference", "tabindex", "title", "transition", "trigger", "visibleArrow", "width", "maxHeight", "listData"]),
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  ref: "root"
};
var _hoisted_2 = ["id", "aria-hidden"];
var _hoisted_3 = ["textContent"];
var _hoisted_4 = ["onClick"];
var _hoisted_5 = ["src"];
var _hoisted_6 = {
  class: "label-view"
};
var _hoisted_7 = {
  class: "main"
};
var _hoisted_8 = {
  key: 0,
  class: "sub"
};
var _hoisted_9 = {
  key: 0,
  class: "line"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "span",
    _hoisted_1,
    [createVNode(Transition, {
      name: _ctx.transition,
      onAfterEnter: _ctx.handleAfterEnter,
      onAfterLeave: _ctx.handleAfterLeave,
      persisted: ""
    }, {
      default: withCtx(function() {
        return [withDirectives(createElementVNode("div", {
          class: normalizeClass(["tiny-mobile-popover tiny-mobile-popper", [_ctx.popperClass, "tiny-mobile-popover__plain", {
            "no-arrow": !_ctx.visibleArrow
          }]]),
          ref: "popper",
          style: normalizeStyle({
            width: _ctx.width === "auto" ? _ctx.width : _ctx.width + "px",
            maxHeight: _ctx.maxHeight === "auto" ? _ctx.maxHeight : _ctx.maxHeight + "px"
          }),
          role: "tooltip",
          id: _ctx.state.tooltipId,
          "aria-hidden": _ctx.disabled || !_ctx.state.showPopper ? "true" : "false"
        }, [_ctx.title ? (openBlock(), createElementBlock("div", {
          key: 0,
          class: "tiny-mobile-popover__title",
          textContent: toDisplayString(_ctx.title)
        }, null, 8, _hoisted_3)) : createCommentVNode("v-if", true), renderSlot(_ctx.$slots, "default", {}, function() {
          return [createElementVNode(
            "div",
            {
              class: "list-content",
              style: normalizeStyle({
                maxHeight: _ctx.maxHeight === "auto" ? _ctx.maxHeight : _ctx.maxHeight + "px"
              })
            },
            [(openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(_ctx.listData, function(item, index) {
                return openBlock(), createElementBlock("div", {
                  key: item.id,
                  class: normalizeClass(["list-view", item.subLabel ? "list-view-height" : ""]),
                  onClick: function onClick($event) {
                    return _ctx.handleItemClick(item);
                  }
                }, [createElementVNode("img", {
                  src: item.icon,
                  class: "icon"
                }, null, 8, _hoisted_5), createElementVNode("div", _hoisted_6, [createElementVNode(
                  "div",
                  _hoisted_7,
                  toDisplayString(item.label),
                  1
                  /* TEXT */
                ), item.subLabel ? (openBlock(), createElementBlock(
                  "div",
                  _hoisted_8,
                  toDisplayString(item.subLabel),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)]), index !== _ctx.listData.length - 1 ? (openBlock(), createElementBlock("div", _hoisted_9)) : createCommentVNode("v-if", true)], 10, _hoisted_4);
              }),
              128
              /* KEYED_FRAGMENT */
            ))],
            4
            /* STYLE */
          )];
        })], 14, _hoisted_2), [[vShow, !_ctx.disabled && _ctx.state.showPopper]])];
      }),
      _: 3
      /* FORWARDED */
    }, 8, ["name", "onAfterEnter", "onAfterLeave"]), createElementVNode(
      "span",
      {
        ref: "wrapper",
        class: normalizeClass(["reference-wrapper", {
          "reference-wrapper-show": !_ctx.disabled && _ctx.state.showPopper
        }])
      },
      [renderSlot(_ctx.$slots, "reference")],
      2
      /* CLASS */
    )],
    512
    /* NEED_PATCH */
  );
}
var mobile = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobile as default
};
