// @opentiny/vue-user-link/lib/pc.js
import { renderless, api } from "@opentiny/vue-renderless/user-link/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import Popover from "@opentiny/vue-popover";
import Loading from "@opentiny/vue-loading";
import { IconChevronDown, IconChevronUp } from "@opentiny/vue-icon";
import { resolveComponent, resolveDirective, openBlock, createElementBlock, Fragment, renderList, toDisplayString, createBlock, withCtx, createElementVNode, withDirectives, createTextVNode, vShow, createVNode } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["fetchUser", "fetchW3Accounts", "getUserImageUrl", "fetchUserByUserId", "userCache", "valueSplit", "card", "textSplit", "textField", "valueField", "modelValue", "service", "cache", "cacheKey", "cacheFields", "batch"]),
  components: {
    TinyPopover: Popover,
    IconChevronDown: IconChevronDown(),
    IconChevronUp: IconChevronUp()
  },
  directives: {
    loading: Loading.directive
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "tiny-userlink"
};
var _hoisted_2 = {
  key: 0
};
var _hoisted_3 = ["onClick"];
var _hoisted_4 = {
  class: "card-box"
};
var _hoisted_5 = /* @__PURE__ */ createElementVNode(
  "div",
  {
    class: "card-box__header"
  },
  null,
  -1
  /* HOISTED */
);
var _hoisted_6 = {
  class: "card-box__body"
};
var _hoisted_7 = {
  class: "card-top-img"
};
var _hoisted_8 = ["src"];
var _hoisted_9 = {
  class: "card-top-text"
};
var _hoisted_10 = {
  class: "card-center"
};
var _hoisted_11 = ["href"];
var _hoisted_12 = {
  class: "card-center"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_IconChevronDown = resolveComponent("IconChevronDown");
  var _component_IconChevronUp = resolveComponent("IconChevronUp");
  var _component_tiny_popover = resolveComponent("tiny-popover");
  var _directive_loading = resolveDirective("loading");
  return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.options, function(user, index) {
      return openBlock(), createElementBlock("li", {
        key: index
      }, [!_ctx.card ? (openBlock(), createElementBlock(
        "span",
        _hoisted_2,
        toDisplayString(user[_ctx.textField] + (index !== _ctx.state.options.length - 1 ? _ctx.textSplit : "")),
        1
        /* TEXT */
      )) : (openBlock(), createBlock(
        _component_tiny_popover,
        {
          key: 1,
          trigger: "click",
          "visible-arrow": false,
          placement: "bottom-start",
          "popper-class": "tiny-user-card"
        },
        {
          reference: withCtx(function() {
            return [createElementVNode("a", {
              onClick: function onClick($event) {
                return _ctx.showCard($event, user);
              }
            }, toDisplayString(user[_ctx.textField] + (index !== _ctx.state.options.length - 1 ? _ctx.textSplit : "")), 9, _hoisted_3)];
          }),
          default: withCtx(function() {
            return [withDirectives((openBlock(), createElementBlock("div", _hoisted_4, [_hoisted_5, createElementVNode("div", _hoisted_6, [createElementVNode("div", _hoisted_7, [createElementVNode("img", {
              src: _ctx.state.imgUrl
            }, null, 8, _hoisted_8)]), createElementVNode("div", _hoisted_9, [createElementVNode(
              "span",
              null,
              toDisplayString(_ctx.state.data.english_Name),
              1
              /* TEXT */
            ), createElementVNode(
              "span",
              null,
              "(" + toDisplayString(_ctx.state.data.last_Name) + ")",
              1
              /* TEXT */
            ), createElementVNode(
              "p",
              null,
              toDisplayString(_ctx.t("ui.usercard.empno")) + ":" + toDisplayString(_ctx.state.data.employee_Number),
              1
              /* TEXT */
            ), createElementVNode(
              "p",
              null,
              toDisplayString(_ctx.state.data.l4_Name) + "(" + toDisplayString(_ctx.state.data.l1_Name) + ")",
              1
              /* TEXT */
            )]), createElementVNode("div", _hoisted_10, [createElementVNode("p", null, [createElementVNode(
              "strong",
              null,
              toDisplayString(_ctx.t("ui.usercard.phone")),
              1
              /* TEXT */
            ), createTextVNode(
              "：" + toDisplayString(_ctx.state.data.person_Phone_Code),
              1
              /* TEXT */
            )]), createElementVNode("p", null, [createElementVNode(
              "strong",
              null,
              toDisplayString(_ctx.t("ui.usercard.internal")),
              1
              /* TEXT */
            ), createTextVNode(
              "：" + toDisplayString(_ctx.state.data.person_Internal_Code),
              1
              /* TEXT */
            )]), createElementVNode("p", null, [createElementVNode(
              "strong",
              null,
              toDisplayString(_ctx.t("ui.usercard.mobile")),
              1
              /* TEXT */
            ), createTextVNode(
              "：" + toDisplayString(_ctx.state.data.person_Mobile_Code),
              1
              /* TEXT */
            )]), createElementVNode("p", null, [createElementVNode(
              "strong",
              null,
              toDisplayString(_ctx.t("ui.usercard.email")),
              1
              /* TEXT */
            ), createTextVNode("："), createElementVNode("a", {
              href: "mailto:" + _ctx.state.data.person_Mail
            }, toDisplayString(_ctx.state.data.person_Mail), 9, _hoisted_11)]), createElementVNode("p", null, [createElementVNode(
              "strong",
              null,
              toDisplayString(_ctx.t("ui.usercard.address")),
              1
              /* TEXT */
            ), createTextVNode(
              "：" + toDisplayString(_ctx.state.data.person_Location),
              1
              /* TEXT */
            )])]), withDirectives(createElementVNode(
              "div",
              _hoisted_12,
              [createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.manager")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.organization_manager),
                1
                /* TEXT */
              )]), createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.viop")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.person_Viop),
                1
                /* TEXT */
              )]), createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.fax")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.person_Fax_Code),
                1
                /* TEXT */
              )]), createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.travelcode")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.person_Travel_Code),
                1
                /* TEXT */
              )]), createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.timezone")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.person_Difference_Time),
                1
                /* TEXT */
              )]), createElementVNode("p", null, [createElementVNode(
                "strong",
                null,
                toDisplayString(_ctx.t("ui.usercard.other")),
                1
                /* TEXT */
              ), createTextVNode(
                "：" + toDisplayString(_ctx.state.data.person_Other),
                1
                /* TEXT */
              )])],
              512
              /* NEED_PATCH */
            ), [[vShow, _ctx.state.expand]])]), createElementVNode("div", {
              class: "box__footer",
              onClick: _cache[0] || (_cache[0] = function() {
                return _ctx.showDetail && _ctx.showDetail.apply(_ctx, arguments);
              })
            }, [withDirectives(createElementVNode(
              "div",
              null,
              [createTextVNode(
                toDisplayString(_ctx.t("ui.usercard.expand")) + " ",
                1
                /* TEXT */
              ), createVNode(_component_IconChevronDown)],
              512
              /* NEED_PATCH */
            ), [[vShow, !_ctx.state.expand]]), withDirectives(createElementVNode(
              "div",
              null,
              [createTextVNode(
                toDisplayString(_ctx.t("ui.usercard.collapse")) + " ",
                1
                /* TEXT */
              ), createVNode(_component_IconChevronUp)],
              512
              /* NEED_PATCH */
            ), [[vShow, _ctx.state.expand]])])])), [[_directive_loading, _ctx.state.spinner]])];
          }),
          _: 2
          /* DYNAMIC */
        },
        1024
        /* DYNAMIC_SLOTS */
      ))]);
    }),
    128
    /* KEYED_FRAGMENT */
  ))]);
}
var pc = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  pc as default
};
