// @opentiny/vue-renderless/action-sheet/vue.js
import "../chunk-G2ADBYYC.js";
import {
  setSheetStyle,
  initScrollMenu,
  visibleHandle,
  watchVisible,
  menuHandle,
  close,
  selectOption,
  confirm,
  actionSelectOption,
  hide
} from "./index";
const api = [
  "state",
  "setSheetStyle",
  "initScrollMenu",
  "visibleHandle",
  "watchVisible",
  "menuHandle",
  "close",
  "selectOption",
  "confirm",
  "actionSelectOption",
  "hide"
];
const renderless = (props, { reactive, watch }, { emit, nextTick, refs, vm }, { BScroll }) => {
  const api2 = {};
  const state = reactive({
    toggle: false,
    active: null,
    sheetMaskStyle: {},
    sheetContentStyle: {},
    scroll: null
  });
  Object.assign(api2, {
    state,
    setSheetStyle: setSheetStyle({ state, props }),
    initScrollMenu: initScrollMenu({ state, nextTick, refs, BScroll }),
    visibleHandle: visibleHandle({ emit, state }),
    watchVisible: watchVisible({ emit, props, state }),
    menuHandle: menuHandle({ state, emit }),
    confirm: confirm({ state, emit }),
    selectOption: selectOption({ emit, props }),
    actionSelectOption: actionSelectOption({ emit }),
    close: close({ emit, vm }),
    hide: hide(emit)
  });
  watch(
    () => props.visible,
    (value) => {
      if (value) {
        api2.setSheetStyle({ state, props });
        api2.initScrollMenu({ state, nextTick, refs, BScroll });
      }
      api2.watchVisible(value);
    }
  );
  watch(() => props.visible, api2.watchVisible, { immediate: true });
  return api2;
};
export {
  api,
  renderless
};
