// @opentiny/vue-credit-card/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { defineComponent, $prefix, $props, $setup } from "@opentiny/vue-common";
import PcTemplate from "./pc.js";
import "@opentiny/vue-theme/credit-card/index.css";
var template = function template2(mode) {
  var _process$env;
  typeof process === "object" ? (_process$env = process.env) == null ? void 0 : _process$env.TINY_MODE : null;
  return PcTemplate;
};
var CreditCard = defineComponent({
  name: $prefix + "CreditCard",
  props: _extends({}, $props, {
    backgroundImage: [String, Object],
    isCardNumberMasked: Boolean,
    fields: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    labels: {
      type: Object,
      default: function _default2() {
        return {
          cardCvv: "",
          cardMonth: "",
          cardName: "",
          cardNumber: "",
          cardYear: ""
        };
      }
    }
  }),
  setup: function setup(props, context) {
    return $setup({
      props,
      context,
      template
    });
  }
});
var version = "3.13.0";
CreditCard.install = function(Vue) {
  Vue.component(CreditCard.name, CreditCard);
};
CreditCard.version = version;
export {
  CreditCard as default
};
