// @opentiny/vue-time-line/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/time-line/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import { IconChevronDown, IconChevronUp } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, Fragment, renderList, normalizeStyle, normalizeClass, createElementVNode, renderSlot, toDisplayString, createCommentVNode, createBlock, resolveDynamicComponent } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  emits: ["click"],
  props: [].concat(props, ["vertical", "nameField", "timeField", "tipsField", "autoColorField", "data", "space", "subField"]),
  components: {
    IconChevronDown: IconChevronDown(),
    IconChevronUp: IconChevronUp()
  },
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "inline-flex flex-col",
  "data-tag": "tiny-timeline"
};
var _hoisted_2 = {
  class: "inline-flex sm:items-start",
  "data-tag": "tiny-timeline-list"
};
var _hoisted_3 = {
  class: "text-color-text-secondary text-left pr-0.5 inline-flex text-xs"
};
var _hoisted_4 = {
  class: "sm:inline-block hidden leading-5 w-32"
};
var _hoisted_5 = {
  class: "ml-1"
};
var _hoisted_6 = {
  class: "sm:hidden inline-flex flex-col leading-5.5 w-20"
};
var _hoisted_7 = {
  class: "text-sm"
};
var _hoisted_8 = {
  class: "text-sm mb-6"
};
var _hoisted_9 = {
  class: "mt-1 w-3 h-3 leading-3 box-border relative text-xs text-center -ml-1.5 cursor-pointer inline-flex items-center justify-center align-top"
};
var _hoisted_10 = {
  key: 1,
  class: /* @__PURE__ */ normalizeClass(["rounded-full block w-1.5 h-1.5 bg-color-brand"])
};
var _hoisted_11 = {
  "data-tag": "tiny-timeline-list-left",
  class: "sm:text-color-text-secondary text-color-text-placeholder text-left sm:pl-3 pl-2 inline-flex flex-col sm:flex-row"
};
var _hoisted_12 = {
  class: "inline-block sm:text-xs text-sm sm:leading-5 leading-5.5"
};
var _hoisted_13 = {
  key: 0,
  class: "sm:hidden inline-block sm:leading-5 leading-5.5 text-sm overflow-hidden text-ellipsis whitespace-nowrap max-w-[theme(spacing.56)] mb-6 text-color-text-primary"
};
var _hoisted_14 = {
  key: 1,
  class: "sm:hidden max-w-[theme(spacing.56)] inline-block text-color-icon-secondary text-xs mt-2.5 leading-5"
};
var _hoisted_15 = {
  key: 0,
  class: "max-w-[theme(spacing.56)] inline-block text-color-icon-secondary sm:text-xs text-sm leading-5 mb-4 break-all break-words"
};
var _hoisted_16 = {
  class: "inline-block leading-5.5 sm:text-xs text-sm overflow-hidden text-ellipsis whitespace-nowrap max-w-[theme(spacing.56)]"
};
var _hoisted_17 = {
  key: 0,
  class: "max-w-[theme(spacing.56)] break-all break-words leading-5 inline-block text-color-icon-secondary text-xs sm:text-xs"
};
var _hoisted_18 = {
  key: 0,
  "data-tag": "tiny-timeline-showdata",
  class: "sm:ml-6 ml-5 sm:m-0 my-1 sm:leading-5 sm:h-5 leading-5.5 inline-flex items-center"
};
var _hoisted_19 = {
  key: 0,
  class: "inline-flex items-center"
};
var _hoisted_20 = {
  key: 1,
  class: "inline-flex items-center"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_chevron_up = resolveComponent("icon-chevron-up");
  var _component_icon_chevron_down = resolveComponent("icon-chevron-down");
  return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(true), createElementBlock(
    Fragment,
    null,
    renderList(_ctx.state.stackNodes, function(node, index) {
      return openBlock(), createElementBlock(
        "div",
        {
          "data-tag": "tiny-timeline-nodes",
          key: index,
          style: normalizeStyle({
            height: index === _ctx.state.stackNodes.length - 1 ? "" : _ctx.space && _ctx.space + "px"
          }),
          class: normalizeClass(["inline-flex", _ctx.getStatusCls(index)])
        },
        [createElementVNode("ul", _hoisted_2, [_ctx.subField ? renderSlot(_ctx.$slots, "left", {
          key: 0,
          slotScope: node
        }, function() {
          return [createElementVNode("li", _hoisted_3, [createElementVNode("div", _hoisted_4, [createElementVNode(
            "span",
            null,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).date),
            1
            /* TEXT */
          ), createElementVNode(
            "span",
            _hoisted_5,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).time),
            1
            /* TEXT */
          )]), createElementVNode("div", _hoisted_6, [createElementVNode(
            "span",
            _hoisted_7,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).date),
            1
            /* TEXT */
          ), createElementVNode(
            "span",
            _hoisted_8,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).time),
            1
            /* TEXT */
          )])])];
        }) : createCommentVNode("v-if", true), createElementVNode(
          "li",
          {
            "data-tag": "tiny-timeline-autocolor",
            style: normalizeStyle({
              height: index === _ctx.state.stackNodes.length - 1 ? "" : _ctx.space && _ctx.space + "px"
            }),
            class: "ml-1.5 relative h-full"
          },
          [createElementVNode(
            "div",
            {
              class: normalizeClass(["sm:w-px absolute mt-4 h-[calc(100%-theme(spacing.3))]", {
                "sm:bg-color-bg-3 sm:border-0 border-l-0.5 border-l-color-bg-3": index !== _ctx.state.stackNodes.length - 1
              }])
            },
            null,
            2
            /* CLASS */
          ), createElementVNode("div", _hoisted_9, [typeof node[_ctx.autoColorField] === "string" ? (openBlock(), createElementBlock(
            "span",
            {
              key: 0,
              class: normalizeClass(["rounded-full block w-1.5 h-1.5", node[_ctx.autoColorField] === "success" && "bg-color-success", node[_ctx.autoColorField] === "error" && "bg-color-error", node[_ctx.autoColorField] === "info" && "bg-color-text-disabled"])
            },
            null,
            2
            /* CLASS */
          )) : !node[_ctx.autoColorField] ? (openBlock(), createElementBlock("span", _hoisted_10)) : (openBlock(), createBlock(resolveDynamicComponent(node[_ctx.autoColorField]), {
            key: 2
          }))])],
          4
          /* STYLE */
        ), !_ctx.subField ? renderSlot(_ctx.$slots, "left", {
          key: 1,
          slotScope: node
        }, function() {
          return [createElementVNode("li", _hoisted_11, [createElementVNode(
            "span",
            _hoisted_12,
            toDisplayString(_ctx.getDate(node[_ctx.timeField]).date) + " " + toDisplayString(_ctx.getDate(node[_ctx.timeField]).time),
            1
            /* TEXT */
          ), node[_ctx.nameField] ? (openBlock(), createElementBlock(
            "span",
            _hoisted_13,
            toDisplayString(node[_ctx.nameField]),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true), node[_ctx.tipsField] ? (openBlock(), createElementBlock(
            "span",
            _hoisted_14,
            toDisplayString(node[_ctx.tipsField]),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)])];
        }) : createCommentVNode("v-if", true), createElementVNode(
          "li",
          {
            class: normalizeClass(["hidden sm:text-xs text-sm text-color-text-primary sm:inline-flex flex-col", _ctx.subField ? "ml-4.5" : "ml-3 "])
          },
          [createElementVNode(
            "span",
            {
              class: normalizeClass(["inline-block leading-5 sm:text-xs text-sm overflow-hidden text-ellipsis whitespace-nowrap max-w-[theme(spacing.56)]", {
                "mb-4": !_ctx.subField
              }])
            },
            toDisplayString(node[_ctx.nameField]),
            3
            /* TEXT, CLASS */
          ), _ctx.subField ? (openBlock(), createElementBlock(
            "span",
            _hoisted_15,
            toDisplayString(node[_ctx.tipsField]),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        ), _ctx.subField ? (openBlock(), createElementBlock(
          "li",
          {
            key: 2,
            class: normalizeClass(["sm:hidden text-xs text-color-text-primary inline-flex flex-col", _ctx.subField ? "ml-4.5" : "ml-3 "])
          },
          [createElementVNode(
            "span",
            _hoisted_16,
            toDisplayString(node[_ctx.nameField]),
            1
            /* TEXT */
          ), node[_ctx.tipsField] ? (openBlock(), createElementBlock(
            "span",
            _hoisted_17,
            toDisplayString(node[_ctx.tipsField]),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)],
          2
          /* CLASS */
        )) : createCommentVNode("v-if", true)])],
        6
        /* CLASS, STYLE */
      );
    }),
    128
    /* KEYED_FRAGMENT */
  )), _ctx.state.showData ? (openBlock(), createElementBlock("div", _hoisted_18, [createElementVNode("div", {
    "data-tag": "tiny-timeline-status",
    class: "sm:h-3 h-3.5 cursor-pointer inline-flex text-color-brand fill-color-brand sm:text-xs text-sm",
    onClick: _cache[0] || (_cache[0] = function() {
      return _ctx.changeStatus && _ctx.changeStatus.apply(_ctx, arguments);
    })
  }, [_ctx.state.showAll ? (openBlock(), createElementBlock("span", _hoisted_19, "收起")) : (openBlock(), createElementBlock("span", _hoisted_20, "展开")), _ctx.state.showAll ? (openBlock(), createBlock(_component_icon_chevron_up, {
    key: 2,
    class: "w-3 h-3 sm:ml-2 ml-1 my-px sm:my-0"
  })) : (openBlock(), createBlock(_component_icon_chevron_down, {
    key: 3,
    class: "w-3 h-3 sm:ml-2 ml-1 my-px sm:my-0"
  }))])])) : createCommentVNode("v-if", true)]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
