// @opentiny/vue-renderless/scrollbar/vue-bar.js
import "../chunk-G2ADBYYC.js";
import {
  BAR_MAP,
  renderThumbStyle,
  clickThumbHandler,
  clickTrackHandler,
  startDrag,
  mouseMoveDocumentHandler,
  mouseUpDocumentHandler
} from "./index";
import { on, off } from "../common/deps/dom";
const api = [
  "state",
  "startDrag",
  "renderThumbStyle",
  "clickTrackHandler",
  "clickThumbHandler",
  "mouseUpDocumentHandler",
  "mouseMoveDocumentHandler"
];
const renderless = (props, { computed, onUnmounted, reactive }, { refs, parent }) => {
  const state = reactive({
    bar: computed(() => BAR_MAP[props.vertical ? "vertical" : "horizontal"]),
    wrap: computed(() => parent.$refs.wrap)
  });
  const api2 = {
    state,
    renderThumbStyle,
    clickTrackHandler: clickTrackHandler({ refs, state }),
    mouseMoveDocumentHandler: mouseMoveDocumentHandler({ refs, state })
  };
  onUnmounted(() => off(document, "mouseup", api2.mouseUpDocumentHandler));
  return Object.assign(api2, {
    startDrag: startDrag({ api: api2, on, state }),
    clickThumbHandler: clickThumbHandler({ api: api2, state }),
    mouseUpDocumentHandler: mouseUpDocumentHandler({ api: api2, off, state })
  });
};
export {
  api,
  renderless
};
