// @opentiny/vue-record/lib/mobile-first.js
import { renderless, api } from "@opentiny/vue-renderless/record/vue";
import { defineComponent, props, setup as _setup } from "@opentiny/vue-common";
import ActionSheet from "@opentiny/vue-action-sheet";
import { IconMic } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, createVNode, withCtx, createElementVNode, toDisplayString, normalizeClass, createBlock } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var _sfc_main = defineComponent({
  props: [].concat(props, ["modelValue", "isHwh5", "lockScroll"]),
  components: {
    TinyActionSheet: ActionSheet,
    IconMic: IconMic()
  },
  inheritAttrs: false,
  setup: function setup(props2, context) {
    return _setup({
      props: props2,
      context,
      renderless,
      api
    });
  }
});
var _hoisted_1 = {
  class: "flex justify-center flex-wrap text-base text-color-text-secondary"
};
var _hoisted_2 = {
  key: 0,
  ref: "canvas",
  class: "w-full h-full"
};
var _hoisted_3 = {
  key: 1,
  class: "absolute left-0 bottom-0 w-full"
};
var _hoisted_4 = {
  key: 1,
  class: "w-6 h-6 bg-color-error rounded"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  var _component_icon_mic = resolveComponent("icon-mic");
  var _component_tiny_action_sheet = resolveComponent("tiny-action-sheet");
  return openBlock(), createElementBlock("div", null, [createVNode(_component_tiny_action_sheet, {
    title: _ctx.t("ui.record.record"),
    visible: _ctx.modelValue,
    "lock-scroll": _ctx.lockScroll,
    "custom-class": "pb-0",
    "mask-closable": false,
    "onUpdate:visible": _cache[3] || (_cache[3] = function($event) {
      return _ctx.$emit("update:modelValue", $event);
    })
  }, {
    "header-left": withCtx(function() {
      return [createElementVNode(
        "div",
        {
          class: "text-base text-color-text-secondary",
          onClick: _cache[0] || (_cache[0] = function() {
            return _ctx.handleCancel && _ctx.handleCancel.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.record.cancel")),
        1
        /* TEXT */
      )];
    }),
    "header-right": withCtx(function() {
      return [createElementVNode(
        "div",
        {
          class: normalizeClass(["text-color-brand", ["recording", "pause"].includes(_ctx.state.record.status) ? "" : "hidden"]),
          onClick: _cache[1] || (_cache[1] = function() {
            return _ctx.handleConfirm && _ctx.handleConfirm.apply(_ctx, arguments);
          })
        },
        toDisplayString(_ctx.t("ui.record.confirm")),
        3
        /* TEXT, CLASS */
      )];
    }),
    default: withCtx(function() {
      return [createElementVNode("div", _hoisted_1, [createElementVNode(
        "div",
        {
          class: normalizeClass(["font-light", ["recording", "pause"].includes(_ctx.state.record.status) ? "visible" : "invisible"])
        },
        toDisplayString(_ctx.state.recodeTime),
        3
        /* TEXT, CLASS */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass(["relative w-full text-center px-4", ["recording"].includes(_ctx.state.record.status) ? "h-24" : "h-10 my-7"])
        },
        [["recording"].includes(_ctx.state.record.status) ? (openBlock(), createElementBlock(
          "canvas",
          _hoisted_2,
          null,
          512
          /* NEED_PATCH */
        )) : (openBlock(), createElementBlock(
          "span",
          _hoisted_3,
          toDisplayString(["ready"].includes(_ctx.state.record.status) ? _ctx.t("ui.record.clickToStartRecording") : _ctx.t("ui.record.clickToResumeRecording")),
          1
          /* TEXT */
        ))],
        2
        /* CLASS */
      ), createElementVNode(
        "div",
        {
          class: normalizeClass(["flex w-16 h-16 justify-center items-center mt-0 m-10 border rounded-full cursor-pointer", [["recording"].includes(_ctx.state.record.status) ? "border-color-border-separator shadow-sm" : "border-color-brand"]])
        },
        [createElementVNode("div", {
          onClick: _cache[2] || (_cache[2] = function() {
            return _ctx.toogleRecord && _ctx.toogleRecord.apply(_ctx, arguments);
          })
        }, [!["recording"].includes(_ctx.state.record.status) ? (openBlock(), createBlock(_component_icon_mic, {
          key: 0,
          "custom-class": "w-10 h-10 fill-color-brand"
        })) : (openBlock(), createElementBlock("div", _hoisted_4))])],
        2
        /* CLASS */
      )])];
    }),
    _: 1
    /* STABLE */
  }, 8, ["title", "visible", "lock-scroll"])]);
}
var mobileFirst = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  mobileFirst as default
};
