// @opentiny/vue-column-list-item/lib/index.js
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
import { renderless, api } from "@opentiny/vue-renderless/column-list-item/vue";
import { defineComponent, $prefix, $props, setup as _setup } from "@opentiny/vue-common";
import Dropdown from "@opentiny/vue-dropdown";
import DropdownMenu from "@opentiny/vue-dropdown-menu";
import DropdownItem from "@opentiny/vue-dropdown-item";
import Checkbox from "@opentiny/vue-checkbox";
import Radio from "@opentiny/vue-radio";
import { IconEllipsis } from "@opentiny/vue-icon";
import { resolveComponent, openBlock, createElementBlock, normalizeClass, createVNode, createCommentVNode, withCtx, renderSlot, createElementVNode, normalizeStyle, Fragment, renderList, withModifiers, createBlock, resolveDynamicComponent, toDisplayString } from "vue";
function _createForOfIteratorHelperLoose_tiny(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it)
    return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray_tiny(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it)
      o = it;
    var i = 0;
    return function() {
      if (i >= o.length)
        return { done: true };
      return { done: false, value: o[i++] };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _unsupportedIterableToArray_tiny(o, minLen) {
  if (!o)
    return;
  if (typeof o === "string")
    return _arrayLikeToArray_tiny(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor)
    n = o.constructor.name;
  if (n === "Map" || n === "Set")
    return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n))
    return _arrayLikeToArray_tiny(o, minLen);
}
function _arrayLikeToArray_tiny(arr, len) {
  if (len == null || len > arr.length)
    len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++)
    arr2[i] = arr[i];
  return arr2;
}
var _export_sfc = function _export_sfc2(sfc, props) {
  var target = sfc.__vccOpts || sfc;
  for (var _iterator = _createForOfIteratorHelperLoose_tiny(props), _step; !(_step = _iterator()).done; ) {
    var _step$value = _step.value, key = _step$value[0], val = _step$value[1];
    target[key] = val;
  }
  return target;
};

var $constants = {
  COLUMN_GROUP: "ColumnListGroup"
};
var _sfc_main = defineComponent({
  name: $prefix + "ColumnListItem",
  emits: ["icon-click", "update:modelValue", "change"],
  components: {
    TinyDropdown: Dropdown,
    TinyDropdownMenu: DropdownMenu,
    TinyDropdownItem: DropdownItem,
    TinyCheckbox: Checkbox,
    TinyRadio: Radio,
    IconEllipsis: IconEllipsis()
  },
  props: _extends({}, $props, {
    _constants: {
      type: Object,
      default: function _default() {
        return $constants;
      }
    },
    modelValue: {},
    label: {},
    options: {
      type: Array,
      default: function _default2() {
        return [];
      }
    },
    flexGrow: {
      type: Array,
      default: function _default3() {
        return [1, 1, 1, 1];
      },
      validator: function validator(value) {
        return value.every(function(val) {
          return typeof val === "number" && val >= 0;
        });
      }
    },
    flexBasis: {
      type: Array,
      default: function _default4() {
        return [];
      }
    },
    image: {
      type: String,
      default: ""
    },
    iconMore: {
      type: [Object, String],
      default: function _default5() {
        return IconEllipsis();
      }
    },
    size: {
      type: String,
      default: "medium"
    },
    showCheckbox: {
      type: Boolean,
      default: false
    },
    showRadio: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    }
  }),
  setup: function setup(props, context) {
    return _setup({
      props,
      context,
      renderless,
      api,
      mono: true
    });
  }
});
var _hoisted_1 = {
  key: 0,
  "data-tag": "tiny-column-list-item-checkbox",
  class: "float-left sm:float-none mr-3 [&_[data-tag=tiny-checkbox]]:p-0 [&_[data-tag=tiny-checkbox]>span>span]:p-0 [&_[data-tag=tiny-checkbox]>span:nth-child(2)]:hidden"
};
var _hoisted_2 = {
  key: 1,
  "data-tag": "tiny-column-list-item-radio",
  class: "float-left sm:float-none mr-1.5 sm:mr-0 [&_[role=radio]]:w-6 [&_[role=radio]]:h-5 [&_[role=radio]_span_span]:p-0"
};
var _hoisted_3 = /* @__PURE__ */ createElementVNode(
  "span",
  null,
  null,
  -1
  /* HOISTED */
);
var _hoisted_4 = ["src"];
var _hoisted_5 = {
  "data-tag": "tiny-column-list-item-column1",
  class: "block text-left sm:flex flex-auto flex-wrap"
};
var _hoisted_6 = {
  key: 3,
  class: "block sm:hidden mt-2.5 -mx-3 border-t-0.5 border-t-color-border-separator"
};
var _hoisted_7 = {
  key: 4,
  "data-tag": "tiny-column-list-item-operatebox",
  class: "w-full sm:w-auto px-0 sm:pl-5 sm:pr-2 h-12 sm:h-auto flex flex-row sm:flex-col items-center justify-around sm:justify-center sm:items-start shadow-none sm:shadow-[-6px_0px_5px_-5px_#e8ebef] shrink-0 text-color-text-primary"
};
var _hoisted_8 = ["onClick"];
var _hoisted_9 = {
  key: 0,
  class: "ml-1 align-middle sm:align-bottom"
};
var _hoisted_10 = {
  key: 0,
  class: "ml-1 align-middle sm:align-bottom"
};
var _hoisted_11 = {
  key: 0,
  class: "ml-1 text-color-text-primary"
};
var _hoisted_12 = ["onClick"];
var _hoisted_13 = {
  key: 0
};
function _sfc_render(_ctx, _cache, $props2, $setup, $data, $options) {
  var _component_tiny_checkbox = resolveComponent("tiny-checkbox");
  var _component_tiny_radio = resolveComponent("tiny-radio");
  var _component_tiny_dropdown_item = resolveComponent("tiny-dropdown-item");
  var _component_tiny_dropdown_menu = resolveComponent("tiny-dropdown-menu");
  var _component_tiny_dropdown = resolveComponent("tiny-dropdown");
  return openBlock(), createElementBlock(
    "div",
    {
      "data-tag": "tiny-column-list-item",
      role: "column-list-item",
      class: normalizeClass(["block sm:flex w-full py-2.5 px-3 sm:pb-2.5 text-sm sm:text-xs border-0.5 sm:border rounded-lg sm:rounded hover:shadow border-color-border-separator", [_ctx.state.effectOptions.length ? "pb-0" : "pb-3", _ctx.image && !_ctx.state.showCheckbox && !_ctx.state.showRadio ? "pl-4" : "pl-3"]]),
      onClick: _cache[4] || (_cache[4] = function($event) {
        return _ctx.$emit("click", $event);
      })
    },
    [_ctx.state.showCheckbox ? (openBlock(), createElementBlock("div", _hoisted_1, [createVNode(_component_tiny_checkbox, {
      modelValue: _ctx.state.model,
      "onUpdate:modelValue": _cache[0] || (_cache[0] = function($event) {
        return _ctx.state.model = $event;
      }),
      disabled: _ctx.state.disabled,
      label: _ctx.label,
      onChange: _ctx.handleChange
    }, null, 8, ["modelValue", "disabled", "label", "onChange"])])) : createCommentVNode("v-if", true), _ctx.state.showRadio ? (openBlock(), createElementBlock("div", _hoisted_2, [createVNode(_component_tiny_radio, {
      modelValue: _ctx.state.model,
      "onUpdate:modelValue": _cache[1] || (_cache[1] = function($event) {
        return _ctx.state.model = $event;
      }),
      disabled: _ctx.state.disabled,
      label: _ctx.label,
      onChange: _ctx.handleChange
    }, {
      default: withCtx(function() {
        return [_hoisted_3];
      }),
      _: 1
      /* STABLE */
    }, 8, ["modelValue", "disabled", "label", "onChange"])])) : createCommentVNode("v-if", true), _ctx.image || _ctx.slots.image && _ctx.slots.image() ? (openBlock(), createElementBlock(
      "div",
      {
        key: 2,
        "data-tag": "tiny-column-list-item-image",
        class: normalizeClass(["float-left block sm:flex mr-3 shrink-0 pt-1", [_ctx.state.size === "small" ? "w-9 h-11" : "w-14 sm:w-11 h-16"]])
      },
      [renderSlot(_ctx.$slots, "image", {}, function() {
        return [createElementVNode("img", {
          src: _ctx.image,
          class: normalizeClass(["rounded", [_ctx.state.size === "small" ? "w-9 h-9" : "w-14 sm:w-11 h-14 sm:h-11"]])
        }, null, 10, _hoisted_4)];
      })],
      2
      /* CLASS */
    )) : createCommentVNode("v-if", true), createElementVNode("div", _hoisted_5, [_ctx.slots.column1 && _ctx.slots.column1() ? (openBlock(), createElementBlock(
      "div",
      {
        key: 0,
        class: "min-w-[theme(spacing.40)] overflow-hidden sm:ml-0 pr-0 sm:pr-5",
        style: normalizeStyle({
          "flex-grow": _ctx.flexGrow[0] || 0,
          "flex-basis": _ctx.flexBasis[0] || "auto"
        })
      },
      [renderSlot(_ctx.$slots, "column1")],
      4
      /* STYLE */
    )) : createCommentVNode("v-if", true), _ctx.slots.column2 && _ctx.slots.column2() ? (openBlock(), createElementBlock(
      "div",
      {
        key: 1,
        "data-tag": "tiny-column-list-item-column2",
        class: normalizeClass(["min-w-[theme(spacing.40)] overflow-hidden pr-0 sm:pr-5 mt-1 sm:mt-0 pt-0 sm:pt-1", [_ctx.state.showCheckbox || _ctx.state.showRadio ? "pl-8 sm:pl-0" : ""]]),
        style: normalizeStyle({
          "flex-grow": _ctx.flexGrow[1] || 0,
          "flex-basis": _ctx.flexBasis[1] || "auto"
        })
      },
      [renderSlot(_ctx.$slots, "column2")],
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true), _ctx.slots.column3 && _ctx.slots.column3() ? (openBlock(), createElementBlock(
      "div",
      {
        key: 2,
        "data-tag": "tiny-column-list-item-column3",
        class: normalizeClass(["min-w-[theme(spacing.40)] overflow-hidden pr-0 sm:pr-5 mt-1 sm:mt-0 pt-0 sm:pt-1", [_ctx.state.showCheckbox || _ctx.state.showRadio ? "pl-8 sm:pl-0" : ""]]),
        style: normalizeStyle({
          "flex-grow": _ctx.flexGrow[2] || 0,
          "flex-basis": _ctx.flexBasis[2] || "auto"
        })
      },
      [renderSlot(_ctx.$slots, "column3")],
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true), _ctx.slots.column4 && _ctx.slots.column4() ? (openBlock(), createElementBlock(
      "div",
      {
        key: 3,
        "data-tag": "tiny-column-list-item-column4",
        class: normalizeClass(["min-w-[theme(spacing.40)] overflow-hidden pr-0 sm:pr-5 mt-1 sm:mt-0 pt-0 sm:pt-1", [_ctx.state.showCheckbox || _ctx.state.showRadio ? "pl-8 sm:pl-0" : ""]]),
        style: normalizeStyle({
          "flex-grow": _ctx.flexGrow[3] || 0,
          "flex-basis": _ctx.flexBasis[3] || "auto"
        })
      },
      [renderSlot(_ctx.$slots, "column4")],
      6
      /* CLASS, STYLE */
    )) : createCommentVNode("v-if", true)]), _ctx.state.effectOptions.length || _ctx.slots.operate && _ctx.slots.operate() ? (openBlock(), createElementBlock("hr", _hoisted_6)) : createCommentVNode("v-if", true), _ctx.state.effectOptions.length || _ctx.slots.operate && _ctx.slots.operate() ? (openBlock(), createElementBlock("div", _hoisted_7, [renderSlot(_ctx.$slots, "operate", {}, function() {
      return [(openBlock(true), createElementBlock(
        Fragment,
        null,
        renderList(_ctx.state.effectOptions.slice(0, _ctx.state.sliceNum), function(item, index) {
          return openBlock(), createElementBlock("div", {
            "data-tag": "tiny-column-list-item-operate",
            class: normalizeClass(["cursor-pointer mb-0 sm:mb-2", [item.disabled ? "text-color-text-disabled cursor-not-allowed" : ""]]),
            key: item.text + index,
            onClick: withModifiers(function($event) {
              return _ctx.handelIconClick(item, index, $event);
            }, ["stop"])
          }, [(openBlock(), createBlock(resolveDynamicComponent(item.icon), {
            class: normalizeClass(["w-4 h-4", [item.disabled ? "fill-color-icon-disabled" : ""]])
          }, null, 8, ["class"])), item.text ? (openBlock(), createElementBlock(
            "span",
            _hoisted_9,
            toDisplayString(item.text),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)], 10, _hoisted_8);
        }),
        128
        /* KEYED_FRAGMENT */
      )), _ctx.state.effectOptions.length === _ctx.state.iconNum ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          "data-tag": "tiny-column-list-item-iconnum",
          class: normalizeClass(["cursor-pointer", [_ctx.state.effectOptions[_ctx.state.sliceNum].disabled ? "text-color-text-disabled cursor-not-allowed" : ""]]),
          onClick: _cache[2] || (_cache[2] = withModifiers(function($event) {
            return _ctx.handelIconClick(_ctx.state.effectOptions[_ctx.state.sliceNum], _ctx.state.sliceNum, $event);
          }, ["stop"]))
        },
        [(openBlock(), createBlock(resolveDynamicComponent(_ctx.state.effectOptions[_ctx.state.sliceNum].icon), {
          class: normalizeClass(["w-4 h-4", [_ctx.state.effectOptions[_ctx.state.sliceNum].disabled ? "fill-color-icon-disabled" : ""]])
        }, null, 8, ["class"])), _ctx.state.effectOptions[_ctx.state.sliceNum].text ? (openBlock(), createElementBlock(
          "span",
          _hoisted_10,
          toDisplayString(_ctx.state.effectOptions[_ctx.state.sliceNum].text),
          1
          /* TEXT */
        )) : createCommentVNode("v-if", true)],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true), _ctx.state.effectOptions.length > _ctx.state.iconNum ? (openBlock(), createBlock(_component_tiny_dropdown, {
        key: 1,
        "show-self-icon": "",
        onClick: _cache[3] || (_cache[3] = withModifiers(function() {
        }, ["stop"]))
      }, {
        dropdown: withCtx(function() {
          return [createVNode(_component_tiny_dropdown_menu, {
            placement: "bottom"
          }, {
            default: withCtx(function() {
              return [(openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(_ctx.state.effectOptions.slice(_ctx.state.sliceNum), function(item, index) {
                  return openBlock(), createBlock(
                    _component_tiny_dropdown_item,
                    {
                      key: item.text + index
                    },
                    {
                      default: withCtx(function() {
                        return [createElementVNode("div", {
                          onClick: withModifiers(function($event) {
                            return _ctx.handelIconClick(item, index + _ctx.state.sliceNum, $event);
                          }, ["stop"]),
                          class: normalizeClass(["px-2", item.disabled ? "text-color-text-disabled cursor-not-allowed" : ""])
                        }, [(openBlock(), createBlock(resolveDynamicComponent(item.icon), {
                          class: normalizeClass(["w-4 h-4", [item.disabled ? "fill-color-icon-disabled" : ""]])
                        }, null, 8, ["class"])), item.text ? (openBlock(), createElementBlock(
                          "span",
                          _hoisted_13,
                          toDisplayString(item.text),
                          1
                          /* TEXT */
                        )) : createCommentVNode("v-if", true)], 10, _hoisted_12)];
                      }),
                      _: 2
                      /* DYNAMIC */
                    },
                    1024
                    /* DYNAMIC_SLOTS */
                  );
                }),
                128
                /* KEYED_FRAGMENT */
              ))];
            }),
            _: 1
            /* STABLE */
          })];
        }),
        default: withCtx(function() {
          return [(openBlock(), createBlock(resolveDynamicComponent(_ctx.iconMore), {
            class: "w-4 h-4"
          })), _ctx.state.effectOptions[0].text ? (openBlock(), createElementBlock(
            "span",
            _hoisted_11,
            toDisplayString(_ctx.t("ui.base.more")),
            1
            /* TEXT */
          )) : createCommentVNode("v-if", true)];
        }),
        _: 1
        /* STABLE */
      })) : createCommentVNode("v-if", true)];
    })])) : createCommentVNode("v-if", true)],
    2
    /* CLASS */
  );
}
var ColumnListItem = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
var version = "3.13.0";
ColumnListItem.model = {
  prop: "modelValue",
  event: "update:modelValue"
};
ColumnListItem.install = function(Vue) {
  Vue.component(ColumnListItem.name, ColumnListItem);
};
ColumnListItem.version = version;
export {
  ColumnListItem as default
};
